{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Recon2\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Imprting Json file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "with urllib.request.urlopen(\"https://raw.githubusercontent.com/sehir-bioinformatics-database-lab/metabolitics/master/metabolitics/datasets/network_models/recon2.json\") as url:\n",
    "    data = json.loads(url.read().decode())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Exploring Recon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Recon items:\n",
      "compartments, genes, id, metabolites, reactions, version, "
     ]
    }
   ],
   "source": [
    "print(\"Recon items:\")\n",
    "for i in data:\n",
    "    print(i, end = ', ')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#reactions: 7785\n",
      "#metabolites: 5324\n",
      "#genes: 1675\n"
     ]
    }
   ],
   "source": [
    "print(\"#reactions:\", len(data[\"reactions\"]))\n",
    "print(\"#metabolites:\", len(data[\"metabolites\"]))\n",
    "print(\"#genes:\",len(data[\"genes\"]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'subsystem': 'beta-Alanine metabolism',\n",
       "  'name': '1,3-Diaminopropane:oxygen oxidoreductase (deaminating)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:549 or HGNC:550 or HGNC:80']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'bamppald_c': 1.0,\n",
       "   'h2o2_c': 1.0,\n",
       "   '13dampp_c': -1.0,\n",
       "   'nh4_c': 1.0},\n",
       "  'id': '13DAMPPOX',\n",
       "  'gene_reaction_rule': 'HGNC:549 or HGNC:550 or HGNC:80'},\n",
       " {'subsystem': 'Vitamin D metabolism',\n",
       "  'name': '1-alpha-Vitamin D-24,25-hydroxylase (D2)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'], 'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'2425dhvitd2_m': -1.0,\n",
       "   'nadph_m': -1.0,\n",
       "   'h2o_m': 1.0,\n",
       "   'nadp_m': 1.0,\n",
       "   'o2_m': -1.0,\n",
       "   '1a2425thvitd2_m': 1.0,\n",
       "   'h_m': -1.0},\n",
       "  'id': '1a_24_25VITD2Hm',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Vitamin D metabolism',\n",
       "  'name': '1-alpha-Vitamin D-24,25-hydroxylase (D3)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'], 'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'2425dhvitd3_m': -1.0,\n",
       "   '1a2425thvitd3_m': 1.0,\n",
       "   'nadph_m': -1.0,\n",
       "   'h2o_m': 1.0,\n",
       "   'nadp_m': 1.0,\n",
       "   'o2_m': -1.0,\n",
       "   'h_m': -1.0},\n",
       "  'id': '1a_24_25VITD3Hm',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Vitamin D metabolism',\n",
       "  'name': '1-alpha,24R,25-Vitamin D-hydroxylase (D2)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'nadph_m': -1.0,\n",
       "   'h2o_m': 1.0,\n",
       "   'nadp_m': 1.0,\n",
       "   'o2_m': -1.0,\n",
       "   '1a2425thvitd2_m': 1.0,\n",
       "   '1a25dhvitd2_m': -1.0,\n",
       "   'h_m': -1.0},\n",
       "  'id': '1a_25VITD2Hm',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Vitamin D metabolism',\n",
       "  'name': '1-alpha,24R,25-Vitamin D-hydroxylase (D3)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'1a2425thvitd3_m': 1.0,\n",
       "   'nadph_m': -1.0,\n",
       "   'h2o_m': 1.0,\n",
       "   'nadp_m': 1.0,\n",
       "   'o2_m': -1.0,\n",
       "   '1a25dhvitd3_m': -1.0,\n",
       "   'h_m': -1.0},\n",
       "  'id': '1a_25VITD3Hm',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Lysine metabolism',\n",
       "  'name': 'delta1-piperideine-2-carboxylate reductase, perixosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['MM'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'nadh_x': -1.0,\n",
       "   '1pipdn2c_x': -1.0,\n",
       "   'nad_x': 1.0,\n",
       "   'h_x': -1.0,\n",
       "   'Lpipecol_x': 1.0},\n",
       "  'id': '1PPDCRp',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Vitamin D metabolism',\n",
       "  'name': '24R-Vitamin D-25-hydroxylase (D2)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['based on Vitamins, G.F.M. Ball,2004, Blackwell publishing, 1st ed (book) pg.196 IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2602']},\n",
       "  'metabolites': {'2425dhvitd2_m': 1.0,\n",
       "   'nadph_m': -1.0,\n",
       "   'h2o_m': 1.0,\n",
       "   'nadp_m': 1.0,\n",
       "   'o2_m': -1.0,\n",
       "   '25hvitd2_m': -1.0,\n",
       "   'h_m': -1.0},\n",
       "  'id': '24_25VITD2Hm',\n",
       "  'gene_reaction_rule': 'HGNC:2602'},\n",
       " {'subsystem': 'Vitamin D metabolism',\n",
       "  'name': '24R-Vitamin D-25-hydroxylase (D3)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['based on Vitamins, G.F.M. Ball,2004, Blackwell publishing, 1st ed (book) pg.196 IT; RE2240M'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2602']},\n",
       "  'metabolites': {'2425dhvitd3_m': 1.0,\n",
       "   'nadph_m': -1.0,\n",
       "   'h2o_m': 1.0,\n",
       "   'nadp_m': 1.0,\n",
       "   'o2_m': -1.0,\n",
       "   '25hvitd3_m': -1.0,\n",
       "   'h_m': -1.0},\n",
       "  'id': '24_25VITD3Hm',\n",
       "  'gene_reaction_rule': 'HGNC:2602'},\n",
       " {'subsystem': 'Vitamin D metabolism',\n",
       "  'name': '1-alpha-Vitamin D-25-hydroxylase (D2)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['based on Vitamins, G.F.M. Ball,2004, Blackwell publishing, 1st ed (book) pg.196 IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2606']},\n",
       "  'metabolites': {'nadph_m': -1.0,\n",
       "   'h2o_m': 1.0,\n",
       "   'nadp_m': 1.0,\n",
       "   'o2_m': -1.0,\n",
       "   '1a25dhvitd2_m': 1.0,\n",
       "   '25hvitd2_m': -1.0,\n",
       "   'h_m': -1.0},\n",
       "  'id': '25VITD2Hm',\n",
       "  'gene_reaction_rule': 'HGNC:2606'},\n",
       " {'subsystem': 'Vitamin D metabolism',\n",
       "  'name': '1-alpha-Vitamin D-25-hydroxylase (D3)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['based on Vitamins, G.F.M. Ball,2004, Blackwell publishing, 1st ed (book) pg.196 IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2606']},\n",
       "  'metabolites': {'nadph_m': -1.0,\n",
       "   'h2o_m': 1.0,\n",
       "   'nadp_m': 1.0,\n",
       "   'o2_m': -1.0,\n",
       "   '1a25dhvitd3_m': 1.0,\n",
       "   '25hvitd3_m': -1.0,\n",
       "   'h_m': -1.0},\n",
       "  'id': '25VITD3Hm',\n",
       "  'gene_reaction_rule': 'HGNC:2606'},\n",
       " {'subsystem': 'Glycine, serine, alanine and threonine metabolism',\n",
       "  'name': '2-Aminoacrylate hydrolysis',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB  see PMID 500639:  this reaction is spontaneous'],\n",
       "   'GENE ASSOCIATION': ['HGNC:10691']},\n",
       "  'metabolites': {'2amac_c': -1.0, 'h2o_c': -1.0, 'pyr_c': 1.0, 'nh4_c': 1.0},\n",
       "  'id': '2AMACHYD',\n",
       "  'gene_reaction_rule': 'HGNC:10691'},\n",
       " {'subsystem': 'Methionine and cysteine metabolism',\n",
       "  'name': '2-Aminoacrylate sulfotransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['-this is a lumped rxn w/ an inferred reduction rxn that occurs in conjunction w/ sulfotransferase step; this reduction step has to occur in order for the transformation (2amac--&gt;Lcyst) to be valid -used NADPH as electron donor since it is a common cofactor involved in biosynthetic steps utilizing oxygen (see Neema for further explanations)  MM'],\n",
       "   'GENE ASSOCIATION': ['HGNC:11455']},\n",
       "  'metabolites': {'Lcyst_c': 1.0,\n",
       "   'nadp_c': 1.0,\n",
       "   'nadph_c': -1.0,\n",
       "   '2amac_c': -1.0,\n",
       "   'paps_c': -1.0,\n",
       "   'pap_c': 1.0},\n",
       "  'id': '2AMACSULT',\n",
       "  'gene_reaction_rule': 'HGNC:11455'},\n",
       " {'subsystem': 'Pyrimidine catabolism',\n",
       "  'name': '2-deoxy-D-ribose 1-phosphate phosphorylase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'h2o_c': -1.0, 'pi_c': 1.0, '2dr1p_c': -1.0, 'drib_c': 1.0},\n",
       "  'id': '2DR1PP',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Propanoate metabolism',\n",
       "  'name': '2-Hydroxybutyrate:NAD+ oxidoreductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:21481 or HGNC:28335 or HGNC:30866 or (HGNC:6535 and HGNC:6541) or HGNC:6535 or HGNC:6541 or HGNC:6544']},\n",
       "  'metabolites': {'2obut_c': 1.0,\n",
       "   'nadh_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'nad_c': -1.0,\n",
       "   '2hb_c': -1.0},\n",
       "  'id': '2HBO',\n",
       "  'gene_reaction_rule': 'HGNC:21481 or HGNC:28335 or HGNC:30866 or (HGNC:6535 and HGNC:6541) or HGNC:6535 or HGNC:6541 or HGNC:6544'},\n",
       " {'subsystem': 'Lysine metabolism',\n",
       "  'name': '2-Oxoadipate:lipoamde 2-oxidoreductase(decarboxylating and acceptor-succinylating) (mitochondria)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:21350 and HGNC:2898 and HGNC:2911 and HGNC:8124']},\n",
       "  'metabolites': {'2oxoadp_m': -1.0,\n",
       "   'nadh_m': 1.0,\n",
       "   'nad_m': -1.0,\n",
       "   'coa_m': -1.0,\n",
       "   'glutcoa_m': 1.0,\n",
       "   'co2_m': 1.0},\n",
       "  'id': '2OXOADOXm',\n",
       "  'gene_reaction_rule': 'HGNC:21350 and HGNC:2898 and HGNC:2911 and HGNC:8124'},\n",
       " {'subsystem': 'Tyrosine metabolism',\n",
       "  'name': '3,4-Dihydroxyphenylethyleneglycol:NAD+ oxidoreductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:16354 or (HGNC:249 and HGNC:250) or (HGNC:249 and HGNC:251) or (HGNC:250 and HGNC:251) or HGNC:252 or HGNC:253 or HGNC:255 or HGNC:256']},\n",
       "  'metabolites': {'34dhoxpeg_c': 1.0,\n",
       "   'nad_c': 1.0,\n",
       "   'nadh_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   '34dhmald_c': -1.0},\n",
       "  'id': '34DHOXPEGOX',\n",
       "  'gene_reaction_rule': 'HGNC:16354 or (HGNC:249 and HGNC:250) or (HGNC:249 and HGNC:251) or (HGNC:250 and HGNC:251) or HGNC:252 or HGNC:253 or HGNC:255 or HGNC:256'},\n",
       " {'subsystem': 'Tyrosine metabolism',\n",
       "  'name': '3,4-Dihydroxyphenylacetate:amet O-methyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2228']},\n",
       "  'metabolites': {'amet_c': -1.0,\n",
       "   'homoval_c': 1.0,\n",
       "   'ahcys_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   '34dhpha_c': -1.0},\n",
       "  'id': '34DHPHAMT',\n",
       "  'gene_reaction_rule': 'HGNC:2228'},\n",
       " {'subsystem': 'Tyrosine metabolism',\n",
       "  'name': '3,4-Dihydroxyphenylacetaldehyde:NAD+ oxidoreductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:405 or HGNC:409 or HGNC:410 or HGNC:411']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   '34dhpac_c': -1.0,\n",
       "   '34dhpha_c': 1.0,\n",
       "   'h_c': 2.0,\n",
       "   'nad_c': -1.0,\n",
       "   'nadh_c': 1.0},\n",
       "  'id': '34DHPLACOX',\n",
       "  'gene_reaction_rule': 'HGNC:405 or HGNC:409 or HGNC:410 or HGNC:411'},\n",
       " {'subsystem': 'Tyrosine metabolism',\n",
       "  'name': '3,4-Dihydroxyphenylacetaldehyde:NADP+ oxidoreductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:405 or HGNC:410 or HGNC:411']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   'nadp_c': -1.0,\n",
       "   'nadph_c': 1.0,\n",
       "   '34dhpac_c': -1.0,\n",
       "   '34dhpha_c': 1.0,\n",
       "   'h_c': 2.0},\n",
       "  'id': '34DHPLACOX_NADP_',\n",
       "  'gene_reaction_rule': 'HGNC:405 or HGNC:410 or HGNC:411'},\n",
       " {'subsystem': 'Tyrosine metabolism',\n",
       "  'name': '3,4-Dihydroxymandelaldehyde:NAD+ oxidoreductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:405 or HGNC:409 or HGNC:410']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   '34dhoxmand_c': 1.0,\n",
       "   '34dhmald_c': -1.0,\n",
       "   'h_c': 2.0,\n",
       "   'nad_c': -1.0,\n",
       "   'nadh_c': 1.0},\n",
       "  'id': '34DHXMANDACOX',\n",
       "  'gene_reaction_rule': 'HGNC:405 or HGNC:409 or HGNC:410'},\n",
       " {'subsystem': 'Tyrosine metabolism',\n",
       "  'name': '3,4-Dihydroxymandelaldehyde:NADP+ oxidoreductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:410 or HGNC:411']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   'nadp_c': -1.0,\n",
       "   'nadph_c': 1.0,\n",
       "   '34dhoxmand_c': 1.0,\n",
       "   '34dhmald_c': -1.0,\n",
       "   'h_c': 2.0},\n",
       "  'id': '34DHXMANDACOX_NADP_',\n",
       "  'gene_reaction_rule': 'HGNC:410 or HGNC:411'},\n",
       " {'subsystem': 'Ubiquinone synthesis',\n",
       "  'name': '3-(4-hydroxyphenyl-)lactate formation',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['IT reaction involved in formation of 4-hydroxybenzoate from tyrosine.'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'34hpl_m': 1.0,\n",
       "   'nad_m': 1.0,\n",
       "   'h_m': -1.0,\n",
       "   '34hpp_m': -1.0,\n",
       "   'nadh_m': -1.0},\n",
       "  'id': '34HPLFM',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Tyrosine metabolism',\n",
       "  'name': '4-Hydroxyphenylpyruvate:oxygen oxidoreductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:5147']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   '34hpp_c': -1.0,\n",
       "   'co2_c': 1.0,\n",
       "   'hgentis_c': 1.0},\n",
       "  'id': '34HPPOR',\n",
       "  'gene_reaction_rule': 'HGNC:5147'},\n",
       " {'subsystem': 'Valine, leucine, and isoleucine metabolism',\n",
       "  'name': 'L-3-aminoisobutyrate transaminase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'glu_L_m': -1.0,\n",
       "   '2mop_m': -1.0,\n",
       "   'akg_m': 1.0,\n",
       "   '3aib_m': 1.0},\n",
       "  'id': '3AIBTm',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Ubiquinone synthesis',\n",
       "  'name': 'Hydroxylation of 3-Decaprenyl-4-hydroxybenzoate (NADH)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['Used to make Ubiquinone-10 enzyme has not been identified yet. IT'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'3dphb_m': -1.0,\n",
       "   'h2o_m': 1.0,\n",
       "   'o2_m': -1.0,\n",
       "   'nadh_m': -1.0,\n",
       "   '3dpdhb_m': 1.0,\n",
       "   'nad_m': 1.0,\n",
       "   'h_m': -1.0},\n",
       "  'id': '3DPHBH1',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Ubiquinone synthesis',\n",
       "  'name': 'Hydroxylation of 3-Decaprenyl-4-hydroxybenzoate (NADPH)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['Used to make Ubiquinone-10 enzyme has not been identified yet. IT'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'3dphb_m': -1.0,\n",
       "   'h2o_m': 1.0,\n",
       "   'nadp_m': 1.0,\n",
       "   'o2_m': -1.0,\n",
       "   '3dpdhb_m': 1.0,\n",
       "   'h_m': -1.0,\n",
       "   'nadph_m': -1.0},\n",
       "  'id': '3DPHBH2',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': '3-Dehydrosphinganine reductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:4021']},\n",
       "  'metabolites': {'nadph_c': -1.0,\n",
       "   'nadp_c': 1.0,\n",
       "   '3dsphgn_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'sphgn_c': 1.0},\n",
       "  'id': '3DSPHR',\n",
       "  'gene_reaction_rule': 'HGNC:4021'},\n",
       " {'subsystem': 'Tryptophan metabolism',\n",
       "  'name': '3-hydroxyanthranilate 3,4-dioxygenase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4796']},\n",
       "  'metabolites': {'3hanthrn_c': -1.0,\n",
       "   'o2_c': -1.0,\n",
       "   'cmusa_c': 1.0,\n",
       "   'h_c': 1.0},\n",
       "  'id': '3HAO',\n",
       "  'gene_reaction_rule': 'HGNC:4796'},\n",
       " {'subsystem': 'Butanoate metabolism',\n",
       "  'name': '3-hydroxybutyryl-CoA dehydratase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD. rat gene that is homologous to the human gene encodes for this activity'],\n",
       "   'GENE ASSOCIATION': ['HGNC:23408']},\n",
       "  'metabolites': {'h2o_m': -1.0, 'b2coa_m': -1.0, '3hbcoa_R_m': 1.0},\n",
       "  'id': '3HBCDm',\n",
       "  'gene_reaction_rule': 'HGNC:23408'},\n",
       " {'subsystem': 'Valine, leucine, and isoleucine metabolism',\n",
       "  'name': '3-hydroxyisobutyryl-CoA hydrolase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['MM'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'3hmp_m': 1.0,\n",
       "   '3hibutcoa_m': -1.0,\n",
       "   'coa_m': 1.0,\n",
       "   'h_m': 1.0,\n",
       "   'h2o_m': -1.0},\n",
       "  'id': '3HBCOAHLm',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Tryptophan metabolism',\n",
       "  'name': '3-Hydroxy-L-kynurenine:2-oxoglutarate aminotransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1564 or HGNC:17929']},\n",
       "  'metabolites': {'akg_c': -1.0,\n",
       "   '42A3HP24DB_c': 1.0,\n",
       "   'hLkynr_c': -1.0,\n",
       "   'glu_L_c': 1.0},\n",
       "  'id': '3HKYNAKGAT',\n",
       "  'gene_reaction_rule': 'HGNC:1564 or HGNC:17929'},\n",
       " {'subsystem': 'Tyrosine metabolism',\n",
       "  'name': '3-Hydroxy-L-tyrosine carboxy-lyase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2719']},\n",
       "  'metabolites': {'co2_c': 1.0, '34dhphe_c': -1.0, 'dopa_c': 1.0, 'h_c': -1.0},\n",
       "  'id': '3HLYTCL',\n",
       "  'gene_reaction_rule': 'HGNC:2719'},\n",
       " {'subsystem': 'beta-Alanine metabolism',\n",
       "  'name': '3-hydroxyisobutyryl-Coenzyme A hydrolase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4908']},\n",
       "  'metabolites': {'3hpcoa_c': -1.0,\n",
       "   '3hpp_c': 1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'h_c': 1.0},\n",
       "  'id': '3HPCOAHYD',\n",
       "  'gene_reaction_rule': 'HGNC:4908'},\n",
       " {'subsystem': 'Propanoate metabolism',\n",
       "  'name': '3-hydroxypropionate dehydrogenase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'msa_c': 1.0,\n",
       "   '3hpp_c': -1.0,\n",
       "   'nadh_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'nad_c': -1.0},\n",
       "  'id': '3HPPD',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Tryptophan metabolism',\n",
       "  'name': '3-Hydroxykynurenamine decarboxy-lyase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['SAB made up name'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2719']},\n",
       "  'metabolites': {'3hxkynam_c': 1.0,\n",
       "   'hLkynr_c': -1.0,\n",
       "   'co2_c': 1.0,\n",
       "   'h_c': -1.0},\n",
       "  'id': '3HXKYNDCL',\n",
       "  'gene_reaction_rule': 'HGNC:2719'},\n",
       " {'subsystem': 'Tryptophan metabolism',\n",
       "  'name': '3-Hydroxykynurenamine:oxygen oxidoreductase(deaminating)(flavin-containing)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:6833 or HGNC:6834']},\n",
       "  'metabolites': {'3hxkynam_c': -1.0,\n",
       "   'h2o2_c': 1.0,\n",
       "   '48dhoxquin_c': 1.0,\n",
       "   'nh4_c': 1.0,\n",
       "   'o2_c': -1.0},\n",
       "  'id': '3HXKYNOXDA',\n",
       "  'gene_reaction_rule': 'HGNC:6833 or HGNC:6834'},\n",
       " {'subsystem': 'Tyrosine metabolism',\n",
       "  'name': '3-Methoxy-4-hydroxyphenylacetaldehyde:NAD+ oxidoreductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:405 or HGNC:409 or HGNC:410 or HGNC:411']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   'homoval_c': 1.0,\n",
       "   'h_c': 2.0,\n",
       "   'nad_c': -1.0,\n",
       "   'nadh_c': 1.0,\n",
       "   '3mox4hpac_c': -1.0},\n",
       "  'id': '3M4HDXPAC',\n",
       "  'gene_reaction_rule': 'HGNC:405 or HGNC:409 or HGNC:410 or HGNC:411'},\n",
       " {'subsystem': 'Tyrosine metabolism',\n",
       "  'name': '3-Methoxy-4-hydroxyphenylglycolaldehyde:NAD+ oxidoreductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:405 or HGNC:409 or HGNC:410 or HGNC:411']},\n",
       "  'metabolites': {'3mox4hoxm_c': 1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   '3m4hpga_c': -1.0,\n",
       "   'h_c': 2.0,\n",
       "   'nad_c': -1.0,\n",
       "   'nadh_c': 1.0},\n",
       "  'id': '3MOX4HOXPGALDOX',\n",
       "  'gene_reaction_rule': 'HGNC:405 or HGNC:409 or HGNC:410 or HGNC:411'},\n",
       " {'subsystem': 'Tyrosine metabolism',\n",
       "  'name': '3-Methoxy-4-hydroxyphenylglycolaldehyde:NADP+ oxidoreductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:405 or HGNC:409 or HGNC:410 or HGNC:411']},\n",
       "  'metabolites': {'3mox4hoxm_c': 1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   '3m4hpga_c': -1.0,\n",
       "   'nadph_c': 1.0,\n",
       "   'nadp_c': -1.0,\n",
       "   'h_c': 2.0},\n",
       "  'id': '3MOX4HOXPGALDOX_NADP_',\n",
       "  'gene_reaction_rule': 'HGNC:405 or HGNC:409 or HGNC:410 or HGNC:411'},\n",
       " {'subsystem': 'Tyrosine metabolism',\n",
       "  'name': '3-Methoxytyramine:oxygen oxidoreductase (deaminating)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:6833 or HGNC:6834']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   '3moxtyr_c': -1.0,\n",
       "   'h2o2_c': 1.0,\n",
       "   'nh4_c': 1.0,\n",
       "   '3mox4hpac_c': 1.0},\n",
       "  'id': '3MOXTYROX',\n",
       "  'gene_reaction_rule': 'HGNC:6833 or HGNC:6834'},\n",
       " {'subsystem': 'Purine catabolism',\n",
       "  'name': '3-nucleotidase (AMP), lysosome',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['MKA AMF IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:123 or HGNC:124 or HGNC:125 or HGNC:29609']},\n",
       "  'metabolites': {'3amp_l': -1.0, 'adn_l': 1.0, 'pi_l': 1.0, 'h2o_l': -1.0},\n",
       "  'id': '3NTD7l',\n",
       "  'gene_reaction_rule': 'HGNC:123 or HGNC:124 or HGNC:125 or HGNC:29609'},\n",
       " {'subsystem': 'Taurine and hypotaurine metabolism',\n",
       "  'name': '3-Sulfino-L-alanine carboxy-lyase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:18966 or HGNC:4092 or HGNC:4093']},\n",
       "  'metabolites': {'co2_c': 1.0,\n",
       "   'hyptaur_c': 1.0,\n",
       "   'h_c': -2.0,\n",
       "   '3sala_c': -1.0},\n",
       "  'id': '3SALACBOXL',\n",
       "  'gene_reaction_rule': 'HGNC:18966 or HGNC:4092 or HGNC:4093'},\n",
       " {'subsystem': 'Taurine and hypotaurine metabolism',\n",
       "  'name': 'cysteinesulfinic acid oxidase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['as described in Bender (Nutritional biochmeistry of the vitamins, 2nd edition)  MM'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'Lcyst_c': 2.0, 'o2_c': -1.0, 'h_c': -2.0, '3sala_c': -2.0},\n",
       "  'id': '3SALAOX',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Methionine and cysteine metabolism',\n",
       "  'name': '3-sulfino-alanine transaminase (irreversible)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['MM. activity measured in rat, 3-sulfinoalanine is substrate found in human brain'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4432']},\n",
       "  'metabolites': {'akg_c': -1.0,\n",
       "   '3snpyr_c': 1.0,\n",
       "   'h_c': -1.0,\n",
       "   'glu_L_c': 1.0,\n",
       "   '3sala_c': -1.0},\n",
       "  'id': '3SALATAi',\n",
       "  'gene_reaction_rule': 'HGNC:4432'},\n",
       " {'subsystem': 'Methionine and cysteine metabolism',\n",
       "  'name': '3-sulfino-alanine transaminase (irreversible), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['MM. activity measured in rat, 3-sulfinoalanine is substrate found in human brain'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4433']},\n",
       "  'metabolites': {'glu_L_m': 1.0,\n",
       "   '3snpyr_m': 1.0,\n",
       "   'h_m': -1.0,\n",
       "   'akg_m': -1.0,\n",
       "   '3sala_m': -1.0},\n",
       "  'id': '3SALATAim',\n",
       "  'gene_reaction_rule': 'HGNC:4433'},\n",
       " {'subsystem': 'Methionine and cysteine metabolism',\n",
       "  'name': '3-sulfinopyruvate hydrolase (spotaneous reaction)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['MM'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'pyr_c': 1.0,\n",
       "   'so3_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   '3snpyr_c': -1.0},\n",
       "  'id': '3SPYRSP',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Methionine and cysteine metabolism',\n",
       "  'name': '3-sulfinopyruvate hydrolase (spotaneous reaction), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['MM'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'3snpyr_m': -1.0,\n",
       "   'h2o_m': -1.0,\n",
       "   'h_m': 1.0,\n",
       "   'pyr_m': 1.0,\n",
       "   'so3_m': 1.0},\n",
       "  'id': '3SPYRSPm',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Tyrosine metabolism',\n",
       "  'name': '4-[(1R)-1-Hydroxy-2-(methylamino)ethyl]-1,2-benzenediol:oxygen oxidoreductase(deaminating)(flavin-containing)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:6833 or HGNC:6834']},\n",
       "  'metabolites': {'mma_c': 1.0,\n",
       "   'o2_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   '34dhmald_c': 1.0,\n",
       "   'h2o2_c': 1.0,\n",
       "   'adrnl_c': -1.0},\n",
       "  'id': '41R1H2MAE12BOOX',\n",
       "  'gene_reaction_rule': 'HGNC:6833 or HGNC:6834'},\n",
       " {'subsystem': 'Tyrosine metabolism',\n",
       "  'name': '4-[(1R)-2-Amino-1-hydroxyethyl]-1,2-benzenediol:oxygen oxidoreductase(deaminating)(flavin-containing)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:6833 or HGNC:6834']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   '34dhmald_c': 1.0,\n",
       "   'nrpphr_c': -1.0,\n",
       "   'h2o2_c': 1.0,\n",
       "   'nh4_c': 1.0},\n",
       "  'id': '41R2A1H12BOOX',\n",
       "  'gene_reaction_rule': 'HGNC:6833 or HGNC:6834'},\n",
       " {'subsystem': 'Tyrosine metabolism',\n",
       "  'name': '4-(2-Aminoethyl)-1,2-benzenediol:oxygen oxidoreductase(deaminating)(flavin-containing)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:549 or HGNC:550 or HGNC:6833 or HGNC:6834 or HGNC:80']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'dopa_c': -1.0,\n",
       "   '34dhpac_c': 1.0,\n",
       "   'h2o2_c': 1.0,\n",
       "   'nh4_c': 1.0},\n",
       "  'id': '42A12BOOX',\n",
       "  'gene_reaction_rule': 'HGNC:549 or HGNC:550 or HGNC:6833 or HGNC:6834 or HGNC:80'},\n",
       " {'subsystem': 'Ubiquinone synthesis',\n",
       "  'name': '4-hydroxybenzoyl-CoA formation',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['IT reaction involved in formation of 4-hydroxybenzoate from tyrosine.'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'coucoa_m': -1.0,\n",
       "   'h2o_m': -1.0,\n",
       "   'accoa_m': 1.0,\n",
       "   'nadh_m': 1.0,\n",
       "   '4hbzcoa_m': 1.0,\n",
       "   'nad_m': -1.0,\n",
       "   'coa_m': -1.0,\n",
       "   'h_m': 1.0},\n",
       "  'id': '4HBZCOAFm',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Ubiquinone synthesis',\n",
       "  'name': '4-hydroxybenzoate formation',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['IT reaction involved in formation of 4-hydroxybenzoate from tyrosine.'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'coa_m': 1.0,\n",
       "   'h_m': 1.0,\n",
       "   'h2o_m': -1.0,\n",
       "   '4hbzcoa_m': -1.0,\n",
       "   '4hbz_m': 1.0},\n",
       "  'id': '4HBZFm',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Arginine and Proline Metabolism',\n",
       "  'name': 'L-4-hydroxyglutamate semialdehyde dehydrogenase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:406']},\n",
       "  'metabolites': {'4hglusa_m': 1.0,\n",
       "   'h2o_m': 1.0,\n",
       "   'e4hglu_m': -1.0,\n",
       "   'nadh_m': -1.0,\n",
       "   'nad_m': 1.0,\n",
       "   'h_m': -2.0},\n",
       "  'id': '4HGLSDm',\n",
       "  'gene_reaction_rule': 'HGNC:406'},\n",
       " {'subsystem': 'Tyrosine metabolism',\n",
       "  'name': '4-Hydroxyphenylacetaldehyde:NADP+ oxidoreductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:405 or HGNC:409 or HGNC:410 or HGNC:411']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   'nadp_c': -1.0,\n",
       "   'nadph_c': 1.0,\n",
       "   'h_c': 2.0,\n",
       "   '4hoxpacd_c': -1.0,\n",
       "   '4hphac_c': 1.0},\n",
       "  'id': '4HOXPACDOX_NADP_',\n",
       "  'gene_reaction_rule': 'HGNC:405 or HGNC:409 or HGNC:410 or HGNC:411'},\n",
       " {'subsystem': 'Cytochrome metabolism',\n",
       "  'name': '4-Nitrophenol Sulfotransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['RS/TV (6/14/05)'],\n",
       "   'GENE ASSOCIATION': ['HGNC:11453 or HGNC:11454 or HGNC:11455']},\n",
       "  'metabolites': {'pap_c': 1.0,\n",
       "   'paps_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   '4nph_c': -1.0,\n",
       "   '4nphsf_c': 1.0},\n",
       "  'id': '4NPHSULT',\n",
       "  'gene_reaction_rule': 'HGNC:11453 or HGNC:11454 or HGNC:11455'},\n",
       " {'subsystem': 'Steroid metabolism',\n",
       "  'name': '5alpha-Dihydrotestosterone sulfotransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:11455 or HGNC:11458']},\n",
       "  'metabolites': {'pap_c': 1.0,\n",
       "   '5adtststerones_c': 1.0,\n",
       "   '5adtststerone_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   'paps_c': -1.0},\n",
       "  'id': '5ADTSTSTERONESULT',\n",
       "  'gene_reaction_rule': 'HGNC:11455 or HGNC:11458'},\n",
       " {'subsystem': 'Tryptophan metabolism',\n",
       "  'name': '5-Hydroxy-L-tryptophan decarboxy-lyase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2719']},\n",
       "  'metabolites': {'srtn_c': 1.0, 'co2_c': 1.0, 'h_c': -1.0, '5htrp_c': -1.0},\n",
       "  'id': '5HLTDL',\n",
       "  'gene_reaction_rule': 'HGNC:2719'},\n",
       " {'subsystem': 'Tryptophan metabolism',\n",
       "  'name': '5-Hydroxyindoleacetaldehyde:oxygen oxidoreductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['SAB questionable--balanced, but h2o2 production is based on an obscure comment in entrez'],\n",
       "   'GENE ASSOCIATION': ['HGNC:553']},\n",
       "  'metabolites': {'5hoxindoa_c': 1.0,\n",
       "   'o2_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   '5hoxindact_c': -1.0,\n",
       "   'h2o2_c': 1.0,\n",
       "   'h_c': 1.0},\n",
       "  'id': '5HOXINDACTO2OX',\n",
       "  'gene_reaction_rule': 'HGNC:553'},\n",
       " {'subsystem': 'Tryptophan metabolism',\n",
       "  'name': '5-Hydroxyindoleacetaldehyde:NAD+ oxidoreductase (mito)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:404 or HGNC:406 or HGNC:407']},\n",
       "  'metabolites': {'5hoxindact_m': -1.0,\n",
       "   'h2o_m': -1.0,\n",
       "   '5hoxindoa_m': 1.0,\n",
       "   'nadh_m': 1.0,\n",
       "   'nad_m': -1.0,\n",
       "   'h_m': 2.0},\n",
       "  'id': '5HOXINDACTOXm',\n",
       "  'gene_reaction_rule': 'HGNC:404 or HGNC:406 or HGNC:407'},\n",
       " {'subsystem': 'Tryptophan metabolism',\n",
       "  'name': '5-Hydroxytryptamine:oxygen oxidoreductase(deaminating)(flavin-containing)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:6833 or HGNC:6834']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'srtn_c': -1.0,\n",
       "   '5hoxindact_c': 1.0,\n",
       "   'h2o2_c': 1.0,\n",
       "   'nh4_c': 1.0},\n",
       "  'id': '5HOXINOXDA',\n",
       "  'gene_reaction_rule': 'HGNC:6833 or HGNC:6834'},\n",
       " {'subsystem': 'Tryptophan metabolism',\n",
       "  'name': '5-Hydroxy-L-tryptophan:oxygen 2,3-dioxygenase (indole-decyclizing)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['SAB. The E.C. number 1.13.11.42 has been deleted in 2006. Instead, replaced with E.C. 1.13.11.11 - tryptophan 2,3-dioxygenase. However, E.C. 1.13.11.11 is for the reaction L-tryptophan + O2 = N-formyl-L-kynurenine. For the reaction 5-Hydroxy-L-tryptophan (Cytosol) + O2 (Cytosol) &lt;=&gt; 5-Hydroxy-N-formylkynurenine, E.C. should be 1.13.11.52. Conclusion: Change E.C. to 1.13.11.52 (source: KEGG).'],\n",
       "   'GENE ASSOCIATION': ['HGNC:6059']},\n",
       "  'metabolites': {'5hoxnfkyn_c': 1.0, '5htrp_c': -1.0, 'o2_c': -1.0},\n",
       "  'id': '5HTRPDOX',\n",
       "  'gene_reaction_rule': 'HGNC:6059'},\n",
       " {'subsystem': 'Tryptophan metabolism',\n",
       "  'name': '5-Hydroxykynurenamine decarboxy-lyase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2719']},\n",
       "  'metabolites': {'5hxkynam_c': 1.0,\n",
       "   'co2_c': 1.0,\n",
       "   'h_c': -1.0,\n",
       "   '5hxkyn_c': -1.0},\n",
       "  'id': '5HXKYNDCL',\n",
       "  'gene_reaction_rule': 'HGNC:2719'},\n",
       " {'subsystem': 'Tryptophan metabolism',\n",
       "  'name': '5-Hydroxykynurenamine:oxygen oxidoreductase(deaminating)(flavin-containing)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:6833 or HGNC:6834']},\n",
       "  'metabolites': {'h2o2_c': 1.0,\n",
       "   '46dhoxquin_c': 1.0,\n",
       "   '5hxkynam_c': -1.0,\n",
       "   'nh4_c': 1.0,\n",
       "   'o2_c': -1.0},\n",
       "  'id': '5HXKYNOXDA',\n",
       "  'gene_reaction_rule': 'HGNC:6833 or HGNC:6834'},\n",
       " {'subsystem': 'N-glycan degradation',\n",
       "  'name': 'alpha-mannosidase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:6827']},\n",
       "  'metabolites': {'m2mn_c': -1.0, 'h2o_c': -2.0, 'mn_c': 1.0, 'man_c': 2.0},\n",
       "  'id': 'A_MANASE',\n",
       "  'gene_reaction_rule': 'HGNC:6827'},\n",
       " {'subsystem': 'N-glycan degradation',\n",
       "  'name': 'alpha-mannosidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:6826']},\n",
       "  'metabolites': {'man_l': 2.0, 'mn_l': 1.0, 'h2o_l': -2.0, 'm2mn_l': -1.0},\n",
       "  'id': 'A_MANASEly',\n",
       "  'gene_reaction_rule': 'HGNC:6826'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'Lactosylceramide 4-alpha-galactosyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['EC: 2.4.1.228 NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:18149']},\n",
       "  'metabolites': {'thcrm_hs_c': 1.0,\n",
       "   'udpgal_c': -1.0,\n",
       "   'udp_c': 1.0,\n",
       "   'galgluside_hs_c': -1.0,\n",
       "   'h_c': 1.0},\n",
       "  'id': 'A4GALTc',\n",
       "  'gene_reaction_rule': 'HGNC:18149'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'Lactosylceramide 4-alpha-galactosyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['EC: 2.4.1.228 NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:18149']},\n",
       "  'metabolites': {'galgluside_hs_g': -1.0,\n",
       "   'h_g': 1.0,\n",
       "   'thcrm_hs_g': 1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0},\n",
       "  'id': 'A4GALTg',\n",
       "  'gene_reaction_rule': 'HGNC:18149'},\n",
       " {'subsystem': 'O-glycan synthesis',\n",
       "  'name': 'alpha 1,4-N-acetylglucosaminyltransferase, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['EC 2.4.1.- NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:17968']},\n",
       "  'metabolites': {'gncore1_g': 1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'T_antigen_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'A4GNT1g',\n",
       "  'gene_reaction_rule': 'HGNC:17968'},\n",
       " {'subsystem': 'O-glycan synthesis',\n",
       "  'name': 'alpha 1,4-N-acetylglucosaminyltransferase, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['EC 2.4.1.- NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:17968']},\n",
       "  'metabolites': {'core2_g': -1.0,\n",
       "   'h_g': 1.0,\n",
       "   'gncore2_g': 1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'udp_g': 1.0},\n",
       "  'id': 'A4GNT2g',\n",
       "  'gene_reaction_rule': 'HGNC:17968'},\n",
       " {'subsystem': 'Propanoate metabolism',\n",
       "  'name': 'Acetoacetyl-CoA:acetate CoA-transferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['also EC 2.8.3.8 NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:21298']},\n",
       "  'metabolites': {'atp_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'aacoa_c': 1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0,\n",
       "   'acac_c': -1.0},\n",
       "  'id': 'AACOAT',\n",
       "  'gene_reaction_rule': 'HGNC:21298'},\n",
       " {'subsystem': 'Glycine, serine, alanine and threonine metabolism',\n",
       "  'name': 'Aminoacetone:oxygen oxidoreductase(deaminating)(flavin-containing)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:550']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'aact_c': -1.0,\n",
       "   'h2o2_c': 1.0,\n",
       "   'mthgxl_c': 1.0,\n",
       "   'nh4_c': 1.0},\n",
       "  'id': 'AACTOOR',\n",
       "  'gene_reaction_rule': 'HGNC:550'},\n",
       " {'subsystem': 'Lysine metabolism',\n",
       "  'name': 'L-aminoadipate-semialdehyde dehydrogenase (NADH), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['MM. Found under reaction synonym name: 2-aminoadipic 6-semialdehyde dehydrogenase; Evidence was from mouse with identification of good human ortholog for U26 in the mouse that catalyzes this reaction.  There was NOT evidence here specific to mitochondria, though.  (Nathan Price)'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'h2o_m': -1.0,\n",
       "   'L2aadp6sa_m': -1.0,\n",
       "   'nadh_m': 1.0,\n",
       "   'nad_m': -1.0,\n",
       "   'h_m': 2.0,\n",
       "   'L2aadp_m': 1.0},\n",
       "  'id': 'AASAD3m',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Lysine metabolism',\n",
       "  'name': '2-aminoadipate transaminase, irreversible',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['MM'],\n",
       "   'GENE ASSOCIATION': ['HGNC:17929']},\n",
       "  'metabolites': {'akg_c': -1.0,\n",
       "   'L2aadp_c': -1.0,\n",
       "   '2oxoadp_c': 1.0,\n",
       "   'glu_L_c': 1.0},\n",
       "  'id': 'AATAi',\n",
       "  'gene_reaction_rule': 'HGNC:17929'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'ABO blood group (transferase A, alpha 1-3-N-acetylgalactosaminyltransferase; transferase B, alpha 1-3-galactosyltransferase)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:79']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'acgalfucgalacglcgalgluside_hs_g': 1.0,\n",
       "   'udpacgal_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'fucgalacglcgalgluside_hs_g': -1.0},\n",
       "  'id': 'ABO1g',\n",
       "  'gene_reaction_rule': 'HGNC:79'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'ABO blood group (transferase A, alpha 1-3-N-acetylgalactosaminyltransferase; transferase B, alpha 1-3-galactosyltransferase)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:79']},\n",
       "  'metabolites': {'galfucgalacglcgalgluside_hs_g': 1.0,\n",
       "   'h_g': 1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'fucgalacglcgalgluside_hs_g': -1.0},\n",
       "  'id': 'ABO2g',\n",
       "  'gene_reaction_rule': 'HGNC:79'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'ABO blood group (transferase A, alpha 1-3-N-acetylgalactosaminyltransferase; transferase B, alpha 1-3-galactosyltransferase)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:79']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'fuc12gal14acglcgalgluside_hs_g': -1.0,\n",
       "   'galfuc12gal14acglcgalgluside_hs_g': 1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0},\n",
       "  'id': 'ABO3g',\n",
       "  'gene_reaction_rule': 'HGNC:79'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'ABO blood group (transferase A, alpha 1-3-N-acetylgalactosaminyltransferase; transferase B, alpha 1-3-galactosyltransferase)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:79']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'acgalfucgalacglcgalgluside_hs_g': 1.0,\n",
       "   'udpacgal_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'fuc12gal14acglcgalgluside_hs_g': -1.0},\n",
       "  'id': 'ABO4g',\n",
       "  'gene_reaction_rule': 'HGNC:79'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'ABO blood group (transferase A, alpha 1-3-N-acetylgalactosaminyltransferase; transferase B, alpha 1-3-galactosyltransferase)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:79']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'udpacgal_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'fucgalacgalfuc12gal14acglcgalgluside_hs_g': -1.0,\n",
       "   'acgalfucgalacgalfuc12gal14acglcgalgluside_hs_g': 1.0},\n",
       "  'id': 'ABO5g',\n",
       "  'gene_reaction_rule': 'HGNC:79'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'ABO blood group (transferase A, alpha 1-3-N-acetylgalactosaminyltransferase; transferase B, alpha 1-3-galactosyltransferase)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:79']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'fucgalacglcgal14acglcgalgluside_hs_g': -1.0,\n",
       "   'udpacgal_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'acgalfucgalacglcgal14acglcgalgluside_hs_g': 1.0},\n",
       "  'id': 'ABO6g',\n",
       "  'gene_reaction_rule': 'HGNC:79'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'ABO blood group (transferase A, alpha 1-3-N-acetylgalactosaminyltransferase; transferase B, alpha 1-3-galactosyltransferase)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:79']},\n",
       "  'metabolites': {'acgalfucgalacgalfucgalacglcgal14acglcgalgluside_hs_g': 1.0,\n",
       "   'h_g': 1.0,\n",
       "   'udpacgal_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'fucgalacgalfucgalacglcgal14acglcgalgluside_hs_g': -1.0},\n",
       "  'id': 'ABO7g',\n",
       "  'gene_reaction_rule': 'HGNC:79'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'ABO blood group (transferase A, alpha 1-3-N-acetylgalactosaminyltransferase; transferase B, alpha 1-3-galactosyltransferase)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:79']},\n",
       "  'metabolites': {'galfucgalacglcgal14acglcgalgluside_hs_g': 1.0,\n",
       "   'fucgalacglcgal14acglcgalgluside_hs_g': -1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'ABO8g',\n",
       "  'gene_reaction_rule': 'HGNC:79'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'ABO blood group (transferase A, alpha 1-3-N-acetylgalactosaminyltransferase; transferase B, alpha 1-3-galactosyltransferase)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:79']},\n",
       "  'metabolites': {'h_g': 2.0,\n",
       "   'fucfucgalacglcgalacglcgal14acglcgalgluside_hs_g': -1.0,\n",
       "   'udpgal_g': -2.0,\n",
       "   'udp_g': 2.0,\n",
       "   'galgalfucfucgalacglcgalacglcgal14acglcgalgluside_hs_g': 1.0},\n",
       "  'id': 'ABO9g',\n",
       "  'gene_reaction_rule': 'HGNC:79'},\n",
       " {'subsystem': 'Glutamate metabolism',\n",
       "  'name': '4-aminobutyrate transaminase, reversible (mitochondrial)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['kh 1/6/03'],\n",
       "   'GENE ASSOCIATION': ['HGNC:23']},\n",
       "  'metabolites': {'glu_L_m': 1.0,\n",
       "   '4abut_m': -1.0,\n",
       "   'sucsal_m': 1.0,\n",
       "   'akg_m': -1.0},\n",
       "  'id': 'ABTArm',\n",
       "  'gene_reaction_rule': 'HGNC:23'},\n",
       " {'subsystem': 'Pentose phosphate pathway',\n",
       "  'name': 'L-arabinitol 4-dehydrogenase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'abt_c': -1.0,\n",
       "   'nadh_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'nad_c': -1.0,\n",
       "   'xylu_L_c': 1.0},\n",
       "  'id': 'ABTD',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'beta-Alanine metabolism',\n",
       "  'name': 'Aminobutyraldehyde dehydrogenase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['JLR'],\n",
       "   'GENE ASSOCIATION': ['HGNC:412']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   '4abutn_c': -1.0,\n",
       "   '4abut_c': 1.0,\n",
       "   'h_c': 2.0,\n",
       "   'nad_c': -1.0,\n",
       "   'nadh_c': 1.0},\n",
       "  'id': 'ABUTD',\n",
       "  'gene_reaction_rule': 'HGNC:412'},\n",
       " {'subsystem': 'Valine, leucine, and isoleucine metabolism',\n",
       "  'name': 'acetyl-CoA C-acetyltransferase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['MM. 2-Methyl-3-acetoacetyl-CoA excreted into urine for patients with disease deficiency'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4803 or HGNC:83 or HGNC:93']},\n",
       "  'metabolites': {'accoa_m': 1.0,\n",
       "   'ppcoa_m': 1.0,\n",
       "   'coa_m': -1.0,\n",
       "   '2maacoa_m': -1.0},\n",
       "  'id': 'ACACT10m',\n",
       "  'gene_reaction_rule': 'HGNC:4803 or HGNC:83 or HGNC:93'},\n",
       " {'subsystem': 'Tryptophan metabolism',\n",
       "  'name': 'acetyl-CoA C-acetyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['JLR- reversible form of ACACT1'],\n",
       "   'GENE ASSOCIATION': ['HGNC:94']},\n",
       "  'metabolites': {'aacoa_c': 1.0, 'coa_c': 1.0, 'accoa_c': -2.0},\n",
       "  'id': 'ACACT1r',\n",
       "  'gene_reaction_rule': 'HGNC:94'},\n",
       " {'subsystem': 'Tryptophan metabolism',\n",
       "  'name': 'acetyl-CoA C-acetyltransferase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['TDV'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:4801 and HGNC:4803) or HGNC:93']},\n",
       "  'metabolites': {'accoa_m': -2.0, 'coa_m': 1.0, 'aacoa_m': 1.0},\n",
       "  'id': 'ACACT1rm',\n",
       "  'gene_reaction_rule': '(HGNC:4801 and HGNC:4803) or HGNC:93'},\n",
       " {'subsystem': 'Cholesterol metabolism',\n",
       "  'name': 'acetyl-CoA C-acetyltransferase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD NJ - added peroxisomal version'],\n",
       "   'GENE ASSOCIATION': ['HGNC:94']},\n",
       "  'metabolites': {'aacoa_x': 1.0, 'accoa_x': -2.0, 'coa_x': 1.0},\n",
       "  'id': 'ACACT1x',\n",
       "  'gene_reaction_rule': 'HGNC:94'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'acetyl-CoA C-acetyltransferase (octanoyl-CoA), peroxisomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:82']},\n",
       "  'metabolites': {'3odcoa_x': -1.0,\n",
       "   'occoa_x': 1.0,\n",
       "   'accoa_x': 1.0,\n",
       "   'coa_x': -1.0},\n",
       "  'id': 'ACACT4p',\n",
       "  'gene_reaction_rule': 'HGNC:82'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'acetyl-CoA C-acyltransferase (decanoyl-CoA), peroxisomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:82']},\n",
       "  'metabolites': {'3oddcoa_x': -1.0,\n",
       "   'accoa_x': 1.0,\n",
       "   'dcacoa_x': 1.0,\n",
       "   'coa_x': -1.0},\n",
       "  'id': 'ACACT5p',\n",
       "  'gene_reaction_rule': 'HGNC:82'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'acetyl-CoA C-acetyltransferase (dodecanoyl), peroxisomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:82']},\n",
       "  'metabolites': {'3otdcoa_x': -1.0,\n",
       "   'ddcacoa_x': 1.0,\n",
       "   'accoa_x': 1.0,\n",
       "   'coa_x': -1.0},\n",
       "  'id': 'ACACT6p',\n",
       "  'gene_reaction_rule': 'HGNC:82'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'acetyl-CoA acyltransferase (tetradecanoyl-CoA), peroxisomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:82']},\n",
       "  'metabolites': {'accoa_x': 1.0,\n",
       "   'coa_x': -1.0,\n",
       "   'tdcoa_x': 1.0,\n",
       "   '3ohdcoa_x': -1.0},\n",
       "  'id': 'ACACT7p',\n",
       "  'gene_reaction_rule': 'HGNC:82'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'acetyl-CoA acyltransferase (hexadecanoyl-CoA), peroxisomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:82']},\n",
       "  'metabolites': {'3ohodcoa_x': -1.0,\n",
       "   'pmtcoa_x': 1.0,\n",
       "   'accoa_x': 1.0,\n",
       "   'coa_x': -1.0},\n",
       "  'id': 'ACACT8p',\n",
       "  'gene_reaction_rule': 'HGNC:82'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'acetyl-CoA acyltransferase (tetracosanoyl-CoA), peroxisomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:82']},\n",
       "  'metabolites': {'3ohxccoa_x': -1.0,\n",
       "   'accoa_x': 1.0,\n",
       "   'coa_x': -1.0,\n",
       "   'ttccoa_x': 1.0},\n",
       "  'id': 'ACACT9p',\n",
       "  'gene_reaction_rule': 'HGNC:82'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'acetyl-CoA carboxylase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'GENE ASSOCIATION': ['HGNC:84 or HGNC:84 or HGNC:84 and HGNC:85 or HGNC:84 and HGNC:85 or HGNC:84 or HGNC:85 and HGNC:84 or HGNC:84 and HGNC:85 or HGNC:84 or HGNC:84 and HGNC:85 or HGNC:84']},\n",
       "  'metabolites': {'malcoa_c': 1.0,\n",
       "   'accoa_c': -1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'hco3_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   'adp_c': 1.0,\n",
       "   'pi_c': 1.0},\n",
       "  'id': 'ACCOAC',\n",
       "  'gene_reaction_rule': 'HGNC:84 or HGNC:84 or HGNC:84 and HGNC:85 or HGNC:84 and HGNC:85 or HGNC:84 or HGNC:85 and HGNC:84 or HGNC:84 and HGNC:85 or HGNC:84 or HGNC:84 and HGNC:85 or HGNC:84'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'Acetyl-Coa carboxylase, beta isoform',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['TDV (9/14/2004)'],\n",
       "   'GENE ASSOCIATION': ['HGNC:85']},\n",
       "  'metabolites': {'hco3_m': -1.0,\n",
       "   'atp_m': -1.0,\n",
       "   'malcoa_m': 1.0,\n",
       "   'adp_m': 1.0,\n",
       "   'pi_m': 1.0,\n",
       "   'accoa_m': -1.0,\n",
       "   'h_m': 1.0},\n",
       "  'id': 'ACCOACm',\n",
       "  'gene_reaction_rule': 'HGNC:85'},\n",
       " {'subsystem': 'Propanoate metabolism',\n",
       "  'name': 'acetate-CoA ligase (ADP-forming)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['JLR- also acts on propanoate'],\n",
       "   'GENE ASSOCIATION': ['HGNC:15814']},\n",
       "  'metabolites': {'adp_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'ppa_c': -1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'ppcoa_c': 1.0,\n",
       "   'pi_c': 1.0},\n",
       "  'id': 'ACCOAL',\n",
       "  'gene_reaction_rule': 'HGNC:15814'},\n",
       " {'subsystem': 'Propanoate metabolism',\n",
       "  'name': 'acetate-CoA ligase (AMP-forming)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['TDV-mitochondrial version JLR- also acts on propanoate AMF- made reaction to create AMP not ADP and changed EC'],\n",
       "   'GENE ASSOCIATION': ['HGNC:16091 or HGNC:24723']},\n",
       "  'metabolites': {'ppi_m': 1.0,\n",
       "   'atp_m': -1.0,\n",
       "   'ppcoa_m': 1.0,\n",
       "   'coa_m': -1.0,\n",
       "   'amp_m': 1.0,\n",
       "   'ppa_m': -1.0},\n",
       "  'id': 'ACCOALm',\n",
       "  'gene_reaction_rule': 'HGNC:16091 or HGNC:24723'},\n",
       " {'subsystem': 'Aminosugar metabolism',\n",
       "  'name': 'N-acetylgalactosamine kinase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'acgal1p_c': 1.0,\n",
       "   'acgal_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'adp_c': 1.0},\n",
       "  'id': 'ACGALK',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Aminosugar metabolism',\n",
       "  'name': 'N-acetylgalactosamine kinase (ITP)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'acgal1p_c': 1.0,\n",
       "   'acgal_c': -1.0,\n",
       "   'itp_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   'idp_c': 1.0},\n",
       "  'id': 'ACGALK2',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Aminosugar metabolism',\n",
       "  'name': 'N-acetylglucosamine 2-epimerase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:9959']},\n",
       "  'metabolites': {'acgam_c': -1.0, 'acmana_c': 1.0},\n",
       "  'id': 'ACGAM2E',\n",
       "  'gene_reaction_rule': 'HGNC:9959'},\n",
       " {'subsystem': 'Aminosugar metabolism',\n",
       "  'name': 'N-acetylglucosamine-6-phosphate synthase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:19980']},\n",
       "  'metabolites': {'coa_c': 1.0,\n",
       "   'gam6p_c': -1.0,\n",
       "   'acgam6p_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'accoa_c': -1.0},\n",
       "  'id': 'ACGAM6PSi',\n",
       "  'gene_reaction_rule': 'HGNC:19980'},\n",
       " {'subsystem': 'Aminosugar metabolism',\n",
       "  'name': 'N-acetylglucosamine kinase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:17174']},\n",
       "  'metabolites': {'acgam_c': -1.0,\n",
       "   'acgam6p_c': 1.0,\n",
       "   'adp_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'h_c': 1.0},\n",
       "  'id': 'ACGAMK',\n",
       "  'gene_reaction_rule': 'HGNC:17174'},\n",
       " {'subsystem': 'Aminosugar metabolism',\n",
       "  'name': 'phosphoacetylglucosamine mutase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:8907']},\n",
       "  'metabolites': {'acgam6p_c': -1.0, 'acgam1p_c': 1.0},\n",
       "  'id': 'ACGAMPM',\n",
       "  'gene_reaction_rule': 'HGNC:8907'},\n",
       " {'subsystem': 'Phosphatidylinositol phosphate metabolism',\n",
       "  'name': 'N-acetylglucosaminylphosphatidylinositol deacetylase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:8966']},\n",
       "  'metabolites': {'ac_c': 1.0,\n",
       "   'gpail_hs_c': 1.0,\n",
       "   'acgpail_hs_c': -1.0,\n",
       "   'h2o_c': -1.0},\n",
       "  'id': 'ACGPID',\n",
       "  'gene_reaction_rule': 'HGNC:8966'},\n",
       " {'subsystem': 'Urea cycle',\n",
       "  'name': 'N-acteylglutamate synthase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:17996']},\n",
       "  'metabolites': {'glu_L_m': -1.0,\n",
       "   'accoa_m': -1.0,\n",
       "   'coa_m': 1.0,\n",
       "   'h_m': 1.0,\n",
       "   'acglu_m': 1.0},\n",
       "  'id': 'ACGSm',\n",
       "  'gene_reaction_rule': 'HGNC:17996'},\n",
       " {'subsystem': 'Glycerophospholipid metabolism',\n",
       "  'name': 'Acetylcholinesterase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:108']},\n",
       "  'metabolites': {'h2o_e': -1.0,\n",
       "   'h_e': 1.0,\n",
       "   'ac_e': 1.0,\n",
       "   'ach_e': -1.0,\n",
       "   'chol_e': 1.0},\n",
       "  'id': 'ACHEe',\n",
       "  'gene_reaction_rule': 'HGNC:108'},\n",
       " {'subsystem': 'Citric acid cycle',\n",
       "  'name': 'ATP-Citrate lyase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['ATP dependent citrate lyase.  Reaction is reversible in KEGG (new EC 2.3.3.8) NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:115']},\n",
       "  'metabolites': {'adp_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'cit_c': -1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'oaa_c': 1.0,\n",
       "   'pi_c': 1.0,\n",
       "   'accoa_c': 1.0},\n",
       "  'id': 'ACITL',\n",
       "  'gene_reaction_rule': 'HGNC:115'},\n",
       " {'subsystem': 'Aminosugar metabolism',\n",
       "  'name': 'N-Acetylneuraminate 9-phosphate pyruvate-lyase (pyruvate-phosphorylating)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['EC-2.5.1.56 NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:19237']},\n",
       "  'metabolites': {'pep_c': -1.0,\n",
       "   'acnamp_c': 1.0,\n",
       "   'pi_c': 1.0,\n",
       "   'acmanap_c': -1.0,\n",
       "   'h2o_c': -1.0},\n",
       "  'id': 'ACNAM9PL',\n",
       "  'gene_reaction_rule': 'HGNC:19237'},\n",
       " {'subsystem': 'Aminosugar metabolism',\n",
       "  'name': 'N-Acetylneuraminate 9-phosphate pyruvate-lyase (pyruvate-phosphorylating)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['EC-2.5.1.56 NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:19237']},\n",
       "  'metabolites': {'pep_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'man6p_c': -1.0,\n",
       "   'pi_c': 1.0,\n",
       "   'kdnp_c': 1.0},\n",
       "  'id': 'ACNAM9PL2',\n",
       "  'gene_reaction_rule': 'HGNC:19237'},\n",
       " {'subsystem': 'Aminosugar metabolism',\n",
       "  'name': 'N-Acetylneuraminate 9-phosphate phosphohydrolase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:124 or HGNC:16140 or HGNC:29609']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   'pi_c': 1.0,\n",
       "   'acnamp_c': -1.0,\n",
       "   'acnam_c': 1.0},\n",
       "  'id': 'ACNAMPH',\n",
       "  'gene_reaction_rule': 'HGNC:124 or HGNC:16140 or HGNC:29609'},\n",
       " {'subsystem': 'Aminosugar metabolism',\n",
       "  'name': 'N-Acetylneuraminate lyase (reversible)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['JLR'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'pyr_c': 1.0, 'acmana_c': 1.0, 'acnam_c': -1.0},\n",
       "  'id': 'ACNMLr',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Valine, leucine, and isoleucine metabolism',\n",
       "  'name': 'acyl-CoA dehydrogenase (2-methylbutanoyl-CoA), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['MM'],\n",
       "   'GENE ASSOCIATION': ['HGNC:89 or HGNC:91']},\n",
       "  'metabolites': {'fad_m': -1.0,\n",
       "   'fadh2_m': 1.0,\n",
       "   '2mbcoa_m': -1.0,\n",
       "   '2mb2coa_m': 1.0},\n",
       "  'id': 'ACOAD10m',\n",
       "  'gene_reaction_rule': 'HGNC:89 or HGNC:91'},\n",
       " {'subsystem': 'Butanoate metabolism',\n",
       "  'name': 'acyl-CoA dehydrogenase (butanoyl-CoA), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Assume NAD/NADH as acceptors Lumped long-chain acyl-CoA dehydrogenase (EC 1.3.99.13) This reaction is therefore equilivalent to trans-2-enoyl-CoA reductase (NAD) (EC 1.3.1.44)'],\n",
       "   'GENE ASSOCIATION': ['HGNC:21497 or HGNC:21597 or HGNC:30211 or HGNC:87 or HGNC:89 or HGNC:90 or HGNC:91']},\n",
       "  'metabolites': {'fad_m': -1.0,\n",
       "   'fadh2_m': 1.0,\n",
       "   'b2coa_m': 1.0,\n",
       "   'btcoa_m': -1.0},\n",
       "  'id': 'ACOAD1fm',\n",
       "  'gene_reaction_rule': 'HGNC:21497 or HGNC:21597 or HGNC:30211 or HGNC:87 or HGNC:89 or HGNC:90 or HGNC:91'},\n",
       " {'subsystem': 'Valine, leucine, and isoleucine metabolism',\n",
       "  'name': 'acyl-CoA dehydrogenase (isovaleryl-CoA), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['MM'],\n",
       "   'GENE ASSOCIATION': ['HGNC:6186']},\n",
       "  'metabolites': {'fad_m': -1.0,\n",
       "   'fadh2_m': 1.0,\n",
       "   '3mb2coa_m': 1.0,\n",
       "   'ivcoa_m': -1.0},\n",
       "  'id': 'ACOAD8m',\n",
       "  'gene_reaction_rule': 'HGNC:6186'},\n",
       " {'subsystem': 'Valine, leucine, and isoleucine metabolism',\n",
       "  'name': 'acyl-CoA dehydrogenase (isobutyryl-CoA), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['MM'],\n",
       "   'GENE ASSOCIATION': ['HGNC:87 or HGNC:89']},\n",
       "  'metabolites': {'fad_m': -1.0,\n",
       "   'ibcoa_m': -1.0,\n",
       "   'fadh2_m': 1.0,\n",
       "   '2mp2coa_m': 1.0},\n",
       "  'id': 'ACOAD9m',\n",
       "  'gene_reaction_rule': 'HGNC:87 or HGNC:89'},\n",
       " {'subsystem': 'Pyruvate metabolism',\n",
       "  'name': 'acetyl-CoA hydrolase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD. measured in rat brain; 78.7 % identical enzyme cloned from human liver cDNA'],\n",
       "   'GENE ASSOCIATION': ['HGNC:24436']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   'ac_c': 1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'accoa_c': -1.0},\n",
       "  'id': 'ACOAHi',\n",
       "  'gene_reaction_rule': 'HGNC:24436'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'acyl-CoA oxidase (hexadecanoyl-CoA), peroxisomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:119']},\n",
       "  'metabolites': {'hdcoa_x': 1.0,\n",
       "   'h2o2_x': 1.0,\n",
       "   'pmtcoa_x': -1.0,\n",
       "   'o2_x': -1.0},\n",
       "  'id': 'ACOAO7p',\n",
       "  'gene_reaction_rule': 'HGNC:119'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'Acetyl-CoA ACP transacylase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['tv rxn is now charge balanced; removed h - NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3594']},\n",
       "  'metabolites': {'acACP_c': 1.0,\n",
       "   'ACP_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'accoa_c': -1.0},\n",
       "  'id': 'ACOATA',\n",
       "  'gene_reaction_rule': 'HGNC:3594'},\n",
       " {'subsystem': 'Urea cycle',\n",
       "  'name': 'acetylornithine deacetylase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'], 'GENE ASSOCIATION': ['HGNC:177']},\n",
       "  'metabolites': {'h2o_c': -1.0, 'orn_c': 1.0, 'ac_c': 1.0, 'acorn_c': -1.0},\n",
       "  'id': 'ACODA',\n",
       "  'gene_reaction_rule': 'HGNC:177'},\n",
       " {'subsystem': 'Citric acid cycle',\n",
       "  'name': 'aconitase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['SMP. Biochemical'],\n",
       "   'GENE ASSOCIATION': ['HGNC:117 or HGNC:6115']},\n",
       "  'metabolites': {'icit_c': 1.0, 'cit_c': -1.0},\n",
       "  'id': 'ACONT',\n",
       "  'gene_reaction_rule': 'HGNC:117 or HGNC:6115'},\n",
       " {'subsystem': 'Citric acid cycle',\n",
       "  'name': 'Aconitate hydratase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['IF. Biochemical'],\n",
       "   'GENE ASSOCIATION': ['HGNC:117 or HGNC:118']},\n",
       "  'metabolites': {'cit_m': -1.0, 'icit_m': 1.0},\n",
       "  'id': 'ACONTm',\n",
       "  'gene_reaction_rule': 'HGNC:117 or HGNC:118'},\n",
       " {'subsystem': 'Bile acid synthesis',\n",
       "  'name': 'acyl-Coenzyme A oxidase 2, branched chain',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['noted in uniprot to have FAD as cofactor - further reaction specific lit evidence not found to support or contradict, see Baumgart et al PNAS (1996) 93: 13748-13753 analog as per KEGG added NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:120']},\n",
       "  'metabolites': {'dhcholoylcoa_x': -1.0,\n",
       "   'fadh2_x': -1.0,\n",
       "   'o2_x': -0.5,\n",
       "   'thcholoylcoa_x': 1.0,\n",
       "   'fad_x': 1.0},\n",
       "  'id': 'ACOX22x',\n",
       "  'gene_reaction_rule': 'HGNC:120'},\n",
       " {'subsystem': 'Bile acid synthesis',\n",
       "  'name': 'acyl-Coenzyme A oxidase 2, branched chain',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['noted in uniprot to have FAD as cofactor - further reaction specific lit evidence not found to support or contradict, see Baumgart et al PNAS (1996) 93: 13748-13753 NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:120']},\n",
       "  'metabolites': {'cholcoads_x': 1.0,\n",
       "   'fad_x': 1.0,\n",
       "   'fadh2_x': -1.0,\n",
       "   'h2o_x': 2.0,\n",
       "   'cholcoas_x': -1.0,\n",
       "   'o2_x': -1.0},\n",
       "  'id': 'ACOX2x',\n",
       "  'gene_reaction_rule': 'HGNC:120'},\n",
       " {'subsystem': 'Vitamin B2 metabolism',\n",
       "  'name': 'acid phosphatase (FMN)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['INVALID ECO TERM: IFH'],\n",
       "   'GENE ASSOCIATION': ['HGNC:122 or HGNC:124 or HGNC:125 or HGNC:29609']},\n",
       "  'metabolites': {'h2o_c': -1.0, 'pi_c': 1.0, 'fmn_c': -1.0, 'ribflv_c': 1.0},\n",
       "  'id': 'ACP1_FMN_',\n",
       "  'gene_reaction_rule': 'HGNC:122 or HGNC:124 or HGNC:125 or HGNC:29609'},\n",
       " {'subsystem': 'Glycolysis/gluconeogenesis',\n",
       "  'name': 'acetyl-CoA synthetase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'GENE ASSOCIATION': ['HGNC:15814 or HGNC:21298']},\n",
       "  'metabolites': {'atp_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0,\n",
       "   'accoa_c': 1.0,\n",
       "   'ac_c': -1.0},\n",
       "  'id': 'ACS',\n",
       "  'gene_reaction_rule': 'HGNC:15814 or HGNC:21298'},\n",
       " {'subsystem': 'Propanoate metabolism',\n",
       "  'name': 'acetyl-CoA synthase (propionate)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['AMF'],\n",
       "   'GENE ASSOCIATION': ['HGNC:15814 or HGNC:21298']},\n",
       "  'metabolites': {'atp_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'ppa_c': -1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0,\n",
       "   'ppcoa_c': 1.0},\n",
       "  'id': 'ACS2',\n",
       "  'gene_reaction_rule': 'HGNC:15814 or HGNC:21298'},\n",
       " {'subsystem': 'Glycolysis/gluconeogenesis',\n",
       "  'name': 'acetyl-CoA synthetase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['IF'],\n",
       "   'GENE ASSOCIATION': ['HGNC:16091']},\n",
       "  'metabolites': {'ppi_m': 1.0,\n",
       "   'atp_m': -1.0,\n",
       "   'ac_m': -1.0,\n",
       "   'accoa_m': 1.0,\n",
       "   'coa_m': -1.0,\n",
       "   'amp_m': 1.0},\n",
       "  'id': 'ACSm',\n",
       "  'gene_reaction_rule': 'HGNC:16091'},\n",
       " {'subsystem': 'Tryptophan metabolism',\n",
       "  'name': 'S-Adenosyl-L-methionine:N-acetylserotonin O-methyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['SAB. Genetic and Biochemical'],\n",
       "   'GENE ASSOCIATION': ['HGNC:750']},\n",
       "  'metabolites': {'5hoxindoa_c': -1.0,\n",
       "   'amet_c': -1.0,\n",
       "   '5moxact_c': 1.0,\n",
       "   'ahcys_c': 1.0},\n",
       "  'id': 'ACSOMT',\n",
       "  'gene_reaction_rule': 'HGNC:750'},\n",
       " {'subsystem': 'Tryptophan metabolism',\n",
       "  'name': 'S-Adenosyl-L-homocysteine:N-acetylserotonin O-methyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:750']},\n",
       "  'metabolites': {'amet_c': -1.0,\n",
       "   'ahcys_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'Nacsertn_c': -1.0,\n",
       "   'melatn_c': 1.0},\n",
       "  'id': 'ACSRTNMT',\n",
       "  'gene_reaction_rule': 'HGNC:750'},\n",
       " {'subsystem': 'Pyruvate metabolism',\n",
       "  'name': 'acetol monooxygenase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['stoichiometry described in Thornalley PJ. Gen Pharmacol. 1996 Jun;27(4):565-73. PMID: 8853285 NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2631']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   'h2o_c': 2.0,\n",
       "   'nadp_c': 1.0,\n",
       "   'nadph_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'mthgxl_c': 1.0,\n",
       "   'acetol_c': -1.0},\n",
       "  'id': 'ACTLMO',\n",
       "  'gene_reaction_rule': 'HGNC:2631'},\n",
       " {'subsystem': 'Pyruvate metabolism',\n",
       "  'name': 'acetone monooxygenase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['stoichiometry described in Thornalley PJ. Gen Pharmacol. 1996 Jun;27(4):565-73. PMID: 8853285 NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2631']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   'h2o_c': 1.0,\n",
       "   'nadp_c': 1.0,\n",
       "   'nadph_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'acetone_c': -1.0,\n",
       "   'acetol_c': 1.0},\n",
       "  'id': 'ACTNMO',\n",
       "  'gene_reaction_rule': 'HGNC:2631'},\n",
       " {'subsystem': 'Glycolysis/gluconeogenesis',\n",
       "  'name': 'acylphosphatase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['Reaction from KEGG: an acylphosphate + H2O -&gt; a carboxylate + phosphate'],\n",
       "   'GENE ASSOCIATION': ['HGNC:179 or HGNC:180 or HGNC:21385']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   'pi_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   '13dpg_c': -1.0,\n",
       "   '3pg_c': 1.0},\n",
       "  'id': 'ACYP',\n",
       "  'gene_reaction_rule': 'HGNC:179 or HGNC:180 or HGNC:21385'},\n",
       " {'subsystem': 'Purine catabolism',\n",
       "  'name': 'Adenosine deaminase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Added EC- JLR'],\n",
       "   'GENE ASSOCIATION': ['HGNC:186']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   'nh4_c': 1.0,\n",
       "   'adn_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'ins_c': 1.0},\n",
       "  'id': 'ADA',\n",
       "  'gene_reaction_rule': 'HGNC:186'},\n",
       " {'subsystem': 'Purine catabolism',\n",
       "  'name': 'Adenosine deaminase, extracellular',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['adenosine deaminase has been found on cell surface in human (Gonzales-Gronow et al, JBC, 2004,279,20993-20998). IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:186']},\n",
       "  'metabolites': {'ins_e': 1.0,\n",
       "   'adn_e': -1.0,\n",
       "   'h_e': -1.0,\n",
       "   'h2o_e': -1.0,\n",
       "   'nh4_e': 1.0},\n",
       "  'id': 'ADAe',\n",
       "  'gene_reaction_rule': 'HGNC:186'},\n",
       " {'subsystem': 'Propanoate metabolism',\n",
       "  'name': 'Acetoacetate decarboxylation (irreversible), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['TDV (Voet &amp; Voet pg 581) Non-enzymatic decarboxylation; spontaneous reaction added &quot;m&quot; suffix - NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'acetone_m': 1.0, 'h_m': -1.0, 'co2_m': 1.0, 'acac_m': -1.0},\n",
       "  'id': 'ADCim',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'adenylate kinase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'GENE ASSOCIATION': ['HGNC:20091 or HGNC:361 or HGNC:362 or HGNC:365']},\n",
       "  'metabolites': {'amp_c': -1.0, 'atp_c': -1.0, 'adp_c': 2.0},\n",
       "  'id': 'ADK1',\n",
       "  'gene_reaction_rule': 'HGNC:20091 or HGNC:361 or HGNC:362 or HGNC:365'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'adenylate kinase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:362 or HGNC:363']},\n",
       "  'metabolites': {'adp_m': 2.0, 'atp_m': -1.0, 'amp_m': -1.0},\n",
       "  'id': 'ADK1m',\n",
       "  'gene_reaction_rule': 'HGNC:362 or HGNC:363'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'adentylate kinase (GTP)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['JLR'],\n",
       "   'GENE ASSOCIATION': ['HGNC:365']},\n",
       "  'metabolites': {'gdp_c': 1.0, 'gtp_c': -1.0, 'amp_c': -1.0, 'adp_c': 1.0},\n",
       "  'id': 'ADK3',\n",
       "  'gene_reaction_rule': 'HGNC:365'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'adentylate kinase (GTP)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:17376']},\n",
       "  'metabolites': {'adp_m': 1.0, 'gdp_m': 1.0, 'amp_m': -1.0, 'gtp_m': -1.0},\n",
       "  'id': 'ADK3m',\n",
       "  'gene_reaction_rule': 'HGNC:17376'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'adenylate kinase (d form)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Inorganic triphosphate can also act as donor  This form was added for Staph based on KEGG evidence (SAB)'],\n",
       "   'GENE ASSOCIATION': ['HGNC:365']},\n",
       "  'metabolites': {'datp_c': -1.0, 'damp_c': -1.0, 'dadp_c': 2.0},\n",
       "  'id': 'ADKd',\n",
       "  'gene_reaction_rule': 'HGNC:365'},\n",
       " {'subsystem': 'Arginine and Proline Metabolism',\n",
       "  'name': 'adenosylmethionine decarboxylase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:457']},\n",
       "  'metabolites': {'amet_c': -1.0, 'co2_c': 1.0, 'h_c': -1.0, 'ametam_c': 1.0},\n",
       "  'id': 'ADMDC',\n",
       "  'gene_reaction_rule': 'HGNC:457'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'adenylate cyclase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['JLR'],\n",
       "   'GENE ASSOCIATION': ['HGNC:21285 or HGNC:232 or HGNC:233 or HGNC:234 or HGNC:235 or HGNC:236 or HGNC:237 or HGNC:238 or HGNC:239 or HGNC:240']},\n",
       "  'metabolites': {'ppi_c': 1.0, 'camp_c': 1.0, 'atp_c': -1.0},\n",
       "  'id': 'ADNCYC',\n",
       "  'gene_reaction_rule': 'HGNC:21285 or HGNC:232 or HGNC:233 or HGNC:234 or HGNC:235 or HGNC:236 or HGNC:237 or HGNC:238 or HGNC:239 or HGNC:240'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'adenosine kinase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:257']},\n",
       "  'metabolites': {'amp_c': 1.0,\n",
       "   'adn_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'adp_c': 1.0},\n",
       "  'id': 'ADNK1',\n",
       "  'gene_reaction_rule': 'HGNC:257'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'adenosine kinase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'adp_m': 1.0,\n",
       "   'atp_m': -1.0,\n",
       "   'adn_m': -1.0,\n",
       "   'amp_m': 1.0,\n",
       "   'h_m': 1.0},\n",
       "  'id': 'ADNK1m',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'ADPglucose diphosphatase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Based on Yang  et al, JBC, 2000, 275, 8844-8853'],\n",
       "   'GENE ASSOCIATION': ['HGNC:8052']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   'g1p_c': 1.0,\n",
       "   'h_c': 2.0,\n",
       "   'adpglc_c': -1.0,\n",
       "   'amp_c': 1.0},\n",
       "  'id': 'ADPGLC',\n",
       "  'gene_reaction_rule': 'HGNC:8052'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'ADPmannose diphosphatase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Based on Yang  et al, JBC, 2000, 275, 8844-8853 IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:8052']},\n",
       "  'metabolites': {'adpman_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'man1p_c': 1.0,\n",
       "   'h_c': 2.0,\n",
       "   'amp_c': 1.0},\n",
       "  'id': 'ADPMAN',\n",
       "  'gene_reaction_rule': 'HGNC:8052'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'ADPribose diphosphatase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:8052']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   'r5p_c': 1.0,\n",
       "   'h_c': 2.0,\n",
       "   'adprib_c': -1.0,\n",
       "   'amp_c': 1.0},\n",
       "  'id': 'ADPRDP',\n",
       "  'gene_reaction_rule': 'HGNC:8052'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'ADPribose diphosphatase (mitochondrial)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:8056']},\n",
       "  'metabolites': {'adprib_m': -1.0,\n",
       "   'h2o_m': -1.0,\n",
       "   'amp_m': 1.0,\n",
       "   'r5p_m': 1.0,\n",
       "   'h_m': 2.0},\n",
       "  'id': 'ADPRDPm',\n",
       "  'gene_reaction_rule': 'HGNC:8056'},\n",
       " {'subsystem': 'Nucleotide salvage pathway',\n",
       "  'name': 'adenine phosphoribosyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:626']},\n",
       "  'metabolites': {'ppi_c': 1.0, 'ade_c': -1.0, 'amp_c': 1.0, 'prpp_c': -1.0},\n",
       "  'id': 'ADPT',\n",
       "  'gene_reaction_rule': 'HGNC:626'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'carnitine O-palmitoyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:18540 or HGNC:2328 or HGNC:2329']},\n",
       "  'metabolites': {'crn_c': -1.0,\n",
       "   'adrncoa_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'adrncrn_c': 1.0},\n",
       "  'id': 'ADRNCPT1',\n",
       "  'gene_reaction_rule': 'HGNC:18540 or HGNC:2328 or HGNC:2329'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'carnitine transferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2330']},\n",
       "  'metabolites': {'coa_m': -1.0,\n",
       "   'adrncoa_m': 1.0,\n",
       "   'crn_m': 1.0,\n",
       "   'adrncrn_m': -1.0},\n",
       "  'id': 'ADRNCPT2',\n",
       "  'gene_reaction_rule': 'HGNC:2330'},\n",
       " {'subsystem': 'Selenoamino acid metabolism',\n",
       "  'name': 'adenylyl-selenate kinase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:8603 or HGNC:8604']},\n",
       "  'metabolites': {'adsel_c': -1.0,\n",
       "   '3padsel_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'adp_c': 1.0},\n",
       "  'id': 'ADSELK',\n",
       "  'gene_reaction_rule': 'HGNC:8603 or HGNC:8604'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'adenylyl-sulfate kinase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'GENE ASSOCIATION': ['HGNC:8603 or HGNC:8604']},\n",
       "  'metabolites': {'adp_c': 1.0,\n",
       "   'paps_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'aps_c': -1.0},\n",
       "  'id': 'ADSK',\n",
       "  'gene_reaction_rule': 'HGNC:8603 or HGNC:8604'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'adenylosuccinate lyase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:291']},\n",
       "  'metabolites': {'fum_c': 1.0, 'amp_c': 1.0, 'dcamp_c': -1.0},\n",
       "  'id': 'ADSL1',\n",
       "  'gene_reaction_rule': 'HGNC:291'},\n",
       " {'subsystem': 'Purine synthesis',\n",
       "  'name': 'adenylosuccinate lyase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:291']},\n",
       "  'metabolites': {'aicar_c': 1.0, 'fum_c': 1.0, '25aics_c': -1.0},\n",
       "  'id': 'ADSL2',\n",
       "  'gene_reaction_rule': 'HGNC:291'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'adenylosuccinate synthase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['r0277'],\n",
       "   'GENE ASSOCIATION': ['HGNC:20093 or HGNC:292']},\n",
       "  'metabolites': {'asp_L_c': -1.0,\n",
       "   'gdp_c': 1.0,\n",
       "   'gtp_c': -1.0,\n",
       "   'h_c': 2.0,\n",
       "   'pi_c': 1.0,\n",
       "   'imp_c': -1.0,\n",
       "   'dcamp_c': 1.0},\n",
       "  'id': 'ADSS',\n",
       "  'gene_reaction_rule': 'HGNC:20093 or HGNC:292'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'N-acetyllactosaminide beta-1,3-N-acetylglucosaminyltransferase, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:15683 or HGNC:15629 or HGNC:15685 or HGNC:21727 or HGNC:24139 or HGNC:18811 or HGNC:24139 or HGNC:13528 or HGNC:15629']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'ksi_pre19_g': 1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'ksi_pre18_g': -1.0},\n",
       "  'id': 'AG13T10g',\n",
       "  'gene_reaction_rule': 'HGNC:15683 or HGNC:15629 or HGNC:15685 or HGNC:21727 or HGNC:24139 or HGNC:18811 or HGNC:24139 or HGNC:13528 or HGNC:15629'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'N-acetyllactosaminide beta-1,3-N-acetylglucosaminyltransferase, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:15683 or HGNC:24139 or HGNC:15685 or HGNC:21727 or HGNC:24139 or HGNC:18811 or HGNC:13528 or HGNC:15629 or HGNC:15629']},\n",
       "  'metabolites': {'ksi_pre22_g': 1.0,\n",
       "   'h_g': 1.0,\n",
       "   'ksi_pre21_g': -1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'udp_g': 1.0},\n",
       "  'id': 'AG13T11g',\n",
       "  'gene_reaction_rule': 'HGNC:15683 or HGNC:24139 or HGNC:15685 or HGNC:21727 or HGNC:24139 or HGNC:18811 or HGNC:13528 or HGNC:15629 or HGNC:15629'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'N-acetyllactosaminide beta-1,3-N-acetylglucosaminyltransferase, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:15683 or HGNC:24139 or HGNC:15685 or HGNC:21727 or HGNC:24139 or HGNC:18811 or HGNC:13528 or HGNC:15629 or HGNC:15629']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'ksi_pre24_g': -1.0,\n",
       "   'ksi_pre25_g': 1.0},\n",
       "  'id': 'AG13T12g',\n",
       "  'gene_reaction_rule': 'HGNC:15683 or HGNC:24139 or HGNC:15685 or HGNC:21727 or HGNC:24139 or HGNC:18811 or HGNC:13528 or HGNC:15629 or HGNC:15629'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'N-acetyllactosaminide beta-1,3-N-acetylglucosaminyltransferase, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:15683 or HGNC:24139 or HGNC:15685 or HGNC:21727 or HGNC:24139 or HGNC:18811 or HGNC:13528 or HGNC:15629 or HGNC:15629']},\n",
       "  'metabolites': {'ksi_pre27_g': -1.0,\n",
       "   'ksi_pre28_g': 1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'AG13T13g',\n",
       "  'gene_reaction_rule': 'HGNC:15683 or HGNC:24139 or HGNC:15685 or HGNC:21727 or HGNC:24139 or HGNC:18811 or HGNC:13528 or HGNC:15629 or HGNC:15629'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'N-acetyllactosaminide beta-1,3-N-acetylglucosaminyltransferase, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:15683 or HGNC:24139 or HGNC:15685 or HGNC:21727 or HGNC:24139 or HGNC:18811 or HGNC:13528 or HGNC:15629 or HGNC:15629']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'ksi_pre31_g': 1.0,\n",
       "   'ksi_pre30_g': -1.0,\n",
       "   'udp_g': 1.0},\n",
       "  'id': 'AG13T14g',\n",
       "  'gene_reaction_rule': 'HGNC:15683 or HGNC:24139 or HGNC:15685 or HGNC:21727 or HGNC:24139 or HGNC:18811 or HGNC:13528 or HGNC:15629 or HGNC:15629'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'N-acetyllactosaminide beta-1,3-N-acetylglucosaminyltransferase, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:15683 or HGNC:24139 or HGNC:15685 or HGNC:21727 or HGNC:24139 or HGNC:18811 or HGNC:13528 or HGNC:15629 or HGNC:15629']},\n",
       "  'metabolites': {'ksi_pre34_g': 1.0,\n",
       "   'ksi_pre33_g': -1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'AG13T15g',\n",
       "  'gene_reaction_rule': 'HGNC:15683 or HGNC:24139 or HGNC:15685 or HGNC:21727 or HGNC:24139 or HGNC:18811 or HGNC:13528 or HGNC:15629 or HGNC:15629'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'N-acetyllactosaminide beta-1,3-N-acetylglucosaminyltransferase, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:15683 or HGNC:15629 or HGNC:15685 or HGNC:21727 or HGNC:24139 or HGNC:18811 or HGNC:24139 or HGNC:15629 or HGNC:13528']},\n",
       "  'metabolites': {'uacgam_g': -1.0,\n",
       "   'ksii_core4_pre3_g': 1.0,\n",
       "   'ksii_core4_pre2_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'AG13T16g',\n",
       "  'gene_reaction_rule': 'HGNC:15683 or HGNC:15629 or HGNC:15685 or HGNC:21727 or HGNC:24139 or HGNC:18811 or HGNC:24139 or HGNC:15629 or HGNC:13528'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'N-acetyllactosaminide beta-1,3-N-acetylglucosaminyltransferase, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:15629 or HGNC:15685 or HGNC:15629 or HGNC:24139 or HGNC:18811 or HGNC:24139 or HGNC:21727 or HGNC:15683 or HGNC:13528']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'ksii_core4_pre5_g': 1.0,\n",
       "   'ksii_core4_pre4_g': -1.0},\n",
       "  'id': 'AG13T17g',\n",
       "  'gene_reaction_rule': 'HGNC:15629 or HGNC:15685 or HGNC:15629 or HGNC:24139 or HGNC:18811 or HGNC:24139 or HGNC:21727 or HGNC:15683 or HGNC:13528'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'N-acetyllactosaminide beta-1,3-N-acetylglucosaminyltransferase, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:15629 or HGNC:15685 or HGNC:15629 or HGNC:24139 or HGNC:18811 or HGNC:24139 or HGNC:21727 or HGNC:15683 or HGNC:13528']},\n",
       "  'metabolites': {'ksii_core4_pre7_g': -1.0,\n",
       "   'h_g': 1.0,\n",
       "   'ksii_core4_pre8_g': 1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'udp_g': 1.0},\n",
       "  'id': 'AG13T18g',\n",
       "  'gene_reaction_rule': 'HGNC:15629 or HGNC:15685 or HGNC:15629 or HGNC:24139 or HGNC:18811 or HGNC:24139 or HGNC:21727 or HGNC:15683 or HGNC:13528'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'N-acetyllactosaminide beta-1,3-N-acetylglucosaminyltransferase, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:15685 or HGNC:15629 or HGNC:18811 or HGNC:24139 or HGNC:24139 or HGNC:21727 or HGNC:13528 or HGNC:15629 or HGNC:15683']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'ksii_core2_pre3_g': 1.0,\n",
       "   'ksii_core2_pre2_g': -1.0},\n",
       "  'id': 'AG13T1g',\n",
       "  'gene_reaction_rule': 'HGNC:15685 or HGNC:15629 or HGNC:18811 or HGNC:24139 or HGNC:24139 or HGNC:21727 or HGNC:13528 or HGNC:15629 or HGNC:15683'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'N-acetyllactosaminide beta-1,3-N-acetylglucosaminyltransferase, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:15683 or HGNC:24139 or HGNC:15685 or HGNC:21727 or HGNC:24139 or HGNC:18811 or HGNC:13528 or HGNC:15629 or HGNC:15629']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'ksii_core2_pre5_g': 1.0,\n",
       "   'ksii_core2_pre4_g': -1.0,\n",
       "   'udp_g': 1.0},\n",
       "  'id': 'AG13T2g',\n",
       "  'gene_reaction_rule': 'HGNC:15683 or HGNC:24139 or HGNC:15685 or HGNC:21727 or HGNC:24139 or HGNC:18811 or HGNC:13528 or HGNC:15629 or HGNC:15629'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'N-acetyllactosaminide beta-1,3-N-acetylglucosaminyltransferase, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:13528 or HGNC:15685 or HGNC:24139 or HGNC:15629 or HGNC:15683 or HGNC:24139 or HGNC:15629 or HGNC:18811 or HGNC:21727']},\n",
       "  'metabolites': {'ksii_core2_pre8_g': 1.0,\n",
       "   'ksii_core2_pre7_g': -1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'AG13T3g',\n",
       "  'gene_reaction_rule': 'HGNC:13528 or HGNC:15685 or HGNC:24139 or HGNC:15629 or HGNC:15683 or HGNC:24139 or HGNC:15629 or HGNC:18811 or HGNC:21727'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'N-acetyllactosaminide beta-1,3-N-acetylglucosaminyltransferase, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:15685 or HGNC:15629 or HGNC:18811 or HGNC:13528 or HGNC:24139 or HGNC:21727 or HGNC:24139 or HGNC:15629 or HGNC:15683']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'ksi_pre1_g': -1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'ksi_pre2_g': 1.0,\n",
       "   'udp_g': 1.0},\n",
       "  'id': 'AG13T4g',\n",
       "  'gene_reaction_rule': 'HGNC:15685 or HGNC:15629 or HGNC:18811 or HGNC:13528 or HGNC:24139 or HGNC:21727 or HGNC:24139 or HGNC:15629 or HGNC:15683'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'N-acetyllactosaminide beta-1,3-N-acetylglucosaminyltransferase, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:15683 or HGNC:24139 or HGNC:15685 or HGNC:21727 or HGNC:24139 or HGNC:18811 or HGNC:13528 or HGNC:15629 or HGNC:15629']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'ksi_pre3_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'ksi_pre4_g': 1.0},\n",
       "  'id': 'AG13T5g',\n",
       "  'gene_reaction_rule': 'HGNC:15683 or HGNC:24139 or HGNC:15685 or HGNC:21727 or HGNC:24139 or HGNC:18811 or HGNC:13528 or HGNC:15629 or HGNC:15629'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'N-acetyllactosaminide beta-1,3-N-acetylglucosaminyltransferase, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:15683 or HGNC:24139 or HGNC:15685 or HGNC:21727 or HGNC:24139 or HGNC:18811 or HGNC:13528 or HGNC:15629 or HGNC:15629']},\n",
       "  'metabolites': {'ksi_pre7_g': 1.0,\n",
       "   'ksi_pre6_g': -1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'AG13T6g',\n",
       "  'gene_reaction_rule': 'HGNC:15683 or HGNC:24139 or HGNC:15685 or HGNC:21727 or HGNC:24139 or HGNC:18811 or HGNC:13528 or HGNC:15629 or HGNC:15629'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'N-acetyllactosaminide beta-1,3-N-acetylglucosaminyltransferase, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:15683 or HGNC:24139 or HGNC:15685 or HGNC:21727 or HGNC:24139 or HGNC:18811 or HGNC:13528 or HGNC:15629 or HGNC:15629']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'ksi_pre9_g': -1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'ksi_pre10_g': 1.0},\n",
       "  'id': 'AG13T7g',\n",
       "  'gene_reaction_rule': 'HGNC:15683 or HGNC:24139 or HGNC:15685 or HGNC:21727 or HGNC:24139 or HGNC:18811 or HGNC:13528 or HGNC:15629 or HGNC:15629'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'N-acetyllactosaminide beta-1,3-N-acetylglucosaminyltransferase, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:15683 or HGNC:24139 or HGNC:15685 or HGNC:21727 or HGNC:24139 or HGNC:18811 or HGNC:13528 or HGNC:15629 or HGNC:15629']},\n",
       "  'metabolites': {'ksi_pre13_g': 1.0,\n",
       "   'ksi_pre12_g': -1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'AG13T8g',\n",
       "  'gene_reaction_rule': 'HGNC:15683 or HGNC:24139 or HGNC:15685 or HGNC:21727 or HGNC:24139 or HGNC:18811 or HGNC:13528 or HGNC:15629 or HGNC:15629'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'N-acetyllactosaminide beta-1,3-N-acetylglucosaminyltransferase, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:21727 or HGNC:15683 or HGNC:13528 or HGNC:24139 or HGNC:15629 or HGNC:24139 or HGNC:15629 or HGNC:15685 or HGNC:18811']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'ksi_pre15_g': -1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'ksi_pre16_g': 1.0,\n",
       "   'udp_g': 1.0},\n",
       "  'id': 'AG13T9g',\n",
       "  'gene_reaction_rule': 'HGNC:21727 or HGNC:15683 or HGNC:13528 or HGNC:24139 or HGNC:15629 or HGNC:24139 or HGNC:15629 or HGNC:15685 or HGNC:18811'},\n",
       " {'subsystem': 'Aminosugar metabolism',\n",
       "  'name': 'N-acetylglucosamine-6-phosphate deacetylase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['acetylglucosamine phosphate deacetylase; acetylaminodeoxyglucosephosphate acetylhydrolase; 2-acetamido-2-deoxy-D-glucose-6-phosphate amidohydrolase'],\n",
       "   'GENE ASSOCIATION': ['HGNC:24262']},\n",
       "  'metabolites': {'ac_c': 1.0,\n",
       "   'acgam6p_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'gam6p_c': 1.0},\n",
       "  'id': 'AGDC',\n",
       "  'gene_reaction_rule': 'HGNC:24262'},\n",
       " {'subsystem': 'Glycerophospholipid metabolism',\n",
       "  'name': 'alkyl acylglycerol phosphocholine transferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'ak2gchol_hs_c': 1.0,\n",
       "   'cdpchol_c': -1.0,\n",
       "   'ak2g_hs_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   'cmp_c': 1.0},\n",
       "  'id': 'AGLPC',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Glycerophospholipid metabolism',\n",
       "  'name': 'alkyl acyl glycerophosphoethanolamine desaturase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   'h2o_c': 2.0,\n",
       "   'nadp_c': 1.0,\n",
       "   'nadph_c': -1.0,\n",
       "   'ak2gpe_hs_c': -1.0,\n",
       "   'dak2gpe_hs_c': 1.0,\n",
       "   'h_c': -1.0},\n",
       "  'id': 'AGLPED',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Glycerophospholipid metabolism',\n",
       "  'name': 'alkyl acylglycerol phosphoethanolamine transferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['see Harper&apos;s Biochemistry &apos;Metabolism of Acylglycerols and Sphingolipids&apos; chapter currently charge imbalanced (+2) - charging of individual molecules consistent w/ known info at pH 7.4 NJ'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'cdpea_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   'ak2g_hs_c': -1.0,\n",
       "   'ak2gpe_hs_c': 1.0,\n",
       "   'cmp_c': 1.0},\n",
       "  'id': 'AGLPET',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Glycerophospholipid metabolism',\n",
       "  'name': 'alkyl glycerol phosphate hydrolase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'ak2gp_hs_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'pi_c': 1.0,\n",
       "   'ak2g_hs_c': 1.0},\n",
       "  'id': 'AGLPH',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Glycerophospholipid metabolism',\n",
       "  'name': 'alkyl glycerol phosphate reductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['takes place on outer peroxisomal membrane NJ'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'akgp_hs_c': 1.0,\n",
       "   'akdhap_hs_c': -1.0,\n",
       "   'nadp_c': -1.0,\n",
       "   'nadph_c': 1.0,\n",
       "   'h_c': 3.0},\n",
       "  'id': 'AGLPR',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Glycerophospholipid metabolism',\n",
       "  'name': 'alkyl glycerol phosphate acyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'Rtotalcoa_c': -1.0,\n",
       "   'nadp_c': 1.0,\n",
       "   'nadph_c': -1.0,\n",
       "   'ak2gp_hs_c': 1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'h_c': 3.0,\n",
       "   'akgp_hs_c': -1.0},\n",
       "  'id': 'AGLPT',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Arginine and Proline Metabolism',\n",
       "  'name': 'agmatinase (m)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB m'],\n",
       "   'GENE ASSOCIATION': ['HGNC:18407']},\n",
       "  'metabolites': {'h2o_m': -1.0, 'agm_m': -1.0, 'urea_m': 1.0, 'ptrc_m': 1.0},\n",
       "  'id': 'AGMTm',\n",
       "  'gene_reaction_rule': 'HGNC:18407'},\n",
       " {'subsystem': 'Triacylglycerol synthesis',\n",
       "  'name': '1-acylglycerol-3-phosphate O-acyltransferase 1',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['FULLR corresponds to acyl/R groups in texts. NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:20880 or HGNC:20885 or HGNC:20886 or HGNC:25193 or HGNC:324 or HGNC:325 or HGNC:326']},\n",
       "  'metabolites': {'pa_hs_c': 1.0,\n",
       "   'Rtotalcoa_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'alpa_hs_c': -1.0,\n",
       "   'h_c': 6.0},\n",
       "  'id': 'AGPAT1',\n",
       "  'gene_reaction_rule': 'HGNC:20880 or HGNC:20885 or HGNC:20886 or HGNC:25193 or HGNC:324 or HGNC:325 or HGNC:326'},\n",
       " {'subsystem': 'Urea cycle',\n",
       "  'name': 'N-acetyl-g-glutamyl-phosphate reductase, irreversible, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:9722']},\n",
       "  'metabolites': {'acg5sa_m': 1.0,\n",
       "   'acg5p_m': -1.0,\n",
       "   'nadph_m': -1.0,\n",
       "   'nadp_m': 1.0,\n",
       "   'pi_m': 1.0,\n",
       "   'h_m': -1.0},\n",
       "  'id': 'AGPRim',\n",
       "  'gene_reaction_rule': 'HGNC:9722'},\n",
       " {'subsystem': 'Alanine and aspartate metabolism',\n",
       "  'name': 'alanine-glyoxylate transaminase (irreversible), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:14404 or HGNC:14412']},\n",
       "  'metabolites': {'ala_L_m': -1.0, 'gly_m': 1.0, 'pyr_m': 1.0, 'glx_m': -1.0},\n",
       "  'id': 'AGTim',\n",
       "  'gene_reaction_rule': 'HGNC:14404 or HGNC:14412'},\n",
       " {'subsystem': 'Alanine and aspartate metabolism',\n",
       "  'name': 'alanine-glyoxylate transaminase (irreversible), (peroxisomal)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:341']},\n",
       "  'metabolites': {'pyr_x': 1.0, 'ala_L_x': -1.0, 'gly_x': 1.0, 'glx_x': -1.0},\n",
       "  'id': 'AGTix',\n",
       "  'gene_reaction_rule': 'HGNC:341'},\n",
       " {'subsystem': 'Methionine and cysteine metabolism',\n",
       "  'name': 'adenosylhomocysteinase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'GENE ASSOCIATION': ['HGNC:343 or HGNC:344']},\n",
       "  'metabolites': {'adn_c': 1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'ahcys_c': -1.0,\n",
       "   'hcys_L_c': 1.0},\n",
       "  'id': 'AHC',\n",
       "  'gene_reaction_rule': 'HGNC:343 or HGNC:344'},\n",
       " {'subsystem': 'N-glycan degradation',\n",
       "  'name': 'beta-N-acetylhexosaminidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:4878 and HGNC:4879) or HGNC:4878 or HGNC:4879']},\n",
       "  'metabolites': {'acgam_l': 3.0,\n",
       "   'n2m2nmn_l': -1.0,\n",
       "   'h2o_l': -3.0,\n",
       "   'm2mn_l': 1.0},\n",
       "  'id': 'AHEXASE2ly',\n",
       "  'gene_reaction_rule': '(HGNC:4878 and HGNC:4879) or HGNC:4878 or HGNC:4879'},\n",
       " {'subsystem': 'N-glycan degradation',\n",
       "  'name': 'beta-N-acetylhexosaminidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:4878 and HGNC:4879) or HGNC:4878 or HGNC:4879']},\n",
       "  'metabolites': {'n2m2mn_l': -1.0,\n",
       "   'acgam_l': 2.0,\n",
       "   'h2o_l': -2.0,\n",
       "   'm2mn_l': 1.0},\n",
       "  'id': 'AHEXASEly',\n",
       "  'gene_reaction_rule': '(HGNC:4878 and HGNC:4879) or HGNC:4878 or HGNC:4879'},\n",
       " {'subsystem': 'Purine synthesis',\n",
       "  'name': 'phosphoribosylaminoimidazolecarboxamide formyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:794']},\n",
       "  'metabolites': {'aicar_c': -1.0,\n",
       "   'fprica_c': 1.0,\n",
       "   '10fthf_c': -1.0,\n",
       "   'thf_c': 1.0},\n",
       "  'id': 'AICART',\n",
       "  'gene_reaction_rule': 'HGNC:794'},\n",
       " {'subsystem': 'Citric acid cycle',\n",
       "  'name': '2-oxoglutarate dehydrogenase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['EC taken from Ligand TV. Genetic'],\n",
       "   'GENE ASSOCIATION': ['HGNC:21350 and HGNC:2898 and HGNC:2911 and HGNC:8124']},\n",
       "  'metabolites': {'succoa_m': 1.0,\n",
       "   'akg_m': -1.0,\n",
       "   'co2_m': 1.0,\n",
       "   'nad_m': -1.0,\n",
       "   'coa_m': -1.0,\n",
       "   'nadh_m': 1.0},\n",
       "  'id': 'AKGDm',\n",
       "  'gene_reaction_rule': 'HGNC:21350 and HGNC:2898 and HGNC:2911 and HGNC:8124'},\n",
       " {'subsystem': 'Steroid metabolism',\n",
       "  'name': 'aldo-keto reductase family 1, member C1 (chlordecone reductase; 3-alpha hydroxysteroid dehydrogenase, type I; dihydrodiol dehydrogenase 4)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:384']},\n",
       "  'metabolites': {'aprgstrn_c': 1.0,\n",
       "   'nadph_c': -1.0,\n",
       "   'prgstrn_c': -1.0,\n",
       "   'nadp_c': 1.0,\n",
       "   'h_c': -1.0},\n",
       "  'id': 'AKR1C1',\n",
       "  'gene_reaction_rule': 'HGNC:384'},\n",
       " {'subsystem': 'Bile acid synthesis',\n",
       "  'name': 'aldo-keto reductase family 1, member C4 (chlordecone reductase; 3-alpha hydroxysteroid dehydrogenase, type I; dihydrodiol dehydrogenase 4)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:387']},\n",
       "  'metabolites': {'nadh_c': -1.0,\n",
       "   'xol7ah2_c': 1.0,\n",
       "   'h_c': -1.0,\n",
       "   'nad_c': 1.0,\n",
       "   'xol7ah_c': -1.0},\n",
       "  'id': 'AKR1C41',\n",
       "  'gene_reaction_rule': 'HGNC:387'},\n",
       " {'subsystem': 'Bile acid synthesis',\n",
       "  'name': 'aldo-keto reductase family 1, member C4 (chlordecone reductase; 3-alpha hydroxysteroid dehydrogenase, type I; dihydrodiol dehydrogenase 4)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:387']},\n",
       "  'metabolites': {'xoltriol_c': 1.0,\n",
       "   'nadh_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'nad_c': 1.0,\n",
       "   'xoldioloneh_c': -1.0},\n",
       "  'id': 'AKR1C42',\n",
       "  'gene_reaction_rule': 'HGNC:387'},\n",
       " {'subsystem': 'Bile acid synthesis',\n",
       "  'name': 'aldo-keto reductase family 1, member D1 (delta 4-3-ketosteroid-5-beta-reductase)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['refs: uniprot, KEGG DW Russell bile acid synthesis review paper (see human recon journal citations) NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:388']},\n",
       "  'metabolites': {'xol7aone_c': -1.0,\n",
       "   'nadp_c': 1.0,\n",
       "   'nadph_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'xol7ah_c': 1.0},\n",
       "  'id': 'AKR1D',\n",
       "  'gene_reaction_rule': 'HGNC:388'},\n",
       " {'subsystem': 'Bile acid synthesis',\n",
       "  'name': 'aldo-keto reductase family 1, member D1 (delta 4-3-ketosteroid-5-beta-reductase)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['refs: uniprot, KEGG DW Russell bile acid synthesis review paper (see human recon journal citations) NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:388']},\n",
       "  'metabolites': {'xoldiolone_c': -1.0,\n",
       "   'nadp_c': 1.0,\n",
       "   'nadph_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'xoldioloneh_c': 1.0},\n",
       "  'id': 'AKR1D2',\n",
       "  'gene_reaction_rule': 'HGNC:388'},\n",
       " {'subsystem': 'Alanine and aspartate metabolism',\n",
       "  'name': 'alanine racemase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'], 'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'ala_L_c': -1.0, 'ala_D_c': 1.0},\n",
       "  'id': 'ALAR',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Glycine, serine, alanine and threonine metabolism',\n",
       "  'name': '5-aminolevulinate synthase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:396 or HGNC:397 or HGNC:4188']},\n",
       "  'metabolites': {'succoa_m': -1.0,\n",
       "   'gly_m': -1.0,\n",
       "   'co2_m': 1.0,\n",
       "   'coa_m': 1.0,\n",
       "   'h_m': -1.0,\n",
       "   '5aop_m': 1.0},\n",
       "  'id': 'ALASm',\n",
       "  'gene_reaction_rule': 'HGNC:396 or HGNC:397 or HGNC:4188'},\n",
       " {'subsystem': 'Glutamate metabolism',\n",
       "  'name': 'L-alanine transaminase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'GENE ASSOCIATION': ['HGNC:18062 or HGNC:4552']},\n",
       "  'metabolites': {'akg_c': -1.0,\n",
       "   'pyr_c': 1.0,\n",
       "   'ala_L_c': -1.0,\n",
       "   'glu_L_c': 1.0},\n",
       "  'id': 'ALATA_L',\n",
       "  'gene_reaction_rule': 'HGNC:18062 or HGNC:4552'},\n",
       " {'subsystem': 'Miscellaneous',\n",
       "  'name': 'alcohol dehydrogenase (methanol)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'GENE ASSOCIATION': ['HGNC:16354 or (HGNC:249 and HGNC:250) or (HGNC:249 and HGNC:251) or HGNC:249 or (HGNC:250 and HGNC:251) or HGNC:250 or HGNC:251 or HGNC:252 or HGNC:253 or HGNC:255 or HGNC:256 or HGNC:28697']},\n",
       "  'metabolites': {'nad_c': -1.0,\n",
       "   'fald_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'nadh_c': 1.0,\n",
       "   'meoh_c': -1.0},\n",
       "  'id': 'ALCD1',\n",
       "  'gene_reaction_rule': 'HGNC:16354 or (HGNC:249 and HGNC:250) or (HGNC:249 and HGNC:251) or HGNC:249 or (HGNC:250 and HGNC:251) or HGNC:250 or HGNC:251 or HGNC:252 or HGNC:253 or HGNC:255 or HGNC:256 or HGNC:28697'},\n",
       " {'subsystem': 'Pyruvate metabolism',\n",
       "  'name': 'alcohol dehydrogenase (L-1,2-propanediol)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:16354 or (HGNC:249 and HGNC:250) or (HGNC:249 and HGNC:251) or HGNC:249 or (HGNC:250 and HGNC:251) or HGNC:250 or HGNC:251 or HGNC:252 or HGNC:253 or HGNC:255 or HGNC:256 or HGNC:28697']},\n",
       "  'metabolites': {'12ppd_S_c': -1.0,\n",
       "   'nad_c': -1.0,\n",
       "   'nadh_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'lald_L_c': 1.0},\n",
       "  'id': 'ALCD21_L',\n",
       "  'gene_reaction_rule': 'HGNC:16354 or (HGNC:249 and HGNC:250) or (HGNC:249 and HGNC:251) or HGNC:249 or (HGNC:250 and HGNC:251) or HGNC:250 or HGNC:251 or HGNC:252 or HGNC:253 or HGNC:255 or HGNC:256 or HGNC:28697'},\n",
       " {'subsystem': 'Pyruvate metabolism',\n",
       "  'name': 'alcohol dehydrogenase (L-lactaldehyde)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:16354 or (HGNC:249 and HGNC:250) or (HGNC:249 and HGNC:251) or HGNC:249 or (HGNC:250 and HGNC:251) or HGNC:250 or HGNC:251 or HGNC:252 or HGNC:253 or HGNC:255 or HGNC:256 or HGNC:28697']},\n",
       "  'metabolites': {'lald_L_c': -1.0,\n",
       "   'nad_c': -1.0,\n",
       "   'nadh_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'mthgxl_c': 1.0},\n",
       "  'id': 'ALCD22_L',\n",
       "  'gene_reaction_rule': 'HGNC:16354 or (HGNC:249 and HGNC:250) or (HGNC:249 and HGNC:251) or HGNC:249 or (HGNC:250 and HGNC:251) or HGNC:250 or HGNC:251 or HGNC:252 or HGNC:253 or HGNC:255 or HGNC:256 or HGNC:28697'},\n",
       " {'subsystem': 'Glycolysis/gluconeogenesis',\n",
       "  'name': 'alcohol dehydrogenase, forward rxn (ethanol -> acetaldehyde)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:16354 or (HGNC:249 and HGNC:250) or (HGNC:249 and HGNC:251) or HGNC:249 or (HGNC:250 and HGNC:251) or HGNC:250 or HGNC:251 or HGNC:252 or HGNC:253 or HGNC:255 or HGNC:256 or HGNC:28697']},\n",
       "  'metabolites': {'acald_c': 1.0,\n",
       "   'nad_c': -1.0,\n",
       "   'nadh_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'etoh_c': -1.0},\n",
       "  'id': 'ALCD2if',\n",
       "  'gene_reaction_rule': 'HGNC:16354 or (HGNC:249 and HGNC:250) or (HGNC:249 and HGNC:251) or HGNC:249 or (HGNC:250 and HGNC:251) or HGNC:250 or HGNC:251 or HGNC:252 or HGNC:253 or HGNC:255 or HGNC:256 or HGNC:28697'},\n",
       " {'subsystem': 'Glycolysis/gluconeogenesis',\n",
       "  'name': 'alcohol dehydrogenase (ethanol, NADP), forward reaction',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:380']},\n",
       "  'metabolites': {'acald_c': 1.0,\n",
       "   'nadp_c': -1.0,\n",
       "   'nadph_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'etoh_c': -1.0},\n",
       "  'id': 'ALCD2yf',\n",
       "  'gene_reaction_rule': 'HGNC:380'},\n",
       " {'subsystem': 'Tryptophan metabolism',\n",
       "  'name': 'aldehyde dehydrogenase (indole-3-acetaldehyde, NAD)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['NCD. The formation rate of IAA from IAL was used to measure ALDH activity (9884322). Another ref. 2437936 for human blood aldehyde dehydrogenase'],\n",
       "   'GENE ASSOCIATION': ['HGNC:15472 or HGNC:402 or HGNC:403 or HGNC:412 or HGNC:877']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   'h_c': 2.0,\n",
       "   'nad_c': -1.0,\n",
       "   'nadh_c': 1.0,\n",
       "   'ind3ac_c': 1.0,\n",
       "   'id3acald_c': -1.0},\n",
       "  'id': 'ALDD20x',\n",
       "  'gene_reaction_rule': 'HGNC:15472 or HGNC:402 or HGNC:403 or HGNC:412 or HGNC:877'},\n",
       " {'subsystem': 'Tryptophan metabolism',\n",
       "  'name': 'aldehyde dehydrogenase (indole-3-acetaldehyde, NAD), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['NCD. Presence of cytoplasmic and mitochondrial isozymes of human aldehyde dehydrogenase (E.C. 1.2.1.3). Genetic ref. 8621661 for Delta-1-pyrroline-5-carboxylate dehydrogenase'],\n",
       "   'GENE ASSOCIATION': ['HGNC:404 or HGNC:406 or HGNC:407']},\n",
       "  'metabolites': {'h2o_m': -1.0,\n",
       "   'id3acald_m': -1.0,\n",
       "   'ind3ac_m': 1.0,\n",
       "   'nadh_m': 1.0,\n",
       "   'nad_m': -1.0,\n",
       "   'h_m': 2.0},\n",
       "  'id': 'ALDD20xm',\n",
       "  'gene_reaction_rule': 'HGNC:404 or HGNC:406 or HGNC:407'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'aldehyde dehydrogenase (pristanal, NAD)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:403']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   'nadh_c': 1.0,\n",
       "   'h_c': 2.0,\n",
       "   'nad_c': -1.0,\n",
       "   'prist_c': 1.0,\n",
       "   'pristanal_c': -1.0},\n",
       "  'id': 'ALDD21',\n",
       "  'gene_reaction_rule': 'HGNC:403'},\n",
       " {'subsystem': 'Glycolysis/gluconeogenesis',\n",
       "  'name': 'aldehyde dehydrogenase (acetaldehyde, NAD)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['Only acetaldehyde considered for now. See also 1.2.1.4. Wide specificity, including oxidation of D-glucuronolactone to D-glucarate. Formerly EC 1.1.1.70; r0176'],\n",
       "   'GENE ASSOCIATION': ['HGNC:15472 or HGNC:402 or HGNC:403 or HGNC:405 or HGNC:407 or HGNC:409 or HGNC:410 or HGNC:411 or HGNC:412 or HGNC:877']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   'acald_c': -1.0,\n",
       "   'h_c': 2.0,\n",
       "   'nad_c': -1.0,\n",
       "   'ac_c': 1.0,\n",
       "   'nadh_c': 1.0},\n",
       "  'id': 'ALDD2x',\n",
       "  'gene_reaction_rule': 'HGNC:15472 or HGNC:402 or HGNC:403 or HGNC:405 or HGNC:407 or HGNC:409 or HGNC:410 or HGNC:411 or HGNC:412 or HGNC:877'},\n",
       " {'subsystem': 'Glycolysis/gluconeogenesis',\n",
       "  'name': 'aldehyde dehydrogenase (acetylaldehyde, NAD), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD. For NAD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:404 or HGNC:407']},\n",
       "  'metabolites': {'acald_m': -1.0,\n",
       "   'h2o_m': -1.0,\n",
       "   'ac_m': 1.0,\n",
       "   'nadh_m': 1.0,\n",
       "   'nad_m': -1.0,\n",
       "   'h_m': 2.0},\n",
       "  'id': 'ALDD2xm',\n",
       "  'gene_reaction_rule': 'HGNC:404 or HGNC:407'},\n",
       " {'subsystem': 'Glycolysis/gluconeogenesis',\n",
       "  'name': 'aldehyde dehydrogenase (acetaldehyde, NADP)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['See also ALDH1 (1.2.1.3); r0177'],\n",
       "   'GENE ASSOCIATION': ['HGNC:15472 or HGNC:403 or HGNC:405 or HGNC:407 or HGNC:410 or HGNC:411 or HGNC:412']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   'nadp_c': -1.0,\n",
       "   'nadph_c': 1.0,\n",
       "   'acald_c': -1.0,\n",
       "   'h_c': 2.0,\n",
       "   'ac_c': 1.0},\n",
       "  'id': 'ALDD2y',\n",
       "  'gene_reaction_rule': 'HGNC:15472 or HGNC:403 or HGNC:405 or HGNC:407 or HGNC:410 or HGNC:411 or HGNC:412'},\n",
       " {'subsystem': 'Glycerophospholipid metabolism',\n",
       "  'name': 'alkaline phosphatase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['added for staph (SAB)'],\n",
       "   'GENE ASSOCIATION': ['HGNC:437 or HGNC:438 or HGNC:439 or HGNC:441']},\n",
       "  'metabolites': {'h2o_c': -1.0, 'dhap_c': -1.0, 'pi_c': 1.0, 'dha_c': 1.0},\n",
       "  'id': 'ALKP',\n",
       "  'gene_reaction_rule': 'HGNC:437 or HGNC:438 or HGNC:439 or HGNC:441'},\n",
       " {'subsystem': 'Eicosanoid metabolism',\n",
       "  'name': 'Arachidonate 12-lipoxygenase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:429']},\n",
       "  'metabolites': {'o2_c': -1.0, '12HPET_c': 1.0, 'arachd_c': -1.0},\n",
       "  'id': 'ALOX12',\n",
       "  'gene_reaction_rule': 'HGNC:429'},\n",
       " {'subsystem': 'Eicosanoid metabolism',\n",
       "  'name': 'Arachidonate 12-lipoxygenase R',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:430']},\n",
       "  'metabolites': {'o2_c': -1.0, '12RHPET_c': 1.0, 'arachd_c': -1.0},\n",
       "  'id': 'ALOX12R',\n",
       "  'gene_reaction_rule': 'HGNC:430'},\n",
       " {'subsystem': 'Eicosanoid metabolism',\n",
       "  'name': 'Arachidonate 15-lipoxygenase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Converts arachidonic acid to 15S-hydroperoxyeicosatetraenoic acid. Also acts on C-12 of arachidonate as well as on linoleic acid. NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:433 or HGNC:434']},\n",
       "  'metabolites': {'o2_c': -1.0, 'arachd_c': -1.0, '15HPET_c': 1.0},\n",
       "  'id': 'ALOX15',\n",
       "  'gene_reaction_rule': 'HGNC:433 or HGNC:434'},\n",
       " {'subsystem': 'Eicosanoid metabolism',\n",
       "  'name': 'Arachidonate 5-lipoxygenase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['cofactors + usage still not clear see version ALOX5b for ATP concomittant hydrolysis - literature notes in vitro requirement for Ca and ATP for activity in addition to Fe'],\n",
       "   'GENE ASSOCIATION': ['HGNC:435']},\n",
       "  'metabolites': {'o2_c': -1.0, '5HPET_c': 1.0, 'arachd_c': -1.0},\n",
       "  'id': 'ALOX5',\n",
       "  'gene_reaction_rule': 'HGNC:435'},\n",
       " {'subsystem': 'Eicosanoid metabolism',\n",
       "  'name': 'Arachidonate 5-lipoxygenase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['cofactors + usage still not clear unknown mechanism - biochemistry supported by  PMID: 2829172, 11160764'],\n",
       "   'GENE ASSOCIATION': ['HGNC:435']},\n",
       "  'metabolites': {'h2o_c': 1.0, 'leuktrA4_c': 1.0, '5HPET_c': -1.0},\n",
       "  'id': 'ALOX52',\n",
       "  'gene_reaction_rule': 'HGNC:435'},\n",
       " {'subsystem': 'Pyruvate metabolism',\n",
       "  'name': 'aldose reductase (methylglyoxal)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:380 or HGNC:381 or HGNC:389']},\n",
       "  'metabolites': {'nadph_c': -1.0,\n",
       "   'nadp_c': 1.0,\n",
       "   'acetol_c': 1.0,\n",
       "   'h_c': -1.0,\n",
       "   'mthgxl_c': -1.0},\n",
       "  'id': 'ALR2',\n",
       "  'gene_reaction_rule': 'HGNC:380 or HGNC:381 or HGNC:389'},\n",
       " {'subsystem': 'Pyruvate metabolism',\n",
       "  'name': 'aldose reductase (acetol)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:380 or HGNC:381 or HGNC:389']},\n",
       "  'metabolites': {'nadph_c': -1.0,\n",
       "   '12ppd_S_c': 1.0,\n",
       "   'nadp_c': 1.0,\n",
       "   'acetol_c': -1.0,\n",
       "   'h_c': -1.0},\n",
       "  'id': 'ALR3',\n",
       "  'gene_reaction_rule': 'HGNC:380 or HGNC:381 or HGNC:389'},\n",
       " {'subsystem': 'Bile acid synthesis',\n",
       "  'name': 'alpha-methylacyl-CoA racemase (reductase)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['should be analogous to AMACRp as per KEGG - in this case: reductase not a racemase NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:451']},\n",
       "  'metabolites': {'dhcholoylcoa_x': 1.0,\n",
       "   'dhcholestancoa_x': -1.0,\n",
       "   'o2_x': -0.5,\n",
       "   'h2o_x': 1.0},\n",
       "  'id': 'AMACR2p',\n",
       "  'gene_reaction_rule': 'HGNC:451'},\n",
       " {'subsystem': 'Bile acid synthesis',\n",
       "  'name': 'alpha-methylacyl-CoA racemase (reductase)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['should be analogous to AMACRr as per KEGG - in this case: reductase not a racemase NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:451']},\n",
       "  'metabolites': {'o2_r': -0.5,\n",
       "   'dhcholoylcoa_r': 1.0,\n",
       "   'dhcholestancoa_r': -1.0,\n",
       "   'h2o_r': 1.0},\n",
       "  'id': 'AMACR2r',\n",
       "  'gene_reaction_rule': 'HGNC:451'},\n",
       " {'subsystem': 'Bile acid synthesis',\n",
       "  'name': 'alpha-methylacyl-CoA racemase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['see Russell bile acid biosynthesis review uniprot NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:451']},\n",
       "  'metabolites': {'cholcoar_x': -1.0, 'cholcoas_x': 1.0},\n",
       "  'id': 'AMACRp',\n",
       "  'gene_reaction_rule': 'HGNC:451'},\n",
       " {'subsystem': 'Bile acid synthesis',\n",
       "  'name': 'alpha-methylacyl-CoA racemase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['see Russell bile acid biosynthesis review uniprot NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:451']},\n",
       "  'metabolites': {'cholcoar_r': -1.0, 'cholcoas_r': 1.0},\n",
       "  'id': 'AMACRr',\n",
       "  'gene_reaction_rule': 'HGNC:451'},\n",
       " {'subsystem': 'Aminosugar metabolism',\n",
       "  'name': 'N-acetyl-D-mannosamine kinase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['JLR - Kegg R02705'],\n",
       "   'GENE ASSOCIATION': ['HGNC:17174 or HGNC:23657']},\n",
       "  'metabolites': {'acmana_c': -1.0,\n",
       "   'acmanap_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'adp_c': 1.0},\n",
       "  'id': 'AMANK',\n",
       "  'gene_reaction_rule': 'HGNC:17174 or HGNC:23657'},\n",
       " {'subsystem': 'Tryptophan metabolism',\n",
       "  'name': '2-aminomuconate reductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['not sure about name NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   'nadp_c': 1.0,\n",
       "   '2oxoadp_c': 1.0,\n",
       "   'nadph_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'amuco_c': -1.0,\n",
       "   'nh4_c': 1.0},\n",
       "  'id': 'AMCOXO',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'Adenosine monophosphate deaminase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'GENE ASSOCIATION': ['HGNC:468 or HGNC:469 or HGNC:470']},\n",
       "  'metabolites': {'amp_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'nh4_c': 1.0,\n",
       "   'h_c': -1.0,\n",
       "   'imp_c': 1.0},\n",
       "  'id': 'AMPDA',\n",
       "  'gene_reaction_rule': 'HGNC:468 or HGNC:469 or HGNC:470'},\n",
       " {'subsystem': 'Glutathione metabolism',\n",
       "  'name': 'alanyl aminopeptidase (cys-gly)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['SAB. activity measured in cytosol rat liver; IT IS leucine aminopeptidase (EC3.4.11.1) NOT alanyl aminopeptidase EC3.4.11.2 which shows &gt; 80% of the activity; sequence homology rat-human not yet verified; annotation &apos;extracellular&apos; under B column ERROR'],\n",
       "   'GENE ASSOCIATION': ['HGNC:500']},\n",
       "  'metabolites': {'h2o_c': -1.0, 'cys_L_c': 1.0, 'gly_c': 1.0, 'cgly_c': -1.0},\n",
       "  'id': 'AMPTASECG',\n",
       "  'gene_reaction_rule': 'HGNC:500'},\n",
       " {'subsystem': 'Glutathione metabolism',\n",
       "  'name': 'alanyl aminopeptidase (cys-gly) (e)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB. rat lumen renal proximal tubule preparation cysteinylglycine metabolism measured due to aminopeptidase M (=N); the annotation in BiGG of strong biochemical evidence in a textbook could not be found in the Google Books version search of this book; intensively sought for human metabolism; human metabolism is likely though not proven directly'],\n",
       "   'GENE ASSOCIATION': ['HGNC:500']},\n",
       "  'metabolites': {'cgly_e': -1.0, 'gly_e': 1.0, 'h2o_e': -1.0, 'cys_L_e': 1.0},\n",
       "  'id': 'AMPTASECGe',\n",
       "  'gene_reaction_rule': 'HGNC:500'},\n",
       " {'subsystem': 'Starch and sucrose metabolism',\n",
       "  'name': 'alpha-amylase, extracellular (strch1 -> strch2)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:478 or HGNC:476 or HGNC:474 or HGNC:475 or HGNC:474 or HGNC:477']},\n",
       "  'metabolites': {'strch2_e': 1.0,\n",
       "   'h2o_e': -8.0,\n",
       "   'strch1_e': -1.0,\n",
       "   'glc_D_e': 8.0},\n",
       "  'id': 'AMY1e',\n",
       "  'gene_reaction_rule': 'HGNC:478 or HGNC:476 or HGNC:474 or HGNC:475 or HGNC:474 or HGNC:477'},\n",
       " {'subsystem': 'Starch and sucrose metabolism',\n",
       "  'name': 'alpha-amylase, extracellular (glygn2 -> glygn4)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:478 or HGNC:476 or HGNC:474 or HGNC:475 or HGNC:474 or HGNC:477']},\n",
       "  'metabolites': {'h2o_e': -8.0,\n",
       "   'glygn2_e': -1.0,\n",
       "   'glc_D_e': 8.0,\n",
       "   'glygn4_e': 1.0},\n",
       "  'id': 'AMY2e',\n",
       "  'gene_reaction_rule': 'HGNC:478 or HGNC:476 or HGNC:474 or HGNC:475 or HGNC:474 or HGNC:477'},\n",
       " {'subsystem': 'Glycine, serine, alanine and threonine metabolism',\n",
       "  'name': 'L-2-amino-3-oxobutanoate decarboxylation, mitochondrial  (spontaneous)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['Spontaneous reaction'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'aact_m': 1.0, '2aobut_m': -1.0, 'h_m': -1.0, 'co2_m': 1.0},\n",
       "  'id': 'AOBUTDsm',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'Ap4A hydrolase, asymmetrically',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:8049']},\n",
       "  'metabolites': {'h_c': 2.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'ap4a_c': -1.0,\n",
       "   'atp_c': 1.0,\n",
       "   'amp_c': 1.0},\n",
       "  'id': 'AP4AH1',\n",
       "  'gene_reaction_rule': 'HGNC:8049'},\n",
       " {'subsystem': 'beta-Alanine metabolism',\n",
       "  'name': '3-Aminopropanoate:2-oxoglutarate aminotransferase (m)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:23']},\n",
       "  'metabolites': {'ala_B_m': -1.0,\n",
       "   'msa_m': 1.0,\n",
       "   'glu_L_m': 1.0,\n",
       "   'akg_m': -1.0},\n",
       "  'id': 'APAT2rm',\n",
       "  'gene_reaction_rule': 'HGNC:23'},\n",
       " {'subsystem': 'Biotin metabolism',\n",
       "  'name': 'Apocarboxylase formation',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['IT faked reaction to balance reaction in human biotin metabolism'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'apoC_c': -1.0,\n",
       "   'h2o_c': 1.0,\n",
       "   'lys_L_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   'apoC_Lys_c': 1.0},\n",
       "  'id': 'APOCF',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Biotin metabolism',\n",
       "  'name': 'Apocarboxylase formation, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['IT faked reaction to balance reaction in human biotin metabolism'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'apoC_Lys_m': 1.0,\n",
       "   'h_m': 1.0,\n",
       "   'h2o_m': 1.0,\n",
       "   'lys_L_m': -1.0,\n",
       "   'apoC_m': -1.0},\n",
       "  'id': 'APOCFm',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Biotin metabolism',\n",
       "  'name': 'proteolysis of apoC-Lys-btn',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'], 'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'apoC_c': 1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'biocyt_c': 1.0,\n",
       "   'apoC_Lys_btn_c': -1.0},\n",
       "  'id': 'APOC_LYS_BTNP',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Biotin metabolism',\n",
       "  'name': 'proteolysis of apoC-Lys-btn, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'], 'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'biocyt_m': 1.0,\n",
       "   'apoC_Lys_btn_m': -1.0,\n",
       "   'h2o_m': -1.0,\n",
       "   'apoC_m': 1.0},\n",
       "  'id': 'APOC_LYS_BTNPm',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Arginine and Proline Metabolism',\n",
       "  'name': 'N-acetylputrescine: oxygen oxireductase (deaminating)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:6833 or HGNC:6834']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'n4abutn_c': 1.0,\n",
       "   'h2o2_c': 1.0,\n",
       "   'aprut_c': -1.0,\n",
       "   'nh4_c': 1.0},\n",
       "  'id': 'APRTO2',\n",
       "  'gene_reaction_rule': 'HGNC:6833 or HGNC:6834'},\n",
       " {'subsystem': 'Pentose phosphate pathway',\n",
       "  'name': 'arabinose reductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'GENE ASSOCIATION': ['HGNC:380 or HGNC:381 or HGNC:389']},\n",
       "  'metabolites': {'abt_c': 1.0,\n",
       "   'nadph_c': -1.0,\n",
       "   'arab_L_c': -1.0,\n",
       "   'nadp_c': 1.0,\n",
       "   'h_c': -1.0},\n",
       "  'id': 'ARABR',\n",
       "  'gene_reaction_rule': 'HGNC:380 or HGNC:381 or HGNC:389'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'carnitine acyltransferase I',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:18540 or HGNC:2328 or HGNC:2329']},\n",
       "  'metabolites': {'crn_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'arachcrn_c': 1.0,\n",
       "   'arachcoa_c': -1.0},\n",
       "  'id': 'ARACHCPT1',\n",
       "  'gene_reaction_rule': 'HGNC:18540 or HGNC:2328 or HGNC:2329'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'carnitine acyltransferase II',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2330']},\n",
       "  'metabolites': {'arachcoa_m': 1.0,\n",
       "   'arachcrn_m': -1.0,\n",
       "   'coa_m': -1.0,\n",
       "   'crn_m': 1.0},\n",
       "  'id': 'ARACHCPT2',\n",
       "  'gene_reaction_rule': 'HGNC:2330'},\n",
       " {'subsystem': 'Arginine and Proline Metabolism',\n",
       "  'name': 'arginine decarboxylase (m)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB m'],\n",
       "   'GENE ASSOCIATION': ['HGNC:29957']},\n",
       "  'metabolites': {'arg_L_m': -1.0, 'h_m': -1.0, 'agm_m': 1.0, 'co2_m': 1.0},\n",
       "  'id': 'ARGDCm',\n",
       "  'gene_reaction_rule': 'HGNC:29957'},\n",
       " {'subsystem': 'Urea cycle',\n",
       "  'name': 'arginase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'GENE ASSOCIATION': ['HGNC:663 or HGNC:664']},\n",
       "  'metabolites': {'arg_L_c': -1.0, 'h2o_c': -1.0, 'urea_c': 1.0, 'orn_c': 1.0},\n",
       "  'id': 'ARGN',\n",
       "  'gene_reaction_rule': 'HGNC:663 or HGNC:664'},\n",
       " {'subsystem': 'Urea cycle',\n",
       "  'name': 'arginase (m)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB m'],\n",
       "   'GENE ASSOCIATION': ['HGNC:664']},\n",
       "  'metabolites': {'orn_m': 1.0, 'arg_L_m': -1.0, 'h2o_m': -1.0, 'urea_m': 1.0},\n",
       "  'id': 'ARGNm',\n",
       "  'gene_reaction_rule': 'HGNC:664'},\n",
       " {'subsystem': 'Alanine and aspartate metabolism',\n",
       "  'name': 'argininosuccinate lyase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:746']},\n",
       "  'metabolites': {'argsuc_c': -1.0, 'fum_c': 1.0, 'arg_L_c': 1.0},\n",
       "  'id': 'ARGSL',\n",
       "  'gene_reaction_rule': 'HGNC:746'},\n",
       " {'subsystem': 'Alanine and aspartate metabolism',\n",
       "  'name': 'argininosuccinate synthase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:758']},\n",
       "  'metabolites': {'asp_L_c': -1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'argsuc_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'citr_L_c': -1.0,\n",
       "   'amp_c': 1.0},\n",
       "  'id': 'ARGSS',\n",
       "  'gene_reaction_rule': 'HGNC:758'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'Arylsulfatase A',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:713']},\n",
       "  'metabolites': {'galside_hs_l': 1.0,\n",
       "   'h_l': 1.0,\n",
       "   'sgalside_hs_l': -1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   'so4_l': 1.0},\n",
       "  'id': 'ARSA',\n",
       "  'gene_reaction_rule': 'HGNC:713'},\n",
       " {'subsystem': 'R group synthesis',\n",
       "  'name': 'R group coenzyme a ligase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['HR fatty acid metabolism - R groups = acyl groups, Rtotal, Rtotal2, Rtotal3 correspond to three possible acyl groups added to glycerol Carbon stoichiometry normalized wrt C16 R1 - C16 groups R2 - essential FA + associated derivatives R3 - non-essential, dietary FA (C18) R4 - monounsaturated FA R5 -other (C26, C24) NJ'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'Rtotalcoa_c': -1.0,\n",
       "   'Rtotal_c': 1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'h_c': 5.0,\n",
       "   'h2o_c': -1.0},\n",
       "  'id': 'ARTCOAL3',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'name': 'R group phosphotase 3',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'Rtotalcoa_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'h_c': -4.0,\n",
       "   'Rtotal_c': -1.0,\n",
       "   'amp_c': 1.0},\n",
       "  'id': 'R_group_phosphotase_3',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'R group synthesis',\n",
       "  'name': 'R group artificial flux',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'], 'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'h_c': -4.0, 'arachcoa_c': -1.0, 'R2coa_hs_c': 1.0},\n",
       "  'id': 'ARTFR207',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'R group synthesis',\n",
       "  'name': 'R group artificial flux',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'], 'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'R3coa_hs_c': 1.0, 'stcoa_c': -1.0, 'h_c': -4.0},\n",
       "  'id': 'ARTFR31',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'R group synthesis',\n",
       "  'name': 'R group artificial flux',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'], 'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'ttccoa_c': -1.0, 'R5coa_hs_c': 1.0, 'h_c': -4.0},\n",
       "  'id': 'ARTFR51',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'R group synthesis',\n",
       "  'name': 'R group to palmitate conversion',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['cytosolic conversion of pooled FA into palmitate (so subsequent transport into mit w/ FA oxidation can occur). Since Carbon stoichiometry normalized wrt C16 ratio between Rgroupcoa and pmtcoa is 1:1 HR fatty acid metabolism - R groups = acyl groups, Rtotal, Rtotal2, Rtotal3 correspond to three possible acyl groups added to glycerol R1 - C16 groups R2 - essential FA + associated derivatives R3 - non-essential, dietary FA (C18) R4 - monounsaturated FA R5 -other (C26, C24) NJ'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'pmtcoa_c': 1.0, 'Rtotalcoa_c': -1.0, 'h_c': 4.0},\n",
       "  'id': 'ARTPLM3',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'R group synthesis',\n",
       "  'name': 'R group to palmitate conversion',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['cytosolic conversion of pooled FA into palmitate (so subsequent transport into mit w/ FA oxidation can occur). Since Carbon stoichiometry normalized wrt C16 ratio between Rgroupcoa and pmtcoa is 1:1 HR fatty acid metabolism - R groups = acyl groups, Rtotal, Rtotal2, Rtotal3 correspond to three possible acyl groups added to glycerol R1 - C16 groups R2 - essential FA + associated derivatives R3 - non-essential, dietary FA (C18) R4 - monounsaturated FA R5 -other (C26, C24) NJ'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'h_m': 4.0, 'Rtotalcoa_m': -1.0, 'pmtcoa_m': 1.0},\n",
       "  'id': 'ARTPLM3m',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'N-acylsphingosine amidohydrolase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:735']},\n",
       "  'metabolites': {'crm_hs_l': -1.0,\n",
       "   'sphings_l': 1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   'Rtotal_l': 1.0},\n",
       "  'id': 'ASAH1',\n",
       "  'gene_reaction_rule': 'HGNC:735'},\n",
       " {'subsystem': 'Vitamin C metabolism',\n",
       "  'name': 'ascorbic acid oxidase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'dhdascb_c': 1.0,\n",
       "   'o2s_c': -2.0,\n",
       "   'ascb_L_c': -1.0,\n",
       "   'h_c': -2.0,\n",
       "   'h2o2_c': 2.0},\n",
       "  'id': 'ASCBOX',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Alanine and aspartate metabolism',\n",
       "  'name': 'L-asparaginase (mitochondrial)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['PMID: 11984834 (denotes it as potentially mitochondrial)  mm'],\n",
       "   'GENE ASSOCIATION': ['HGNC:16448']},\n",
       "  'metabolites': {'asn_L_m': -1.0,\n",
       "   'nh4_m': 1.0,\n",
       "   'h2o_m': -1.0,\n",
       "   'asp_L_m': 1.0},\n",
       "  'id': 'ASNNm',\n",
       "  'gene_reaction_rule': 'HGNC:16448'},\n",
       " {'subsystem': 'Alanine and aspartate metabolism',\n",
       "  'name': 'asparagine synthase (glutamine-hydrolysing)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:753']},\n",
       "  'metabolites': {'asp_L_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'asn_L_c': 1.0,\n",
       "   'glu_L_c': 1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'gln_L_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   'amp_c': 1.0,\n",
       "   'atp_c': -1.0},\n",
       "  'id': 'ASNS1',\n",
       "  'gene_reaction_rule': 'HGNC:753'},\n",
       " {'subsystem': 'beta-Alanine metabolism',\n",
       "  'name': 'aspartate 1-decarboxylase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['The E. coli enzyme contains a pyruvoyl group.'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4092']},\n",
       "  'metabolites': {'asp_L_c': -1.0, 'co2_c': 1.0, 'h_c': -1.0, 'ala_B_c': 1.0},\n",
       "  'id': 'ASP1DC',\n",
       "  'gene_reaction_rule': 'HGNC:4092'},\n",
       " {'subsystem': 'Pyrimidine synthesis',\n",
       "  'name': 'aspartate carbamoyltransferase (reversible)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:1424']},\n",
       "  'metabolites': {'asp_L_c': -1.0,\n",
       "   'cbasp_c': 1.0,\n",
       "   'cbp_c': -1.0,\n",
       "   'pi_c': 1.0,\n",
       "   'h_c': 1.0},\n",
       "  'id': 'ASPCTr',\n",
       "  'gene_reaction_rule': 'HGNC:1424'},\n",
       " {'subsystem': 'Alanine and aspartate metabolism',\n",
       "  'name': 'aspartate N-acetyltransferase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['MM'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'accoa_m': -1.0,\n",
       "   'coa_m': 1.0,\n",
       "   'h_m': 1.0,\n",
       "   'asp_L_m': -1.0,\n",
       "   'Nacasp_m': 1.0},\n",
       "  'id': 'ASPNATm',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Alanine and aspartate metabolism',\n",
       "  'name': 'aspartate transaminase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:4432']},\n",
       "  'metabolites': {'asp_L_c': -1.0,\n",
       "   'akg_c': -1.0,\n",
       "   'glu_L_c': 1.0,\n",
       "   'oaa_c': 1.0},\n",
       "  'id': 'ASPTA',\n",
       "  'gene_reaction_rule': 'HGNC:4432'},\n",
       " {'subsystem': 'Alanine and aspartate metabolism',\n",
       "  'name': 'aspartate transaminase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4433']},\n",
       "  'metabolites': {'glu_L_m': 1.0,\n",
       "   'oaa_m': 1.0,\n",
       "   'asp_L_m': -1.0,\n",
       "   'akg_m': -1.0},\n",
       "  'id': 'ASPTAm',\n",
       "  'gene_reaction_rule': 'HGNC:4433'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'ATP diphosphohydrolase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3363 or HGNC:3364 or HGNC:3365 or HGNC:3367 or HGNC:3368']},\n",
       "  'metabolites': {'atp_e': -1.0,\n",
       "   'h2o_e': -2.0,\n",
       "   'amp_e': 1.0,\n",
       "   'h_e': 2.0,\n",
       "   'pi_e': 2.0},\n",
       "  'id': 'ATPH1e',\n",
       "  'gene_reaction_rule': 'HGNC:3363 or HGNC:3364 or HGNC:3365 or HGNC:3367 or HGNC:3368'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'ATP diphosphohydrolase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:24860 or HGNC:3363 or HGNC:3364 or HGNC:3365 or HGNC:3367 or HGNC:3368']},\n",
       "  'metabolites': {'adp_e': -1.0,\n",
       "   'h2o_e': -1.0,\n",
       "   'amp_e': 1.0,\n",
       "   'h_e': 1.0,\n",
       "   'pi_e': 1.0},\n",
       "  'id': 'ATPH2e',\n",
       "  'gene_reaction_rule': 'HGNC:24860 or HGNC:3363 or HGNC:3364 or HGNC:3365 or HGNC:3367 or HGNC:3368'},\n",
       " {'subsystem': 'N-glycan degradation',\n",
       "  'name': 'beta-mannosidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:6831']},\n",
       "  'metabolites': {'man_l': 1.0, 'mn_l': -1.0, 'h2o_l': -1.0, 'acgam_l': 1.0},\n",
       "  'id': 'B_MANNASEly',\n",
       "  'gene_reaction_rule': 'HGNC:6831'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'Beta-1,3-galactosyltransferase 3',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:918']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'udpacgal_g': -1.0,\n",
       "   'thcrm_hs_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'gbside_hs_g': 1.0},\n",
       "  'id': 'B3GALT3g',\n",
       "  'gene_reaction_rule': 'HGNC:918'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'Beta-1,3-galactosyltransferase 4',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:919']},\n",
       "  'metabolites': {'ga2_hs_g': -1.0,\n",
       "   'h_g': 1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'ga1_hs_g': 1.0},\n",
       "  'id': 'B3GALT41g',\n",
       "  'gene_reaction_rule': 'HGNC:919'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'Beta-1,3-galactosyltransferase 4',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:919']},\n",
       "  'metabolites': {'gm1_hs_g': 1.0,\n",
       "   'gm2_hs_g': -1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0},\n",
       "  'id': 'B3GALT42g',\n",
       "  'gene_reaction_rule': 'HGNC:919'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'Beta-1,3-galactosyltransferase 4',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:919']},\n",
       "  'metabolites': {'gd2_hs_g': -1.0,\n",
       "   'gd1b_hs_g': 1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'h_g': 3.0},\n",
       "  'id': 'B3GALT43g',\n",
       "  'gene_reaction_rule': 'HGNC:919'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'Beta-1,3-galactosyltransferase 4',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:919']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'gt1c_hs_g': 1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'gt2_hs_g': -1.0},\n",
       "  'id': 'B3GALT44g',\n",
       "  'gene_reaction_rule': 'HGNC:919'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'Beta-1,3-galactosyltransferase 5',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:920']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'galgbside_hs_g': 1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'gbside_hs_g': -1.0},\n",
       "  'id': 'B3GALT5g',\n",
       "  'gene_reaction_rule': 'HGNC:920'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'Beta galactosyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['general reaction - no gene association get NJ'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'galacglcgalgluside_hs_g': 1.0,\n",
       "   'h_g': 1.0,\n",
       "   'acglcgalgluside_hs_g': -1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0},\n",
       "  'id': 'B3GALTg',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'UDP-GlcNAc:betaGal beta-1,3-N-acetylglucosaminyltransferase 1',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:15629']},\n",
       "  'metabolites': {'gal14acglcgalgluside_hs_g': -1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'acglcgal14acglcgalgluside_hs_g': 1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'B3GNT11g',\n",
       "  'gene_reaction_rule': 'HGNC:15629'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'UDP-GlcNAc:betaGal beta-1,3-N-acetylglucosaminyltransferase 1',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:15629']},\n",
       "  'metabolites': {'acglc13galacglcgal14acglcgalgluside_hs_g': 1.0,\n",
       "   'galacglcgal14acglcgalgluside_hs_g': -1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'B3GNT12g',\n",
       "  'gene_reaction_rule': 'HGNC:15629'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'UDP-GlcNAc:betaGal beta-1,3-N-acetylglucosaminyltransferase 3, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:13528']},\n",
       "  'metabolites': {'gal14acglcgalgluside_hs_g': 1.0,\n",
       "   'acglcgalgluside_hs_g': -1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'B3GNT310g',\n",
       "  'gene_reaction_rule': 'HGNC:13528'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'UDP-GlcNAc:betaGal beta-1,3-N-acetylglucosaminyltransferase 3, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:13528']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'acgalfucgalacglcgalgluside_hs_g': -1.0,\n",
       "   'galacgalfuc12gal14acglcgalgluside_hs_g': 1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0},\n",
       "  'id': 'B3GNT311g',\n",
       "  'gene_reaction_rule': 'HGNC:13528'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'UDP-GlcNAc:betaGal beta-1,3-N-acetylglucosaminyltransferase 3, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:13528']},\n",
       "  'metabolites': {'udp_g': 1.0,\n",
       "   'galacglcgal14acglcgalgluside_hs_g': 1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'acglcgal14acglcgalgluside_hs_g': -1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'B3GNT312g',\n",
       "  'gene_reaction_rule': 'HGNC:13528'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'UDP-GlcNAc:betaGal beta-1,3-N-acetylglucosaminyltransferase 3, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:13528']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'galacgalfucgalacglcgal14acglcgalgluside_hs_g': 1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'acgalfucgalacglcgal14acglcgalgluside_hs_g': -1.0},\n",
       "  'id': 'B3GNT313g',\n",
       "  'gene_reaction_rule': 'HGNC:13528'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'UDP-GlcNAc:betaGal beta-1,3-N-acetylglucosaminyltransferase 3, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:13528']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'acglcgalacglcgal14acglcgalgluside_hs_g': -1.0,\n",
       "   'galacglcgalacglcgal14acglcgalgluside_hs_g': 1.0},\n",
       "  'id': 'B3GNT314g',\n",
       "  'gene_reaction_rule': 'HGNC:13528'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'UDP-GlcNAc:betaGal beta-1,3-N-acetylglucosaminyltransferase 3, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:13528']},\n",
       "  'metabolites': {'acglc13galacglcgal14acglcgalgluside_hs_g': -1.0,\n",
       "   'h_g': 1.0,\n",
       "   'galacglc13galacglcgal14acglcgalgluside_hs_g': 1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0},\n",
       "  'id': 'B3GNT315g',\n",
       "  'gene_reaction_rule': 'HGNC:13528'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'UDP-GlcNAc:betaGal beta-1,3-N-acetylglucosaminyltransferase 3, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:13528']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'galthcrm_hs_g': 1.0,\n",
       "   'thcrm_hs_g': -1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0},\n",
       "  'id': 'B3GNT31g',\n",
       "  'gene_reaction_rule': 'HGNC:13528'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'UDP-GlcNAc:betaGal beta-1,3-N-acetylglucosaminyltransferase 3, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:13528']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'galgalthcrm_hs_g': 1.0,\n",
       "   'galthcrm_hs_g': -1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0},\n",
       "  'id': 'B3GNT32g',\n",
       "  'gene_reaction_rule': 'HGNC:13528'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'UDP-GlcNAc:betaGal beta-1,3-N-acetylglucosaminyltransferase 3, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:13528']},\n",
       "  'metabolites': {'udp_g': 1.0,\n",
       "   'galgalgalthcrm_hs_g': 1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'galgalthcrm_hs_g': -1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'B3GNT33g',\n",
       "  'gene_reaction_rule': 'HGNC:13528'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'UDP-GlcNAc:betaGal beta-1,3-N-acetylglucosaminyltransferase 3, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:13528']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'galgbside_hs_g': -1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'acglcgalgbside_hs_g': 1.0},\n",
       "  'id': 'B3GNT34g',\n",
       "  'gene_reaction_rule': 'HGNC:13528'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'UDP-GlcNAc:betaGal beta-1,3-N-acetylglucosaminyltransferase 3, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:13528']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'galacglcgalgbside_hs_g': 1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'acglcgalgbside_hs_g': -1.0},\n",
       "  'id': 'B3GNT35g',\n",
       "  'gene_reaction_rule': 'HGNC:13528'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'UDP-GlcNAc:betaGal beta-1,3-N-acetylglucosaminyltransferase 3, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:13528']},\n",
       "  'metabolites': {'acgbgbside_hs_g': 1.0,\n",
       "   'acgal_g': -1.0,\n",
       "   'h2o_g': 1.0,\n",
       "   'gbside_hs_g': -1.0},\n",
       "  'id': 'B3GNT36g',\n",
       "  'gene_reaction_rule': 'HGNC:13528'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'UDP-GlcNAc:betaGal beta-1,3-N-acetylglucosaminyltransferase 3, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:13528']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'gm2a_hs_g': -1.0,\n",
       "   'gm1a_hs_g': 1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'udpgal_g': -1.0},\n",
       "  'id': 'B3GNT37g',\n",
       "  'gene_reaction_rule': 'HGNC:13528'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'UDP-GlcNAc:betaGal beta-1,3-N-acetylglucosaminyltransferase 3, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:13528']},\n",
       "  'metabolites': {'galgluside_hs_g': 1.0,\n",
       "   'h_g': 1.0,\n",
       "   'gluside_hs_g': -1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0},\n",
       "  'id': 'B3GNT39g',\n",
       "  'gene_reaction_rule': 'HGNC:13528'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'UDP-GlcNAc:betaGal beta-1,3-N-acetylglucosaminyltransferase 5',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:15684']},\n",
       "  'metabolites': {'galgluside_hs_g': -1.0,\n",
       "   'h_g': 1.0,\n",
       "   'acglcgalgluside_hs_g': 1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'udp_g': 1.0},\n",
       "  'id': 'B3GNT51g',\n",
       "  'gene_reaction_rule': 'HGNC:15684'},\n",
       " {'subsystem': 'Bile acid synthesis',\n",
       "  'name': 'bile acid Coenzyme A: amino acid N-acyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['glycine or taurine as substrates, produces glycocholic or taurocholic acids respectively NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:932']},\n",
       "  'metabolites': {'gchola_x': 1.0,\n",
       "   'cholcoa_x': -1.0,\n",
       "   'gly_x': -1.0,\n",
       "   'coa_x': 1.0},\n",
       "  'id': 'BAAT1x',\n",
       "  'gene_reaction_rule': 'HGNC:932'},\n",
       " {'subsystem': 'Bile acid synthesis',\n",
       "  'name': 'bile acid Coenzyme A: amino acid N-acyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['glycine or taurine as substrates, produces glycocholic or taurocholic acids respectively NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:932']},\n",
       "  'metabolites': {'tchola_x': 1.0,\n",
       "   'cholcoa_x': -1.0,\n",
       "   'coa_x': 1.0,\n",
       "   'taur_x': -1.0},\n",
       "  'id': 'BAAT2x',\n",
       "  'gene_reaction_rule': 'HGNC:932'},\n",
       " {'subsystem': 'Bile acid synthesis',\n",
       "  'name': 'bile acid Coenzyme A: amino acid N-acyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['fourth version analog added in similarity to BAAT1x, BAAT2x NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:932']},\n",
       "  'metabolites': {'dgchol_x': 1.0,\n",
       "   'dcholcoa_x': -1.0,\n",
       "   'gly_x': -1.0,\n",
       "   'coa_x': 1.0},\n",
       "  'id': 'BAAT4x',\n",
       "  'gene_reaction_rule': 'HGNC:932'},\n",
       " {'subsystem': 'Biotin metabolism',\n",
       "  'name': 'biotin-[acetyl-CoA-carboxylase] ligase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4976']},\n",
       "  'metabolites': {'btamp_c': 1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'btn_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'atp_c': -1.0},\n",
       "  'id': 'BACCL',\n",
       "  'gene_reaction_rule': 'HGNC:4976'},\n",
       " {'subsystem': 'Biotin metabolism',\n",
       "  'name': 'biotin-[acetyl-CoA-carboxylase] ligase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4976']},\n",
       "  'metabolites': {'atp_m': -1.0,\n",
       "   'h_m': -1.0,\n",
       "   'ppi_m': 1.0,\n",
       "   'btn_m': -1.0,\n",
       "   'btamp_m': 1.0},\n",
       "  'id': 'BACCLm',\n",
       "  'gene_reaction_rule': 'HGNC:4976'},\n",
       " {'subsystem': 'beta-Alanine metabolism',\n",
       "  'name': 'beta-Aminopropion aldehyde:NAD+ oxidoreductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:15472 or HGNC:402 or HGNC:403 or HGNC:412 or HGNC:877']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   'ala_B_c': 1.0,\n",
       "   'h_c': 2.0,\n",
       "   'nad_c': -1.0,\n",
       "   'nadh_c': 1.0,\n",
       "   'bamppald_c': -1.0},\n",
       "  'id': 'BAMPPALDOX',\n",
       "  'gene_reaction_rule': 'HGNC:15472 or HGNC:402 or HGNC:403 or HGNC:412 or HGNC:877'},\n",
       " {'subsystem': 'beta-Alanine metabolism',\n",
       "  'name': 'beta-Aminopropion aldehyde:NAD+ oxidoreductase (m)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:404 or HGNC:406 or HGNC:407']},\n",
       "  'metabolites': {'h2o_m': -1.0,\n",
       "   'ala_B_m': 1.0,\n",
       "   'bamppald_m': -1.0,\n",
       "   'nadh_m': 1.0,\n",
       "   'nad_m': -1.0,\n",
       "   'h_m': 2.0},\n",
       "  'id': 'BAMPPALDOXm',\n",
       "  'gene_reaction_rule': 'HGNC:404 or HGNC:406 or HGNC:407'},\n",
       " {'subsystem': 'Lysine metabolism',\n",
       "  'name': '4-Trimethylammoniobutanoate,2-oxoglutarate:oxygen oxidoreductase (3-hydroxylating)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['MM'],\n",
       "   'GENE ASSOCIATION': ['HGNC:964']},\n",
       "  'metabolites': {'akg_c': -1.0,\n",
       "   'o2_c': -1.0,\n",
       "   'co2_c': 1.0,\n",
       "   'succ_c': 1.0,\n",
       "   '4tmeabutn_c': -1.0,\n",
       "   'crn_c': 1.0},\n",
       "  'id': 'BBHOX',\n",
       "  'gene_reaction_rule': 'HGNC:964'},\n",
       " {'subsystem': 'Vitamin A metabolism',\n",
       "  'name': 'beta-Carotene dioxygenase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:13815']},\n",
       "  'metabolites': {'retinal_c': 2.0, 'o2_c': -1.0, 'caro_c': -1.0},\n",
       "  'id': 'BCDO',\n",
       "  'gene_reaction_rule': 'HGNC:13815'},\n",
       " {'subsystem': 'Stilbene, coumarine and lignin synthesis',\n",
       "  'name': 'beta-D-Glucosyl-2-coumarinate glucohydrolase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'bdg2hc_c': -1.0,\n",
       "   'glc_D_c': 1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   '2coum_c': 1.0},\n",
       "  'id': 'BDG2HCGHD',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Butanoate metabolism',\n",
       "  'name': '(R)-3-Hydroxybutanoate:NAD+ oxidoreductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1027']},\n",
       "  'metabolites': {'nad_m': -1.0,\n",
       "   'h_m': 1.0,\n",
       "   'nadh_m': 1.0,\n",
       "   'acac_m': 1.0,\n",
       "   'bhb_m': -1.0},\n",
       "  'id': 'BDHm',\n",
       "  'gene_reaction_rule': 'HGNC:1027'},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'GDPmannose:chitobiosyldiphosphodolichol beta-D-mannosyltransferase (uterus)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'gdp_c': 1.0,\n",
       "   'gdpmann_c': -1.0,\n",
       "   'chito2pdol_U_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   'mpdol_U_c': 1.0},\n",
       "  'id': 'BDMT_U',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Glycine, serine, alanine and threonine metabolism',\n",
       "  'name': 'betaine-aldehyde dehydrogenase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['MM'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'glyb_m': 1.0,\n",
       "   'h2o_m': -1.0,\n",
       "   'nadh_m': 1.0,\n",
       "   'nad_m': -1.0,\n",
       "   'betald_m': -1.0,\n",
       "   'h_m': 2.0},\n",
       "  'id': 'BETALDHxm',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Glycine, serine, alanine and threonine metabolism',\n",
       "  'name': 'betaine-homocysteine S-methyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1047 or HGNC:1048']},\n",
       "  'metabolites': {'dmgly_c': 1.0,\n",
       "   'glyb_c': -1.0,\n",
       "   'met_L_c': 1.0,\n",
       "   'hcys_L_c': -1.0},\n",
       "  'id': 'BHMT',\n",
       "  'gene_reaction_rule': 'HGNC:1047 or HGNC:1048'},\n",
       " {'subsystem': 'Heme degradation',\n",
       "  'name': 'Nad(p)h biliverdin reductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['TDV'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1062 or HGNC:1063']},\n",
       "  'metabolites': {'biliverd_c': -1.0,\n",
       "   'nadp_c': 1.0,\n",
       "   'nadph_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'bilirub_c': 1.0},\n",
       "  'id': 'BILIRED',\n",
       "  'gene_reaction_rule': 'HGNC:1062 or HGNC:1063'},\n",
       " {'subsystem': 'Phosphatidylinositol phosphate metabolism',\n",
       "  'name': 'B mannosyltransferase, endoplasmic reticulum',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'memgacpail_hs_r': -1.0,\n",
       "   'dolp_U_r': 1.0,\n",
       "   'h_r': 2.0,\n",
       "   'dolmanp_U_r': -1.0,\n",
       "   'm2emgacpail_hs_r': 1.0},\n",
       "  'id': 'BMTer_U',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': '3,5-bisphosphate nucleotidase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:1096']},\n",
       "  'metabolites': {'h2o_c': -1.0, 'pi_c': 1.0, 'pap_c': -1.0, 'amp_c': 1.0},\n",
       "  'id': 'BPNT',\n",
       "  'gene_reaction_rule': 'HGNC:1096'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': '3,5-bisphosphate nucleotidase (paps)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:1096']},\n",
       "  'metabolites': {'h2o_c': -1.0, 'pi_c': 1.0, 'aps_c': 1.0, 'paps_c': -1.0},\n",
       "  'id': 'BPNT2',\n",
       "  'gene_reaction_rule': 'HGNC:1096'},\n",
       " {'subsystem': 'Biotin metabolism',\n",
       "  'name': 'biotinidase (biotin)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1122']},\n",
       "  'metabolites': {'biocyt_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'btn_c': 1.0,\n",
       "   'lys_L_c': 1.0},\n",
       "  'id': 'BTND1',\n",
       "  'gene_reaction_rule': 'HGNC:1122'},\n",
       " {'subsystem': 'Biotin metabolism',\n",
       "  'name': 'biotinidase (biotin), nuclear',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1122']},\n",
       "  'metabolites': {'biocyt_n': -1.0,\n",
       "   'h2o_n': -1.0,\n",
       "   'btn_n': 1.0,\n",
       "   'lys_L_n': 1.0},\n",
       "  'id': 'BTND1n',\n",
       "  'gene_reaction_rule': 'HGNC:1122'},\n",
       " {'subsystem': 'Biotin metabolism',\n",
       "  'name': 'biotinidase (biotin), extracellular',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Also acts on biotin esters IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1122']},\n",
       "  'metabolites': {'lys_L_e': 1.0,\n",
       "   'h2o_e': -1.0,\n",
       "   'biocyt_e': -1.0,\n",
       "   'btn_e': 1.0},\n",
       "  'id': 'BTNDe',\n",
       "  'gene_reaction_rule': 'HGNC:1122'},\n",
       " {'subsystem': 'Biotin metabolism',\n",
       "  'name': 'biotinidase (biotin), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Also acts on biotin esters IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1122']},\n",
       "  'metabolites': {'biocyt_m': -1.0,\n",
       "   'h2o_m': -1.0,\n",
       "   'lys_L_m': 1.0,\n",
       "   'btn_m': 1.0},\n",
       "  'id': 'BTNDm',\n",
       "  'gene_reaction_rule': 'HGNC:1122'},\n",
       " {'subsystem': 'Biotin metabolism',\n",
       "  'name': 'holocarboxylase synthestase (biotin protein ligase)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT for human biotin metabolism'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4976']},\n",
       "  'metabolites': {'btamp_c': -1.0,\n",
       "   'apoC_Lys_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   'apoC_Lys_btn_c': 1.0,\n",
       "   'amp_c': 1.0},\n",
       "  'id': 'BTNPL',\n",
       "  'gene_reaction_rule': 'HGNC:4976'},\n",
       " {'subsystem': 'Biotin metabolism',\n",
       "  'name': 'holocarboxylase synthestase (biotin protein ligase), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT for human biotin metabolism'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4976']},\n",
       "  'metabolites': {'apoC_Lys_btn_m': 1.0,\n",
       "   'apoC_Lys_m': -1.0,\n",
       "   'amp_m': 1.0,\n",
       "   'h_m': 1.0,\n",
       "   'btamp_m': -1.0},\n",
       "  'id': 'BTNPLm',\n",
       "  'gene_reaction_rule': 'HGNC:4976'},\n",
       " {'subsystem': 'Pyrimidine catabolism',\n",
       "  'name': 'beta-ureidopropionase (D-3-amino-isobutanoate forming)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['MM'],\n",
       "   'GENE ASSOCIATION': ['HGNC:16297']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   '3aib_D_c': 1.0,\n",
       "   'h_c': -2.0,\n",
       "   'co2_c': 1.0,\n",
       "   'nh4_c': 1.0,\n",
       "   '3uib_c': -1.0},\n",
       "  'id': 'BUP2',\n",
       "  'gene_reaction_rule': 'HGNC:16297'},\n",
       " {'subsystem': 'Cholesterol metabolism',\n",
       "  'name': 'C-14 sterol reductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD ER version NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:11863']},\n",
       "  'metabolites': {'44mzym_r': 1.0,\n",
       "   'nadp_r': 1.0,\n",
       "   'h_r': -1.0,\n",
       "   'nadph_r': -1.0,\n",
       "   '44mctr_r': -1.0},\n",
       "  'id': 'C14STRr',\n",
       "  'gene_reaction_rule': 'HGNC:11863'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'carnitine acyltransferase I',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'GENE ASSOCIATION': ['HGNC:18540 or HGNC:2328 or HGNC:2329']},\n",
       "  'metabolites': {'pmtcrn_c': 1.0,\n",
       "   'pmtcoa_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'crn_c': -1.0},\n",
       "  'id': 'C160CPT1',\n",
       "  'gene_reaction_rule': 'HGNC:18540 or HGNC:2328 or HGNC:2329'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'carnitine acyltransferase II',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['TV'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2330']},\n",
       "  'metabolites': {'coa_m': -1.0,\n",
       "   'crn_m': 1.0,\n",
       "   'pmtcoa_m': 1.0,\n",
       "   'pmtcrn_m': -1.0},\n",
       "  'id': 'C160CPT2',\n",
       "  'gene_reaction_rule': 'HGNC:2330'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'production of palmitoleoylcarnitine',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['TDV'],\n",
       "   'GENE ASSOCIATION': ['HGNC:18540 or HGNC:2328 or HGNC:2329']},\n",
       "  'metabolites': {'crn_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'hdcoa_c': -1.0,\n",
       "   'hdcecrn_c': 1.0},\n",
       "  'id': 'C161CPT1',\n",
       "  'gene_reaction_rule': 'HGNC:18540 or HGNC:2328 or HGNC:2329'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'carnitine O-palmitoyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:18540 or HGNC:2328 or HGNC:2329']},\n",
       "  'metabolites': {'crn_c': -1.0,\n",
       "   'hdcoa_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'hdd2crn_c': 1.0},\n",
       "  'id': 'C161CPT12',\n",
       "  'gene_reaction_rule': 'HGNC:18540 or HGNC:2328 or HGNC:2329'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'transport of Hexadecenoyl-CoA into mitochondrial matrix',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['TDV'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2330']},\n",
       "  'metabolites': {'hdcecrn_m': -1.0,\n",
       "   'hdcoa_m': 1.0,\n",
       "   'coa_m': -1.0,\n",
       "   'crn_m': 1.0},\n",
       "  'id': 'C161CPT2',\n",
       "  'gene_reaction_rule': 'HGNC:2330'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'C161 transport into the mitochondria',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2330']},\n",
       "  'metabolites': {'hdd2crn_m': -1.0,\n",
       "   'coa_m': -1.0,\n",
       "   'hdcoa_m': 1.0,\n",
       "   'crn_m': 1.0},\n",
       "  'id': 'C161CPT22',\n",
       "  'gene_reaction_rule': 'HGNC:2330'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'carnitine acyltransferase I',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['TV (see C160CRN)'],\n",
       "   'GENE ASSOCIATION': ['HGNC:18540 or HGNC:2328 or HGNC:2329']},\n",
       "  'metabolites': {'crn_c': -1.0,\n",
       "   'stcoa_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'stcrn_c': 1.0},\n",
       "  'id': 'C180CPT1',\n",
       "  'gene_reaction_rule': 'HGNC:18540 or HGNC:2328 or HGNC:2329'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'carnitine acyltransferase II',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['TV'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2330']},\n",
       "  'metabolites': {'coa_m': -1.0,\n",
       "   'stcrn_m': -1.0,\n",
       "   'crn_m': 1.0,\n",
       "   'stcoa_m': 1.0},\n",
       "  'id': 'C180CPT2',\n",
       "  'gene_reaction_rule': 'HGNC:2330'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'transport of Octadecenoyl-CoA into mitochondrial matrix',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['TDV'],\n",
       "   'GENE ASSOCIATION': ['HGNC:18540 or HGNC:2329 or HGNC:2328 or HGNC:2329 or HGNC:2329 or HGNC:2329']},\n",
       "  'metabolites': {'crn_c': -1.0,\n",
       "   'octdececoa_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'elaidcrn_c': 1.0},\n",
       "  'id': 'C181CPT1',\n",
       "  'gene_reaction_rule': 'HGNC:18540 or HGNC:2329 or HGNC:2328 or HGNC:2329 or HGNC:2329 or HGNC:2329'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'transport of Octadecenoyl-CoA into mitochondrial matrix',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['TDV'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2330']},\n",
       "  'metabolites': {'elaidcrn_m': -1.0,\n",
       "   'coa_m': -1.0,\n",
       "   'crn_m': 1.0,\n",
       "   'octdececoa_m': 1.0},\n",
       "  'id': 'C181CPT2',\n",
       "  'gene_reaction_rule': 'HGNC:2330'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'carnitine C20:4 transferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['TDV'],\n",
       "   'GENE ASSOCIATION': ['HGNC:18540 or HGNC:2328 or HGNC:2329']},\n",
       "  'metabolites': {'crn_c': -1.0,\n",
       "   'arachdcoa_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'arachdcrn_c': 1.0},\n",
       "  'id': 'C204CPT1',\n",
       "  'gene_reaction_rule': 'HGNC:18540 or HGNC:2328 or HGNC:2329'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'arachidonic acid transport into the mitochondria',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['TDV C20:4 transport'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2330']},\n",
       "  'metabolites': {'coa_m': -1.0,\n",
       "   'arachdcrn_m': -1.0,\n",
       "   'crn_m': 1.0,\n",
       "   'arachdcoa_m': 1.0},\n",
       "  'id': 'C204CPT2',\n",
       "  'gene_reaction_rule': 'HGNC:2330'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'carnitine C22:6 transferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['TDV'],\n",
       "   'GENE ASSOCIATION': ['HGNC:18540 or HGNC:2328 or HGNC:2329']},\n",
       "  'metabolites': {'crn_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'c226coa_c': -1.0,\n",
       "   'c226crn_c': 1.0},\n",
       "  'id': 'C226CPT1',\n",
       "  'gene_reaction_rule': 'HGNC:18540 or HGNC:2328 or HGNC:2329'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'C226 transport into the mitochondria',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['TDV'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2330']},\n",
       "  'metabolites': {'coa_m': -1.0,\n",
       "   'c226coa_m': 1.0,\n",
       "   'crn_m': 1.0,\n",
       "   'c226crn_m': -1.0},\n",
       "  'id': 'C226CPT2',\n",
       "  'gene_reaction_rule': 'HGNC:2330'},\n",
       " {'subsystem': 'Limonene and pinene degradation',\n",
       "  'name': 'cis-2-Methyl-5-isopropylhexa-2,5-dienoyl-CoA hydro-lyase (m)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:4801 and HGNC:4803) or HGNC:890']},\n",
       "  'metabolites': {'3h26dm5coa_m': 1.0, 'c2m26dcoa_m': -1.0, 'h2o_m': -1.0},\n",
       "  'id': 'C2M26DCOAHLm',\n",
       "  'gene_reaction_rule': '(HGNC:4801 and HGNC:4803) or HGNC:890'},\n",
       " {'subsystem': 'Limonene and pinene degradation',\n",
       "  'name': 'cis-2-Methyl-5-isopropylhexa-2,5-dienoyl-CoA hydro-lyase (x)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3149 or HGNC:3247']},\n",
       "  'metabolites': {'c2m26dcoa_x': -1.0, '3h26dm5coa_x': 1.0, 'h2o_x': -1.0},\n",
       "  'id': 'C2M26DCOAHLx',\n",
       "  'gene_reaction_rule': 'HGNC:3149 or HGNC:3247'},\n",
       " {'subsystem': 'Cholesterol metabolism',\n",
       "  'name': 'C-3 sterol dehydrogenase (4-methylzymosterol)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD ER version NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:13398']},\n",
       "  'metabolites': {'h_r': 1.0,\n",
       "   '4mzym_int2_r': 1.0,\n",
       "   'nadp_r': -1.0,\n",
       "   '4mzym_int1_r': -1.0,\n",
       "   'co2_r': 1.0,\n",
       "   'nadph_r': 1.0},\n",
       "  'id': 'C3STDH1Pr',\n",
       "  'gene_reaction_rule': 'HGNC:13398'},\n",
       " {'subsystem': 'Cholesterol metabolism',\n",
       "  'name': 'C-3 sterol dehydrogenase (4-methylzymosterol)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD ER version NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:13398']},\n",
       "  'metabolites': {'h_r': 1.0,\n",
       "   'co2_r': 1.0,\n",
       "   'nad_r': -1.0,\n",
       "   'nadh_r': 1.0,\n",
       "   '4mzym_int1_r': -1.0,\n",
       "   '4mzym_int2_r': 1.0},\n",
       "  'id': 'C3STDH1r',\n",
       "  'gene_reaction_rule': 'HGNC:13398'},\n",
       " {'subsystem': 'Cholesterol metabolism',\n",
       "  'name': 'C-3 sterol keto reductase (zymosterol)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD ER - NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:5213']},\n",
       "  'metabolites': {'zym_int2_r': -1.0,\n",
       "   'nadp_r': 1.0,\n",
       "   'h_r': -1.0,\n",
       "   'zymst_r': 1.0,\n",
       "   'nadph_r': -1.0},\n",
       "  'id': 'C3STKR2r',\n",
       "  'gene_reaction_rule': 'HGNC:5213'},\n",
       " {'subsystem': 'Cholesterol metabolism',\n",
       "  'name': 'C-4 sterol methyl oxidase (4,4-dimethylzymosterol)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['EC 1.-.-.- NCD ER version NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:10545']},\n",
       "  'metabolites': {'h_r': -3.0,\n",
       "   '44mzym_r': -1.0,\n",
       "   'o2_r': -3.0,\n",
       "   'nadp_r': 3.0,\n",
       "   'h2o_r': 4.0,\n",
       "   'nadph_r': -3.0,\n",
       "   '4mzym_int1_r': 1.0},\n",
       "  'id': 'C4STMO1r',\n",
       "  'gene_reaction_rule': 'HGNC:10545'},\n",
       " {'subsystem': 'Cholesterol metabolism',\n",
       "  'name': 'C-4 methyl sterol oxidase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['double check rxn specifics - see lit ref: Marijanovic,  Molecular Endocrinology 2003 17:1715-1725 ER version NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:13398']},\n",
       "  'metabolites': {'h_r': 1.0,\n",
       "   '4mzym_int2_r': -1.0,\n",
       "   'o2_r': -1.0,\n",
       "   'nadp_r': -1.0,\n",
       "   'nadph_r': 1.0,\n",
       "   'zym_int2_r': 1.0,\n",
       "   'co2_r': 1.0},\n",
       "  'id': 'C4STMO2Pr',\n",
       "  'gene_reaction_rule': 'HGNC:13398'},\n",
       " {'subsystem': 'Cholesterol metabolism',\n",
       "  'name': 'C-4 methyl sterol oxidase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['double check rxn specifics - see lit ref: Marijanovic,  Molecular Endocrinology 2003 17:1715-1725 ER version NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:10545 or HGNC:13398']},\n",
       "  'metabolites': {'h_r': 1.0,\n",
       "   'nad_r': -1.0,\n",
       "   '4mzym_int2_r': -1.0,\n",
       "   'o2_r': -1.0,\n",
       "   'nadh_r': 1.0,\n",
       "   'zym_int2_r': 1.0,\n",
       "   'co2_r': 1.0},\n",
       "  'id': 'C4STMO2r',\n",
       "  'gene_reaction_rule': 'HGNC:10545 or HGNC:13398'},\n",
       " {'subsystem': 'Glycolysis/gluconeogenesis',\n",
       "  'name': 'catalase A, peroxisomal (ethanol)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1516']},\n",
       "  'metabolites': {'h2o_x': 2.0,\n",
       "   'h2o2_x': -1.0,\n",
       "   'acald_x': 1.0,\n",
       "   'etoh_x': -1.0},\n",
       "  'id': 'CAT2p',\n",
       "  'gene_reaction_rule': 'HGNC:1516'},\n",
       " {'subsystem': 'ROS detoxification',\n",
       "  'name': 'catalase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['TV'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1516']},\n",
       "  'metabolites': {'o2_m': 1.0, 'h2o_m': 2.0, 'h2o2_m': -2.0},\n",
       "  'id': 'CATm',\n",
       "  'gene_reaction_rule': 'HGNC:1516'},\n",
       " {'subsystem': 'Miscellaneous',\n",
       "  'name': 'catalase A, peroxisomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1516']},\n",
       "  'metabolites': {'h2o_x': 2.0, 'h2o2_x': -2.0, 'o2_x': 1.0},\n",
       "  'id': 'CATp',\n",
       "  'gene_reaction_rule': 'HGNC:1516'},\n",
       " {'subsystem': 'Vitamin B12 metabolism',\n",
       "  'name': 'NADH:cob(III)alamine oxidoreductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'h2o_c': 2.0,\n",
       "   'cbl2_c': 2.0,\n",
       "   'aqcobal_c': -2.0,\n",
       "   'h_c': -1.0,\n",
       "   'nad_c': 1.0,\n",
       "   'nadh_c': -1.0},\n",
       "  'id': 'CBL2OR',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Vitamin B12 metabolism',\n",
       "  'name': 'cob(I)alamin adenosyltransferase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['JLR (EC 2.5.1.17) IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:18871']},\n",
       "  'metabolites': {'atp_m': -1.0,\n",
       "   'pppi_m': 1.0,\n",
       "   'cbl1_m': -1.0,\n",
       "   'h_m': -1.0,\n",
       "   'adocbl_m': 1.0},\n",
       "  'id': 'CBLATm',\n",
       "  'gene_reaction_rule': 'HGNC:18871'},\n",
       " {'subsystem': 'Glycolysis/gluconeogenesis',\n",
       "  'name': 'Carbamoyl phosphate phosphotransferase, endoplasmic reticulum',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['See PMID: 11946414 NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:24861 or HGNC:28906 or HGNC:4056']},\n",
       "  'metabolites': {'h_r': -1.0,\n",
       "   'co2_r': 1.0,\n",
       "   'g6p_r': 1.0,\n",
       "   'nh4_r': 1.0,\n",
       "   'glc_D_r': -1.0,\n",
       "   'cbp_r': -1.0},\n",
       "  'id': 'CBPPer',\n",
       "  'gene_reaction_rule': 'HGNC:24861 or HGNC:28906 or HGNC:4056'},\n",
       " {'subsystem': 'Pyrimidine synthesis',\n",
       "  'name': 'carbamoyl-phosphate synthase (glutamine-hydrolysing)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:1424']},\n",
       "  'metabolites': {'cbp_c': 1.0,\n",
       "   'gln_L_c': -1.0,\n",
       "   'atp_c': -2.0,\n",
       "   'hco3_c': -1.0,\n",
       "   'glu_L_c': 1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'h_c': 2.0,\n",
       "   'adp_c': 2.0,\n",
       "   'pi_c': 1.0},\n",
       "  'id': 'CBPS',\n",
       "  'gene_reaction_rule': 'HGNC:1424'},\n",
       " {'subsystem': 'Glutamate metabolism',\n",
       "  'name': 'carbamoyl-phosphate synthase (ammonia) (mitochondria',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['MM'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2323']},\n",
       "  'metabolites': {'hco3_m': -1.0,\n",
       "   'atp_m': -2.0,\n",
       "   'cbp_m': 1.0,\n",
       "   'nh4_m': -1.0,\n",
       "   'pi_m': 1.0,\n",
       "   'adp_m': 2.0,\n",
       "   'h_m': 2.0},\n",
       "  'id': 'CBPSam',\n",
       "  'gene_reaction_rule': 'HGNC:2323'},\n",
       " {'subsystem': 'Eicosanoid metabolism',\n",
       "  'name': 'Carbonyl reductase [NADPH] 1',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['multiple EC EC 1.1.1.189 EC 1.1.1.197 NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1548']},\n",
       "  'metabolites': {'prostge1_c': 1.0,\n",
       "   'nadp_c': 1.0,\n",
       "   'nadph_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'prostg1_c': -1.0},\n",
       "  'id': 'CBR1',\n",
       "  'gene_reaction_rule': 'HGNC:1548'},\n",
       " {'subsystem': 'Eicosanoid metabolism',\n",
       "  'name': 'Carbonyl reductase [NADPH] 1',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['multiple EC EC 1.1.1.189 EC 1.1.1.197 NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1548']},\n",
       "  'metabolites': {'nadp_c': 1.0,\n",
       "   'nadph_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'prostge2_c': -1.0,\n",
       "   'prostgf2_c': 1.0},\n",
       "  'id': 'CBR2',\n",
       "  'gene_reaction_rule': 'HGNC:1548'},\n",
       " {'subsystem': 'Glycerophospholipid metabolism',\n",
       "  'name': 'phosphatidylinositol synthase (Homo sapiens)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ - reversibility of enzyme found in lit by NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1769']},\n",
       "  'metabolites': {'pail_hs_c': 1.0,\n",
       "   'inost_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   'cdpdag_hs_c': -1.0,\n",
       "   'cmp_c': 1.0},\n",
       "  'id': 'CDIPTr',\n",
       "  'gene_reaction_rule': 'HGNC:1769'},\n",
       " {'subsystem': 'Glycerophospholipid metabolism',\n",
       "  'name': 'phosphatidate cytidylyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['ER membrane - but on cytosolic surface NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1800 or HGNC:1801']},\n",
       "  'metabolites': {'ctp_c': -1.0,\n",
       "   'pa_hs_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'h_c': -1.0,\n",
       "   'cdpdag_hs_c': 1.0},\n",
       "  'id': 'CDS',\n",
       "  'gene_reaction_rule': 'HGNC:1800 or HGNC:1801'},\n",
       " {'subsystem': 'Glycerophospholipid metabolism',\n",
       "  'name': 'phosphatidate cytidylyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['inner mitochondrial membrane NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1800 or HGNC:1801']},\n",
       "  'metabolites': {'cdpdag_hs_m': 1.0,\n",
       "   'pa_hs_m': -1.0,\n",
       "   'h_m': -1.0,\n",
       "   'ppi_m': 1.0,\n",
       "   'ctp_m': -1.0},\n",
       "  'id': 'CDSm',\n",
       "  'gene_reaction_rule': 'HGNC:1800 or HGNC:1801'},\n",
       " {'subsystem': 'Glycerophospholipid metabolism',\n",
       "  'name': 'choline phosphotransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:17852 or HGNC:24289']},\n",
       "  'metabolites': {'pchol_hs_c': 1.0,\n",
       "   'dag_hs_c': -1.0,\n",
       "   'cdpchol_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   'cmp_c': 1.0},\n",
       "  'id': 'CEPTC',\n",
       "  'gene_reaction_rule': 'HGNC:17852 or HGNC:24289'},\n",
       " {'subsystem': 'Glycerophospholipid metabolism',\n",
       "  'name': 'ethanolamine phosphotransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:24289']},\n",
       "  'metabolites': {'cdpea_c': -1.0,\n",
       "   'dag_hs_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   'pe_hs_c': 1.0,\n",
       "   'cmp_c': 1.0},\n",
       "  'id': 'CEPTE',\n",
       "  'gene_reaction_rule': 'HGNC:24289'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'Ceramide kinase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:19256']},\n",
       "  'metabolites': {'adp_c': 1.0,\n",
       "   'crm_hs_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'crmp_hs_c': 1.0},\n",
       "  'id': 'CERK',\n",
       "  'gene_reaction_rule': 'HGNC:19256'},\n",
       " {'subsystem': 'Bile acid synthesis',\n",
       "  'name': 'cholesterol 25-hydroxylase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1907']},\n",
       "  'metabolites': {'h_r': -1.0,\n",
       "   'o2_r': -1.0,\n",
       "   'nadp_r': 1.0,\n",
       "   'h2o_r': 1.0,\n",
       "   'chsterol_r': -1.0,\n",
       "   'xol25oh_r': 1.0,\n",
       "   'nadph_r': -1.0},\n",
       "  'id': 'CH25H',\n",
       "  'gene_reaction_rule': 'HGNC:1907'},\n",
       " {'subsystem': 'Glycerophospholipid metabolism',\n",
       "  'name': 'Choline O-acetyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['rev by uniprot NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1912']},\n",
       "  'metabolites': {'chol_c': -1.0, 'ach_c': 1.0, 'coa_c': 1.0, 'accoa_c': -1.0},\n",
       "  'id': 'CHAT',\n",
       "  'gene_reaction_rule': 'HGNC:1912'},\n",
       " {'subsystem': 'Glycerophospholipid metabolism',\n",
       "  'name': 'Choline O-acetyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1912']},\n",
       "  'metabolites': {'coa_n': 1.0, 'accoa_n': -1.0, 'chol_n': -1.0, 'ach_n': 1.0},\n",
       "  'id': 'CHATn',\n",
       "  'gene_reaction_rule': 'HGNC:1912'},\n",
       " {'subsystem': 'Glycerophospholipid metabolism',\n",
       "  'name': 'choline phosphate phosphatase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['no high energy phosphate compounds found associated w/ rxn NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:16815']},\n",
       "  'metabolites': {'chol_c': 1.0, 'cholp_c': -1.0, 'h2o_c': -1.0, 'pi_c': 1.0},\n",
       "  'id': 'CHLP',\n",
       "  'gene_reaction_rule': 'HGNC:16815'},\n",
       " {'subsystem': 'Glycerophospholipid metabolism',\n",
       "  'name': 'choline phosphate cytididyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:8754 or HGNC:8755']},\n",
       "  'metabolites': {'ctp_c': -1.0,\n",
       "   'cholp_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'cdpchol_c': 1.0,\n",
       "   'h_c': -1.0},\n",
       "  'id': 'CHLPCTD',\n",
       "  'gene_reaction_rule': 'HGNC:8754 or HGNC:8755'},\n",
       " {'subsystem': 'Glycine, serine, alanine and threonine metabolism',\n",
       "  'name': 'choline dehydrogenase (FAD acceptor), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:24288']},\n",
       "  'metabolites': {'fad_m': -1.0,\n",
       "   'betald_m': 1.0,\n",
       "   'fadh2_m': 1.0,\n",
       "   'chol_m': -1.0},\n",
       "  'id': 'CHOLD2m',\n",
       "  'gene_reaction_rule': 'HGNC:24288'},\n",
       " {'subsystem': 'Glycerophospholipid metabolism',\n",
       "  'name': 'Choline kinase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1937 or HGNC:1938']},\n",
       "  'metabolites': {'chol_c': -1.0,\n",
       "   'cholp_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'adp_c': 1.0},\n",
       "  'id': 'CHOLK',\n",
       "  'gene_reaction_rule': 'HGNC:1937 or HGNC:1938'},\n",
       " {'subsystem': 'Steroid metabolism',\n",
       "  'name': 'steroid sulfotransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:11455 or HGNC:11459']},\n",
       "  'metabolites': {'pap_c': 1.0,\n",
       "   'chsterols_c': 1.0,\n",
       "   'chsterol_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   'paps_c': -1.0},\n",
       "  'id': 'CHSTEROLSULT',\n",
       "  'gene_reaction_rule': 'HGNC:11455 or HGNC:11459'},\n",
       " {'subsystem': 'Aminosugar metabolism',\n",
       "  'name': 'chitinase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:17432 or HGNC:1936']},\n",
       "  'metabolites': {'acgam_c': 3.0, 'h2o_c': -2.0, 'chtn_c': -1.0},\n",
       "  'id': 'CHTNASE',\n",
       "  'gene_reaction_rule': 'HGNC:17432 or HGNC:1936'},\n",
       " {'subsystem': 'Aminosugar metabolism',\n",
       "  'name': 'chitinase, extracellular',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:17432']},\n",
       "  'metabolites': {'chtn_e': -1.0, 'h2o_e': -2.0, 'acgam_e': 3.0},\n",
       "  'id': 'CHTNASEe',\n",
       "  'gene_reaction_rule': 'HGNC:17432'},\n",
       " {'subsystem': 'Citric acid cycle',\n",
       "  'name': 'Citrate lyase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['JLR'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'ac_c': 1.0, 'oaa_c': 1.0, 'cit_c': -1.0},\n",
       "  'id': 'CITL',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'C5-branched dibasic acid metabolism',\n",
       "  'name': 'L-Citramalyl-CoA hydro-lyase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'itaccoa_m': -1.0, 'h2o_m': -1.0, 'citmcoa_L_m': 1.0},\n",
       "  'id': 'CITMCOAHm',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'C5-branched dibasic acid metabolism',\n",
       "  'name': 'L-Citramalyl-CoA pyruvate-lyase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['also EC 4.1.3.22 NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'accoa_m': 1.0, 'pyr_m': 1.0, 'citmcoa_L_m': -1.0},\n",
       "  'id': 'CITMCOALm',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Urea cycle',\n",
       "  'name': 'ATP Creatine kinase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['kegg C02305 MJC'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1995 or HGNC:1996 or HGNC:31736']},\n",
       "  'metabolites': {'adp_m': 1.0,\n",
       "   'atp_m': -1.0,\n",
       "   'creat_m': -1.0,\n",
       "   'h_m': 1.0,\n",
       "   'pcreat_m': 1.0},\n",
       "  'id': 'CK',\n",
       "  'gene_reaction_rule': 'HGNC:1995 or HGNC:1996 or HGNC:31736'},\n",
       " {'subsystem': 'Urea cycle',\n",
       "  'name': 'ATP Creatine kinase (c)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['kegg C02305 MJC SAB c'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:1991 and HGNC:1994) or HGNC:1991 or HGNC:1994']},\n",
       "  'metabolites': {'adp_c': 1.0,\n",
       "   'pcreat_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'creat_c': -1.0},\n",
       "  'id': 'CKc',\n",
       "  'gene_reaction_rule': '(HGNC:1991 and HGNC:1994) or HGNC:1991 or HGNC:1994'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'carnitine O-palmitoyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:18540 or HGNC:2328 or HGNC:2329']},\n",
       "  'metabolites': {'crn_c': -1.0,\n",
       "   'clpndcoa_c': -1.0,\n",
       "   'clpndcrn_c': 1.0,\n",
       "   'coa_c': 1.0},\n",
       "  'id': 'CLPNDCPT1',\n",
       "  'gene_reaction_rule': 'HGNC:18540 or HGNC:2328 or HGNC:2329'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'carnitine transferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2330']},\n",
       "  'metabolites': {'clpndcoa_m': 1.0,\n",
       "   'coa_m': -1.0,\n",
       "   'crn_m': 1.0,\n",
       "   'clpndcrn_m': -1.0},\n",
       "  'id': 'CLPNDCPT2',\n",
       "  'gene_reaction_rule': 'HGNC:2330'},\n",
       " {'subsystem': 'Glycerophospholipid metabolism',\n",
       "  'name': 'cardiolipin synthase (homo sapiens)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:16148']},\n",
       "  'metabolites': {'clpn_hs_c': 1.0,\n",
       "   'pglyc_hs_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'cdpdag_hs_c': -1.0,\n",
       "   'cmp_c': 1.0},\n",
       "  'id': 'CLS_hs',\n",
       "  'gene_reaction_rule': 'HGNC:16148'},\n",
       " {'subsystem': 'Aminosugar metabolism',\n",
       "  'name': 'CMP sialic acid synthase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:18290']},\n",
       "  'metabolites': {'ctp_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'cmpacna_c': 1.0,\n",
       "   'acnam_c': -1.0},\n",
       "  'id': 'CMPSAS',\n",
       "  'gene_reaction_rule': 'HGNC:18290'},\n",
       " {'subsystem': 'Aminosugar metabolism',\n",
       "  'name': 'CMP sialic acid synthase, nuclear',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:18290']},\n",
       "  'metabolites': {'acnam_n': -1.0,\n",
       "   'cmpacna_n': 1.0,\n",
       "   'ctp_n': -1.0,\n",
       "   'ppi_n': 1.0},\n",
       "  'id': 'CMPSASn',\n",
       "  'gene_reaction_rule': 'HGNC:18290'},\n",
       " {'subsystem': 'Alkaloid synthesis',\n",
       "  'name': 'Carboxylesterase (cocaine) (er)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1864']},\n",
       "  'metabolites': {'bz_r': 1.0,\n",
       "   'coke_r': -1.0,\n",
       "   'h_r': 1.0,\n",
       "   'h2o_r': -1.0,\n",
       "   'egme_r': 1.0},\n",
       "  'id': 'COKECBESr',\n",
       "  'gene_reaction_rule': 'HGNC:1864'},\n",
       " {'subsystem': 'Ubiquinone synthesis',\n",
       "  'name': 'methyltransferase COQ3',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['_2.1.1.114 IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:18175']},\n",
       "  'metabolites': {'2dpmhobq_m': -1.0,\n",
       "   'h_m': 1.0,\n",
       "   'ahcys_m': 1.0,\n",
       "   'q10_m': 1.0,\n",
       "   'amet_m': -1.0},\n",
       "  'id': 'COQ3m',\n",
       "  'gene_reaction_rule': 'HGNC:18175'},\n",
       " {'subsystem': 'Ubiquinone synthesis',\n",
       "  'name': 'Ubiquinone biosynthesis methyltransferase COQ5',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['_2.1.1.- IT'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'2dp6mobq_me_m': 1.0,\n",
       "   'ahcys_m': 1.0,\n",
       "   'h_m': 1.0,\n",
       "   '2dp6mobq_m': -1.0,\n",
       "   'amet_m': -1.0},\n",
       "  'id': 'COQ5m',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Ubiquinone synthesis',\n",
       "  'name': 'Ubiquinone biosynthesis monooxgenase COQ6',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['_1.13.12.- IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:20233']},\n",
       "  'metabolites': {'o2_m': -1.0,\n",
       "   'h2o_m': 1.0,\n",
       "   '2dp6mobq_m': 1.0,\n",
       "   '2dp6mep_m': -1.0},\n",
       "  'id': 'COQ6m',\n",
       "  'gene_reaction_rule': 'HGNC:20233'},\n",
       " {'subsystem': 'Ubiquinone synthesis',\n",
       "  'name': 'Ubiquinone biosynthesis COQ7',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['_1.14.13.- IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2244']},\n",
       "  'metabolites': {'nadph_m': -1.0,\n",
       "   'h2o_m': 1.0,\n",
       "   'nadp_m': 1.0,\n",
       "   'o2_m': -1.0,\n",
       "   '2dp6mobq_me_m': -1.0,\n",
       "   '2dpmhobq_m': 1.0,\n",
       "   'h_m': -1.0},\n",
       "  'id': 'COQ7m',\n",
       "  'gene_reaction_rule': 'HGNC:2244'},\n",
       " {'subsystem': 'O-glycan synthesis',\n",
       "  'name': 'Core 2 acetylglucosaminyltransferase, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:17973 or HGNC:4203 or HGNC:4205']},\n",
       "  'metabolites': {'core2_g': 1.0,\n",
       "   'h_g': 1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'T_antigen_g': -1.0,\n",
       "   'udp_g': 1.0},\n",
       "  'id': 'CORE2GTg',\n",
       "  'gene_reaction_rule': 'HGNC:17973 or HGNC:4203 or HGNC:4205'},\n",
       " {'subsystem': 'O-glycan synthesis',\n",
       "  'name': 'Core 3 beta-GlcNAc-transferase, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:24141']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'core3_g': 1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'Tn_antigen_g': -1.0},\n",
       "  'id': 'CORE3GTg',\n",
       "  'gene_reaction_rule': 'HGNC:24141'},\n",
       " {'subsystem': 'O-glycan synthesis',\n",
       "  'name': 'Core 4 beta6-GalNAc-transferase, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4205']},\n",
       "  'metabolites': {'core3_g': -1.0,\n",
       "   'core4_g': 1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'CORE4GTg',\n",
       "  'gene_reaction_rule': 'HGNC:4205'},\n",
       " {'subsystem': 'O-glycan synthesis',\n",
       "  'name': 'Core 5 alpha-GalNAc-transferase, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['EC 2.4.1.- NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'udpacgal_g': -1.0,\n",
       "   'core5_g': 1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'Tn_antigen_g': -1.0},\n",
       "  'id': 'CORE5GTg',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'O-glycan synthesis',\n",
       "  'name': 'Core 6 beta-GlcNAc-transferase A, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4203']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'core6_g': 1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'Tn_antigen_g': -1.0},\n",
       "  'id': 'CORE6GTg',\n",
       "  'gene_reaction_rule': 'HGNC:4203'},\n",
       " {'subsystem': 'O-glycan synthesis',\n",
       "  'name': 'Core 7 alpha-GalNAc-transferase, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['EC 2.4.1.- NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'udpacgal_g': -1.0,\n",
       "   'core7_g': 1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'Tn_antigen_g': -1.0},\n",
       "  'id': 'CORE7GTg',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'O-glycan synthesis',\n",
       "  'name': 'N-acetylgalactosamine 3-alpha-galactosyltransferase, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['EC 2.4.1.- NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'core8_g': 1.0,\n",
       "   'h_g': 1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'Tn_antigen_g': -1.0},\n",
       "  'id': 'CORE8GTg',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Ubiquinone synthesis',\n",
       "  'name': 'p-coumaroyl-CoA formation',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['IT reaction involved in formation of 4-hydroxybenzoate from tyrosine. (EC 6.2.1.12) .'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'coucoa_m': 1.0,\n",
       "   'T4hcinnm_m': -1.0,\n",
       "   'ppi_m': 1.0,\n",
       "   'atp_m': -1.0,\n",
       "   'coa_m': -1.0,\n",
       "   'amp_m': 1.0},\n",
       "  'id': 'COUCOAFm',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Glycerophospholipid metabolism',\n",
       "  'name': 'choline-phosphate cytidylyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:8754 or HGNC:8755']},\n",
       "  'metabolites': {'ctp_c': -1.0,\n",
       "   'cmpntm2amep_c': 1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'ntm2amep_c': -1.0,\n",
       "   'h_c': -1.0},\n",
       "  'id': 'CPCTDTX',\n",
       "  'gene_reaction_rule': 'HGNC:8754 or HGNC:8755'},\n",
       " {'subsystem': 'Heme synthesis',\n",
       "  'name': 'coproporphyrinogen oxidase (O2 required)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['r0608, JLR'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2321']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   'h2o_c': 2.0,\n",
       "   'pppg9_c': 1.0,\n",
       "   'cpppg3_c': -1.0,\n",
       "   'h_c': -2.0,\n",
       "   'co2_c': 2.0},\n",
       "  'id': 'CPPPGO',\n",
       "  'gene_reaction_rule': 'HGNC:2321'},\n",
       " {'subsystem': 'Urea cycle',\n",
       "  'name': 'Creatinine synthase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'pcreat_c': -1.0, 'pi_c': 1.0, 'crtn_c': 1.0},\n",
       "  'id': 'CRTNsyn',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Chondroitin sulfate degradation',\n",
       "  'name': 'chondroitin sulfate A proteoglycan protease, lysosome (endosome)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['Note that this reaction occurs in the endosome during endocytosis, but has been compartmentalized as lysosomal. NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'cspg_a_l': -1.0,\n",
       "   'cs_a_l': 1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   'Ser_Gly_Ala_X_Gly_l': 1.0},\n",
       "  'id': 'CSAPASEly',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Chondroitin sulfate degradation',\n",
       "  'name': 'chondroitin sulfate B proteoglycan protease, lysosome (endosome)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['Note that this reaction occurs in the endosome during endocytosis, but has been compartmentalized as lysosomal. NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'Ser_Gly_Ala_X_Gly_l': 1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   'cs_b_l': 1.0,\n",
       "   'cspg_b_l': -1.0},\n",
       "  'id': 'CSBPASEly',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Chondroitin sulfate degradation',\n",
       "  'name': 'chondroitin sulfate C proteoglycan protease, lysosome (endosome)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['Note that this reaction occurs in the endosome during endocytosis, but has been compartmentalized as lysosomal. NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'Ser_Gly_Ala_X_Gly_l': 1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   'cspg_c_l': -1.0,\n",
       "   'cs_c_l': 1.0},\n",
       "  'id': 'CSCPASEly',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Chondroitin sulfate degradation',\n",
       "  'name': 'chondroitin sulfate D proteoglycan protease, lysosome (endosome)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['Note that this reaction occurs in the endosome during endocytosis, but has been compartmentalized as lysosomal. NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'Ser_Gly_Ala_X_Gly_l': 1.0,\n",
       "   'cs_d_l': 1.0,\n",
       "   'cspg_d_l': -1.0,\n",
       "   'h2o_l': -1.0},\n",
       "  'id': 'CSDPASEly',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Chondroitin sulfate degradation',\n",
       "  'name': 'chondroitin sulfate E proteoglycan protease, lysosome (endosome)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['Note that this reaction occurs in the endosome during endocytosis, but has been compartmentalized as lysosomal. NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'Ser_Gly_Ala_X_Gly_l': 1.0,\n",
       "   'cspg_e_l': -1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   'cs_e_l': 1.0},\n",
       "  'id': 'CSEPASEly',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Citric acid cycle',\n",
       "  'name': 'citrate synthase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['IF old EC number (4.1.3.7) is obsolete -- transferred to 2.3.3.1 NCD. Biochemical'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2422']},\n",
       "  'metabolites': {'cit_m': 1.0,\n",
       "   'h2o_m': -1.0,\n",
       "   'accoa_m': -1.0,\n",
       "   'oaa_m': -1.0,\n",
       "   'coa_m': 1.0,\n",
       "   'h_m': 1.0},\n",
       "  'id': 'CSm',\n",
       "  'gene_reaction_rule': 'HGNC:2422'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'carnitine O-aceyltransferase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['TDV'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2342']},\n",
       "  'metabolites': {'ppcoa_m': 1.0, 'coa_m': -1.0, 'pcrn_m': -1.0, 'crn_m': 1.0},\n",
       "  'id': 'CSNAT2m',\n",
       "  'gene_reaction_rule': 'HGNC:2342'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'carnitine dimethyl nonanoyl transferase, revsible, peroxisomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2342 or HGNC:2366']},\n",
       "  'metabolites': {'dmnoncrn_x': 1.0,\n",
       "   'crn_x': -1.0,\n",
       "   'dmnoncoa_x': -1.0,\n",
       "   'coa_x': 1.0},\n",
       "  'id': 'CSNAT2x',\n",
       "  'gene_reaction_rule': 'HGNC:2342 or HGNC:2366'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'carnitine O-aceyltransferase, peroxisomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2342']},\n",
       "  'metabolites': {'pcrn_x': 1.0, 'crn_x': -1.0, 'ppcoa_x': -1.0, 'coa_x': 1.0},\n",
       "  'id': 'CSNAT3x',\n",
       "  'gene_reaction_rule': 'HGNC:2342'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'carnitine O-acetyltransferase, endolasmic reticulum',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['mm'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2342']},\n",
       "  'metabolites': {'crn_r': -1.0, 'acrn_r': 1.0, 'coa_r': 1.0, 'accoa_r': -1.0},\n",
       "  'id': 'CSNATer',\n",
       "  'gene_reaction_rule': 'HGNC:2342'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'carnitine O-aceyltransferase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['TDV'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2342']},\n",
       "  'metabolites': {'accoa_m': 1.0, 'coa_m': -1.0, 'crn_m': 1.0, 'acrn_m': -1.0},\n",
       "  'id': 'CSNATm',\n",
       "  'gene_reaction_rule': 'HGNC:2342'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'carnitine O-acetyltransferase, reverse direction, peroxisomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2342']},\n",
       "  'metabolites': {'acrn_x': 1.0, 'accoa_x': -1.0, 'crn_x': -1.0, 'coa_x': 1.0},\n",
       "  'id': 'CSNATp',\n",
       "  'gene_reaction_rule': 'HGNC:2342'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'carnitine O-acetyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IF'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2342']},\n",
       "  'metabolites': {'acrn_c': 1.0, 'crn_c': -1.0, 'coa_c': 1.0, 'accoa_c': -1.0},\n",
       "  'id': 'CSNATr',\n",
       "  'gene_reaction_rule': 'HGNC:2342'},\n",
       " {'subsystem': 'Pyrimidine catabolism',\n",
       "  'name': 'Cytosine deaminase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['JLR. Part of a famous suicide gene combination with 5-fluorocytosine'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'ura_c': 1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'csn_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'nh4_c': 1.0},\n",
       "  'id': 'CSND',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'CTP synthase (NH3)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2519 or HGNC:2520']},\n",
       "  'metabolites': {'utp_c': -1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'h_c': 2.0,\n",
       "   'adp_c': 1.0,\n",
       "   'ctp_c': 1.0,\n",
       "   'pi_c': 1.0,\n",
       "   'nh4_c': -1.0},\n",
       "  'id': 'CTPS1',\n",
       "  'gene_reaction_rule': 'HGNC:2519 or HGNC:2520'},\n",
       " {'subsystem': 'Pyrimidine synthesis',\n",
       "  'name': 'CTP synthase (glutamine)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2519 or HGNC:2520']},\n",
       "  'metabolites': {'utp_c': -1.0,\n",
       "   'gln_L_c': -1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'glu_L_c': 1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'h_c': 2.0,\n",
       "   'adp_c': 1.0,\n",
       "   'ctp_c': 1.0,\n",
       "   'pi_c': 1.0},\n",
       "  'id': 'CTPS2',\n",
       "  'gene_reaction_rule': 'HGNC:2519 or HGNC:2520'},\n",
       " {'subsystem': 'Methionine and cysteine metabolism',\n",
       "  'name': 'Glutathione:cystine oxidoreductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['reaction according to KEGG  MM'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'cys_L_c': 2.0,\n",
       "   'gthrd_c': -2.0,\n",
       "   'gthox_c': 1.0,\n",
       "   'Lcystin_c': -1.0},\n",
       "  'id': 'CYSGLTH',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Methionine and cysteine metabolism',\n",
       "  'name': 'L-Cystine Lysteine-lyase (deaminating)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['according to reaction in KEGG  MM'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   'pyr_c': 1.0,\n",
       "   'nh4_c': 1.0,\n",
       "   'thcys_c': 1.0,\n",
       "   'Lcystin_c': -1.0},\n",
       "  'id': 'CYSLYSL',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Taurine and hypotaurine metabolism',\n",
       "  'name': 'cysteine oxidase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1795']},\n",
       "  'metabolites': {'o2_c': -1.0, 'cys_L_c': -1.0, 'h_c': 2.0, '3sala_c': 1.0},\n",
       "  'id': 'CYSO',\n",
       "  'gene_reaction_rule': 'HGNC:1795'},\n",
       " {'subsystem': 'Methionine and cysteine metabolism',\n",
       "  'name': 'cysteine transaminase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['MM'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4432']},\n",
       "  'metabolites': {'akg_c': -1.0,\n",
       "   'cys_L_c': -1.0,\n",
       "   'mercppyr_c': 1.0,\n",
       "   'glu_L_c': 1.0},\n",
       "  'id': 'CYSTA',\n",
       "  'gene_reaction_rule': 'HGNC:4432'},\n",
       " {'subsystem': 'Methionine and cysteine metabolism',\n",
       "  'name': 'cysteine transaminase (mitochondrial)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['MM'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4433']},\n",
       "  'metabolites': {'glu_L_m': 1.0,\n",
       "   'mercppyr_m': 1.0,\n",
       "   'cys_L_m': -1.0,\n",
       "   'akg_m': -1.0},\n",
       "  'id': 'CYSTAm',\n",
       "  'gene_reaction_rule': 'HGNC:4433'},\n",
       " {'subsystem': 'Methionine and cysteine metabolism',\n",
       "  'name': 'cystathionine g-lyase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2501']},\n",
       "  'metabolites': {'2obut_c': 1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'cys_L_c': 1.0,\n",
       "   'nh4_c': 1.0,\n",
       "   'cyst_L_c': -1.0},\n",
       "  'id': 'CYSTGL',\n",
       "  'gene_reaction_rule': 'HGNC:2501'},\n",
       " {'subsystem': 'Methionine and cysteine metabolism',\n",
       "  'name': 'cystathionine beta-synthase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1550']},\n",
       "  'metabolites': {'h2o_c': 1.0,\n",
       "   'hcys_L_c': -1.0,\n",
       "   'cyst_L_c': 1.0,\n",
       "   'ser_L_c': -1.0},\n",
       "  'id': 'CYSTS',\n",
       "  'gene_reaction_rule': 'HGNC:1550'},\n",
       " {'subsystem': 'Pyrimidine catabolism',\n",
       "  'name': 'cytidine deaminase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1712 or HGNC:13203']},\n",
       "  'metabolites': {'cytd_c': -1.0,\n",
       "   'uri_c': 1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'nh4_c': 1.0},\n",
       "  'id': 'CYTD',\n",
       "  'gene_reaction_rule': 'HGNC:1712 or HGNC:13203'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'ATPcytidine 5-phosphotransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'GENE ASSOCIATION': ['HGNC:12562 or HGNC:14859']},\n",
       "  'metabolites': {'cytd_c': -1.0,\n",
       "   'cmp_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'adp_c': 1.0},\n",
       "  'id': 'CYTDK1',\n",
       "  'gene_reaction_rule': 'HGNC:12562 or HGNC:14859'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidine kinase (ATP), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'adp_m': 1.0,\n",
       "   'cmp_m': 1.0,\n",
       "   'atp_m': -1.0,\n",
       "   'h_m': 1.0,\n",
       "   'cytd_m': -1.0},\n",
       "  'id': 'CYTDK2m',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidine deaminase, nuclear',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'], 'GENE ASSOCIATION': ['HGNC:13203']},\n",
       "  'metabolites': {'nh4_n': 1.0,\n",
       "   'h_n': -1.0,\n",
       "   'uri_n': 1.0,\n",
       "   'cytd_n': -1.0,\n",
       "   'h2o_n': -1.0},\n",
       "  'id': 'CYTDn',\n",
       "  'gene_reaction_rule': 'HGNC:13203'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidylate kinase (CMP)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:18170']},\n",
       "  'metabolites': {'cdp_c': 1.0, 'adp_c': 1.0, 'atp_c': -1.0, 'cmp_c': -1.0},\n",
       "  'id': 'CYTK1',\n",
       "  'gene_reaction_rule': 'HGNC:18170'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidylate kinase (CMP,dGTP)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:18170']},\n",
       "  'metabolites': {'cdp_c': 1.0, 'dgdp_c': 1.0, 'dgtp_c': -1.0, 'cmp_c': -1.0},\n",
       "  'id': 'CYTK10',\n",
       "  'gene_reaction_rule': 'HGNC:18170'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidylate kinase (CMP,dGTP),nuclear',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:18170']},\n",
       "  'metabolites': {'cmp_n': -1.0, 'dgdp_n': 1.0, 'cdp_n': 1.0, 'dgtp_n': -1.0},\n",
       "  'id': 'CYTK10n',\n",
       "  'gene_reaction_rule': 'HGNC:18170'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidylate kinase (dCMP,dGTP)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:18170']},\n",
       "  'metabolites': {'dcmp_c': -1.0,\n",
       "   'dgdp_c': 1.0,\n",
       "   'dgtp_c': -1.0,\n",
       "   'dcdp_c': 1.0},\n",
       "  'id': 'CYTK11',\n",
       "  'gene_reaction_rule': 'HGNC:18170'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidylate kinase (dCMP,dGTP),nuclear',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:18170']},\n",
       "  'metabolites': {'dcdp_n': 1.0,\n",
       "   'dgdp_n': 1.0,\n",
       "   'dcmp_n': -1.0,\n",
       "   'dgtp_n': -1.0},\n",
       "  'id': 'CYTK11n',\n",
       "  'gene_reaction_rule': 'HGNC:18170'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidylate kinase (dCMP,dCTP)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:18170']},\n",
       "  'metabolites': {'dcmp_c': -1.0, 'dcdp_c': 2.0, 'dctp_c': -1.0},\n",
       "  'id': 'CYTK12',\n",
       "  'gene_reaction_rule': 'HGNC:18170'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidylate kinase (dCMP,dCTP),nuclear',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:18170']},\n",
       "  'metabolites': {'dctp_n': -1.0, 'dcdp_n': 2.0, 'dcmp_n': -1.0},\n",
       "  'id': 'CYTK12n',\n",
       "  'gene_reaction_rule': 'HGNC:18170'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidylate kinase (dCMP,dATP)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:18170']},\n",
       "  'metabolites': {'dcmp_c': -1.0,\n",
       "   'datp_c': -1.0,\n",
       "   'dadp_c': 1.0,\n",
       "   'dcdp_c': 1.0},\n",
       "  'id': 'CYTK13',\n",
       "  'gene_reaction_rule': 'HGNC:18170'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidylate kinase (dCMP,dATP),nuclear',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:18170']},\n",
       "  'metabolites': {'dadp_n': 1.0,\n",
       "   'dcdp_n': 1.0,\n",
       "   'datp_n': -1.0,\n",
       "   'dcmp_n': -1.0},\n",
       "  'id': 'CYTK13n',\n",
       "  'gene_reaction_rule': 'HGNC:18170'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidylate kinase (dCMP,UTP)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:18170']},\n",
       "  'metabolites': {'dcmp_c': -1.0, 'utp_c': -1.0, 'udp_c': 1.0, 'dcdp_c': 1.0},\n",
       "  'id': 'CYTK14',\n",
       "  'gene_reaction_rule': 'HGNC:18170'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidylate kinase (dCMP,CTP),nuclear',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:18170']},\n",
       "  'metabolites': {'dcdp_n': 1.0, 'udp_n': 1.0, 'dcmp_n': -1.0, 'utp_n': -1.0},\n",
       "  'id': 'CYTK14n',\n",
       "  'gene_reaction_rule': 'HGNC:18170'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidylate kinase (CMP),mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['No evidence in the mitochondrion'],\n",
       "   'GENE ASSOCIATION': ['HGNC:27015']},\n",
       "  'metabolites': {'adp_m': 1.0, 'cmp_m': -1.0, 'atp_m': -1.0, 'cdp_m': 1.0},\n",
       "  'id': 'CYTK1m',\n",
       "  'gene_reaction_rule': 'HGNC:27015'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidylate kinase (CMP),nuclear',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:18170']},\n",
       "  'metabolites': {'cmp_n': -1.0, 'atp_n': -1.0, 'adp_n': 1.0, 'cdp_n': 1.0},\n",
       "  'id': 'CYTK1n',\n",
       "  'gene_reaction_rule': 'HGNC:18170'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidylate kinase (dCMP)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:18170']},\n",
       "  'metabolites': {'dcmp_c': -1.0, 'adp_c': 1.0, 'atp_c': -1.0, 'dcdp_c': 1.0},\n",
       "  'id': 'CYTK2',\n",
       "  'gene_reaction_rule': 'HGNC:18170'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidylate kinase (dCMP),nuclear',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:18170']},\n",
       "  'metabolites': {'dcdp_n': 1.0, 'atp_n': -1.0, 'adp_n': 1.0, 'dcmp_n': -1.0},\n",
       "  'id': 'CYTK2n',\n",
       "  'gene_reaction_rule': 'HGNC:18170'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidylate kinase (CMP)(GTP)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'GENE ASSOCIATION': ['HGNC:365 or HGNC:365 or HGNC:18170 or HGNC:27015']},\n",
       "  'metabolites': {'cdp_c': 1.0, 'gtp_c': -1.0, 'gdp_c': 1.0, 'cmp_c': -1.0},\n",
       "  'id': 'CYTK3',\n",
       "  'gene_reaction_rule': 'HGNC:365 or HGNC:365 or HGNC:18170 or HGNC:27015'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidylate kinase (dCMP,CTP),nuclear',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:18170']},\n",
       "  'metabolites': {'dcdp_n': 1.0, 'cdp_n': 1.0, 'dcmp_n': -1.0, 'ctp_n': -1.0},\n",
       "  'id': 'CYTK3n',\n",
       "  'gene_reaction_rule': 'HGNC:18170'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidylate kinase (dCMP)(GTP)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'GENE ASSOCIATION': ['HGNC:365 or HGNC:365 or HGNC:18170 or HGNC:27015']},\n",
       "  'metabolites': {'dcmp_c': -1.0, 'gtp_c': -1.0, 'gdp_c': 1.0, 'dcdp_c': 1.0},\n",
       "  'id': 'CYTK4',\n",
       "  'gene_reaction_rule': 'HGNC:365 or HGNC:365 or HGNC:18170 or HGNC:27015'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidylate kinase (dCMP,GTP),nuclear',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:18170']},\n",
       "  'metabolites': {'gtp_n': -1.0, 'dcdp_n': 1.0, 'dcmp_n': -1.0, 'gdp_n': 1.0},\n",
       "  'id': 'CYTK4n',\n",
       "  'gene_reaction_rule': 'HGNC:18170'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidylate kinase (dCMP)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:18170']},\n",
       "  'metabolites': {'ctp_c': -1.0, 'cdp_c': 1.0, 'dcmp_c': -1.0, 'dcdp_c': 1.0},\n",
       "  'id': 'CYTK5',\n",
       "  'gene_reaction_rule': 'HGNC:18170'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidylate kinase (CMP),nuclear',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:18170']},\n",
       "  'metabolites': {'gtp_n': -1.0, 'cmp_n': -1.0, 'cdp_n': 1.0, 'gdp_n': 1.0},\n",
       "  'id': 'CYTK5n',\n",
       "  'gene_reaction_rule': 'HGNC:18170'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidylate kinase (CMP,CTP)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:18170']},\n",
       "  'metabolites': {'cdp_c': 2.0, 'ctp_c': -1.0, 'cmp_c': -1.0},\n",
       "  'id': 'CYTK6',\n",
       "  'gene_reaction_rule': 'HGNC:18170'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidylate kinase (CMP,CTP),nuclear',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:18170']},\n",
       "  'metabolites': {'cmp_n': -1.0, 'cdp_n': 2.0, 'ctp_n': -1.0},\n",
       "  'id': 'CYTK6n',\n",
       "  'gene_reaction_rule': 'HGNC:18170'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidylate kinase (CMP,UTP)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:18170']},\n",
       "  'metabolites': {'cdp_c': 1.0, 'utp_c': -1.0, 'udp_c': 1.0, 'cmp_c': -1.0},\n",
       "  'id': 'CYTK7',\n",
       "  'gene_reaction_rule': 'HGNC:18170'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidylate kinase (CMP,UTP),nuclear',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:18170']},\n",
       "  'metabolites': {'cmp_n': -1.0, 'udp_n': 1.0, 'cdp_n': 1.0, 'utp_n': -1.0},\n",
       "  'id': 'CYTK7n',\n",
       "  'gene_reaction_rule': 'HGNC:18170'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidylate kinase (CMP,dATP)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:18170']},\n",
       "  'metabolites': {'cdp_c': 1.0, 'datp_c': -1.0, 'dadp_c': 1.0, 'cmp_c': -1.0},\n",
       "  'id': 'CYTK8',\n",
       "  'gene_reaction_rule': 'HGNC:18170'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidylate kinase (CMP,dATP),nuclear',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:18170']},\n",
       "  'metabolites': {'dadp_n': 1.0, 'cmp_n': -1.0, 'datp_n': -1.0, 'cdp_n': 1.0},\n",
       "  'id': 'CYTK8n',\n",
       "  'gene_reaction_rule': 'HGNC:18170'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidylate kinase (CMP,dCTP)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:18170']},\n",
       "  'metabolites': {'cdp_c': 1.0, 'dcdp_c': 1.0, 'dctp_c': -1.0, 'cmp_c': -1.0},\n",
       "  'id': 'CYTK9',\n",
       "  'gene_reaction_rule': 'HGNC:18170'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'cytidylate kinase (CMP,dCTP),nuclear',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:18170']},\n",
       "  'metabolites': {'dctp_n': -1.0, 'cmp_n': -1.0, 'cdp_n': 1.0, 'dcdp_n': 1.0},\n",
       "  'id': 'CYTK9n',\n",
       "  'gene_reaction_rule': 'HGNC:18170'},\n",
       " {'subsystem': 'Pyrimidine catabolism',\n",
       "  'name': 'D-3-Amino-isobutanoate:pyruvate aminotransferase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['MM'],\n",
       "   'GENE ASSOCIATION': ['HGNC:14412']},\n",
       "  'metabolites': {'2mop_m': 1.0,\n",
       "   '3aib_D_m': -1.0,\n",
       "   'ala_L_m': 1.0,\n",
       "   'pyr_m': -1.0},\n",
       "  'id': 'D3AIBTm',\n",
       "  'gene_reaction_rule': 'HGNC:14412'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'Deoxyadenosine deaminase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['JLR'],\n",
       "   'GENE ASSOCIATION': ['HGNC:186']},\n",
       "  'metabolites': {'din_c': 1.0,\n",
       "   'dad_2_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'nh4_c': 1.0},\n",
       "  'id': 'DADA',\n",
       "  'gene_reaction_rule': 'HGNC:186'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'Deoxyadenosine deaminase, extracellular',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['adenosine deaminase has been found on cell surface in human (Gonzales-Gronow et al, JBC, 2004,279,20993-20998). IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:186']},\n",
       "  'metabolites': {'din_e': 1.0,\n",
       "   'h_e': -1.0,\n",
       "   'h2o_e': -1.0,\n",
       "   'nh4_e': 1.0,\n",
       "   'dad_2_e': -1.0},\n",
       "  'id': 'DADAe',\n",
       "  'gene_reaction_rule': 'HGNC:186'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'deoxyadenosine kinase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Deoxyguanosine can also act as acceptor. Possibly identical with EC 2.7.1.74 deoxycytidine kinase.'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'dad_2_c': -1.0,\n",
       "   'damp_c': 1.0,\n",
       "   'adp_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'h_c': 1.0},\n",
       "  'id': 'DADNK',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Glycerophospholipid metabolism',\n",
       "  'name': 'Diacylglycerol phosphate kinase (homo sapiens)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2849 or HGNC:2850 or HGNC:2851 or HGNC:2852 or HGNC:2853 or HGNC:2854 or HGNC:2855 or HGNC:2856 or HGNC:2857']},\n",
       "  'metabolites': {'adp_c': 1.0,\n",
       "   'pa_hs_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'dag_hs_c': -1.0},\n",
       "  'id': 'DAGK_hs',\n",
       "  'gene_reaction_rule': 'HGNC:2849 or HGNC:2850 or HGNC:2851 or HGNC:2852 or HGNC:2853 or HGNC:2854 or HGNC:2855 or HGNC:2856 or HGNC:2857'},\n",
       " {'subsystem': 'Glycerophospholipid metabolism',\n",
       "  'name': 'Diacylglycerol phosphate kinase (homo sapiens)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2855 or HGNC:2857']},\n",
       "  'metabolites': {'h_n': 1.0,\n",
       "   'atp_n': -1.0,\n",
       "   'pa_hs_n': 1.0,\n",
       "   'adp_n': 1.0,\n",
       "   'dag_hs_n': -1.0},\n",
       "  'id': 'DAGKn_hs',\n",
       "  'gene_reaction_rule': 'HGNC:2855 or HGNC:2857'},\n",
       " {'subsystem': 'D-alanine metabolism',\n",
       "  'name': 'D-Alanine Oxidase (x)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2671']},\n",
       "  'metabolites': {'nh4_x': 1.0,\n",
       "   'h2o2_x': 1.0,\n",
       "   'pyr_x': 1.0,\n",
       "   'h2o_x': -1.0,\n",
       "   'o2_x': -1.0,\n",
       "   'ala_D_x': -1.0},\n",
       "  'id': 'DALAOXx',\n",
       "  'gene_reaction_rule': 'HGNC:2671'},\n",
       " {'subsystem': 'Arginine and Proline Metabolism',\n",
       "  'name': 'D-arginine oxidase, perixosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2671']},\n",
       "  'metabolites': {'nh4_x': 1.0,\n",
       "   'h2o2_x': 1.0,\n",
       "   'arg_D_x': -1.0,\n",
       "   'h2o_x': -1.0,\n",
       "   'o2_x': -1.0,\n",
       "   '5g2oxpt_x': 1.0},\n",
       "  'id': 'DARGOp',\n",
       "  'gene_reaction_rule': 'HGNC:2671'},\n",
       " {'subsystem': 'Vitamin C metabolism',\n",
       "  'name': 'L-Dehydroascorbate lactonohydrolase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'dhdascb_c': -1.0, 'h2o_c': -1.0, '23doguln_c': 1.0},\n",
       "  'id': 'DASCBH',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Vitamin C metabolism',\n",
       "  'name': 'dehydroascorbate reductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'GENE ASSOCIATION': ['HGNC:16065 or HGNC:4330 or HGNC:13312']},\n",
       "  'metabolites': {'dhdascb_c': -1.0,\n",
       "   'ascb_L_c': 1.0,\n",
       "   'nadp_c': 1.0,\n",
       "   'nadph_c': -1.0,\n",
       "   'h_c': -1.0},\n",
       "  'id': 'DASCBR',\n",
       "  'gene_reaction_rule': 'HGNC:16065 or HGNC:4330 or HGNC:13312'},\n",
       " {'subsystem': 'Alanine and aspartate metabolism',\n",
       "  'name': 'D-aspartate oxidase, peroxisomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['MM'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2727']},\n",
       "  'metabolites': {'asp_D_x': -1.0,\n",
       "   'nh4_x': 1.0,\n",
       "   'h2o2_x': 1.0,\n",
       "   'oaa_x': 1.0,\n",
       "   'h2o_x': -1.0,\n",
       "   'o2_x': -1.0},\n",
       "  'id': 'DASPO1p',\n",
       "  'gene_reaction_rule': 'HGNC:2727'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'dodecenoyl-Coenzyme A delta isomerase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:14601 or HGNC:2703']},\n",
       "  'metabolites': {'dd2coa_m': -1.0, 'dd3coa_m': 1.0},\n",
       "  'id': 'DCIm',\n",
       "  'gene_reaction_rule': 'HGNC:14601 or HGNC:2703'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'Deoxycytidine kinase, mitochondrial (ATP)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT (2.7.1.74)'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'adp_m': 1.0,\n",
       "   'atp_m': -1.0,\n",
       "   'h_m': 1.0,\n",
       "   'dcyt_m': -1.0,\n",
       "   'dcmp_m': 1.0},\n",
       "  'id': 'DCK1m',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'Deoxycytidine kinase, nuclear (ATP)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT (2.7.1.74)'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2704']},\n",
       "  'metabolites': {'h_n': 1.0,\n",
       "   'atp_n': -1.0,\n",
       "   'adp_n': 1.0,\n",
       "   'dcyt_n': -1.0,\n",
       "   'dcmp_n': 1.0},\n",
       "  'id': 'DCK1n',\n",
       "  'gene_reaction_rule': 'HGNC:2704'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'Deoxycytidine kinase, nuclear (UTP)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT (2.7.1.74)'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2704']},\n",
       "  'metabolites': {'h_n': 1.0,\n",
       "   'udp_n': 1.0,\n",
       "   'dcyt_n': -1.0,\n",
       "   'dcmp_n': 1.0,\n",
       "   'utp_n': -1.0},\n",
       "  'id': 'DCK2n',\n",
       "  'gene_reaction_rule': 'HGNC:2704'},\n",
       " {'subsystem': 'Pyrimidine catabolism',\n",
       "  'name': 'dCMP deaminase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2710']},\n",
       "  'metabolites': {'dcmp_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'dump_c': 1.0,\n",
       "   'nh4_c': 1.0},\n",
       "  'id': 'DCMPDA',\n",
       "  'gene_reaction_rule': 'HGNC:2710'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'carnitine O-palmitoyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:18540 or HGNC:2328 or HGNC:2329']},\n",
       "  'metabolites': {'crn_c': -1.0,\n",
       "   'dcsptn1crn_c': 1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'dcsptn1coa_c': -1.0},\n",
       "  'id': 'DCSPTN1CPT1',\n",
       "  'gene_reaction_rule': 'HGNC:18540 or HGNC:2328 or HGNC:2329'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'carnitine transferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2330']},\n",
       "  'metabolites': {'dcsptn1coa_m': 1.0,\n",
       "   'coa_m': -1.0,\n",
       "   'dcsptn1crn_m': -1.0,\n",
       "   'crn_m': 1.0},\n",
       "  'id': 'DCSPTN1CPT2',\n",
       "  'gene_reaction_rule': 'HGNC:2330'},\n",
       " {'subsystem': 'Tyrosine metabolism',\n",
       "  'name': 'Dopachrome tautomerase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['ISOMERASE SAB added EC and adjusted compounds for balance'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2709']},\n",
       "  'metabolites': {'56dihindlcrbxlt_c': 1.0, 'L_dpchrm_c': -1.0},\n",
       "  'id': 'DCT',\n",
       "  'gene_reaction_rule': 'HGNC:2709'},\n",
       " {'subsystem': 'Pyrimidine catabolism',\n",
       "  'name': 'deoxycytidine deaminase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1712 or HGNC:13203']},\n",
       "  'metabolites': {'dcyt_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'duri_c': 1.0,\n",
       "   'nh4_c': 1.0},\n",
       "  'id': 'DCYTD',\n",
       "  'gene_reaction_rule': 'HGNC:1712 or HGNC:13203'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'deoxycytidine deaminase, nuclear',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:13203']},\n",
       "  'metabolites': {'nh4_n': 1.0,\n",
       "   'h_n': -1.0,\n",
       "   'duri_n': 1.0,\n",
       "   'dcyt_n': -1.0,\n",
       "   'h2o_n': -1.0},\n",
       "  'id': 'DCYTDn',\n",
       "  'gene_reaction_rule': 'HGNC:13203'},\n",
       " {'subsystem': 'Arginine and Proline Metabolism',\n",
       "  'name': '2-dehydro-3-deoxy-phosphogluconate aldolase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['IT rxn is done by 4.1.2.14 and 4.1.3.16 mitochondrial isozyme - NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'4h2oglt_m': -1.0, 'glx_m': 1.0, 'pyr_m': 1.0},\n",
       "  'id': 'DDPGAm',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'dehydrodolichol diphosphate phosphatase (uterus)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'], 'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   'dedoldp_U_c': -1.0,\n",
       "   'h_c': 2.0,\n",
       "   'dedolp_U_c': 1.0,\n",
       "   'pi_c': 1.0},\n",
       "  'id': 'DEDOLP1_U',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'dehydrodolichol phosphate phosphatase (uterus)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'], 'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'h_c': 2.0,\n",
       "   'dedolp_U_c': -1.0,\n",
       "   'pi_c': 1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'dedol_U_c': 1.0},\n",
       "  'id': 'DEDOLP2_U',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'dehydrodolichol reductase (uterus)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'], 'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'dolichol_U_c': 1.0,\n",
       "   'nadp_c': 1.0,\n",
       "   'nadph_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'dedol_U_c': -1.0},\n",
       "  'id': 'DEDOLR_U',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Fatty acid synthesis',\n",
       "  'name': 'palmitoyl-CoA desaturase (n-C16:0CoA -> n-C16:1CoA)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['for homo sapiens, noted to actually be NADH-cyt b5, using NADH at present NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:10571']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   'hdcoa_c': 1.0,\n",
       "   'h2o_c': 2.0,\n",
       "   'pmtcoa_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'nad_c': 1.0,\n",
       "   'nadh_c': -1.0},\n",
       "  'id': 'DESAT16_2',\n",
       "  'gene_reaction_rule': 'HGNC:10571'},\n",
       " {'subsystem': 'Fatty acid synthesis',\n",
       "  'name': 'fatty acyl-CoA desaturase (n-C18:2CoA -> n-C18:3CoA)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['for homo sapiens, noted to actually be NADH-cyt b5, using NADH at present NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3575']},\n",
       "  'metabolites': {'strdnccoa_c': 1.0,\n",
       "   'o2_c': -1.0,\n",
       "   'h2o_c': 2.0,\n",
       "   'h_c': -1.0,\n",
       "   'nad_c': 1.0,\n",
       "   'lnlncacoa_c': -1.0,\n",
       "   'nadh_c': -1.0},\n",
       "  'id': 'DESAT18_10',\n",
       "  'gene_reaction_rule': 'HGNC:3575'},\n",
       " {'subsystem': 'Fatty acid synthesis',\n",
       "  'name': 'stearoyl-CoA desaturase (n-C18:0CoA -> n-C18:1CoA)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['for homo sapiens, noted to actually be NADH-cyt b5, using NADH at present NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:10571']},\n",
       "  'metabolites': {'octdececoa_c': 1.0,\n",
       "   'o2_c': -1.0,\n",
       "   'h2o_c': 2.0,\n",
       "   'stcoa_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'nad_c': 1.0,\n",
       "   'nadh_c': -1.0},\n",
       "  'id': 'DESAT18_3',\n",
       "  'gene_reaction_rule': 'HGNC:10571'},\n",
       " {'subsystem': 'Fatty acid synthesis',\n",
       "  'name': 'stearoyl-CoA desaturase (n-C18:0CoA -> n-C18:1CoA)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['for homo sapiens, noted to actually be NADH-cyt b5, using NADH at present NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:10571']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   'octd11ecoa_c': 1.0,\n",
       "   'h2o_c': 2.0,\n",
       "   'stcoa_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'nad_c': 1.0,\n",
       "   'nadh_c': -1.0},\n",
       "  'id': 'DESAT18_4',\n",
       "  'gene_reaction_rule': 'HGNC:10571'},\n",
       " {'subsystem': 'Fatty acid synthesis',\n",
       "  'name': 'stearoyl-CoA desaturase (n-C18:0CoA -> n-C18:1CoA)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['for homo sapiens, noted to actually be NADH-cyt b5, using NADH at present NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:10571']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   'od2coa_c': 1.0,\n",
       "   'h2o_c': 2.0,\n",
       "   'stcoa_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'nad_c': 1.0,\n",
       "   'nadh_c': -1.0},\n",
       "  'id': 'DESAT18_5',\n",
       "  'gene_reaction_rule': 'HGNC:10571'},\n",
       " {'subsystem': 'Fatty acid synthesis',\n",
       "  'name': 'fatty acyl-CoA desaturase (n-C18:1CoA -> n-C18:2CoA)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['for homo sapiens, noted to actually be NADH-cyt b5, using NADH at present NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3575']},\n",
       "  'metabolites': {'octdececoa_c': -1.0,\n",
       "   'o2_c': -1.0,\n",
       "   'h2o_c': 2.0,\n",
       "   'lneldccoa_c': 1.0,\n",
       "   'h_c': -1.0,\n",
       "   'nad_c': 1.0,\n",
       "   'nadh_c': -1.0},\n",
       "  'id': 'DESAT18_6',\n",
       "  'gene_reaction_rule': 'HGNC:3575'},\n",
       " {'subsystem': 'Fatty acid synthesis',\n",
       "  'name': 'fatty acyl-CoA desaturase (n-C18:1CoA -> n-C18:2CoA)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['for homo sapiens, noted to actually be NADH-cyt b5, using NADH at present NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3575']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   'octd11ecoa_c': -1.0,\n",
       "   'h2o_c': 2.0,\n",
       "   'lneldccoa_c': 1.0,\n",
       "   'h_c': -1.0,\n",
       "   'nad_c': 1.0,\n",
       "   'nadh_c': -1.0},\n",
       "  'id': 'DESAT18_7',\n",
       "  'gene_reaction_rule': 'HGNC:3575'},\n",
       " {'subsystem': 'Fatty acid synthesis',\n",
       "  'name': 'fatty acyl-CoA desaturase (n-C18:1CoA -> n-C18:2CoA)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['for homo sapiens, noted to actually be NADH-cyt b5, using NADH at present NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3575']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   'od2coa_c': -1.0,\n",
       "   'h2o_c': 2.0,\n",
       "   'lneldccoa_c': 1.0,\n",
       "   'h_c': -1.0,\n",
       "   'nad_c': 1.0,\n",
       "   'nadh_c': -1.0},\n",
       "  'id': 'DESAT18_8',\n",
       "  'gene_reaction_rule': 'HGNC:3575'},\n",
       " {'subsystem': 'Fatty acid synthesis',\n",
       "  'name': 'fatty acyl-CoA desaturase (n-C18:2CoA -> n-C18:3CoA)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['for homo sapiens, noted to actually be NADH-cyt b5, using NADH at present NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3574 or HGNC:3575']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   'h2o_c': 2.0,\n",
       "   'lnlccoa_c': -1.0,\n",
       "   'lnlncgcoa_c': 1.0,\n",
       "   'h_c': -1.0,\n",
       "   'nad_c': 1.0,\n",
       "   'nadh_c': -1.0},\n",
       "  'id': 'DESAT18_9',\n",
       "  'gene_reaction_rule': 'HGNC:3574 or HGNC:3575'},\n",
       " {'subsystem': 'Fatty acid synthesis',\n",
       "  'name': 'fatty acyl-CoA desaturase (n-C20:3CoA -> n-C20:4CoA)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['for homo sapiens, noted to actually be NADH-cyt b5, using NADH at present NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3574']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   'h2o_c': 2.0,\n",
       "   'dlnlcgcoa_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'nad_c': 1.0,\n",
       "   'arachdcoa_c': 1.0,\n",
       "   'nadh_c': -1.0},\n",
       "  'id': 'DESAT20_1',\n",
       "  'gene_reaction_rule': 'HGNC:3574'},\n",
       " {'subsystem': 'Fatty acid synthesis',\n",
       "  'name': 'fatty acyl-CoA desaturase (n-C20:4CoA -> n-C20:5CoA)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['for homo sapiens, noted to actually be NADH-cyt b5, using NADH at present NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3574']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   'h2o_c': 2.0,\n",
       "   'tmndnccoa_c': 1.0,\n",
       "   'eicostetcoa_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'nad_c': 1.0,\n",
       "   'nadh_c': -1.0},\n",
       "  'id': 'DESAT20_2',\n",
       "  'gene_reaction_rule': 'HGNC:3574'},\n",
       " {'subsystem': 'Fatty acid synthesis',\n",
       "  'name': 'fatty acyl-CoA desaturase (n-C22:4CoA -> n-C22:5CoA)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'nadh_x': -1.0,\n",
       "   'dcsptn1coa_x': 1.0,\n",
       "   'nad_x': 1.0,\n",
       "   'h_x': -1.0,\n",
       "   'adrncoa_x': -1.0,\n",
       "   'h2o_x': 2.0,\n",
       "   'o2_x': -1.0},\n",
       "  'id': 'DESAT22_1p',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Fatty acid synthesis',\n",
       "  'name': 'fatty acyl-CoA desaturase (n-C22:5CoA -> n-C22:6CoA)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'nadh_x': -1.0,\n",
       "   'nad_x': 1.0,\n",
       "   'h_x': -1.0,\n",
       "   'clpndcoa_x': -1.0,\n",
       "   'h2o_x': 2.0,\n",
       "   'o2_x': -1.0,\n",
       "   'c226coa_x': 1.0},\n",
       "  'id': 'DESAT22_2p',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Fatty acid synthesis',\n",
       "  'name': 'fatty acyl-CoA desaturase (n-C24:5CoA -> n-C24:6CoA)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['for homo sapiens, noted to actually be NADH-cyt b5, using NADH at present NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3575']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   'h2o_c': 2.0,\n",
       "   'tethex3coa_c': 1.0,\n",
       "   'h_c': -1.0,\n",
       "   'nad_c': 1.0,\n",
       "   'tetpent3coa_c': -1.0,\n",
       "   'nadh_c': -1.0},\n",
       "  'id': 'DESAT24_1',\n",
       "  'gene_reaction_rule': 'HGNC:3575'},\n",
       " {'subsystem': 'Triacylglycerol synthesis',\n",
       "  'name': 'diacylglycerol acyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['FULLR corresponds to acyl/R groups in texts. NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:16940 or HGNC:2843']},\n",
       "  'metabolites': {'h_c': 4.0,\n",
       "   'Rtotalcoa_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'tag_hs_c': 1.0,\n",
       "   'dag_hs_c': -1.0},\n",
       "  'id': 'DGAT',\n",
       "  'gene_reaction_rule': 'HGNC:16940 or HGNC:2843'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'deoxyguanylate kinase (dGMP:ATP)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:4693']},\n",
       "  'metabolites': {'dgdp_c': 1.0, 'dgmp_c': -1.0, 'adp_c': 1.0, 'atp_c': -1.0},\n",
       "  'id': 'DGK1',\n",
       "  'gene_reaction_rule': 'HGNC:4693'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'deoxyguanylate kinase (dGMP:dATP) (mitochondrial)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['Added EC- JLR IT'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'dgmp_m': -1.0,\n",
       "   'dadp_m': 1.0,\n",
       "   'dgdp_m': 1.0,\n",
       "   'datp_m': -1.0},\n",
       "  'id': 'DGK2m',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'deoxyguanosine kinase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:2858']},\n",
       "  'metabolites': {'adp_m': 1.0,\n",
       "   'atp_m': -1.0,\n",
       "   'dgmp_m': 1.0,\n",
       "   'h_m': 1.0,\n",
       "   'dgsn_m': -1.0},\n",
       "  'id': 'DGNSKm',\n",
       "  'gene_reaction_rule': 'HGNC:2858'},\n",
       " {'subsystem': 'Pentose phosphate pathway',\n",
       "  'name': 'dehydro-L-gulonate decarboxylase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'co2_c': 1.0,\n",
       "   'h_c': -1.0,\n",
       "   '3dhguln_c': -1.0,\n",
       "   'xylu_L_c': 1.0},\n",
       "  'id': 'DGULND',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Triacylglycerol synthesis',\n",
       "  'name': 'dihydroxyacetone phosphate acyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4416']},\n",
       "  'metabolites': {'Rtotalcoa_c': -1.0,\n",
       "   'dhap_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'adhap_hs_c': 1.0,\n",
       "   'h_c': 4.0},\n",
       "  'id': 'DHAPA',\n",
       "  'gene_reaction_rule': 'HGNC:4416'},\n",
       " {'subsystem': 'Triacylglycerol synthesis',\n",
       "  'name': 'dihydroxyacetone phosphate acyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4416']},\n",
       "  'metabolites': {'dhap_x': -1.0,\n",
       "   'adhap_hs_x': 1.0,\n",
       "   'Rtotalcoa_x': -1.0,\n",
       "   'h_x': 4.0,\n",
       "   'coa_x': 1.0},\n",
       "  'id': 'DHAPAx',\n",
       "  'gene_reaction_rule': 'HGNC:4416'},\n",
       " {'subsystem': 'Cholesterol metabolism',\n",
       "  'name': '24-dehydrocholesterol reductase [Precursor]',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['lanosterol -&gt; cholesterol pathway FAD known to be cofactor, other details missing NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2859']},\n",
       "  'metabolites': {'fadh2_r': -1.0,\n",
       "   'zymst_r': -1.0,\n",
       "   'zymstnl_r': 1.0,\n",
       "   'fad_r': 1.0},\n",
       "  'id': 'DHCR241r',\n",
       "  'gene_reaction_rule': 'HGNC:2859'},\n",
       " {'subsystem': 'Cholesterol metabolism',\n",
       "  'name': '24-dehydrocholesterol reductase [Precursor]',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['lanosterol -&gt; cholesterol pathway FAD known to be cofactor, other details missing NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2651 or HGNC:2859']},\n",
       "  'metabolites': {'chlstol_r': -1.0,\n",
       "   'lthstrl_r': 1.0,\n",
       "   'fadh2_r': -1.0,\n",
       "   'fad_r': 1.0},\n",
       "  'id': 'DHCR242r',\n",
       "  'gene_reaction_rule': 'HGNC:2651 or HGNC:2859'},\n",
       " {'subsystem': 'Cholesterol metabolism',\n",
       "  'name': '24-dehydrocholesterol reductase [Precursor]',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['lanosterol -&gt; cholesterol pathway FAD known to be cofactor, other details missing NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2859']},\n",
       "  'metabolites': {'dsmsterol_r': -1.0,\n",
       "   'fadh2_r': -1.0,\n",
       "   'chsterol_r': 1.0,\n",
       "   'fad_r': 1.0},\n",
       "  'id': 'DHCR243r',\n",
       "  'gene_reaction_rule': 'HGNC:2859'},\n",
       " {'subsystem': 'Cholesterol metabolism',\n",
       "  'name': '7-dehydrocholesterol reductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['lanosterol -&gt; cholesterol pathway see paper refs Moebius et al PNAS 95:1899-1902 - check/re-check lit for more specific details (NADPH known to be cofactor - also NADH?, possibly rev?) NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2860']},\n",
       "  'metabolites': {'dsmsterol_r': 1.0,\n",
       "   'ddsmsterol_r': -1.0,\n",
       "   'h_r': -1.0,\n",
       "   'nadph_r': -1.0,\n",
       "   'nadp_r': 1.0},\n",
       "  'id': 'DHCR71r',\n",
       "  'gene_reaction_rule': 'HGNC:2860'},\n",
       " {'subsystem': 'Cholesterol metabolism',\n",
       "  'name': '7-dehydrocholesterol reductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['lanosterol -&gt; cholesterol pathway see paper refs Moebius et al PNAS 95:1899-1902 - check/re-check lit for more specific details (NADPH known to be cofactor - also NADH?, possibly rev?) NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2860']},\n",
       "  'metabolites': {'7dhchsterol_r': -1.0,\n",
       "   'nadp_r': 1.0,\n",
       "   'h_r': -1.0,\n",
       "   'chsterol_r': 1.0,\n",
       "   'nadph_r': -1.0},\n",
       "  'id': 'DHCR72r',\n",
       "  'gene_reaction_rule': 'HGNC:2860'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'dihydroceramide desaturase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:13709 or HGNC:13709 or HGNC:20113']},\n",
       "  'metabolites': {'dhcrm_hs_c': -1.0,\n",
       "   'nadp_c': -1.0,\n",
       "   'nadph_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'crm_hs_c': 1.0},\n",
       "  'id': 'DHCRD1',\n",
       "  'gene_reaction_rule': 'HGNC:13709 or HGNC:13709 or HGNC:20113'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'dihydroceramide desaturase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:13709 or HGNC:13709 or HGNC:20113']},\n",
       "  'metabolites': {'dhcrm_hs_c': -1.0,\n",
       "   'fadh2_c': 1.0,\n",
       "   'crm_hs_c': 1.0,\n",
       "   'fad_c': -1.0},\n",
       "  'id': 'DHCRD2',\n",
       "  'gene_reaction_rule': 'HGNC:13709 or HGNC:13709 or HGNC:20113'},\n",
       " {'subsystem': 'Ubiquinone synthesis',\n",
       "  'name': 'dihydroxydecaprenylbenzoate methyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Used to make Ubiquinone-10 EC  2.1.1.114 IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:18175']},\n",
       "  'metabolites': {'3dpdhb_me_m': 1.0,\n",
       "   'amet_m': -1.0,\n",
       "   'h_m': 1.0,\n",
       "   'ahcys_m': 1.0,\n",
       "   '3dpdhb_m': -1.0},\n",
       "  'id': 'DHDPBMTm',\n",
       "  'gene_reaction_rule': 'HGNC:18175'},\n",
       " {'subsystem': 'Steroid metabolism',\n",
       "  'name': 'Dehydroepiandrosterone sulfotransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:11377 or HGNC:11455 or HGNC:11458']},\n",
       "  'metabolites': {'dhea_c': -1.0,\n",
       "   'pap_c': 1.0,\n",
       "   'dheas_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'paps_c': -1.0},\n",
       "  'id': 'DHEASULT',\n",
       "  'gene_reaction_rule': 'HGNC:11377 or HGNC:11455 or HGNC:11458'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'dihydrofolate reductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'], 'GENE ASSOCIATION': ['HGNC:2861']},\n",
       "  'metabolites': {'dhf_c': -1.0,\n",
       "   'nadp_c': 1.0,\n",
       "   'nadph_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'thf_c': 1.0},\n",
       "  'id': 'DHFR',\n",
       "  'gene_reaction_rule': 'HGNC:2861'},\n",
       " {'subsystem': 'Pyrimidine synthesis',\n",
       "  'name': 'dihydroorotase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:1424']},\n",
       "  'metabolites': {'cbasp_c': 1.0, 'h2o_c': -1.0, 'h_c': 1.0, 'dhor_S_c': -1.0},\n",
       "  'id': 'DHORTS',\n",
       "  'gene_reaction_rule': 'HGNC:1424'},\n",
       " {'subsystem': 'Pyrimidine catabolism',\n",
       "  'name': 'dihydropyrimidinase (5,6-dihydrouracil)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3013 or HGNC:3014 or HGNC:3015']},\n",
       "  'metabolites': {'56dura_c': -1.0, 'h2o_c': -1.0, 'h_c': 1.0, 'cala_c': 1.0},\n",
       "  'id': 'DHPM1',\n",
       "  'gene_reaction_rule': 'HGNC:3013 or HGNC:3014 or HGNC:3015'},\n",
       " {'subsystem': 'Pyrimidine catabolism',\n",
       "  'name': 'dihydropyrimidinase (dihydrothymine)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['Also acts on dihydrothymine and hydantoin. Found not specific evidence in PubMed, but Dihydropyrimidinase is supposed to be in cytosol and catalyze this reaction. Old information'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3013']},\n",
       "  'metabolites': {'56dthm_c': -1.0, 'h2o_c': -1.0, 'h_c': 1.0, '3uib_c': 1.0},\n",
       "  'id': 'DHPM2',\n",
       "  'gene_reaction_rule': 'HGNC:3013'},\n",
       " {'subsystem': 'Tetrahydrobiopterin metabolism',\n",
       "  'name': '6,7-dihydropteridine reductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:9752']},\n",
       "  'metabolites': {'dhbpt_c': -1.0,\n",
       "   'nadh_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'nad_c': 1.0,\n",
       "   'thbpt_c': 1.0},\n",
       "  'id': 'DHPR',\n",
       "  'gene_reaction_rule': 'HGNC:9752'},\n",
       " {'subsystem': 'Tetrahydrobiopterin metabolism',\n",
       "  'name': '6,7-dihydropteridine reduction',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'5mthf_c': -1.0,\n",
       "   'dhbpt_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   'mlthf_c': 1.0,\n",
       "   'thbpt_c': 1.0},\n",
       "  'id': 'DHPR2',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Arginine and Proline Metabolism',\n",
       "  'name': '2,3-diketo-5-methylthio-1-phosphopentane degradation reaction',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['JLR- based on B. subtilis and K. pneumoniae reactions (not sure about the h2o and h)'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'dkmpp_c': -1.0,\n",
       "   '2kmb_c': 1.0,\n",
       "   'h_c': 2.0,\n",
       "   'for_c': 1.0,\n",
       "   'pi_c': 1.0},\n",
       "  'id': 'DKMPPD',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'carnitine O-palmitoyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:18540 or HGNC:2328 or HGNC:2329']},\n",
       "  'metabolites': {'crn_c': -1.0,\n",
       "   'dlnlcgcrn_c': 1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'dlnlcgcoa_c': -1.0},\n",
       "  'id': 'DLNLCGCPT1',\n",
       "  'gene_reaction_rule': 'HGNC:18540 or HGNC:2328 or HGNC:2329'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'carnitine transferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2330']},\n",
       "  'metabolites': {'dlnlcgcoa_m': 1.0,\n",
       "   'coa_m': -1.0,\n",
       "   'dlnlcgcrn_m': -1.0,\n",
       "   'crn_m': 1.0},\n",
       "  'id': 'DLNLCGCPT2',\n",
       "  'gene_reaction_rule': 'HGNC:2330'},\n",
       " {'subsystem': 'Exchange/demand reaction',\n",
       "  'name': 'DM atp(c)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'], 'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   'pi_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'adp_c': 1.0},\n",
       "  'id': 'DM_atp_c_',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Cholesterol metabolism',\n",
       "  'name': 'dimethylallyltranstransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['Human Y/N: Y'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4249']},\n",
       "  'metabolites': {'dmpp_c': -1.0, 'ppi_c': 1.0, 'ipdp_c': -1.0, 'grdp_c': 1.0},\n",
       "  'id': 'DMATT',\n",
       "  'gene_reaction_rule': 'HGNC:4249'},\n",
       " {'subsystem': 'Cholesterol metabolism',\n",
       "  'name': 'dimethylallyltranstransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['peroxisomal version - NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3631']},\n",
       "  'metabolites': {'ipdp_x': -1.0, 'ppi_x': 1.0, 'dmpp_x': -1.0, 'grdp_x': 1.0},\n",
       "  'id': 'DMATTx',\n",
       "  'gene_reaction_rule': 'HGNC:3631'},\n",
       " {'subsystem': 'Glycine, serine, alanine and threonine metabolism',\n",
       "  'name': 'dimethylglycine dehydrogenase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['MM'],\n",
       "   'GENE ASSOCIATION': ['HGNC:24475']},\n",
       "  'metabolites': {'fald_m': 1.0,\n",
       "   'sarcs_m': 1.0,\n",
       "   'h2o_m': -1.0,\n",
       "   'fad_m': -1.0,\n",
       "   'dmgly_m': -1.0,\n",
       "   'fadh2_m': 1.0},\n",
       "  'id': 'DMGDHm',\n",
       "  'gene_reaction_rule': 'HGNC:24475'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'carnitine fatty-acyl transferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['made reversible since Keq ~ 0.6. dmhptcoa is  a special case because its biochemical fate has not be elucidated. NJ'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'crn_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'dmhptcrn_c': 1.0,\n",
       "   'dmhptcoa_c': -1.0},\n",
       "  'id': 'DMHPTCRNCPT1',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': '2,6 dimethylheptanoly CoA carnitine transferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['made reversible since Keq ~ 0.6. dmhptcoa is  a special case because its biochemical fate has not be elucidated. NJ'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'dmhptcrn_m': -1.0,\n",
       "   'dmhptcoa_m': 1.0,\n",
       "   'coa_m': -1.0,\n",
       "   'crn_m': 1.0},\n",
       "  'id': 'DMHPTCRNCPT2',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'DMNONCOACRNCPT1',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'], 'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'crn_c': -1.0,\n",
       "   'dmnoncoa_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'dmnoncrn_c': 1.0},\n",
       "  'id': 'DMNONCOACRNCPT1',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'DMNONCRNCPT2',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'], 'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'coa_m': -1.0,\n",
       "   'dmnoncrn_m': -1.0,\n",
       "   'crn_m': 1.0,\n",
       "   'dmnoncoa_m': 1.0},\n",
       "  'id': 'DMNONCRNCPT2',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Methionine and cysteine metabolism',\n",
       "  'name': 'DNA (cytosine-5-)-methyltransferase, nucleus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2976 or HGNC:2977 or HGNC:2978 or HGNC:2979']},\n",
       "  'metabolites': {'h_n': 1.0,\n",
       "   'dna5mtc_n': 1.0,\n",
       "   'amet_n': -1.0,\n",
       "   'dna_n': -1.0,\n",
       "   'ahcys_n': 1.0},\n",
       "  'id': 'DNAMTn',\n",
       "  'gene_reaction_rule': 'HGNC:2976 or HGNC:2977 or HGNC:2978 or HGNC:2979'},\n",
       " {'subsystem': 'Selenoamino acid metabolism',\n",
       "  'name': 'DNA (cytosine-5-)-methyltransferase, nucleus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['SAB copied from DNAMTn'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2976 or HGNC:2977 or HGNC:2978 or HGNC:2979']},\n",
       "  'metabolites': {'h_n': 1.0,\n",
       "   'seasmet_n': -1.0,\n",
       "   'dna5mtc_n': 1.0,\n",
       "   'dna_n': -1.0,\n",
       "   'seahcys_n': 1.0},\n",
       "  'id': 'DNAMTSEn',\n",
       "  'gene_reaction_rule': 'HGNC:2976 or HGNC:2977 or HGNC:2978 or HGNC:2979'},\n",
       " {'subsystem': 'Vitamin C metabolism',\n",
       "  'name': '2,3-dioxo-L-gulonate decarboxylase (L-lyxonate-forming)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['EC-4.1.1.- NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'23doguln_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'co2_c': 1.0,\n",
       "   'lyxnt_c': 1.0},\n",
       "  'id': 'DOGULND1',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Vitamin C metabolism',\n",
       "  'name': '2,3-dioxo-L-gulonate decarboxylase (L-xylonate-forming)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['EC-4.1.1.- NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'xylnt_c': 1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'co2_c': 1.0,\n",
       "   '23doguln_c': -1.0},\n",
       "  'id': 'DOGULND2',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Vitamin C metabolism',\n",
       "  'name': '2,3-dioxo-L-gulonate:hydrogen peroxide oxireductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'h2o2_c': -1.0,\n",
       "   '23doguln_c': -1.0,\n",
       "   'h_c': 2.0,\n",
       "   'oxa_c': 1.0,\n",
       "   'thrnt_c': 1.0},\n",
       "  'id': 'DOGULNO1',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Vitamin C metabolism',\n",
       "  'name': '2,3-dioxo-L-gulonate:hydrogen peroxide oxireductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'oxa_c': 1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   '23doguln_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   'eryth_c': 1.0},\n",
       "  'id': 'DOGULNO2',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'Dolichyl-diphosphooligosaccharide:protein-L-asparagine oligopolysaccharidotransferase (uterus)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'g3m8mpdol_U_r': -1.0,\n",
       "   'Asn_X_Ser_Thr_r': -1.0,\n",
       "   'g3m8masn_r': 1.0,\n",
       "   'doldp_U_r': 1.0},\n",
       "  'id': 'DOLASNT_Uer',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'Dolichyl-diphosphate phosphohydrolase, human (uterus)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'dolp_U_r': 1.0,\n",
       "   'pi_r': 1.0,\n",
       "   'h2o_r': -1.0,\n",
       "   'h_r': 4.0,\n",
       "   'doldp_U_r': -1.0},\n",
       "  'id': 'DOLDPP_Uer',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'Dolichyl-beta-D-glucosyl-phosphate dolichylphosphohydrolase (uterus)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'], 'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'dolp_U_r': 1.0,\n",
       "   'glc_D_r': 1.0,\n",
       "   'h2o_r': -1.0,\n",
       "   'h_r': 2.0,\n",
       "   'dolglcp_U_r': -1.0},\n",
       "  'id': 'DOLGPP_Uer',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'Dolichol kinase, human (uterus)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'], 'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'ctp_c': -1.0,\n",
       "   'dolp_U_c': 1.0,\n",
       "   'dolichol_U_c': -1.0,\n",
       "   'cdp_c': 1.0,\n",
       "   'h_c': 1.0},\n",
       "  'id': 'DOLK_U',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'dolichyl-phosphate-glucose-glycolipid alpha-glucosyltransferase (liver)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['EC 2.4.1.- NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:23157']},\n",
       "  'metabolites': {'dolp_U_r': 1.0,\n",
       "   'm8mpdol_U_r': -1.0,\n",
       "   'g1m8mpdol_L_r': 1.0,\n",
       "   'h_r': 2.0,\n",
       "   'dolglcp_U_r': -1.0},\n",
       "  'id': 'DOLPGT1_Ler',\n",
       "  'gene_reaction_rule': 'HGNC:23157'},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'dolichyl-phosphate-glucose-glycolipid alpha-glucosyltransferase (uterus)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['EC 2.4.1.- NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'dolp_U_r': 1.0,\n",
       "   'm8mpdol_U_r': -1.0,\n",
       "   'g1m8mpdol_U_r': 1.0,\n",
       "   'h_r': 2.0,\n",
       "   'dolglcp_U_r': -1.0},\n",
       "  'id': 'DOLPGT1_Uer',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'dolichyl-phosphate-glucose-glycolipid alpha-glucosyltransferase (liver)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['EC 2.4.1.- NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:23161']},\n",
       "  'metabolites': {'dolp_U_r': 1.0,\n",
       "   'dolglcp_U_r': -1.0,\n",
       "   'g1m8mpdol_L_r': -1.0,\n",
       "   'g2m8mpdol_L_r': 1.0,\n",
       "   'h_r': 2.0},\n",
       "  'id': 'DOLPGT2_Ler',\n",
       "  'gene_reaction_rule': 'HGNC:23161'},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'dolichyl-phosphate-glucose-glycolipid alpha-glucosyltransferase (uterus)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['EC 2.4.1.- NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'dolp_U_r': 1.0,\n",
       "   'dolglcp_U_r': -1.0,\n",
       "   'g2m8mpdol_U_r': 1.0,\n",
       "   'g1m8mpdol_U_r': -1.0,\n",
       "   'h_r': 2.0},\n",
       "  'id': 'DOLPGT2_Uer',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'dolichyl-phosphate-glucose-glycolipid alpha-glucosyltransferase (liver)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['EC 2.4.1.- NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:23162']},\n",
       "  'metabolites': {'g3m8mpdol_L_r': 1.0,\n",
       "   'dolp_U_r': 1.0,\n",
       "   'g2m8mpdol_L_r': -1.0,\n",
       "   'h_r': 2.0,\n",
       "   'dolglcp_U_r': -1.0},\n",
       "  'id': 'DOLPGT3_Ler',\n",
       "  'gene_reaction_rule': 'HGNC:23162'},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'dolichyl-phosphate-glucose-glycolipid alpha-glucosyltransferase (uterus)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['EC 2.4.1.- NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'dolp_U_r': 1.0,\n",
       "   'g3m8mpdol_U_r': 1.0,\n",
       "   'g2m8mpdol_U_r': -1.0,\n",
       "   'h_r': 2.0,\n",
       "   'dolglcp_U_r': -1.0},\n",
       "  'id': 'DOLPGT3_Uer',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'Dolichol-phosphate phosphohydrolase, human (uterus)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'], 'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'dolp_U_r': -1.0,\n",
       "   'pi_r': 1.0,\n",
       "   'h2o_r': -1.0,\n",
       "   'dolichol_U_r': 1.0},\n",
       "  'id': 'DOLPH_Uer',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'Dolichyl-phosphate D-mannosyltransferase (uterus)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'dolp_U_c': -1.0,\n",
       "   'gdpmann_c': -1.0,\n",
       "   'dolmanp_U_c': 1.0,\n",
       "   'h_c': -1.0,\n",
       "   'gdp_c': 1.0},\n",
       "  'id': 'DOLPMT_U',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'dolichyl-phosphate-mannose-glycolipid alpha-mannosyltransferase (uterus)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'dolp_U_r': 1.0,\n",
       "   'h_r': 2.0,\n",
       "   'dolmanp_U_r': -1.0,\n",
       "   'm5mpdol_U_r': 1.0,\n",
       "   'm4mpdol_U_r': -1.0},\n",
       "  'id': 'DOLPMT1_Uer',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'dolichyl-phosphate-mannose-glycolipid alpha-mannosyltransferase (uterus)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'], 'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'dolp_U_r': 1.0,\n",
       "   'm6mpdol_U_r': 1.0,\n",
       "   'h_r': 2.0,\n",
       "   'dolmanp_U_r': -1.0,\n",
       "   'm5mpdol_U_r': -1.0},\n",
       "  'id': 'DOLPMT2_Uer',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'dolichyl-phosphate-mannose-glycolipid alpha-mannosyltransferase (uterus)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'dolp_U_r': 1.0,\n",
       "   'h_r': 2.0,\n",
       "   'm6mpdol_U_r': -1.0,\n",
       "   'm7mpdol_U_r': 1.0,\n",
       "   'dolmanp_U_r': -1.0},\n",
       "  'id': 'DOLPMT3_Uer',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'dolichyl-phosphate-mannose-glycolipid alpha-mannosyltransferase (uterus)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'], 'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'dolp_U_r': 1.0,\n",
       "   'm7mpdol_U_r': -1.0,\n",
       "   'h_r': 2.0,\n",
       "   'dolmanp_U_r': -1.0,\n",
       "   'm8mpdol_U_r': 1.0},\n",
       "  'id': 'DOLPMT4_Uer',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Tyrosine metabolism',\n",
       "  'name': 'dopamine beta-monooxygenase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:21063 or HGNC:2689']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   'h2o_c': 1.0,\n",
       "   'dopa_c': -1.0,\n",
       "   'dhdascb_c': 1.0,\n",
       "   'ascb_L_c': -1.0,\n",
       "   'nrpphr_c': 1.0},\n",
       "  'id': 'DOPABMO',\n",
       "  'gene_reaction_rule': 'HGNC:21063 or HGNC:2689'},\n",
       " {'subsystem': 'Tyrosine metabolism',\n",
       "  'name': 'L-dopachrome isomerase 1',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:12442']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   'h2o_c': 2.0,\n",
       "   '2c23dh56dhoxin_c': -2.0,\n",
       "   'L_dpchrm_c': 2.0},\n",
       "  'id': 'DOPACHRMISO',\n",
       "  'gene_reaction_rule': 'HGNC:12442'},\n",
       " {'subsystem': 'Tyrosine metabolism',\n",
       "  'name': 'Dopamine:amet O-methyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2228']},\n",
       "  'metabolites': {'amet_c': -1.0,\n",
       "   'ahcys_c': 1.0,\n",
       "   '3moxtyr_c': 1.0,\n",
       "   'dopa_c': -1.0,\n",
       "   'h_c': 1.0},\n",
       "  'id': 'DOPAMT',\n",
       "  'gene_reaction_rule': 'HGNC:2228'},\n",
       " {'subsystem': 'Tyrosine metabolism',\n",
       "  'name': 'Dopaquinone isomerase 1',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'dopaqn_c': -1.0, '2c23dh56dhoxin_c': 1.0, 'h_c': 1.0},\n",
       "  'id': 'DOPAQNISO1',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Tyrosine metabolism',\n",
       "  'name': 'Dopamine Sulfotransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['RS/TV (6/14/2005)'],\n",
       "   'GENE ASSOCIATION': ['HGNC:11453 or HGNC:11454 or HGNC:11455']},\n",
       "  'metabolites': {'h_c': 1.0,\n",
       "   'pap_c': 1.0,\n",
       "   'paps_c': -1.0,\n",
       "   'dopa_c': -1.0,\n",
       "   'dopasf_c': 1.0},\n",
       "  'id': 'DOPASULT',\n",
       "  'gene_reaction_rule': 'HGNC:11453 or HGNC:11454 or HGNC:11455'},\n",
       " {'subsystem': 'Arginine and Proline Metabolism',\n",
       "  'name': 'D-ornithine oxidase, perixosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2671']},\n",
       "  'metabolites': {'nh4_x': 1.0,\n",
       "   'orn_D_x': -1.0,\n",
       "   'h2o2_x': 1.0,\n",
       "   '5a2opntn_x': 1.0,\n",
       "   'h2o_x': -1.0,\n",
       "   'o2_x': -1.0},\n",
       "  'id': 'DORNOp',\n",
       "  'gene_reaction_rule': 'HGNC:2671'},\n",
       " {'subsystem': 'CoA synthesis',\n",
       "  'name': 'dephospho-CoA kinase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:29932']},\n",
       "  'metabolites': {'dpcoa_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'adp_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'h_c': 1.0},\n",
       "  'id': 'DPCOAK',\n",
       "  'gene_reaction_rule': 'HGNC:29932'},\n",
       " {'subsystem': 'Glycolysis/gluconeogenesis',\n",
       "  'name': 'Diphosphoglycerate phosphatase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['added EC num; also EC 5.4.2.4 -- NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1093 or HGNC:8888 or HGNC:8889']},\n",
       "  'metabolites': {'h2o_c': -1.0, 'pi_c': 1.0, '23dpg_c': -1.0, '3pg_c': 1.0},\n",
       "  'id': 'DPGase',\n",
       "  'gene_reaction_rule': 'HGNC:1093 or HGNC:8888 or HGNC:8889'},\n",
       " {'subsystem': 'Glycolysis/gluconeogenesis',\n",
       "  'name': 'Diphosphoglyceromutase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1093 or HGNC:8888 or HGNC:8889']},\n",
       "  'metabolites': {'h_c': 1.0, '13dpg_c': -1.0, '23dpg_c': 1.0},\n",
       "  'id': 'DPGM',\n",
       "  'gene_reaction_rule': 'HGNC:1093 or HGNC:8888 or HGNC:8889'},\n",
       " {'subsystem': 'Ubiquinone synthesis',\n",
       "  'name': '3-Decaprenyl-4-hydroxy-5-methoxybenzoate  decarboxylation',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['enzyme has not been identified yet IT'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'3dpdhb_me_m': -1.0,\n",
       "   'h_m': -1.0,\n",
       "   'co2_m': 1.0,\n",
       "   '2dp6mep_m': 1.0},\n",
       "  'id': 'DPHMBDCm',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Cholesterol metabolism',\n",
       "  'name': 'diphosphomevalonate decarboxylase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['peroxisomal version - NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:7529']},\n",
       "  'metabolites': {'co2_x': 1.0,\n",
       "   'pi_x': 1.0,\n",
       "   'adp_x': 1.0,\n",
       "   'ipdp_x': 1.0,\n",
       "   '5dpmev_x': -1.0,\n",
       "   'atp_x': -1.0},\n",
       "  'id': 'DPMVDx',\n",
       "  'gene_reaction_rule': 'HGNC:7529'},\n",
       " {'subsystem': 'Miscellaneous',\n",
       "  'name': 'decaprenyl synthase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['by analogy to other dimethyl transferase rxns. See PMID: 10586494, human interemidate (from C15 -&gt;C50) not explicitly determined yet, however it is known to occur since this pathway is essential in Q10 synthesis. For human metabolism: may be associated w/ GeneID: 23590 TPRT - recheck lit in future IT and NJ'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'ppi_c': 7.0,\n",
       "   'ipdp_c': -7.0,\n",
       "   'frdp_c': -1.0,\n",
       "   'decdp_c': 1.0},\n",
       "  'id': 'DPPS',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Arginine and Proline Metabolism',\n",
       "  'name': 'D-proline oxidase, perixosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['MM. rat brain activity, presumed to be in mammalians'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2671']},\n",
       "  'metabolites': {'o2_x': -1.0,\n",
       "   '1p2cbxl_x': 1.0,\n",
       "   'pro_D_x': -1.0,\n",
       "   'h_x': 1.0,\n",
       "   'h2o2_x': 1.0},\n",
       "  'id': 'DPROOp',\n",
       "  'gene_reaction_rule': 'HGNC:2671'},\n",
       " {'subsystem': 'Pentose phosphate pathway',\n",
       "  'name': 'Deoxyribokinase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:30325']},\n",
       "  'metabolites': {'2dr5p_c': 1.0,\n",
       "   'adp_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'drib_c': -1.0},\n",
       "  'id': 'DRBK',\n",
       "  'gene_reaction_rule': 'HGNC:30325'},\n",
       " {'subsystem': 'Pentose phosphate pathway',\n",
       "  'name': 'deoxyribose-phosphate aldolase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'], 'GENE ASSOCIATION': ['HGNC:24269']},\n",
       "  'metabolites': {'acald_c': 1.0, 'g3p_c': 1.0, '2dr5p_c': -1.0},\n",
       "  'id': 'DRPA',\n",
       "  'gene_reaction_rule': 'HGNC:24269'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'dihydrosphingosine N-acyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['CoA transfer not explicitly in rxn (accounted for in Rtotal rxn -&gt; ARTF) NJ'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'dhcrm_hs_c': 1.0,\n",
       "   'Rtotalcoa_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'sphgn_c': -1.0,\n",
       "   'h_c': 5.0},\n",
       "  'id': 'DSAT',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'dTMP kinase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:3061']},\n",
       "  'metabolites': {'dtmp_c': -1.0, 'dtdp_c': 1.0, 'adp_c': 1.0, 'atp_c': -1.0},\n",
       "  'id': 'DTMPK',\n",
       "  'gene_reaction_rule': 'HGNC:3061'},\n",
       " {'subsystem': 'Pyrimidine catabolism',\n",
       "  'name': 'dihydrouracil dehydrogenase (NADP)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['_01-26-05 IT based on Lu et al., JBC,1992,24,17102-17109'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3012']},\n",
       "  'metabolites': {'ura_c': 1.0,\n",
       "   'nadp_c': -1.0,\n",
       "   'nadph_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   '56dura_c': -1.0},\n",
       "  'id': 'DURAD',\n",
       "  'gene_reaction_rule': 'HGNC:3012'},\n",
       " {'subsystem': 'Pyrimidine catabolism',\n",
       "  'name': 'dihydrothymin dehydrogenase (NADP)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['_01-26-05 IT based on Lu et al., JBC,1992,24,17102-17109'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3012']},\n",
       "  'metabolites': {'56dthm_c': -1.0,\n",
       "   'nadp_c': -1.0,\n",
       "   'nadph_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'thym_c': 1.0},\n",
       "  'id': 'DURAD2',\n",
       "  'gene_reaction_rule': 'HGNC:3012'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'deoxyuridine kinase (ATP:Deoxyuridine)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'GENE ASSOCIATION': ['HGNC:11830 or HGNC:11831']},\n",
       "  'metabolites': {'dump_c': 1.0,\n",
       "   'duri_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'adp_c': 1.0},\n",
       "  'id': 'DURIK1',\n",
       "  'gene_reaction_rule': 'HGNC:11830 or HGNC:11831'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'deoxyuridine kinase (ATP:Deoxyuridine), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'], 'GENE ASSOCIATION': ['HGNC:11831']},\n",
       "  'metabolites': {'dump_m': 1.0,\n",
       "   'duri_m': -1.0,\n",
       "   'atp_m': -1.0,\n",
       "   'adp_m': 1.0,\n",
       "   'h_m': 1.0},\n",
       "  'id': 'DURIK1m',\n",
       "  'gene_reaction_rule': 'HGNC:11831'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'deoxyuridine phosphorylase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['JLR'],\n",
       "   'GENE ASSOCIATION': ['HGNC:23061 or HGNC:7892']},\n",
       "  'metabolites': {'ura_c': 1.0, '2dr1p_c': 1.0, 'pi_c': -1.0, 'duri_c': -1.0},\n",
       "  'id': 'DURIPP',\n",
       "  'gene_reaction_rule': 'HGNC:23061 or HGNC:7892'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'dUTP diphosphatase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3078 or HGNC:6176']},\n",
       "  'metabolites': {'dump_m': 1.0,\n",
       "   'h2o_m': -1.0,\n",
       "   'h_m': 1.0,\n",
       "   'ppi_m': 1.0,\n",
       "   'dutp_m': -1.0},\n",
       "  'id': 'DUTPDPm',\n",
       "  'gene_reaction_rule': 'HGNC:3078 or HGNC:6176'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'dUTP diphosphatase, nuclear',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:3078']},\n",
       "  'metabolites': {'dutp_n': -1.0,\n",
       "   'h_n': 1.0,\n",
       "   'dump_n': 1.0,\n",
       "   'h2o_n': -1.0,\n",
       "   'ppi_n': 1.0},\n",
       "  'id': 'DUTPDPn',\n",
       "  'gene_reaction_rule': 'HGNC:3078'},\n",
       " {'subsystem': 'Cholesterol metabolism',\n",
       "  'name': '3-beta-hydroxysteroid-delta(8),delta(7)-isomerase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['involved in lanosterol -&gt; cholesterol pathway NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3133']},\n",
       "  'metabolites': {'chlstol_r': 1.0, 'zymst_r': -1.0},\n",
       "  'id': 'EBP1r',\n",
       "  'gene_reaction_rule': 'HGNC:3133'},\n",
       " {'subsystem': 'Cholesterol metabolism',\n",
       "  'name': '3-beta-hydroxysteroid-delta(8),delta(7)-isomerase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['involved in lanosterol -&gt; cholesterol pathway NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3133']},\n",
       "  'metabolites': {'lthstrl_r': 1.0, 'zymstnl_r': -1.0},\n",
       "  'id': 'EBP2r',\n",
       "  'gene_reaction_rule': 'HGNC:3133'},\n",
       " {'subsystem': 'Alkaloid synthesis',\n",
       "  'name': 'Ecgonine isomerase (ER)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['SAB  reaction name uncertain'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'pecgon_r': 1.0, 'ecgon_r': -1.0},\n",
       "  'id': 'ECGISOr',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Valine, leucine, and isoleucine metabolism',\n",
       "  'name': '3-hydroxyacyl-CoA dehydratase (3-hydroxyisobutyryl-CoA) (mitochondria)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['MM. based on assigned short chain enoyl-CoA hydratase activity'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3151 or (HGNC:4801 and HGNC:4803)']},\n",
       "  'metabolites': {'3hibutcoa_m': 1.0, 'h2o_m': -1.0, '2mp2coa_m': -1.0},\n",
       "  'id': 'ECOAH12m',\n",
       "  'gene_reaction_rule': 'HGNC:3151 or (HGNC:4801 and HGNC:4803)'},\n",
       " {'subsystem': 'Tryptophan metabolism',\n",
       "  'name': '3-hydroxyacyl-CoA dehydratase (3-hydroxybutanoyl-CoA) (mitochondria)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['SAB for mito'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3151 or (HGNC:4801 and HGNC:4803) or HGNC:890']},\n",
       "  'metabolites': {'h2o_m': 1.0, 'b2coa_m': 1.0, '3hbcoa_m': -1.0},\n",
       "  'id': 'ECOAH1m',\n",
       "  'gene_reaction_rule': 'HGNC:3151 or (HGNC:4801 and HGNC:4803) or HGNC:890'},\n",
       " {'subsystem': 'Tryptophan metabolism',\n",
       "  'name': '3-hydroxyacyl-CoA dehydratase (3-hydroxybutanoyl-CoA) (perioxisome)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['SAB for x'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3149 or HGNC:3247']},\n",
       "  'metabolites': {'h2o_x': 1.0, '3hbcoa_x': -1.0, 'b2coa_x': 1.0},\n",
       "  'id': 'ECOAH1x',\n",
       "  'gene_reaction_rule': 'HGNC:3149 or HGNC:3247'},\n",
       " {'subsystem': 'Valine, leucine, and isoleucine metabolism',\n",
       "  'name': '2-Methylprop-2-enoyl-CoA (2-Methylbut-2-enoyl-CoA), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['MM. based on assigned short chain enoyl-CoA hydratase activity'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3151 or (HGNC:4801 and HGNC:4803)']},\n",
       "  'metabolites': {'h2o_m': -1.0, '3hmbcoa_m': 1.0, '2mb2coa_m': -1.0},\n",
       "  'id': 'ECOAH9m',\n",
       "  'gene_reaction_rule': 'HGNC:3151 or (HGNC:4801 and HGNC:4803)'},\n",
       " {'subsystem': 'Alkaloid synthesis',\n",
       "  'name': 'ecgonine methyl esterase (ER)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['SAB  EC 3.1.1.-'],\n",
       "   'GENE ASSOCIATION': ['HGNC:17']},\n",
       "  'metabolites': {'ecgon_r': 1.0,\n",
       "   'h2o_r': -1.0,\n",
       "   'h_r': 1.0,\n",
       "   'meoh_r': 1.0,\n",
       "   'egme_r': -1.0},\n",
       "  'id': 'EGMESTr',\n",
       "  'gene_reaction_rule': 'HGNC:17'},\n",
       " {'subsystem': 'Arginine and Proline Metabolism',\n",
       "  'name': 'L-erythro-4-Hydroxyglutamate:2-oxoglutarate aminotransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['NCD. measured in rat liver and activity is presumed to be in mammalians; substrate erythro-4-hydroxyglutamate is found in humans'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4432']},\n",
       "  'metabolites': {'akg_c': -1.0,\n",
       "   'e4hglu_c': -1.0,\n",
       "   'glu_L_c': 1.0,\n",
       "   '4h2oglt_c': 1.0},\n",
       "  'id': 'EHGLAT',\n",
       "  'gene_reaction_rule': 'HGNC:4432'},\n",
       " {'subsystem': 'Arginine and Proline Metabolism',\n",
       "  'name': 'L-erythro-4-Hydroxyglutamate:2-oxoglutarate aminotransferase 2, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD SAB oaa -&gt; asp-L'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4433']},\n",
       "  'metabolites': {'oaa_m': -1.0,\n",
       "   '4h2oglt_m': 1.0,\n",
       "   'e4hglu_m': -1.0,\n",
       "   'asp_L_m': 1.0},\n",
       "  'id': 'EHGLAT2m',\n",
       "  'gene_reaction_rule': 'HGNC:4433'},\n",
       " {'subsystem': 'Arginine and Proline Metabolism',\n",
       "  'name': 'L-erythro-4-Hydroxyglutamate:2-oxoglutarate aminotransferase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4433']},\n",
       "  'metabolites': {'glu_L_m': 1.0,\n",
       "   '4h2oglt_m': 1.0,\n",
       "   'e4hglu_m': -1.0,\n",
       "   'akg_m': -1.0},\n",
       "  'id': 'EHGLATm',\n",
       "  'gene_reaction_rule': 'HGNC:4433'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'carnitine O-palmitoyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:18540 or HGNC:2328 or HGNC:2329']},\n",
       "  'metabolites': {'crn_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'eicostetcoa_c': -1.0,\n",
       "   'eicostetcrn_c': 1.0},\n",
       "  'id': 'EICOSTETCPT1',\n",
       "  'gene_reaction_rule': 'HGNC:18540 or HGNC:2328 or HGNC:2329'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'carnitine transferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2330']},\n",
       "  'metabolites': {'eicostetcrn_m': -1.0,\n",
       "   'eicostetcoa_m': 1.0,\n",
       "   'coa_m': -1.0,\n",
       "   'crn_m': 1.0},\n",
       "  'id': 'EICOSTETCPT2',\n",
       "  'gene_reaction_rule': 'HGNC:2330'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'carnitine O-palmitoyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:18540 or HGNC:2328 or HGNC:2329']},\n",
       "  'metabolites': {'crn_c': -1.0,\n",
       "   'od2coa_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'elaidcrn_c': 1.0},\n",
       "  'id': 'ELAIDCPT1',\n",
       "  'gene_reaction_rule': 'HGNC:18540 or HGNC:2328 or HGNC:2329'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'carnitine transferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2330']},\n",
       "  'metabolites': {'elaidcrn_m': -1.0,\n",
       "   'coa_m': -1.0,\n",
       "   'od2coa_m': 1.0,\n",
       "   'crn_m': 1.0},\n",
       "  'id': 'ELAIDCPT2',\n",
       "  'gene_reaction_rule': 'HGNC:2330'},\n",
       " {'subsystem': 'N-glycan degradation',\n",
       "  'name': 'endo-beta-N-acetylglucosaminidase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:24622']},\n",
       "  'metabolites': {'acgam_c': 1.0,\n",
       "   's2l2n2m2m_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   's2l2n2m2mn_c': 1.0},\n",
       "  'id': 'ENGASE',\n",
       "  'gene_reaction_rule': 'HGNC:24622'},\n",
       " {'subsystem': 'N-glycan degradation',\n",
       "  'name': 'endo-beta-N-acetylglucosaminidase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:24622']},\n",
       "  'metabolites': {'acgam_c': 1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'n2m2nmn_c': 1.0,\n",
       "   'n2m2nm_c': -1.0},\n",
       "  'id': 'ENGASE2',\n",
       "  'gene_reaction_rule': 'HGNC:24622'},\n",
       " {'subsystem': 'N-glycan degradation',\n",
       "  'name': 'endo-beta-N-acetylglucosaminidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'n2m2nm_l': -1.0,\n",
       "   'n2m2nmn_l': 1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   'acgam_l': 1.0},\n",
       "  'id': 'ENGASE2ly',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Keratan sulfate degradation',\n",
       "  'name': 'endo-beta-N-acetylglucosaminidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'ksi_deg3_l': 1.0,\n",
       "   'ksi_deg2_l': -1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   'acgam_l': 1.0},\n",
       "  'id': 'ENGASE3ly',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'N-glycan degradation',\n",
       "  'name': 'endo-beta-N-acetylglucosaminidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'acgam_l': 1.0,\n",
       "   's2l2n2m2mn_l': 1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   's2l2n2m2m_l': -1.0},\n",
       "  'id': 'ENGASEly',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'Endomannosidase (glc1man-producing), Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'glc1man_g': 1.0,\n",
       "   'm7masnA_g': 1.0,\n",
       "   'h2o_g': -1.0,\n",
       "   'g1m8masn_g': -1.0},\n",
       "  'id': 'ENMAN1g',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'Endomannosidase (glc2man-producing), Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'m7masnA_g': 1.0,\n",
       "   'g2m8masn_g': -1.0,\n",
       "   'h2o_g': -1.0,\n",
       "   'glc2man_g': 1.0},\n",
       "  'id': 'ENMAN2g',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'Endomannosidase (glc3man-producing), Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'m7masnA_g': 1.0,\n",
       "   'glc3man_g': 1.0,\n",
       "   'g3m8masn_g': -1.0,\n",
       "   'h2o_g': -1.0},\n",
       "  'id': 'ENMAN3g',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'Endomannosidase (m6masnC-producing), Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'glc1man_g': 1.0,\n",
       "   'g1m7masnC_g': -1.0,\n",
       "   'm6masnC_g': 1.0,\n",
       "   'h2o_g': -1.0},\n",
       "  'id': 'ENMAN4g',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'Endomannosidase (m6masnB2-producing), Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'glc1man_g': 1.0,\n",
       "   'g1m7masnB_g': -1.0,\n",
       "   'm6masnB2_g': 1.0,\n",
       "   'h2o_g': -1.0},\n",
       "  'id': 'ENMAN5g',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'Endomannosidase (m5masnB1-producing), Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'glc1man_g': 1.0,\n",
       "   'g1m6masnB1_g': -1.0,\n",
       "   'm5masnB1_g': 1.0,\n",
       "   'h2o_g': -1.0},\n",
       "  'id': 'ENMAN6g',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Glycolysis/gluconeogenesis',\n",
       "  'name': 'enolase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['SMP'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3350 or HGNC:3354 or HGNC:3354 or HGNC:3353']},\n",
       "  'metabolites': {'pep_c': 1.0, 'h2o_c': 1.0, '2pg_c': -1.0},\n",
       "  'id': 'ENO',\n",
       "  'gene_reaction_rule': 'HGNC:3350 or HGNC:3354 or HGNC:3354 or HGNC:3353'},\n",
       " {'subsystem': 'Glycerophospholipid metabolism',\n",
       "  'name': 'ethanolamine-phosphate cytidylyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:8756']},\n",
       "  'metabolites': {'2ameph_c': -1.0,\n",
       "   'ctp_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'cmp2amep_c': 1.0,\n",
       "   'h_c': -1.0},\n",
       "  'id': 'EPCTX',\n",
       "  'gene_reaction_rule': 'HGNC:8756'},\n",
       " {'subsystem': 'Steroid metabolism',\n",
       "  'name': 'Estrogen sulfotransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:11377 or HGNC:11453 or HGNC:11455']},\n",
       "  'metabolites': {'estrone_c': -1.0,\n",
       "   'pap_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'paps_c': -1.0,\n",
       "   'estrones_c': 1.0},\n",
       "  'id': 'ESTSULT',\n",
       "  'gene_reaction_rule': 'HGNC:11377 or HGNC:11453 or HGNC:11455'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'electron transfer flavoprotein',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['TV (5/4/05) Voet p.571'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3481 and HGNC:3482']},\n",
       "  'metabolites': {'fad_m': 1.0,\n",
       "   'fadh2_m': -1.0,\n",
       "   'etfox_m': -1.0,\n",
       "   'etfrd_m': 1.0},\n",
       "  'id': 'ETF',\n",
       "  'gene_reaction_rule': 'HGNC:3481 and HGNC:3482'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'Electron transfer flavoprotein-ubiquinone oxidoreductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['TV Ruzicka F, JBC, 1977'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3483']},\n",
       "  'metabolites': {'q10h2_m': 1.0,\n",
       "   'etfox_m': 1.0,\n",
       "   'etfrd_m': -1.0,\n",
       "   'q10_m': -1.0},\n",
       "  'id': 'ETFQO',\n",
       "  'gene_reaction_rule': 'HGNC:3483'},\n",
       " {'subsystem': 'Glycerophospholipid metabolism',\n",
       "  'name': 'Ethanolamine kinase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1937 or HGNC:24649 or HGNC:1937 or HGNC:25575 or HGNC:1938 or HGNC:1938']},\n",
       "  'metabolites': {'etha_c': -1.0,\n",
       "   'ethamp_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'adp_c': 1.0},\n",
       "  'id': 'ETHAK',\n",
       "  'gene_reaction_rule': 'HGNC:1937 or HGNC:24649 or HGNC:1937 or HGNC:25575 or HGNC:1938 or HGNC:1938'},\n",
       " {'subsystem': 'Glycerophospholipid metabolism',\n",
       "  'name': 'ethanolamine phosphate phosphatase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['no high energy phosphate compounds found associated w/ rxn NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:16815']},\n",
       "  'metabolites': {'h2o_c': -1.0, 'pi_c': 1.0, 'etha_c': 1.0, 'ethamp_c': -1.0},\n",
       "  'id': 'ETHP',\n",
       "  'gene_reaction_rule': 'HGNC:16815'},\n",
       " {'subsystem': 'Glycolysis/gluconeogenesis',\n",
       "  'name': 'ethanol monooxygenase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD. 10620320 - expression, fluorescence; 17084997 - biochemical'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2631']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   'h2o_c': 2.0,\n",
       "   'nadp_c': 1.0,\n",
       "   'nadph_c': -1.0,\n",
       "   'acald_c': 1.0,\n",
       "   'h_c': -1.0,\n",
       "   'etoh_c': -1.0},\n",
       "  'id': 'ETOHMO',\n",
       "  'gene_reaction_rule': 'HGNC:2631'},\n",
       " {'subsystem': 'Fructose and mannose metabolism',\n",
       "  'name': 'fucose-1-phosphate guanylyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:19661 or HGNC:3825']},\n",
       "  'metabolites': {'gdpfuc_c': 1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'gtp_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'fuc1p_L_c': -1.0},\n",
       "  'id': 'F1PGT',\n",
       "  'gene_reaction_rule': 'HGNC:19661 or HGNC:3825'},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'glycoprotein 6-alpha-L-fucosyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4019']},\n",
       "  'metabolites': {'gdp_g': 1.0,\n",
       "   'fn2m2masn_g': 1.0,\n",
       "   'n2m2masn_g': -1.0,\n",
       "   'h2o_g': -1.0,\n",
       "   'h_g': 1.0,\n",
       "   'gdpfuc_g': -1.0},\n",
       "  'id': 'F6Tg',\n",
       "  'gene_reaction_rule': 'HGNC:4019'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acyl-ACP hydrolase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD rxn is now charge balanced - NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:25625']},\n",
       "  'metabolites': {'ddca_c': 1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'ACP_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'ddcaACP_c': -1.0},\n",
       "  'id': 'FA120ACPH',\n",
       "  'gene_reaction_rule': 'HGNC:25625'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acyl-ACP hydrolase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD rxn is now charge balanced - NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:25625']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   'ACP_c': 1.0,\n",
       "   'ttdca_c': 1.0,\n",
       "   'myrsACP_c': -1.0,\n",
       "   'h_c': 1.0},\n",
       "  'id': 'FA140ACPH',\n",
       "  'gene_reaction_rule': 'HGNC:25625'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acyl-ACP hydrolase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:25625']},\n",
       "  'metabolites': {'tdeACP_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'ACP_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'ttdcea_c': 1.0},\n",
       "  'id': 'FA141ACPH',\n",
       "  'gene_reaction_rule': 'HGNC:25625'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acyl-ACP hydrolase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:25625 or HGNC:3594']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   'ACP_c': 1.0,\n",
       "   'hdca_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'palmACP_c': -1.0},\n",
       "  'id': 'FA160ACPH',\n",
       "  'gene_reaction_rule': 'HGNC:25625 or HGNC:3594'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acyl-ACP hydrolase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:25625']},\n",
       "  'metabolites': {'hdcea_c': 1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'ACP_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'hdeACP_c': -1.0},\n",
       "  'id': 'FA161ACPH',\n",
       "  'gene_reaction_rule': 'HGNC:25625'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acyl-ACP hydrolase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:25625']},\n",
       "  'metabolites': {'ocdcaACP_c': -1.0,\n",
       "   'ACP_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'ocdca_c': 1.0},\n",
       "  'id': 'FA180ACPH',\n",
       "  'gene_reaction_rule': 'HGNC:25625'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acyl-ACP hydrolase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:25625']},\n",
       "  'metabolites': {'octeACP_c': -1.0,\n",
       "   'ocdcea_c': 1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'ACP_c': 1.0,\n",
       "   'h_c': 1.0},\n",
       "  'id': 'FA181ACPH',\n",
       "  'gene_reaction_rule': 'HGNC:25625'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acyl-ACP hydrolase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'lnlcACP_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'ACP_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'lnlc_c': 1.0},\n",
       "  'id': 'FA1821ACPH',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acyl-ACP hydrolase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'lneldc_c': 1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'ACP_c': 1.0,\n",
       "   'lneldcACP_c': -1.0,\n",
       "   'h_c': 1.0},\n",
       "  'id': 'FA1822ACPH',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acyl-ACP hydrolase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:25625']},\n",
       "  'metabolites': {'ocdcyaACP_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'ACP_c': 1.0,\n",
       "   'ocdcya_c': 1.0,\n",
       "   'h_c': 1.0},\n",
       "  'id': 'FA182ACPH',\n",
       "  'gene_reaction_rule': 'HGNC:25625'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase (myristate, n-C14:0)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['JLR'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3570']},\n",
       "  'metabolites': {'atp_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0,\n",
       "   'ttdca_c': -1.0,\n",
       "   'tdcoa_c': 1.0},\n",
       "  'id': 'FACOAL140i',\n",
       "  'gene_reaction_rule': 'HGNC:3570'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['AR'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3571 or HGNC:3569 or HGNC:3570 or HGNC:3570 or HGNC:16496 or HGNC:16496']},\n",
       "  'metabolites': {'atp_c': -1.0,\n",
       "   'ptdca_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'ptdcacoa_c': 1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0},\n",
       "  'id': 'FACOAL150',\n",
       "  'gene_reaction_rule': 'HGNC:3571 or HGNC:3569 or HGNC:3570 or HGNC:3570 or HGNC:16496 or HGNC:16496'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase (palmitate, n-C16:0)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['TV catalyzed by acyl-CoA synthetases'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3571 or HGNC:3569 or HGNC:3570 or HGNC:3570 or HGNC:16496 or HGNC:16496']},\n",
       "  'metabolites': {'atp_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'pmtcoa_c': 1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0,\n",
       "   'hdca_c': -1.0},\n",
       "  'id': 'FACOAL160i',\n",
       "  'gene_reaction_rule': 'HGNC:3571 or HGNC:3569 or HGNC:3570 or HGNC:3570 or HGNC:16496 or HGNC:16496'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase (hexadecenoate)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'GENE ASSOCIATION': ['HGNC:24174 or HGNC:3569']},\n",
       "  'metabolites': {'hdcoa_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'hdcea_c': -1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0},\n",
       "  'id': 'FACOAL161',\n",
       "  'gene_reaction_rule': 'HGNC:24174 or HGNC:3569'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['AR'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3571 or HGNC:3569 or HGNC:3570 or HGNC:3570 or HGNC:16496 or HGNC:16496']},\n",
       "  'metabolites': {'atp_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0,\n",
       "   'hpdcacoa_c': 1.0,\n",
       "   'hpdca_c': -1.0},\n",
       "  'id': 'FACOAL170',\n",
       "  'gene_reaction_rule': 'HGNC:3571 or HGNC:3569 or HGNC:3570 or HGNC:3570 or HGNC:16496 or HGNC:16496'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase (stearate, n-C18:0)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['TV catalyzed by acyl-CoA synthetases'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3571 or HGNC:3569 or HGNC:3570 or HGNC:3570 or HGNC:16496 or HGNC:16496']},\n",
       "  'metabolites': {'atp_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'stcoa_c': 1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0,\n",
       "   'ocdca_c': -1.0},\n",
       "  'id': 'FACOAL180i',\n",
       "  'gene_reaction_rule': 'HGNC:3571 or HGNC:3569 or HGNC:3570 or HGNC:3570 or HGNC:16496 or HGNC:16496'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3569']},\n",
       "  'metabolites': {'octd11ecoa_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0,\n",
       "   'vacc_c': -1.0},\n",
       "  'id': 'FACOAL1812',\n",
       "  'gene_reaction_rule': 'HGNC:3569'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3569']},\n",
       "  'metabolites': {'od2coa_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'elaid_c': -1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0,\n",
       "   'ppi_c': 1.0},\n",
       "  'id': 'FACOAL1813',\n",
       "  'gene_reaction_rule': 'HGNC:3569'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'C181 fatty acid activation',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['TV catalyzed by acyl-CoA synthetases'],\n",
       "   'GENE ASSOCIATION': ['HGNC:24174 or HGNC:3569 or HGNC:3570 or HGNC:3571']},\n",
       "  'metabolites': {'octdececoa_c': 1.0,\n",
       "   'ocdcea_c': -1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0},\n",
       "  'id': 'FACOAL181i',\n",
       "  'gene_reaction_rule': 'HGNC:24174 or HGNC:3569 or HGNC:3570 or HGNC:3571'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ, RE1514C'],\n",
       "   'GENE ASSOCIATION': ['HGNC:24174 or HGNC:3569 or HGNC:3570 or HGNC:3571']},\n",
       "  'metabolites': {'lnlc_c': -1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'lnlccoa_c': 1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0},\n",
       "  'id': 'FACOAL1821',\n",
       "  'gene_reaction_rule': 'HGNC:24174 or HGNC:3569 or HGNC:3570 or HGNC:3571'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3569']},\n",
       "  'metabolites': {'atp_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'lneldccoa_c': 1.0,\n",
       "   'amp_c': 1.0,\n",
       "   'lneldc_c': -1.0,\n",
       "   'coa_c': -1.0},\n",
       "  'id': 'FACOAL1822',\n",
       "  'gene_reaction_rule': 'HGNC:3569'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:16496 or HGNC:16526 or HGNC:24174 or HGNC:3569 or HGNC:3570 or HGNC:3571']},\n",
       "  'metabolites': {'atp_c': -1.0,\n",
       "   'lnlncgcoa_c': 1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'lnlncg_c': -1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0},\n",
       "  'id': 'FACOAL1831',\n",
       "  'gene_reaction_rule': 'HGNC:16496 or HGNC:16526 or HGNC:24174 or HGNC:3569 or HGNC:3570 or HGNC:3571'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ, RE3114C'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3569']},\n",
       "  'metabolites': {'atp_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0,\n",
       "   'lnlnca_c': -1.0,\n",
       "   'lnlncacoa_c': 1.0},\n",
       "  'id': 'FACOAL1832',\n",
       "  'gene_reaction_rule': 'HGNC:3569'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3569']},\n",
       "  'metabolites': {'strdnccoa_c': 1.0,\n",
       "   'strdnc_c': -1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0},\n",
       "  'id': 'FACOAL184',\n",
       "  'gene_reaction_rule': 'HGNC:3569'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['current models suggest COAL activity occurs on outer peroxisomal membrane NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:10996']},\n",
       "  'metabolites': {'atp_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'pristcoa_c': 1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0,\n",
       "   'prist_c': -1.0},\n",
       "  'id': 'FACOAL191',\n",
       "  'gene_reaction_rule': 'HGNC:10996'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase (arachidate, n-C20:0)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3569']},\n",
       "  'metabolites': {'atp_c': -1.0,\n",
       "   'arach_c': -1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'arachcoa_c': 1.0},\n",
       "  'id': 'FACOAL200',\n",
       "  'gene_reaction_rule': 'HGNC:3569'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3569']},\n",
       "  'metabolites': {'dlnlcg_c': -1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'dlnlcgcoa_c': 1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0},\n",
       "  'id': 'FACOAL203',\n",
       "  'gene_reaction_rule': 'HGNC:3569'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['rev version  so additional rxn doesn&apos;t need to be added to &quot;unload&quot; AA NJ. Merged GPR from FACOAL204 and FACOAL204i - (May 2011) IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3570 or HGNC:3570 or HGNC:3569 or HGNC:3571 or HGNC:3571 or HGNC:16496 or HGNC:16496']},\n",
       "  'metabolites': {'atp_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0,\n",
       "   'arachdcoa_c': 1.0,\n",
       "   'arachd_c': -1.0},\n",
       "  'id': 'FACOAL204',\n",
       "  'gene_reaction_rule': 'HGNC:3570 or HGNC:3570 or HGNC:3569 or HGNC:3571 or HGNC:3571 or HGNC:16496 or HGNC:16496'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['rev version  so additional rxn doesn&apos;t need to be added to &quot;unload&quot; AA alternate to arachidonic acid (n-3 rather than n-6) NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3569']},\n",
       "  'metabolites': {'atp_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'eicostetcoa_c': 1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'eicostet_c': -1.0,\n",
       "   'amp_c': 1.0},\n",
       "  'id': 'FACOAL2042',\n",
       "  'gene_reaction_rule': 'HGNC:3569'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3569 or HGNC:16496']},\n",
       "  'metabolites': {'atp_c': -1.0,\n",
       "   'tmndnccoa_c': 1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'tmndnc_c': -1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0},\n",
       "  'id': 'FACOAL205',\n",
       "  'gene_reaction_rule': 'HGNC:3569 or HGNC:16496'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:10996']},\n",
       "  'metabolites': {'atp_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'phytcoa_c': 1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0,\n",
       "   'phyt_c': -1.0},\n",
       "  'id': 'FACOAL206',\n",
       "  'gene_reaction_rule': 'HGNC:10996'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3569']},\n",
       "  'metabolites': {'atp_c': -1.0,\n",
       "   'adrncoa_c': 1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'adrn_c': -1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0},\n",
       "  'id': 'FACOAL224',\n",
       "  'gene_reaction_rule': 'HGNC:3569'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3569']},\n",
       "  'metabolites': {'atp_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'dcsptn1_c': -1.0,\n",
       "   'dcsptn1coa_c': 1.0,\n",
       "   'amp_c': 1.0},\n",
       "  'id': 'FACOAL2251',\n",
       "  'gene_reaction_rule': 'HGNC:3569'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3569']},\n",
       "  'metabolites': {'atp_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'clpndcoa_c': 1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0,\n",
       "   'clpnd_c': -1.0},\n",
       "  'id': 'FACOAL2252',\n",
       "  'gene_reaction_rule': 'HGNC:3569'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3569']},\n",
       "  'metabolites': {'c226coa_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0,\n",
       "   'crvnc_c': -1.0},\n",
       "  'id': 'FACOAL226',\n",
       "  'gene_reaction_rule': 'HGNC:3569'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3569']},\n",
       "  'metabolites': {'atp_c': -1.0,\n",
       "   'ttccoa_c': 1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'lgnc_c': -1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0},\n",
       "  'id': 'FACOAL240',\n",
       "  'gene_reaction_rule': 'HGNC:3569'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3569']},\n",
       "  'metabolites': {'nrvnc_c': -1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'nrvnccoa_c': 1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0},\n",
       "  'id': 'FACOAL241',\n",
       "  'gene_reaction_rule': 'HGNC:3569'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3569']},\n",
       "  'metabolites': {'atp_c': -1.0,\n",
       "   'tettet6coa_c': 1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'tettet6_c': -1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0},\n",
       "  'id': 'FACOAL244_1',\n",
       "  'gene_reaction_rule': 'HGNC:3569'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3569 or HGNC:16496']},\n",
       "  'metabolites': {'atp_c': -1.0,\n",
       "   'tetpent6coa_c': 1.0,\n",
       "   'tetpent6_c': -1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0,\n",
       "   'ppi_c': 1.0},\n",
       "  'id': 'FACOAL245_1',\n",
       "  'gene_reaction_rule': 'HGNC:3569 or HGNC:16496'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3569']},\n",
       "  'metabolites': {'atp_c': -1.0,\n",
       "   'tetpent3_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0,\n",
       "   'tetpent3coa_c': 1.0},\n",
       "  'id': 'FACOAL245_2',\n",
       "  'gene_reaction_rule': 'HGNC:3569'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3569 or HGNC:16496']},\n",
       "  'metabolites': {'tethex3coa_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'tethex3_c': -1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0},\n",
       "  'id': 'FACOAL246_1',\n",
       "  'gene_reaction_rule': 'HGNC:3569 or HGNC:16496'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase (n-C26:0)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['cytoplasmic version -NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3569']},\n",
       "  'metabolites': {'atp_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'hexccoa_c': 1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0,\n",
       "   'hexc_c': -1.0},\n",
       "  'id': 'FACOAL260',\n",
       "  'gene_reaction_rule': 'HGNC:3569'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase (butanoate, n-C4:0), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['JLR changed EC from 6.2.1.3 (long chain fatty-acyl CoA ligase) to 6.2.1.2 (butyrate-CoA ligase) - NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:18049']},\n",
       "  'metabolites': {'but_m': -1.0,\n",
       "   'ppi_m': 1.0,\n",
       "   'atp_m': -1.0,\n",
       "   'btcoa_m': 1.0,\n",
       "   'coa_m': -1.0,\n",
       "   'amp_m': 1.0},\n",
       "  'id': 'FACOAL40im',\n",
       "  'gene_reaction_rule': 'HGNC:18049'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty-acid--CoA ligase (octanoate, n-C8:0)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['JLR Also 6.2.1.3'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3570']},\n",
       "  'metabolites': {'atp_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'octa_c': -1.0,\n",
       "   'coa_c': -1.0,\n",
       "   'amp_c': 1.0,\n",
       "   'occoa_c': 1.0},\n",
       "  'id': 'FACOAL80i',\n",
       "  'gene_reaction_rule': 'HGNC:3570'},\n",
       " {'subsystem': 'Vitamin B2 metabolism',\n",
       "  'name': 'FAD diphosphatase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3356 or HGNC:3358']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   'fmn_c': 1.0,\n",
       "   'h_c': 2.0,\n",
       "   'fad_c': -1.0,\n",
       "   'amp_c': 1.0},\n",
       "  'id': 'FADDP',\n",
       "  'gene_reaction_rule': 'HGNC:3356 or HGNC:3358'},\n",
       " {'subsystem': 'Fatty acid synthesis',\n",
       "  'name': 'fatty-acyl-CoA elongation (n-C18:3CoA)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['fatty acid elongation step NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:14415 or HGNC:14416 or HGNC:15829 or HGNC:21308']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   'h2o_c': 3.0,\n",
       "   'nadp_c': 4.0,\n",
       "   'nadph_c': -4.0,\n",
       "   'dlnlcgcoa_c': 1.0,\n",
       "   'lnlncgcoa_c': -1.0,\n",
       "   'malcoa_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'h_c': -5.0,\n",
       "   'co2_c': 1.0},\n",
       "  'id': 'FAEL183',\n",
       "  'gene_reaction_rule': 'HGNC:14415 or HGNC:14416 or HGNC:15829 or HGNC:21308'},\n",
       " {'subsystem': 'Fatty acid synthesis',\n",
       "  'name': 'fatty-acyl-CoA elongation (n-C20:4CoA)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['fatty acid elongation step NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:14415 or HGNC:14416 or HGNC:15829 or HGNC:21308']},\n",
       "  'metabolites': {'strdnccoa_c': -1.0,\n",
       "   'o2_c': -1.0,\n",
       "   'h2o_c': 3.0,\n",
       "   'nadp_c': 4.0,\n",
       "   'nadph_c': -4.0,\n",
       "   'eicostetcoa_c': 1.0,\n",
       "   'malcoa_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'h_c': -5.0,\n",
       "   'co2_c': 1.0},\n",
       "  'id': 'FAEL184',\n",
       "  'gene_reaction_rule': 'HGNC:14415 or HGNC:14416 or HGNC:15829 or HGNC:21308'},\n",
       " {'subsystem': 'Fatty acid synthesis',\n",
       "  'name': 'fatty-acyl-CoA elongation (n-C20:4CoA)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['fatty acid elongation step NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:14415 or HGNC:14416 or HGNC:15829 or HGNC:21308']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   'h2o_c': 3.0,\n",
       "   'nadp_c': 4.0,\n",
       "   'nadph_c': -4.0,\n",
       "   'adrncoa_c': 1.0,\n",
       "   'malcoa_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'h_c': -5.0,\n",
       "   'arachdcoa_c': -1.0,\n",
       "   'co2_c': 1.0},\n",
       "  'id': 'FAEL204',\n",
       "  'gene_reaction_rule': 'HGNC:14415 or HGNC:14416 or HGNC:15829 or HGNC:21308'},\n",
       " {'subsystem': 'Fatty acid synthesis',\n",
       "  'name': 'fatty-acyl-CoA elongation (n-C20:5CoA)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['fatty acid elongation step NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:14415 or HGNC:14416 or HGNC:15829 or HGNC:21308']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   'h2o_c': 3.0,\n",
       "   'nadp_c': 4.0,\n",
       "   'nadph_c': -4.0,\n",
       "   'tmndnccoa_c': -1.0,\n",
       "   'clpndcoa_c': 1.0,\n",
       "   'malcoa_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'h_c': -5.0,\n",
       "   'co2_c': 1.0},\n",
       "  'id': 'FAEL205',\n",
       "  'gene_reaction_rule': 'HGNC:14415 or HGNC:14416 or HGNC:15829 or HGNC:21308'},\n",
       " {'subsystem': 'Cytochrome metabolism',\n",
       "  'name': 'Fatty acid omega-hydroxylase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['fatty acid hydroxylation NJ'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:23198 and HGNC:9208) or (HGNC:2642 and HGNC:9208) or (HGNC:2644 and HGNC:9208)']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   'h2o_c': 1.0,\n",
       "   'whddca_c': 1.0,\n",
       "   'nadph_c': -1.0,\n",
       "   'ddca_c': -1.0,\n",
       "   'nadp_c': 1.0,\n",
       "   'h_c': -1.0},\n",
       "  'id': 'FAH1',\n",
       "  'gene_reaction_rule': '(HGNC:23198 and HGNC:9208) or (HGNC:2642 and HGNC:9208) or (HGNC:2644 and HGNC:9208)'},\n",
       " {'subsystem': 'Cytochrome metabolism',\n",
       "  'name': 'Fatty acid omega-hydroxylase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['fatty acid omega hydroxylation NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2642']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   'h2o_c': 1.0,\n",
       "   'nadp_c': 1.0,\n",
       "   'nadph_c': -1.0,\n",
       "   'whttdca_c': 1.0,\n",
       "   'h_c': -1.0,\n",
       "   'ttdca_c': -1.0},\n",
       "  'id': 'FAH2',\n",
       "  'gene_reaction_rule': 'HGNC:2642'},\n",
       " {'subsystem': 'Cytochrome metabolism',\n",
       "  'name': 'Fatty acid omega-hydroxylase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['fatty acid omega hydroxylation NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2642']},\n",
       "  'metabolites': {'o2_c': -1.0,\n",
       "   'h2o_c': 1.0,\n",
       "   'nadp_c': 1.0,\n",
       "   'nadph_c': -1.0,\n",
       "   'whhdca_c': 1.0,\n",
       "   'h_c': -1.0,\n",
       "   'hdca_c': -1.0},\n",
       "  'id': 'FAH3',\n",
       "  'gene_reaction_rule': 'HGNC:2642'},\n",
       " {'subsystem': 'Tyrosine metabolism',\n",
       "  'name': 'formaldehyde dehydrogenase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:253']},\n",
       "  'metabolites': {'Sfglutth_c': 1.0,\n",
       "   'fald_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   'nad_c': -1.0,\n",
       "   'nadh_c': 1.0,\n",
       "   'gthrd_c': -1.0},\n",
       "  'id': 'FALDH',\n",
       "  'gene_reaction_rule': 'HGNC:253'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty acid beta oxidation(C18-->C16)m',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['TV Other enzymes: 1.3.3.6, 4.2.1.74, 4.2.1.17, 1.1.1.35, 1.1.1.211'],\n",
       "   'GENE ASSOCIATION': ['HGNC:88']},\n",
       "  'metabolites': {'h2o_m': -1.0,\n",
       "   'fad_m': -1.0,\n",
       "   'accoa_m': 1.0,\n",
       "   'nadh_m': 1.0,\n",
       "   'nad_m': -1.0,\n",
       "   'coa_m': -1.0,\n",
       "   'h_m': 1.0,\n",
       "   'fadh2_m': 1.0,\n",
       "   'pmtcoa_m': 1.0,\n",
       "   'stcoa_m': -1.0},\n",
       "  'id': 'FAOXC180',\n",
       "  'gene_reaction_rule': 'HGNC:88'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'Beta oxidation of long chain fatty acid',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['- in peroxisome, no fad, 1ste set of beta-ox is by acyl-coa oxidase (passes electrons directly to o2 -&gt; h2o2) NJ'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:10606 and HGNC:119 and (HGNC:3247 or HGNC:5213)) or (HGNC:119 and (HGNC:3247 or HGNC:5213) and HGNC:82)']},\n",
       "  'metabolites': {'nadh_x': 1.0,\n",
       "   'stcoa_x': -1.0,\n",
       "   'pmtcoa_x': 1.0,\n",
       "   'nad_x': -1.0,\n",
       "   'h_x': 1.0,\n",
       "   'coa_x': -1.0,\n",
       "   'accoa_x': 1.0,\n",
       "   'h2o_x': -1.0,\n",
       "   'o2_x': -1.0,\n",
       "   'h2o2_x': 1.0},\n",
       "  'id': 'FAOXC180x',\n",
       "  'gene_reaction_rule': '(HGNC:10606 and HGNC:119 and (HGNC:3247 or HGNC:5213)) or (HGNC:119 and (HGNC:3247 or HGNC:5213) and HGNC:82)'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'Beta oxidation  fatty acid',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['_5 rounds of beta ox, 1 loss of degree of unsat -&gt; 4 fad oxidized c18:1 -&gt; c8:0 4 step FAO lumped for 2 rounds  also includes EC 1.1.1.35, 1.1.1.211, 1.3.3.6, 4.2.1.17, 4.2.1.74 NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:89 or HGNC:90']},\n",
       "  'metabolites': {'h2o_m': -5.0,\n",
       "   'od2coa_m': -1.0,\n",
       "   'fad_m': -4.0,\n",
       "   'accoa_m': 5.0,\n",
       "   'occoa_m': 1.0,\n",
       "   'nadh_m': 5.0,\n",
       "   'nad_m': -5.0,\n",
       "   'coa_m': -5.0,\n",
       "   'h_m': 5.0,\n",
       "   'fadh2_m': 4.0},\n",
       "  'id': 'FAOXC1811601m',\n",
       "  'gene_reaction_rule': 'HGNC:89 or HGNC:90'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'Beta oxidation of fatty acid',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['c18:2,n-6 -&gt; c8:0 - 5 rounds of beta ox, lose 2 degrees of unsat - only 3 fad oxidized 4 step FAO lumped also includes EC 1.1.1.35, 1.1.1.211, 1.3.3.6, 4.2.1.17, 4.2.1.74 NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:89 or HGNC:90']},\n",
       "  'metabolites': {'lnlccoa_m': -1.0,\n",
       "   'h2o_m': -5.0,\n",
       "   'fad_m': -3.0,\n",
       "   'accoa_m': 5.0,\n",
       "   'occoa_m': 1.0,\n",
       "   'nadh_m': 5.0,\n",
       "   'nad_m': -5.0,\n",
       "   'coa_m': -5.0,\n",
       "   'h_m': 5.0,\n",
       "   'fadh2_m': 3.0},\n",
       "  'id': 'FAOXC182806m',\n",
       "  'gene_reaction_rule': 'HGNC:89 or HGNC:90'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'Beta oxidation of fatty acid',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['_5 rounds of beta ox, 3 degrees loss of saturation -&gt; only 2 fad oxidized to fadh2 - in peroxisome, no fad, 1ste set of beta-ox is by acyl-coa oxidase (passes electrons directly to o2 -&gt; h2o2) peroxisomal (very long chain FA) c18:3,n-6 -&gt; c8:0, 4 step FAO lumped for 2 rounds  also includes EC 1.1.1.35, 1.1.1.211, 1.3.3.6, 4.2.1.17, 4.2.1.74 NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:89 or HGNC:90']},\n",
       "  'metabolites': {'lnlncgcoa_m': -1.0,\n",
       "   'h2o_m': -5.0,\n",
       "   'o2_m': -2.0,\n",
       "   'accoa_m': 5.0,\n",
       "   'occoa_m': 1.0,\n",
       "   'nadh_m': 5.0,\n",
       "   'nad_m': -5.0,\n",
       "   'coa_m': -5.0,\n",
       "   'h_m': 5.0,\n",
       "   'h2o2_m': 2.0},\n",
       "  'id': 'FAOXC183806m',\n",
       "  'gene_reaction_rule': 'HGNC:89 or HGNC:90'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'Beta oxidation of long chain fatty acid',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['_5 rounds of beta ox, 3 degrees loss of saturation -&gt; only 2 fad oxidized to fadh2 - in peroxisome, no fad, 1ste set of beta-ox is by acyl-coa oxidase (passes electrons directly to o2 -&gt; h2o2) peroxisomal (very long chain FA) c18:3,n-6 -&gt; c8:0, 4 step FAO lumped for 2 rounds  also includes EC 1.1.1.35, 1.1.1.211, 1.3.3.6, 4.2.1.17, 4.2.1.74 NJ'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:119 and HGNC:3247 and HGNC:82) or (HGNC:119 and HGNC:5213 and HGNC:82)']},\n",
       "  'metabolites': {'nadh_x': 5.0,\n",
       "   'occoa_x': 1.0,\n",
       "   'h2o2_x': 2.0,\n",
       "   'nad_x': -5.0,\n",
       "   'h_x': 5.0,\n",
       "   'accoa_x': 5.0,\n",
       "   'lnlncgcoa_x': -1.0,\n",
       "   'h2o_x': -5.0,\n",
       "   'o2_x': -2.0,\n",
       "   'coa_x': -5.0},\n",
       "  'id': 'FAOXC183806x',\n",
       "  'gene_reaction_rule': '(HGNC:119 and HGNC:3247 and HGNC:82) or (HGNC:119 and HGNC:5213 and HGNC:82)'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'fatty acid beta oxidation(C20-->C18)m',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['c20:0 -&gt; c18:0 4 step FAO lumped also includes EC 1.1.1.35, 1.1.1.211, 1.3.3.6, 4.2.1.17, 4.2.1.74 NJ'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:21497 and HGNC:4801 and HGNC:4803) or (HGNC:30211 and HGNC:4801 and HGNC:4803) or (HGNC:4801 and HGNC:4803 and HGNC:88) or (HGNC:4801 and HGNC:4803 and HGNC:92)']},\n",
       "  'metabolites': {'fad_m': -1.0,\n",
       "   'h2o_m': -1.0,\n",
       "   'arachcoa_m': -1.0,\n",
       "   'accoa_m': 1.0,\n",
       "   'nadh_m': 1.0,\n",
       "   'nad_m': -1.0,\n",
       "   'coa_m': -1.0,\n",
       "   'h_m': 1.0,\n",
       "   'fadh2_m': 1.0,\n",
       "   'stcoa_m': 1.0},\n",
       "  'id': 'FAOXC200180m',\n",
       "  'gene_reaction_rule': '(HGNC:21497 and HGNC:4801 and HGNC:4803) or (HGNC:30211 and HGNC:4801 and HGNC:4803) or (HGNC:4801 and HGNC:4803 and HGNC:88) or (HGNC:4801 and HGNC:4803 and HGNC:92)'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'Beta oxidation of long chain fatty acid',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['peroxisomal (very long chain FA) - in peroxisome, no fad, 1ste set of beta-ox is by acyl-coa oxidase (passes electrons directly to o2 -&gt; h2o2) c20:0 -&gt; c18:0 4 step FAO lumped also includes EC 1.1.1.35, 1.1.1.211, 1.3.3.6, 4.2.1.17, 4.2.1.74 NJ'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:10606 and HGNC:119 and (HGNC:3247 or HGNC:5213)) or (HGNC:119 and (HGNC:3247 or HGNC:5213) and HGNC:82)']},\n",
       "  'metabolites': {'nadh_x': 1.0,\n",
       "   'arachcoa_x': -1.0,\n",
       "   'h2o2_x': 1.0,\n",
       "   'nad_x': -1.0,\n",
       "   'h_x': 1.0,\n",
       "   'accoa_x': 1.0,\n",
       "   'h2o_x': -1.0,\n",
       "   'o2_x': -1.0,\n",
       "   'coa_x': -1.0,\n",
       "   'stcoa_x': 1.0},\n",
       "  'id': 'FAOXC200180x',\n",
       "  'gene_reaction_rule': '(HGNC:10606 and HGNC:119 and (HGNC:3247 or HGNC:5213)) or (HGNC:119 and (HGNC:3247 or HGNC:5213) and HGNC:82)'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'Beta oxidation of fatty acid',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['c20:3 -&gt; c18:3 4 step FAO lumped also includes EC 1.1.1.35, 1.1.1.211, 1.3.3.6, 4.2.1.17, 4.2.1.74 NJ'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:21497 and HGNC:4801 and HGNC:4803) or (HGNC:30211 and HGNC:4801 and HGNC:4803) or (HGNC:4801 and HGNC:4803 and HGNC:88) or (HGNC:4801 and HGNC:4803 and HGNC:92)']},\n",
       "  'metabolites': {'lnlncgcoa_m': 1.0,\n",
       "   'dlnlcgcoa_m': -1.0,\n",
       "   'h2o_m': -1.0,\n",
       "   'fad_m': -1.0,\n",
       "   'accoa_m': 1.0,\n",
       "   'nadh_m': 1.0,\n",
       "   'nad_m': -1.0,\n",
       "   'coa_m': -1.0,\n",
       "   'h_m': 1.0,\n",
       "   'fadh2_m': 1.0},\n",
       "  'id': 'FAOXC2031836m',\n",
       "  'gene_reaction_rule': '(HGNC:21497 and HGNC:4801 and HGNC:4803) or (HGNC:30211 and HGNC:4801 and HGNC:4803) or (HGNC:4801 and HGNC:4803 and HGNC:88) or (HGNC:4801 and HGNC:4803 and HGNC:92)'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'Beta oxidation of fatty acid',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['_1 round beta ox, 1 degree loss of sat -&gt; no fad oxidation peroxisomal (very long chain FA) c20:5,n-3 -&gt; c18:4, n-3 - final &apos;3&apos; in abbreviation indicates omega3 FA 4 step FAO lumped for 2 rounds  also includes EC 1.1.1.35, 1.1.1.211, 1.3.3.6, 4.2.1.17, 4.2.1.74 NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:89 or HGNC:90']},\n",
       "  'metabolites': {'tmndnccoa_m': -1.0,\n",
       "   'h2o_m': -1.0,\n",
       "   'strdnccoa_m': 1.0,\n",
       "   'accoa_m': 1.0,\n",
       "   'nadh_m': 1.0,\n",
       "   'nad_m': -1.0,\n",
       "   'coa_m': -1.0,\n",
       "   'h_m': 1.0},\n",
       "  'id': 'FAOXC2051843m',\n",
       "  'gene_reaction_rule': 'HGNC:89 or HGNC:90'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'Beta oxidation of long chain fatty acid',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['_1 round beta ox, 1 degree loss of sat -&gt; no fad oxidation - in peroxisome, no fad, 1ste set of beta-ox is by acyl-coa oxidase (passes electrons directly to o2 -&gt; h2o2) peroxisomal (very long chain FA) c20:5,n-3 -&gt; c18:4, n-3 - final &apos;3&apos; in abbreviation indicates omega3 FA 4 step FAO lumped for 2 rounds  also includes EC 1.1.1.35, 1.1.1.211, 1.3.3.6, 4.2.1.17, 4.2.1.74 NJ'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:119 and HGNC:3247 and HGNC:82) or (HGNC:119 and HGNC:5213 and HGNC:82)']},\n",
       "  'metabolites': {'nadh_x': 1.0,\n",
       "   'nad_x': -1.0,\n",
       "   'h_x': 1.0,\n",
       "   'accoa_x': 1.0,\n",
       "   'tmndnccoa_x': -1.0,\n",
       "   'h2o_x': -1.0,\n",
       "   'strdnccoa_x': 1.0,\n",
       "   'coa_x': -1.0},\n",
       "  'id': 'FAOXC2051843x',\n",
       "  'gene_reaction_rule': '(HGNC:119 and HGNC:3247 and HGNC:82) or (HGNC:119 and HGNC:5213 and HGNC:82)'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'Beta oxidation of long chain fatty acid',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['_1 round beta ox peroxisomal (very long chain FA) c22:4,n-6 -&gt; c20:4, n-6 - final &apos;6&apos; in abbreviation indicates omega6 FA 4 step FAO lumped for 2 rounds  also includes EC 1.1.1.35, 1.1.1.211, 1.3.3.6, 4.2.1.17, 4.2.1.74 NJ'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:21497 and HGNC:4801 and HGNC:4803) or (HGNC:30211 and HGNC:4801 and HGNC:4803) or (HGNC:4801 and HGNC:4803 and HGNC:88) or (HGNC:4801 and HGNC:4803 and HGNC:92)']},\n",
       "  'metabolites': {'adrncoa_m': -1.0,\n",
       "   'h2o_m': -1.0,\n",
       "   'fad_m': -1.0,\n",
       "   'accoa_m': 1.0,\n",
       "   'nadh_m': 1.0,\n",
       "   'arachdcoa_m': 1.0,\n",
       "   'nad_m': -1.0,\n",
       "   'coa_m': -1.0,\n",
       "   'h_m': 1.0,\n",
       "   'fadh2_m': 1.0},\n",
       "  'id': 'FAOXC2242046m',\n",
       "  'gene_reaction_rule': '(HGNC:21497 and HGNC:4801 and HGNC:4803) or (HGNC:30211 and HGNC:4801 and HGNC:4803) or (HGNC:4801 and HGNC:4803 and HGNC:88) or (HGNC:4801 and HGNC:4803 and HGNC:92)'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'Beta oxidation of long chain fatty acid',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['_1 round beta ox - in peroxisome, no fad, 1ste set of beta-ox is by acyl-coa oxidase (passes electrons directly to o2 -&gt; h2o2) peroxisomal (very long chain FA) c22:4,n-6 -&gt; c20:4, n-6 - final &apos;6&apos; in abbreviation indicates omega6 FA 4 step FAO lumped for 2 rounds  also includes EC 1.1.1.35, 1.1.1.211, 1.3.3.6, 4.2.1.17, 4.2.1.74 NJ'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:10606 and HGNC:119 and (HGNC:3247 or HGNC:5213)) or (HGNC:119 and (HGNC:3247 or HGNC:5213) and HGNC:82)']},\n",
       "  'metabolites': {'nadh_x': 1.0,\n",
       "   'arachdcoa_x': 1.0,\n",
       "   'h2o2_x': 1.0,\n",
       "   'nad_x': -1.0,\n",
       "   'h_x': 1.0,\n",
       "   'coa_x': -1.0,\n",
       "   'accoa_x': 1.0,\n",
       "   'adrncoa_x': -1.0,\n",
       "   'h2o_x': -1.0,\n",
       "   'o2_x': -1.0},\n",
       "  'id': 'FAOXC2242046x',\n",
       "  'gene_reaction_rule': '(HGNC:10606 and HGNC:119 and (HGNC:3247 or HGNC:5213)) or (HGNC:119 and (HGNC:3247 or HGNC:5213) and HGNC:82)'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'Beta oxidation of fatty acid',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['_2 rounds of beta-ox, plus a loss of 2 degrees of desatuartion, so no fad is oxidized peroxisomal (very long chain FA) c22:5,n-6 -&gt; c18:3, n-6 - final &apos;6&apos; in abbreviation indicates omega6 FA 4 step FAO lumped for 2 rounds  also includes EC 1.1.1.35, 1.1.1.211, 1.3.3.6, 4.2.1.17, 4.2.1.74 NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:89 or HGNC:90']},\n",
       "  'metabolites': {'lnlncgcoa_m': 1.0,\n",
       "   'h2o_m': -2.0,\n",
       "   'dcsptn1coa_m': -1.0,\n",
       "   'accoa_m': 2.0,\n",
       "   'nadh_m': 2.0,\n",
       "   'nad_m': -2.0,\n",
       "   'coa_m': -2.0,\n",
       "   'h_m': 2.0},\n",
       "  'id': 'FAOXC2251836m',\n",
       "  'gene_reaction_rule': 'HGNC:89 or HGNC:90'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'Beta oxidation of long chain fatty acid',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['_2 rounds of beta-ox, plus a loss of 2 degrees of desatuartion, so no fad is oxidized - in peroxisome, no fad, 1ste set of beta-ox is by acyl-coa oxidase (passes electrons directly to o2 -&gt; h2o2) peroxisomal (very long chain FA) c22:5,n-6 -&gt; c18:3, n-6 - final &apos;6&apos; in abbreviation indicates omega6 FA 4 step FAO lumped for 2 rounds  also includes EC 1.1.1.35, 1.1.1.211, 1.3.3.6, 4.2.1.17, 4.2.1.74 NJ'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:119 and HGNC:3247 and HGNC:82) or (HGNC:119 and HGNC:5213 and HGNC:82)']},\n",
       "  'metabolites': {'nadh_x': 2.0,\n",
       "   'dcsptn1coa_x': -1.0,\n",
       "   'nad_x': -2.0,\n",
       "   'h_x': 2.0,\n",
       "   'coa_x': -2.0,\n",
       "   'accoa_x': 2.0,\n",
       "   'lnlncgcoa_x': 1.0,\n",
       "   'h2o_x': -2.0},\n",
       "  'id': 'FAOXC2251836x',\n",
       "  'gene_reaction_rule': '(HGNC:119 and HGNC:3247 and HGNC:82) or (HGNC:119 and HGNC:5213 and HGNC:82)'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'Beta oxidation of long chain fatty acid',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['_1 round beta ox - c22:5,n-3 -&gt; c20:5, n-3 - final &apos;3&apos; in abbreviation indicates omega3 FA 4 step FAO lumped for 2 rounds  also includes EC 1.1.1.35, 1.1.1.211, 1.3.3.6, 4.2.1.17, 4.2.1.74 NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:89 or HGNC:90']},\n",
       "  'metabolites': {'tmndnccoa_m': 1.0,\n",
       "   'h2o_m': -1.0,\n",
       "   'o2_m': -1.0,\n",
       "   'dcsptn1coa_m': -1.0,\n",
       "   'accoa_m': 1.0,\n",
       "   'nadh_m': 1.0,\n",
       "   'nad_m': -1.0,\n",
       "   'coa_m': -1.0,\n",
       "   'h_m': 1.0,\n",
       "   'h2o2_m': 1.0},\n",
       "  'id': 'FAOXC2252053m',\n",
       "  'gene_reaction_rule': 'HGNC:89 or HGNC:90'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'Beta oxidation of long chain fatty acid',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['_1 round beta ox - in peroxisome, no fad, 1ste set of beta-ox is by acyl-coa oxidase (passes electrons directly to o2 -&gt; h2o2) peroxisomal (very long chain FA) c22:5,n-3 -&gt; c20:5, n-3 - final &apos;3&apos; in abbreviation indicates omega3 FA 4 step FAO lumped for 2 rounds  also includes EC 1.1.1.35, 1.1.1.211, 1.3.3.6, 4.2.1.17, 4.2.1.74 NJ'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:119 and HGNC:3247 and HGNC:82) or (HGNC:119 and HGNC:5213 and HGNC:82)']},\n",
       "  'metabolites': {'nadh_x': 1.0,\n",
       "   'dcsptn1coa_x': -1.0,\n",
       "   'h2o2_x': 1.0,\n",
       "   'nad_x': -1.0,\n",
       "   'h_x': 1.0,\n",
       "   'coa_x': -1.0,\n",
       "   'accoa_x': 1.0,\n",
       "   'tmndnccoa_x': 1.0,\n",
       "   'h2o_x': -1.0,\n",
       "   'o2_x': -1.0},\n",
       "  'id': 'FAOXC2252053x',\n",
       "  'gene_reaction_rule': '(HGNC:119 and HGNC:3247 and HGNC:82) or (HGNC:119 and HGNC:5213 and HGNC:82)'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'Beta oxidation of long chain fatty acid',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['_1 round of beta-oxidation, loses degree of desat, so no fad peroxisomal (very long chain FA) c22:6,n-3 -&gt; c20:5, n-3 4 step FAO lumped also includes EC 1.1.1.35, 1.1.1.211, 1.3.3.6, 4.2.1.17, 4.2.1.74 NJ'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:14601 and HGNC:21497 and HGNC:2753 and HGNC:4801 and HGNC:4803) or (HGNC:14601 and HGNC:2753 and HGNC:30211 and HGNC:4801 and HGNC:4803) or (HGNC:14601 and HGNC:2753 and HGNC:4801 and HGNC:4803 and HGNC:88) or (HGNC:14601 and HGNC:2753 and HGNC:4801 and HGNC:4803 and HGNC:92) or (HGNC:21497 and HGNC:2703 and HGNC:2753 and HGNC:4801 and HGNC:4803) or (HGNC:2703 and HGNC:2753 and HGNC:30211 and HGNC:4801 and HGNC:4803) or (HGNC:2703 and HGNC:2753 and HGNC:4801 and HGNC:4803 and HGNC:88) or (HGNC:2703 and HGNC:2753 and HGNC:4801 and HGNC:4803 and HGNC:92)']},\n",
       "  'metabolites': {'tmndnccoa_m': 1.0,\n",
       "   'nadph_m': -1.0,\n",
       "   'c226coa_m': -1.0,\n",
       "   'h2o_m': -1.0,\n",
       "   'nadp_m': 1.0,\n",
       "   'fad_m': -1.0,\n",
       "   'accoa_m': 1.0,\n",
       "   'nadh_m': 1.0,\n",
       "   'nad_m': -1.0,\n",
       "   'coa_m': -1.0,\n",
       "   'fadh2_m': 1.0},\n",
       "  'id': 'FAOXC226205m',\n",
       "  'gene_reaction_rule': '(HGNC:14601 and HGNC:21497 and HGNC:2753 and HGNC:4801 and HGNC:4803) or (HGNC:14601 and HGNC:2753 and HGNC:30211 and HGNC:4801 and HGNC:4803) or (HGNC:14601 and HGNC:2753 and HGNC:4801 and HGNC:4803 and HGNC:88) or (HGNC:14601 and HGNC:2753 and HGNC:4801 and HGNC:4803 and HGNC:92) or (HGNC:21497 and HGNC:2703 and HGNC:2753 and HGNC:4801 and HGNC:4803) or (HGNC:2703 and HGNC:2753 and HGNC:30211 and HGNC:4801 and HGNC:4803) or (HGNC:2703 and HGNC:2753 and HGNC:4801 and HGNC:4803 and HGNC:88) or (HGNC:2703 and HGNC:2753 and HGNC:4801 and HGNC:4803 and HGNC:92)'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'Beta oxidation of long chain fatty acid',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['_1 round of beta-oxidation, loses degree of desat, so no fad - in peroxisome, no fad, 1ste set of beta-ox is by acyl-coa oxidase (passes electrons directly to o2 -&gt; h2o2) peroxisomal (very long chain FA) c22:6,n-3 -&gt; c20:5, n-3 4 step FAO lumped also includes EC 1.1.1.35, 1.1.1.211, 1.3.3.6, 4.2.1.17, 4.2.1.74 NJ'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:10606 and HGNC:119 and HGNC:2754 and (HGNC:3247 or HGNC:5213)) or (HGNC:119 and HGNC:2754 and (HGNC:3247 or HGNC:5213) and HGNC:82)']},\n",
       "  'metabolites': {'nadh_x': 1.0,\n",
       "   'nad_x': -1.0,\n",
       "   'fadh2_x': 1.0,\n",
       "   'fad_x': -1.0,\n",
       "   'accoa_x': 1.0,\n",
       "   'coa_x': -1.0,\n",
       "   'tmndnccoa_x': 1.0,\n",
       "   'nadp_x': 1.0,\n",
       "   'h2o_x': -1.0,\n",
       "   'c226coa_x': -1.0,\n",
       "   'nadph_x': -1.0},\n",
       "  'id': 'FAOXC226205x',\n",
       "  'gene_reaction_rule': '(HGNC:10606 and HGNC:119 and HGNC:2754 and (HGNC:3247 or HGNC:5213)) or (HGNC:119 and HGNC:2754 and (HGNC:3247 or HGNC:5213) and HGNC:82)'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'Beta oxidation of long chain fatty acid',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['_1 round beta ox - - in peroxisome, no fad, 1ste set of beta-ox is by acyl-coa oxidase (passes electrons directly to o2 -&gt; h2o2) peroxisomal (very long chain FA) c24:4,n-6 -&gt; c22:4, n-6 - final &apos;6&apos; in abbreviation indicates omega6 FA 4 step FAO lumped for 2 rounds  also includes EC 1.1.1.35, 1.1.1.211, 1.3.3.6, 4.2.1.17, 4.2.1.74 NJ'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:10606 and HGNC:119 and (HGNC:3247 or HGNC:5213)) or (HGNC:119 and (HGNC:3247 or HGNC:5213) and HGNC:82)']},\n",
       "  'metabolites': {'nadh_x': 1.0,\n",
       "   'adrncoa_x': 1.0,\n",
       "   'h2o2_x': 1.0,\n",
       "   'nad_x': -1.0,\n",
       "   'h_x': 1.0,\n",
       "   'accoa_x': 1.0,\n",
       "   'tettet6coa_x': -1.0,\n",
       "   'h2o_x': -1.0,\n",
       "   'o2_x': -1.0,\n",
       "   'coa_x': -1.0},\n",
       "  'id': 'FAOXC2442246x',\n",
       "  'gene_reaction_rule': '(HGNC:10606 and HGNC:119 and (HGNC:3247 or HGNC:5213)) or (HGNC:119 and (HGNC:3247 or HGNC:5213) and HGNC:82)'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'Beta oxidation of long chain fatty acid',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['- in peroxisome, no fad, 1ste set of beta-ox is by acyl-coa oxidase (passes electrons directly to o2 -&gt; h2o2) confusing name, sorry - accounting for omega3 vs omega6 FA peroxisomal (very long chain FA) c24:5,n-3 -&gt; c22:5, n-3 4 step FAO lumped also includes EC 1.1.1.35, 1.1.1.211, 1.3.3.6, 4.2.1.17, 4.2.1.74 NJ'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:10606 and HGNC:119 and (HGNC:3247 or HGNC:5213)) or (HGNC:119 and (HGNC:3247 or HGNC:5213) and HGNC:82)']},\n",
       "  'metabolites': {'nadh_x': 1.0,\n",
       "   'tetpent3coa_x': -1.0,\n",
       "   'h2o2_x': 1.0,\n",
       "   'nad_x': -1.0,\n",
       "   'h_x': 1.0,\n",
       "   'accoa_x': 1.0,\n",
       "   'clpndcoa_x': 1.0,\n",
       "   'h2o_x': -1.0,\n",
       "   'o2_x': -1.0,\n",
       "   'coa_x': -1.0},\n",
       "  'id': 'FAOXC2452253x',\n",
       "  'gene_reaction_rule': '(HGNC:10606 and HGNC:119 and (HGNC:3247 or HGNC:5213)) or (HGNC:119 and (HGNC:3247 or HGNC:5213) and HGNC:82)'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'Beta oxidation of long chain fatty acid',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['peroxisomal (very long chain FA) - - in peroxisome, no fad, 1ste set of beta-ox is by acyl-coa oxidase (passes electrons directly to o2 -&gt; h2o2) c24:5,n-6 -&gt; c22:5, n-6 4 step FAO lumped also includes EC 1.1.1.35, 1.1.1.211, 1.3.3.6, 4.2.1.17, 4.2.1.74 NJ'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:10606 and HGNC:119 and (HGNC:3247 or HGNC:5213)) or (HGNC:119 and (HGNC:3247 or HGNC:5213) and HGNC:82)']},\n",
       "  'metabolites': {'nadh_x': 1.0,\n",
       "   'dcsptn1coa_x': 1.0,\n",
       "   'tetpent6coa_x': -1.0,\n",
       "   'h2o2_x': 1.0,\n",
       "   'nad_x': -1.0,\n",
       "   'h_x': 1.0,\n",
       "   'coa_x': -1.0,\n",
       "   'accoa_x': 1.0,\n",
       "   'h2o_x': -1.0,\n",
       "   'o2_x': -1.0},\n",
       "  'id': 'FAOXC2452256x',\n",
       "  'gene_reaction_rule': '(HGNC:10606 and HGNC:119 and (HGNC:3247 or HGNC:5213)) or (HGNC:119 and (HGNC:3247 or HGNC:5213) and HGNC:82)'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'Beta oxidation of long chain fatty acid',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['peroxisomal (very long chain FA) - - in peroxisome, no fad, 1ste set of beta-ox is by acyl-coa oxidase (passes electrons directly to o2 -&gt; h2o2) c24:6 -&gt; c22:6 4 step FAO lumped also includes EC 1.1.1.35, 1.1.1.211, 1.3.3.6, 4.2.1.17, 4.2.1.74 NJ'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:10606 and HGNC:119 and (HGNC:3247 or HGNC:5213)) or (HGNC:119 and (HGNC:3247 or HGNC:5213) and HGNC:82)']},\n",
       "  'metabolites': {'nadh_x': 1.0,\n",
       "   'h2o2_x': 1.0,\n",
       "   'nad_x': -1.0,\n",
       "   'h_x': 1.0,\n",
       "   'accoa_x': 1.0,\n",
       "   'tethex3coa_x': -1.0,\n",
       "   'h2o_x': -1.0,\n",
       "   'o2_x': -1.0,\n",
       "   'coa_x': -1.0,\n",
       "   'c226coa_x': 1.0},\n",
       "  'id': 'FAOXC246226x',\n",
       "  'gene_reaction_rule': '(HGNC:10606 and HGNC:119 and (HGNC:3247 or HGNC:5213)) or (HGNC:119 and (HGNC:3247 or HGNC:5213) and HGNC:82)'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'Beta oxidation of long chain fatty acid',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['peroxisomal (very long chain FA) - in peroxisome, no fad, 1ste set of beta-ox is by acyl-coa oxidase (passes electrons directly to o2 -&gt; h2o2) c26:0 -&gt; c24:0 4 step FAO lumped also includes EC 1.1.1.35, 1.1.1.211, 1.3.3.6, 4.2.1.17, 4.2.1.74 -&gt; into pmtcoa NJ'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:10606 and HGNC:119 and (HGNC:3247 or HGNC:5213)) or (HGNC:119 and (HGNC:3247 or HGNC:5213) and HGNC:82)']},\n",
       "  'metabolites': {'nadh_x': 1.0,\n",
       "   'hexccoa_x': -1.0,\n",
       "   'h2o2_x': 1.0,\n",
       "   'nad_x': -1.0,\n",
       "   'h_x': 1.0,\n",
       "   'accoa_x': 1.0,\n",
       "   'ttccoa_x': 1.0,\n",
       "   'h2o_x': -1.0,\n",
       "   'o2_x': -1.0,\n",
       "   'coa_x': -1.0},\n",
       "  'id': 'FAOXC260240x',\n",
       "  'gene_reaction_rule': '(HGNC:10606 and HGNC:119 and (HGNC:3247 or HGNC:5213)) or (HGNC:119 and (HGNC:3247 or HGNC:5213) and HGNC:82)'},\n",
       " {'subsystem': 'Fatty acid synthesis',\n",
       "  'name': 'fatty acyl-CoA synthase (n-C10:0CoA)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:14416 and HGNC:15829 and HGNC:21308 and HGNC:3594) or HGNC:3594']},\n",
       "  'metabolites': {'h2o_c': 1.0,\n",
       "   'nadp_c': 2.0,\n",
       "   'nadph_c': -2.0,\n",
       "   'malcoa_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'h_c': -3.0,\n",
       "   'co2_c': 1.0,\n",
       "   'occoa_c': -1.0,\n",
       "   'dcacoa_c': 1.0},\n",
       "  'id': 'FAS100COA',\n",
       "  'gene_reaction_rule': '(HGNC:14416 and HGNC:15829 and HGNC:21308 and HGNC:3594) or HGNC:3594'},\n",
       " {'subsystem': 'Fatty acid synthesis',\n",
       "  'name': 'fatty-acyl-CoA synthase (n-C12:0CoA)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:14416 and HGNC:15829 and HGNC:21308 and HGNC:3594) or HGNC:3594']},\n",
       "  'metabolites': {'h2o_c': 1.0,\n",
       "   'nadp_c': 2.0,\n",
       "   'nadph_c': -2.0,\n",
       "   'ddcacoa_c': 1.0,\n",
       "   'malcoa_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'h_c': -3.0,\n",
       "   'co2_c': 1.0,\n",
       "   'dcacoa_c': -1.0},\n",
       "  'id': 'FAS120COA',\n",
       "  'gene_reaction_rule': '(HGNC:14416 and HGNC:15829 and HGNC:21308 and HGNC:3594) or HGNC:3594'},\n",
       " {'subsystem': 'Fatty acid synthesis',\n",
       "  'name': 'fatty-acyl-CoA synthase (n-C14:0CoA)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:14416 and HGNC:15829 and HGNC:21308 and HGNC:3594) or HGNC:3594']},\n",
       "  'metabolites': {'h2o_c': 1.0,\n",
       "   'nadp_c': 2.0,\n",
       "   'nadph_c': -2.0,\n",
       "   'ddcacoa_c': -1.0,\n",
       "   'malcoa_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'h_c': -3.0,\n",
       "   'co2_c': 1.0,\n",
       "   'tdcoa_c': 1.0},\n",
       "  'id': 'FAS140COA',\n",
       "  'gene_reaction_rule': '(HGNC:14416 and HGNC:15829 and HGNC:21308 and HGNC:3594) or HGNC:3594'},\n",
       " {'subsystem': 'Fatty acid synthesis',\n",
       "  'name': 'fatty-acyl-CoA synthase (n-C16:0CoA)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:14416 and HGNC:15829 and HGNC:21308 and HGNC:3594) or HGNC:3594']},\n",
       "  'metabolites': {'h2o_c': 1.0,\n",
       "   'nadp_c': 2.0,\n",
       "   'nadph_c': -2.0,\n",
       "   'pmtcoa_c': 1.0,\n",
       "   'malcoa_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'h_c': -3.0,\n",
       "   'co2_c': 1.0,\n",
       "   'tdcoa_c': -1.0},\n",
       "  'id': 'FAS160COA',\n",
       "  'gene_reaction_rule': '(HGNC:14416 and HGNC:15829 and HGNC:21308 and HGNC:3594) or HGNC:3594'},\n",
       " {'subsystem': 'Fatty acid synthesis',\n",
       "  'name': 'fatty-acyl-CoA synthase (n-C18:0CoA)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:14416 and HGNC:15829 and HGNC:21308 and HGNC:3594) or HGNC:3594']},\n",
       "  'metabolites': {'h2o_c': 1.0,\n",
       "   'nadp_c': 2.0,\n",
       "   'nadph_c': -2.0,\n",
       "   'pmtcoa_c': -1.0,\n",
       "   'stcoa_c': 1.0,\n",
       "   'malcoa_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'h_c': -3.0,\n",
       "   'co2_c': 1.0},\n",
       "  'id': 'FAS180COA',\n",
       "  'gene_reaction_rule': '(HGNC:14416 and HGNC:15829 and HGNC:21308 and HGNC:3594) or HGNC:3594'},\n",
       " {'subsystem': 'Fatty acid synthesis',\n",
       "  'name': 'fatty acyl-CoA synthase (n-C8:0CoA), lumped reaction',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3594']},\n",
       "  'metabolites': {'h2o_c': 3.0,\n",
       "   'malcoa_c': -3.0,\n",
       "   'nadph_c': -6.0,\n",
       "   'nadp_c': 6.0,\n",
       "   'coa_c': 3.0,\n",
       "   'h_c': -9.0,\n",
       "   'accoa_c': -1.0,\n",
       "   'co2_c': 3.0,\n",
       "   'occoa_c': 1.0},\n",
       "  'id': 'FAS80COA_L',\n",
       "  'gene_reaction_rule': 'HGNC:3594'},\n",
       " {'subsystem': 'Glycolysis/gluconeogenesis',\n",
       "  'name': 'fructose-bisphosphate aldolase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['6765112 - fluorescence of presence in human brain'],\n",
       "   'GENE ASSOCIATION': ['HGNC:414 or HGNC:417 or HGNC:418']},\n",
       "  'metabolites': {'dhap_c': 1.0, 'fdp_c': -1.0, 'g3p_c': 1.0},\n",
       "  'id': 'FBA',\n",
       "  'gene_reaction_rule': 'HGNC:414 or HGNC:417 or HGNC:418'},\n",
       " {'subsystem': 'Fructose and mannose metabolism',\n",
       "  'name': 'D-Fructose 1-phosphate D-glyceraldehyde-3-phosphate-lyase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:414 or HGNC:417 or HGNC:418']},\n",
       "  'metabolites': {'f1p_c': -1.0, 'dhap_c': 1.0, 'glyald_c': 1.0},\n",
       "  'id': 'FBA2',\n",
       "  'gene_reaction_rule': 'HGNC:414 or HGNC:417 or HGNC:418'},\n",
       " {'subsystem': 'Glyoxylate and dicarboxylate metabolism',\n",
       "  'name': 'D-Fructose 1-phosphate D-glyceraldehyde-3-phosphate-lyase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:414 or HGNC:417 or HGNC:418']},\n",
       "  'metabolites': {'dhap_c': 1.0, 'xu1p_D_c': -1.0, 'gcald_c': 1.0},\n",
       "  'id': 'FBA4',\n",
       "  'gene_reaction_rule': 'HGNC:414 or HGNC:417 or HGNC:418'},\n",
       " {'subsystem': 'Galactose metabolism',\n",
       "  'name': 'D-Tagatose 1-phosphate D-glyceraldehyde-3-phosphate-lyase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:417']},\n",
       "  'metabolites': {'dhap_c': 1.0, 'tag1p_D_c': -1.0, 'glyald_c': 1.0},\n",
       "  'id': 'FBA5',\n",
       "  'gene_reaction_rule': 'HGNC:417'},\n",
       " {'subsystem': 'Glycolysis/gluconeogenesis',\n",
       "  'name': 'fructose-bisphosphatase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['SMP'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3606 or HGNC:3607']},\n",
       "  'metabolites': {'h2o_c': -1.0, 'pi_c': 1.0, 'fdp_c': -1.0, 'f6p_c': 1.0},\n",
       "  'id': 'FBP',\n",
       "  'gene_reaction_rule': 'HGNC:3606 or HGNC:3607'},\n",
       " {'subsystem': 'Fructose and mannose metabolism',\n",
       "  'name': 'Fructose-2,6-bisphosphate 2-phosphatase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['IF'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3606 or HGNC:3607 or HGNC:8872 or HGNC:8873 or HGNC:8874 or HGNC:8875']},\n",
       "  'metabolites': {'f26bp_c': -1.0, 'h2o_c': -1.0, 'pi_c': 1.0, 'f6p_c': 1.0},\n",
       "  'id': 'FBP26',\n",
       "  'gene_reaction_rule': 'HGNC:3606 or HGNC:3607 or HGNC:8872 or HGNC:8873 or HGNC:8874 or HGNC:8875'},\n",
       " {'subsystem': 'Heme synthesis',\n",
       "  'name': 'Ferrochelatase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD; r0096'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3647']},\n",
       "  'metabolites': {'pheme_m': 1.0, 'h_m': 2.0, 'fe2_m': -1.0, 'ppp9_m': -1.0},\n",
       "  'id': 'FCLTm',\n",
       "  'gene_reaction_rule': 'HGNC:3647'},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'formyl coa hydrolase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['Note - formcoa formula is uncharged - the correct is FACOAH2 w/ forcoa NJ'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   'for_c': 1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'formcoa_c': -1.0},\n",
       "  'id': 'FCOAH',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Fatty acid oxidation',\n",
       "  'name': 'formate dehydrogenase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'], 'GENE ASSOCIATION': ['HGNC:3978']},\n",
       "  'metabolites': {'nadh_c': 1.0, 'co2_c': 1.0, 'nad_c': -1.0, 'for_c': -1.0},\n",
       "  'id': 'FDH',\n",
       "  'gene_reaction_rule': 'HGNC:3978'},\n",
       " {'subsystem': 'Heme synthesis',\n",
       "  'name': 'Fe(III) reduction (ascorbate)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['stoichiometry unknown; inferred from EC1.16.1.7 NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:20797']},\n",
       "  'metabolites': {'ascb_L_e': -1.0,\n",
       "   'h_e': 2.0,\n",
       "   'dhdascb_e': 1.0,\n",
       "   'fe2_e': 2.0,\n",
       "   'fe3_e': -2.0},\n",
       "  'id': 'FE3R2e',\n",
       "  'gene_reaction_rule': 'HGNC:20797'},\n",
       " {'subsystem': 'Fructose and mannose metabolism',\n",
       "  'name': 'Fucokinase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['kh 1/6/03 NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:29500']},\n",
       "  'metabolites': {'adp_c': 1.0,\n",
       "   'fuc1p_L_c': 1.0,\n",
       "   'fuc_L_c': -1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'h_c': 1.0},\n",
       "  'id': 'FK',\n",
       "  'gene_reaction_rule': 'HGNC:29500'},\n",
       " {'subsystem': 'Tryptophan metabolism',\n",
       "  'name': 'N-Formyl-L-kynurenine amidohydrolase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:20910']},\n",
       "  'metabolites': {'Lfmkynr_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'Lkynr_c': 1.0,\n",
       "   'for_c': 1.0,\n",
       "   'h_c': 1.0},\n",
       "  'id': 'FKYNH',\n",
       "  'gene_reaction_rule': 'HGNC:20910'},\n",
       " {'subsystem': 'Vitamin B2 metabolism',\n",
       "  'name': 'FMN adenylyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['Human Y/N: Y'],\n",
       "   'GENE ASSOCIATION': ['HGNC:24671']},\n",
       "  'metabolites': {'fad_c': 1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'fmn_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'atp_c': -1.0},\n",
       "  'id': 'FMNAT',\n",
       "  'gene_reaction_rule': 'HGNC:24671'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'folate reductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'], 'GENE ASSOCIATION': ['HGNC:2861']},\n",
       "  'metabolites': {'fol_c': -1.0, 'nadp_c': 1.0, 'nadph_c': -1.0, 'dhf_c': 1.0},\n",
       "  'id': 'FOLR2',\n",
       "  'gene_reaction_rule': 'HGNC:2861'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'folylpolyglutamate synthetase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3824']},\n",
       "  'metabolites': {'h2o_c': 3.0,\n",
       "   'glu_L_c': -4.0,\n",
       "   'atp_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   'adp_c': 1.0,\n",
       "   'pi_c': 1.0,\n",
       "   '5thf_c': 1.0,\n",
       "   'thf_c': -1.0},\n",
       "  'id': 'FPGS',\n",
       "  'gene_reaction_rule': 'HGNC:3824'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'folylpolyglutamate synthetase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3824']},\n",
       "  'metabolites': {'6thf_c': 1.0,\n",
       "   'glu_L_c': -1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   'adp_c': 1.0,\n",
       "   'pi_c': 1.0,\n",
       "   '5thf_c': -1.0},\n",
       "  'id': 'FPGS2',\n",
       "  'gene_reaction_rule': 'HGNC:3824'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'folylpolyglutamate synthetase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3824']},\n",
       "  'metabolites': {'glu_L_m': -1.0,\n",
       "   '6thf_m': 1.0,\n",
       "   'atp_m': -1.0,\n",
       "   '5thf_m': -1.0,\n",
       "   'pi_m': 1.0,\n",
       "   'adp_m': 1.0,\n",
       "   'h_m': 1.0},\n",
       "  'id': 'FPGS2m',\n",
       "  'gene_reaction_rule': 'HGNC:3824'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'folylpolyglutamate synthetase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3824']},\n",
       "  'metabolites': {'6thf_c': -1.0,\n",
       "   'glu_L_c': -1.0,\n",
       "   'atp_c': -1.0,\n",
       "   '7thf_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'adp_c': 1.0,\n",
       "   'pi_c': 1.0},\n",
       "  'id': 'FPGS3',\n",
       "  'gene_reaction_rule': 'HGNC:3824'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'folylpolyglutamate synthetase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3824']},\n",
       "  'metabolites': {'7thf_m': 1.0,\n",
       "   'glu_L_m': -1.0,\n",
       "   '6thf_m': -1.0,\n",
       "   'atp_m': -1.0,\n",
       "   'pi_m': 1.0,\n",
       "   'adp_m': 1.0,\n",
       "   'h_m': 1.0},\n",
       "  'id': 'FPGS3m',\n",
       "  'gene_reaction_rule': 'HGNC:3824'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'folylpolyglutamate synthetase (DHF)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3824']},\n",
       "  'metabolites': {'h2o_c': 3.0,\n",
       "   'dhf_c': -1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'glu_L_c': -4.0,\n",
       "   '5dhf_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'adp_c': 1.0,\n",
       "   'pi_c': 1.0},\n",
       "  'id': 'FPGS4',\n",
       "  'gene_reaction_rule': 'HGNC:3824'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'folylpolyglutamate synthetase (DHF), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3824']},\n",
       "  'metabolites': {'5dhf_m': 1.0,\n",
       "   'glu_L_m': -4.0,\n",
       "   'dhf_m': -1.0,\n",
       "   'h2o_m': 3.0,\n",
       "   'atp_m': -1.0,\n",
       "   'pi_m': 1.0,\n",
       "   'adp_m': 1.0,\n",
       "   'h_m': 1.0},\n",
       "  'id': 'FPGS4m',\n",
       "  'gene_reaction_rule': 'HGNC:3824'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'folylpolyglutamate  synthetase (DHF)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3824']},\n",
       "  'metabolites': {'glu_L_c': -1.0,\n",
       "   'atp_c': -1.0,\n",
       "   '5dhf_c': -1.0,\n",
       "   '6dhf_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'adp_c': 1.0,\n",
       "   'pi_c': 1.0},\n",
       "  'id': 'FPGS5',\n",
       "  'gene_reaction_rule': 'HGNC:3824'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'folylpolyglutamate  synthetase (DHF), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3824']},\n",
       "  'metabolites': {'5dhf_m': -1.0,\n",
       "   'glu_L_m': -1.0,\n",
       "   'atp_m': -1.0,\n",
       "   'pi_m': 1.0,\n",
       "   'adp_m': 1.0,\n",
       "   'h_m': 1.0,\n",
       "   '6dhf_m': 1.0},\n",
       "  'id': 'FPGS5m',\n",
       "  'gene_reaction_rule': 'HGNC:3824'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'folylpolyglutamate synthetase (DHF)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3824']},\n",
       "  'metabolites': {'pi_c': 1.0,\n",
       "   'glu_L_c': -1.0,\n",
       "   'atp_c': -1.0,\n",
       "   '6dhf_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   'adp_c': 1.0,\n",
       "   '7dhf_c': 1.0},\n",
       "  'id': 'FPGS6',\n",
       "  'gene_reaction_rule': 'HGNC:3824'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'folylpolyglutamate synthetase (DHF), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3824']},\n",
       "  'metabolites': {'glu_L_m': -1.0,\n",
       "   'atp_m': -1.0,\n",
       "   'pi_m': 1.0,\n",
       "   'adp_m': 1.0,\n",
       "   'h_m': 1.0,\n",
       "   '6dhf_m': -1.0,\n",
       "   '7dhf_m': 1.0},\n",
       "  'id': 'FPGS6m',\n",
       "  'gene_reaction_rule': 'HGNC:3824'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'folylpolyglutamate synthetase (10fthf)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3824']},\n",
       "  'metabolites': {'h2o_c': 3.0,\n",
       "   'glu_L_c': -4.0,\n",
       "   'atp_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   'adp_c': 1.0,\n",
       "   '10fthf5glu_c': 1.0,\n",
       "   'pi_c': 1.0,\n",
       "   '10fthf_c': -1.0},\n",
       "  'id': 'FPGS7',\n",
       "  'gene_reaction_rule': 'HGNC:3824'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'folylpolyglutamate synthetase (10fthf), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3824']},\n",
       "  'metabolites': {'glu_L_m': -4.0,\n",
       "   'h2o_m': 3.0,\n",
       "   'atp_m': -1.0,\n",
       "   '10fthf_m': -1.0,\n",
       "   'pi_m': 1.0,\n",
       "   '10fthf5glu_m': 1.0,\n",
       "   'adp_m': 1.0,\n",
       "   'h_m': 1.0},\n",
       "  'id': 'FPGS7m',\n",
       "  'gene_reaction_rule': 'HGNC:3824'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'folylpolyglutamate synthetase (10fthf)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3824']},\n",
       "  'metabolites': {'atp_c': -1.0,\n",
       "   'glu_L_c': -1.0,\n",
       "   '10fthf6glu_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'adp_c': 1.0,\n",
       "   '10fthf5glu_c': -1.0,\n",
       "   'pi_c': 1.0},\n",
       "  'id': 'FPGS8',\n",
       "  'gene_reaction_rule': 'HGNC:3824'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'folylpolyglutamate synthetase (10fthf), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3824']},\n",
       "  'metabolites': {'10fthf6glu_m': 1.0,\n",
       "   'glu_L_m': -1.0,\n",
       "   'atp_m': -1.0,\n",
       "   'pi_m': 1.0,\n",
       "   '10fthf5glu_m': -1.0,\n",
       "   'adp_m': 1.0,\n",
       "   'h_m': 1.0},\n",
       "  'id': 'FPGS8m',\n",
       "  'gene_reaction_rule': 'HGNC:3824'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'folylpolyglutamate synthetase (10fthf)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3824']},\n",
       "  'metabolites': {'10fthf7glu_c': 1.0,\n",
       "   'glu_L_c': -1.0,\n",
       "   'atp_c': -1.0,\n",
       "   '10fthf6glu_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   'adp_c': 1.0,\n",
       "   'pi_c': 1.0},\n",
       "  'id': 'FPGS9',\n",
       "  'gene_reaction_rule': 'HGNC:3824'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'folylpolyglutamate synthetase (10fthf), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3824']},\n",
       "  'metabolites': {'10fthf6glu_m': -1.0,\n",
       "   'glu_L_m': -1.0,\n",
       "   '10fthf7glu_m': 1.0,\n",
       "   'atp_m': -1.0,\n",
       "   'pi_m': 1.0,\n",
       "   'adp_m': 1.0,\n",
       "   'h_m': 1.0},\n",
       "  'id': 'FPGS9m',\n",
       "  'gene_reaction_rule': 'HGNC:3824'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'folylpolyglutamate synthetase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3824']},\n",
       "  'metabolites': {'h_m': 1.0,\n",
       "   'glu_L_m': -4.0,\n",
       "   'h2o_m': 3.0,\n",
       "   'atp_m': -1.0,\n",
       "   '5thf_m': 1.0,\n",
       "   'pi_m': 1.0,\n",
       "   'adp_m': 1.0,\n",
       "   'thf_m': -1.0},\n",
       "  'id': 'FPGSm',\n",
       "  'gene_reaction_rule': 'HGNC:3824'},\n",
       " {'subsystem': 'Cholesterol metabolism',\n",
       "  'name': 'farnesyltranstransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['Human Y/N: Y'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4249']},\n",
       "  'metabolites': {'ppi_c': 1.0, 'ipdp_c': -1.0, 'frdp_c': -1.0, 'ggdp_c': 1.0},\n",
       "  'id': 'FRTT',\n",
       "  'gene_reaction_rule': 'HGNC:4249'},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'farnesyltranstransferase (trans,trans,cis-geranylgeranyl diphosphate-generating)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['EC 2.5.1.- NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'ttc_ggdp_c': 1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'ipdp_c': -1.0,\n",
       "   'frdp_c': -1.0},\n",
       "  'id': 'FT',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'formimidoyltransferase cyclodeaminase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3974']},\n",
       "  'metabolites': {'nh4_c': 1.0,\n",
       "   'methf_c': 1.0,\n",
       "   'h_c': -2.0,\n",
       "   '5forthf_c': -1.0},\n",
       "  'id': 'FTCD',\n",
       "  'gene_reaction_rule': 'HGNC:3974'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': '5-formethyltetrahydrofolate cyclo-ligase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:7437']},\n",
       "  'metabolites': {'5fthf_c': -1.0,\n",
       "   'pi_c': 1.0,\n",
       "   'methf_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'adp_c': 1.0},\n",
       "  'id': 'FTHFCL',\n",
       "  'gene_reaction_rule': 'HGNC:7437'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'formyltetrahydrofolate dehydrogenase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:26777 or HGNC:3978']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   'nadp_c': -1.0,\n",
       "   'nadph_c': 1.0,\n",
       "   'thf_c': 1.0,\n",
       "   'co2_c': 1.0,\n",
       "   '10fthf_c': -1.0,\n",
       "   'h_c': 1.0},\n",
       "  'id': 'FTHFDH',\n",
       "  'gene_reaction_rule': 'HGNC:26777 or HGNC:3978'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'Formate-tetrahydrofolate ligase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:7432']},\n",
       "  'metabolites': {'adp_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'thf_c': -1.0,\n",
       "   'for_c': -1.0,\n",
       "   'pi_c': 1.0,\n",
       "   '10fthf_c': 1.0},\n",
       "  'id': 'FTHFL',\n",
       "  'gene_reaction_rule': 'HGNC:7432'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'formate-tetrahydrofolate ligase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:21055 or HGNC:7432']},\n",
       "  'metabolites': {'atp_m': -1.0,\n",
       "   'adp_m': 1.0,\n",
       "   '10fthf_m': 1.0,\n",
       "   'pi_m': 1.0,\n",
       "   'for_m': -1.0,\n",
       "   'thf_m': -1.0},\n",
       "  'id': 'FTHFLm',\n",
       "  'gene_reaction_rule': 'HGNC:21055 or HGNC:7432'},\n",
       " {'subsystem': 'Keratan sulfate degradation',\n",
       "  'name': 'alpha-fucosidase, extracellular',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4008']},\n",
       "  'metabolites': {'fuc_L_e': 1.0,\n",
       "   'h2o_e': -1.0,\n",
       "   'ksi_deg1_e': 1.0,\n",
       "   'ksi_e': -1.0},\n",
       "  'id': 'FUCASE2e',\n",
       "  'gene_reaction_rule': 'HGNC:4008'},\n",
       " {'subsystem': 'Keratan sulfate degradation',\n",
       "  'name': 'alpha-fucosidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4006']},\n",
       "  'metabolites': {'fuc_L_l': 1.0,\n",
       "   'ksi_deg1_l': 1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   'ksi_l': -1.0},\n",
       "  'id': 'FUCASE2ly',\n",
       "  'gene_reaction_rule': 'HGNC:4006'},\n",
       " {'subsystem': 'N-glycan degradation',\n",
       "  'name': 'alpha-fucosidase, extracellular',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4008']},\n",
       "  'metabolites': {'fuc_L_e': 1.0,\n",
       "   's2l2n2m2masn_e': 1.0,\n",
       "   'h2o_e': -1.0,\n",
       "   's2l2fn2m2masn_e': -1.0},\n",
       "  'id': 'FUCASEe',\n",
       "  'gene_reaction_rule': 'HGNC:4008'},\n",
       " {'subsystem': 'N-glycan degradation',\n",
       "  'name': 'alpha-fucosidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4006']},\n",
       "  'metabolites': {'fuc_L_l': 1.0,\n",
       "   's2l2n2m2masn_l': 1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   's2l2fn2m2masn_l': -1.0},\n",
       "  'id': 'FUCASEly',\n",
       "  'gene_reaction_rule': 'HGNC:4006'},\n",
       " {'subsystem': 'Citric acid cycle',\n",
       "  'name': 'fumarase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'], 'GENE ASSOCIATION': ['HGNC:3700']},\n",
       "  'metabolites': {'h2o_c': -1.0, 'fum_c': -1.0, 'mal_L_c': 1.0},\n",
       "  'id': 'FUM',\n",
       "  'gene_reaction_rule': 'HGNC:3700'},\n",
       " {'subsystem': 'Tyrosine metabolism',\n",
       "  'name': 'fumarylacetoacetase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3579']},\n",
       "  'metabolites': {'acac_c': 1.0,\n",
       "   '4fumacac_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'fum_c': 1.0,\n",
       "   'h_c': 1.0},\n",
       "  'id': 'FUMAC',\n",
       "  'gene_reaction_rule': 'HGNC:3579'},\n",
       " {'subsystem': 'Citric acid cycle',\n",
       "  'name': 'fumarase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['IF. Biochemical'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3700']},\n",
       "  'metabolites': {'mal_L_m': 1.0, 'h2o_m': -1.0, 'fum_m': -1.0},\n",
       "  'id': 'FUMm',\n",
       "  'gene_reaction_rule': 'HGNC:3700'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'Galactoside 2-alpha-L-fucosyltransferase 1',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4012']},\n",
       "  'metabolites': {'gdp_g': 1.0,\n",
       "   'h_g': 1.0,\n",
       "   'galgbside_hs_g': -1.0,\n",
       "   'fucgalgbside_hs_g': 1.0,\n",
       "   'gdpfuc_g': -1.0},\n",
       "  'id': 'FUT11g',\n",
       "  'gene_reaction_rule': 'HGNC:4012'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'Galactoside 2-alpha-L-fucosyltransferase 1',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4012']},\n",
       "  'metabolites': {'gdp_g': 1.0,\n",
       "   'galacglcgalgluside_hs_g': -1.0,\n",
       "   'fucgalacglcgalgluside_hs_g': 1.0,\n",
       "   'gdpfuc_g': -1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'FUT12g',\n",
       "  'gene_reaction_rule': 'HGNC:4012'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'Galactoside 2-alpha-L-fucosyltransferase 1',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4012']},\n",
       "  'metabolites': {'gdp_g': 1.0,\n",
       "   'gal14acglcgalgluside_hs_g': -1.0,\n",
       "   'fuc12gal14acglcgalgluside_hs_g': 1.0,\n",
       "   'gdpfuc_g': -1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'FUT14g',\n",
       "  'gene_reaction_rule': 'HGNC:4012'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'Galactoside 2-alpha-L-fucosyltransferase 1',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4012']},\n",
       "  'metabolites': {'gdp_g': 1.0,\n",
       "   'h_g': 1.0,\n",
       "   'galacgalfuc12gal14acglcgalgluside_hs_g': -1.0,\n",
       "   'gdpfuc_g': -1.0,\n",
       "   'fucgalacgalfuc12gal14acglcgalgluside_hs_g': 1.0},\n",
       "  'id': 'FUT15g',\n",
       "  'gene_reaction_rule': 'HGNC:4012'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'Galactoside 2-alpha-L-fucosyltransferase 1',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4012']},\n",
       "  'metabolites': {'gdp_g': 1.0,\n",
       "   'galacglcgal14acglcgalgluside_hs_g': -1.0,\n",
       "   'fucgalacglcgal14acglcgalgluside_hs_g': 1.0,\n",
       "   'gdpfuc_g': -1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'FUT16g',\n",
       "  'gene_reaction_rule': 'HGNC:4012'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'Galactoside 2-alpha-L-fucosyltransferase 1',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4012']},\n",
       "  'metabolites': {'gdp_g': 1.0,\n",
       "   'h_g': 1.0,\n",
       "   'galacgalfucgalacglcgal14acglcgalgluside_hs_g': -1.0,\n",
       "   'fucgalacgalfucgalacglcgal14acglcgalgluside_hs_g': 1.0,\n",
       "   'gdpfuc_g': -1.0},\n",
       "  'id': 'FUT17g',\n",
       "  'gene_reaction_rule': 'HGNC:4012'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'Galactoside 2-alpha-L-fucosyltransferase 1',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4012']},\n",
       "  'metabolites': {'gdp_g': 2.0,\n",
       "   'galacglcgalacglcgal14acglcgalgluside_hs_g': -1.0,\n",
       "   'h_g': 2.0,\n",
       "   'gdpfuc_g': -2.0,\n",
       "   'fucfucgalacglcgalacglcgal14acglcgalgluside_hs_g': 1.0},\n",
       "  'id': 'FUT18g',\n",
       "  'gene_reaction_rule': 'HGNC:4012'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'fucosyltransferase 3 (galactoside 3(4)-L-fucosyltransferase, Lewis blood group included) 1',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4014']},\n",
       "  'metabolites': {'gdp_g': 1.0,\n",
       "   'galacglcgalgluside_hs_g': -1.0,\n",
       "   'fuc14galacglcgalgluside_hs_g': 1.0,\n",
       "   'gdpfuc_g': -1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'FUT31g',\n",
       "  'gene_reaction_rule': 'HGNC:4014'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'fucosyltransferase 3 (galactoside 3(4)-L-fucosyltransferase, Lewis blood group included) 1',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4014']},\n",
       "  'metabolites': {'gdp_g': 1.0,\n",
       "   'h_g': 1.0,\n",
       "   'fucacngalacglcgalgluside_hs_g': 1.0,\n",
       "   'gdpfuc_g': -1.0,\n",
       "   'acngalacglcgalgluside_hs_g': -1.0},\n",
       "  'id': 'FUT32g',\n",
       "  'gene_reaction_rule': 'HGNC:4014'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'fucosyltransferase 3 (galactoside 3(4)-L-fucosyltransferase, Lewis blood group included) 1',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4014']},\n",
       "  'metabolites': {'gdp_g': 1.0,\n",
       "   'fucacgalfucgalacglcgalgluside_hs_g': 1.0,\n",
       "   'acgalfucgalacglcgalgluside_hs_g': -1.0,\n",
       "   'gdpfuc_g': -1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'FUT33g',\n",
       "  'gene_reaction_rule': 'HGNC:4014'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'fucosyltransferase 3 (galactoside 3(4)-L-fucosyltransferase, Lewis blood group included) 1',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4014']},\n",
       "  'metabolites': {'gdp_g': 1.0,\n",
       "   'h_g': 1.0,\n",
       "   'fucfucgalacglcgalgluside_hs_g': 1.0,\n",
       "   'gdpfuc_g': -1.0,\n",
       "   'fucgalacglcgalgluside_hs_g': -1.0},\n",
       "  'id': 'FUT34g',\n",
       "  'gene_reaction_rule': 'HGNC:4014'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'fucosyltransferase 3 (galactoside 3(4)-L-fucosyltransferase, Lewis blood group included) 1',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4014']},\n",
       "  'metabolites': {'gdp_g': 1.0,\n",
       "   'galfucgalacglcgalgluside_hs_g': -1.0,\n",
       "   'fucgalfucgalacglcgalgluside_hs_g': 1.0,\n",
       "   'gdpfuc_g': -1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'FUT35g',\n",
       "  'gene_reaction_rule': 'HGNC:4014'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'Alpha-(1,3)-fucosyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4020']},\n",
       "  'metabolites': {'gdp_g': 1.0,\n",
       "   'h_g': 1.0,\n",
       "   'fucgalacglc13galacglcgal14acglcgalgluside_hs_g': -1.0,\n",
       "   'gdpfuc_g': -1.0,\n",
       "   'fucfucgalacglc13galacglcgal14acglcgalgluside_hs_g': 1.0},\n",
       "  'id': 'FUT910g',\n",
       "  'gene_reaction_rule': 'HGNC:4020'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'Alpha-(1,3)-fucosyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4020']},\n",
       "  'metabolites': {'gdp_g': 1.0,\n",
       "   'fucfucgalacglc13galacglcgal14acglcgalgluside_hs_g': -1.0,\n",
       "   'gdpfuc_g': -1.0,\n",
       "   'fucfucfucgalacglc13galacglcgal14acglcgalgluside_hs_g': 1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'FUT911g',\n",
       "  'gene_reaction_rule': 'HGNC:4020'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'Alpha-(1,3)-fucosyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4020']},\n",
       "  'metabolites': {'gdp_g': 1.0,\n",
       "   'gal14acglcgalgluside_hs_g': -1.0,\n",
       "   'fucgal14acglcgalgluside_hs_g': 1.0,\n",
       "   'gdpfuc_g': -1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'FUT91g',\n",
       "  'gene_reaction_rule': 'HGNC:4020'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'Alpha-(1,3)-fucosyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4020']},\n",
       "  'metabolites': {'gdp_g': 1.0,\n",
       "   'h_g': 1.0,\n",
       "   'acngal14acglcgalgluside_hs_g': -1.0,\n",
       "   'gdpfuc_g': -1.0,\n",
       "   'fucacngal14acglcgalgluside_hs_g': 1.0},\n",
       "  'id': 'FUT92g',\n",
       "  'gene_reaction_rule': 'HGNC:4020'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'Alpha-(1,3)-fucosyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4020']},\n",
       "  'metabolites': {'gdp_g': 1.0,\n",
       "   'h_g': 1.0,\n",
       "   'fucfuc12gal14acglcgalgluside_hs_g': 1.0,\n",
       "   'fuc12gal14acglcgalgluside_hs_g': -1.0,\n",
       "   'gdpfuc_g': -1.0},\n",
       "  'id': 'FUT93g',\n",
       "  'gene_reaction_rule': 'HGNC:4020'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'Alpha-(1,3)-fucosyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4020']},\n",
       "  'metabolites': {'gdp_g': 1.0,\n",
       "   'h_g': 1.0,\n",
       "   'fucgalacglcgal14acglcgalgluside_hs_g': -1.0,\n",
       "   'gdpfuc_g': -1.0,\n",
       "   'fucfucgalacglcgal14acglcgalgluside_hs_g': 1.0},\n",
       "  'id': 'FUT94g',\n",
       "  'gene_reaction_rule': 'HGNC:4020'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'Alpha-(1,3)-fucosyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4020']},\n",
       "  'metabolites': {'gdp_g': 1.0,\n",
       "   'h_g': 1.0,\n",
       "   'fucfucfucgalacglcgal14acglcgalgluside_hs_g': 1.0,\n",
       "   'gdpfuc_g': -1.0,\n",
       "   'fucfucgalacglcgal14acglcgalgluside_hs_g': -1.0},\n",
       "  'id': 'FUT95g',\n",
       "  'gene_reaction_rule': 'HGNC:4020'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'Alpha-(1,3)-fucosyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4020']},\n",
       "  'metabolites': {'gdp_g': 1.0,\n",
       "   'galacglcgal14acglcgalgluside_hs_g': -1.0,\n",
       "   'fuc132galacglcgal14acglcgalgluside_hs_g': 1.0,\n",
       "   'gdpfuc_g': -1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'FUT96g',\n",
       "  'gene_reaction_rule': 'HGNC:4020'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'Alpha-(1,3)-fucosyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4020']},\n",
       "  'metabolites': {'gdp_g': 1.0,\n",
       "   'fucfuc132galacglcgal14acglcgalgluside_hs_g': 1.0,\n",
       "   'fuc132galacglcgal14acglcgalgluside_hs_g': -1.0,\n",
       "   'gdpfuc_g': -1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'FUT97g',\n",
       "  'gene_reaction_rule': 'HGNC:4020'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'Alpha-(1,3)-fucosyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4020']},\n",
       "  'metabolites': {'gdp_g': 1.0,\n",
       "   'galacglcgal14acglcgalgluside_hs_g': -1.0,\n",
       "   'fuc13galacglcgal14acglcgalgluside_hs_g': 1.0,\n",
       "   'gdpfuc_g': -1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'FUT98g',\n",
       "  'gene_reaction_rule': 'HGNC:4020'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'Alpha-(1,3)-fucosyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4020']},\n",
       "  'metabolites': {'gdp_g': 1.0,\n",
       "   'h_g': 1.0,\n",
       "   'galacglc13galacglcgal14acglcgalgluside_hs_g': -1.0,\n",
       "   'fucgalacglc13galacglcgal14acglcgalgluside_hs_g': 1.0,\n",
       "   'gdpfuc_g': -1.0},\n",
       "  'id': 'FUT99g',\n",
       "  'gene_reaction_rule': 'HGNC:4020'},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'Glycolipid 1,2-alpha-D-mannosyltransferase (uterus)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'], 'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'m3mpdol_U_c': 1.0,\n",
       "   'gdpmann_c': -1.0,\n",
       "   'gdp_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'm2mpdol_U_c': -1.0},\n",
       "  'id': 'G12MT1_U',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'Glycolipid 1,2-alpha-D-mannosyltransferase (uterus)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'], 'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'gdp_c': 1.0,\n",
       "   'gdpmann_c': -1.0,\n",
       "   'm4mpdol_U_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'm3mpdol_U_c': -1.0},\n",
       "  'id': 'G12MT2_U',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'Glycolipid 1,3-alpha-D-mannosyltransferase (uterus)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'gdpmann_c': -1.0,\n",
       "   'm1mpdol_U_c': 1.0,\n",
       "   'gdp_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'mpdol_U_c': -1.0},\n",
       "  'id': 'G13MT_U',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'beta-N-acetylglucosaminylglycopeptide beta-1,4-galactosyltransferase, Golgi',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Keratan sulfate I biosynthesis NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'ksi_pre15_g': 1.0,\n",
       "   'ksi_pre14_g': -1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0},\n",
       "  'id': 'G14T10g',\n",
       "  'gene_reaction_rule': 'HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'beta-N-acetylglucosaminylglycopeptide beta-1,4-galactosyltransferase, Golgi',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Keratan sulfate I biosynthesis NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'ksi_pre17_g': -1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'ksi_pre18_g': 1.0},\n",
       "  'id': 'G14T11g',\n",
       "  'gene_reaction_rule': 'HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'beta-N-acetylglucosaminylglycopeptide beta-1,4-galactosyltransferase, Golgi',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Keratan sulfate I biosynthesis NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'ksi_pre20_g': -1.0,\n",
       "   'ksi_pre21_g': 1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0},\n",
       "  'id': 'G14T12g',\n",
       "  'gene_reaction_rule': 'HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'beta-N-acetylglucosaminylglycopeptide beta-1,4-galactosyltransferase, Golgi',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Keratan sulfate I biosynthesis NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'ksi_pre24_g': 1.0,\n",
       "   'ksi_pre23_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'udpgal_g': -1.0},\n",
       "  'id': 'G14T13g',\n",
       "  'gene_reaction_rule': 'HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'beta-N-acetylglucosaminylglycopeptide beta-1,4-galactosyltransferase, Golgi',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Keratan sulfate I biosynthesis NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928']},\n",
       "  'metabolites': {'ksi_pre26_g': -1.0,\n",
       "   'h_g': 1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'ksi_pre27_g': 1.0},\n",
       "  'id': 'G14T14g',\n",
       "  'gene_reaction_rule': 'HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'beta-N-acetylglucosaminylglycopeptide beta-1,4-galactosyltransferase, Golgi',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Keratan sulfate I biosynthesis NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'ksi_pre29_g': -1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'ksi_pre30_g': 1.0},\n",
       "  'id': 'G14T15g',\n",
       "  'gene_reaction_rule': 'HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'beta-N-acetylglucosaminylglycopeptide beta-1,4-galactosyltransferase, Golgi',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Keratan sulfate I biosynthesis NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'ksi_pre33_g': 1.0,\n",
       "   'ksi_pre32_g': -1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0},\n",
       "  'id': 'G14T16g',\n",
       "  'gene_reaction_rule': 'HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'beta-N-acetylglucosaminylglycopeptide beta-1,4-galactosyltransferase, Golgi',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Keratan sulfate I biosynthesis NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928']},\n",
       "  'metabolites': {'ksi_pre35_g': -1.0,\n",
       "   'h_g': 1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'ksi_pre36_g': 1.0},\n",
       "  'id': 'G14T17g',\n",
       "  'gene_reaction_rule': 'HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'beta-N-acetylglucosaminylglycopeptide beta-1,4-galactosyltransferase, Golgi',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Keratan sulfate II biosynthesis NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928']},\n",
       "  'metabolites': {'core4_g': -1.0,\n",
       "   'ksii_core4_pre1_g': 1.0,\n",
       "   'udpgal_g': -2.0,\n",
       "   'udp_g': 2.0,\n",
       "   'h_g': 2.0},\n",
       "  'id': 'G14T18g',\n",
       "  'gene_reaction_rule': 'HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'beta-N-acetylglucosaminylglycopeptide beta-1,4-galactosyltransferase, Golgi',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Keratan sulfate II biosynthesis NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'ksii_core4_pre3_g': -1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'ksii_core4_pre4_g': 1.0},\n",
       "  'id': 'G14T19g',\n",
       "  'gene_reaction_rule': 'HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'beta-N-acetylglucosaminylglycopeptide beta-1,4-galactosyltransferase, Golgi',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Keratan sulfate II biosynthesis NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928']},\n",
       "  'metabolites': {'ksii_core4_pre7_g': 1.0,\n",
       "   'ksii_core4_pre6_g': -1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'G14T20g',\n",
       "  'gene_reaction_rule': 'HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'beta-N-acetylglucosaminylglycopeptide beta-1,4-galactosyltransferase, Golgi',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Keratan sulfate II biosynthesis NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'ksii_core4_pre9_g': -1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'ksii_core4_pre10_g': 1.0},\n",
       "  'id': 'G14T21g',\n",
       "  'gene_reaction_rule': 'HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'beta-N-acetylglucosaminylglycopeptide beta-1,4-galactosyltransferase, Golgi',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Keratan sulfate II biosynthesis NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928']},\n",
       "  'metabolites': {'ksii_core2_pre1_g': -1.0,\n",
       "   'h_g': 1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'ksii_core2_pre2_g': 1.0},\n",
       "  'id': 'G14T2g',\n",
       "  'gene_reaction_rule': 'HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'beta-N-acetylglucosaminylglycopeptide beta-1,4-galactosyltransferase, Golgi',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Keratan sulfate II biosynthesis NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928']},\n",
       "  'metabolites': {'ksii_core2_pre4_g': 1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'ksii_core2_pre3_g': -1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'G14T3g',\n",
       "  'gene_reaction_rule': 'HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'beta-N-acetylglucosaminylglycopeptide beta-1,4-galactosyltransferase, Golgi',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Keratan sulfate II biosynthesis NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'ksii_core2_pre7_g': 1.0,\n",
       "   'ksii_core2_pre6_g': -1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0},\n",
       "  'id': 'G14T4g',\n",
       "  'gene_reaction_rule': 'HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'beta-N-acetylglucosaminylglycopeptide beta-1,4-galactosyltransferase, Golgi',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Keratan sulfate II biosynthesis NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928']},\n",
       "  'metabolites': {'ksii_core2_pre9_g': -1.0,\n",
       "   'h_g': 1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'ksii_core2_pre10_g': 1.0},\n",
       "  'id': 'G14T5g',\n",
       "  'gene_reaction_rule': 'HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'beta-N-acetylglucosaminylglycopeptide beta-1,4-galactosyltransferase, Golgi',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Keratan sulfate I biosynthesis NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'ksi_pre3_g': 1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'ksi_pre2_g': -1.0},\n",
       "  'id': 'G14T6g',\n",
       "  'gene_reaction_rule': 'HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'beta-N-acetylglucosaminylglycopeptide beta-1,4-galactosyltransferase, Golgi',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Keratan sulfate I biosynthesis NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'ksi_pre5_g': -1.0,\n",
       "   'ksi_pre6_g': 1.0},\n",
       "  'id': 'G14T7g',\n",
       "  'gene_reaction_rule': 'HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'beta-N-acetylglucosaminylglycopeptide beta-1,4-galactosyltransferase, Golgi',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Keratan sulfate I biosynthesis NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'ksi_pre9_g': 1.0,\n",
       "   'ksi_pre8_g': -1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0},\n",
       "  'id': 'G14T8g',\n",
       "  'gene_reaction_rule': 'HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928'},\n",
       " {'subsystem': 'Keratan sulfate synthesis',\n",
       "  'name': 'beta-N-acetylglucosaminylglycopeptide beta-1,4-galactosyltransferase, Golgi',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Keratan sulfate I biosynthesis NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928']},\n",
       "  'metabolites': {'udp_g': 1.0,\n",
       "   'ksi_pre12_g': 1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'ksi_pre11_g': -1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'G14T9g',\n",
       "  'gene_reaction_rule': 'HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928'},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'beta-N-acetylglucosaminylglycopeptide beta-1,4-galactosyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928']},\n",
       "  'metabolites': {'fn2m2masn_g': -1.0,\n",
       "   'h2o_g': 1.0,\n",
       "   'l2fn2m2masn_g': 1.0,\n",
       "   'h_g': 2.0,\n",
       "   'udpgal_g': -2.0,\n",
       "   'udp_g': 2.0},\n",
       "  'id': 'G14Tg',\n",
       "  'gene_reaction_rule': 'HGNC:924 or HGNC:925 or HGNC:926 or HGNC:928'},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'Glycolipid 1,6-alpha-D-mannosyltransferase (uterus)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'], 'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'gdpmann_c': -1.0,\n",
       "   'm1mpdol_U_c': -1.0,\n",
       "   'gdp_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'm2mpdol_U_c': 1.0},\n",
       "  'id': 'G16MT_U',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Nucleotide sugar metabolism',\n",
       "  'name': 'glucose-1-phosphate thymidylyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['JLR'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'dtdpglu_c': 1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'dttp_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'g1p_c': -1.0},\n",
       "  'id': 'G1PTT',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Glycerophospholipid metabolism',\n",
       "  'name': 'glycerol-3-phosphate dehydrogenase (NAD)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NAD-specific; see also G3PD2'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4455']},\n",
       "  'metabolites': {'nad_c': -1.0,\n",
       "   'glyc3p_c': -1.0,\n",
       "   'dhap_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'nadh_c': 1.0},\n",
       "  'id': 'G3PD1',\n",
       "  'gene_reaction_rule': 'HGNC:4455'},\n",
       " {'subsystem': 'Arginine and Proline Metabolism',\n",
       "  'name': 'L-glutamate 5-semialdehyde dehydratase, reversible, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['spontaneous reaction NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'h_m': 1.0, 'h2o_m': 1.0, '1pyr5c_m': 1.0, 'glu5sa_m': -1.0},\n",
       "  'id': 'G5SADrm',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Urea cycle',\n",
       "  'name': 'glutamate-5-semialdehyde dehydrogenase (m)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['SAB m TDV: name change from G5SDm'],\n",
       "   'GENE ASSOCIATION': ['HGNC:9722']},\n",
       "  'metabolites': {'nadph_m': -1.0,\n",
       "   'nadp_m': 1.0,\n",
       "   'glu5sa_m': 1.0,\n",
       "   'glu5p_m': -1.0,\n",
       "   'pi_m': 1.0,\n",
       "   'h_m': -1.0},\n",
       "  'id': 'G5SDym',\n",
       "  'gene_reaction_rule': 'HGNC:9722'},\n",
       " {'subsystem': 'Aminosugar metabolism',\n",
       "  'name': 'glucosamine-6-phosphate deaminase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Previously EC 5.3.1.10'],\n",
       "   'GENE ASSOCIATION': ['HGNC:21526 or HGNC:4417']},\n",
       "  'metabolites': {'h2o_c': -1.0, 'nh4_c': 1.0, 'gam6p_c': -1.0, 'f6p_c': 1.0},\n",
       "  'id': 'G6PDA',\n",
       "  'gene_reaction_rule': 'HGNC:21526 or HGNC:4417'},\n",
       " {'subsystem': 'Pentose phosphate pathway',\n",
       "  'name': 'glucose-6-phosphate dehydrogenase, endoplasmic reticulum',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['tv, jp NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4795']},\n",
       "  'metabolites': {'6pgl_r': 1.0,\n",
       "   'g6p_r': -1.0,\n",
       "   'h_r': 1.0,\n",
       "   'nadh_r': 1.0,\n",
       "   'nad_r': -1.0},\n",
       "  'id': 'G6PDH1rer',\n",
       "  'gene_reaction_rule': 'HGNC:4795'},\n",
       " {'subsystem': 'Pentose phosphate pathway',\n",
       "  'name': 'glucose 6-phosphate dehydrogenase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['JLR - a reversible reaction that is otherwise identical to ZWF'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4057']},\n",
       "  'metabolites': {'6pgl_c': 1.0,\n",
       "   'g6p_c': -1.0,\n",
       "   'nadp_c': -1.0,\n",
       "   'nadph_c': 1.0,\n",
       "   'h_c': 1.0},\n",
       "  'id': 'G6PDH2r',\n",
       "  'gene_reaction_rule': 'HGNC:4057'},\n",
       " {'subsystem': 'Pentose phosphate pathway',\n",
       "  'name': 'glucose 6-phosphate dehydrogenase, endoplasmic reticulum',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4057 or HGNC:4795']},\n",
       "  'metabolites': {'6pgl_r': 1.0,\n",
       "   'g6p_r': -1.0,\n",
       "   'h_r': 1.0,\n",
       "   'nadph_r': 1.0,\n",
       "   'nadp_r': -1.0},\n",
       "  'id': 'G6PDH2rer',\n",
       "  'gene_reaction_rule': 'HGNC:4057 or HGNC:4795'},\n",
       " {'subsystem': 'Glycolysis/gluconeogenesis',\n",
       "  'name': 'glucose-6-phosphate phosphatase, edoplasmic reticular',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:24861 or HGNC:28906 or HGNC:4056']},\n",
       "  'metabolites': {'glc_D_r': 1.0, 'pi_r': 1.0, 'g6p_r': -1.0, 'h2o_r': -1.0},\n",
       "  'id': 'G6PPer',\n",
       "  'gene_reaction_rule': 'HGNC:24861 or HGNC:28906 or HGNC:4056'},\n",
       " {'subsystem': 'Urea cycle',\n",
       "  'name': 'guanidinoacetate N-methyltransferase (c)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB c'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4136']},\n",
       "  'metabolites': {'amet_c': -1.0,\n",
       "   'ahcys_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'creat_c': 1.0,\n",
       "   'gudac_c': -1.0},\n",
       "  'id': 'GACMTRc',\n",
       "  'gene_reaction_rule': 'HGNC:4136'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'Galactosylceramide sulfotransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:24240']},\n",
       "  'metabolites': {'pap_g': 1.0,\n",
       "   'h_g': 1.0,\n",
       "   'paps_g': -1.0,\n",
       "   'galside_hs_g': -1.0,\n",
       "   'sgalside_hs_g': 1.0},\n",
       "  'id': 'GAL3ST11',\n",
       "  'gene_reaction_rule': 'HGNC:24240'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'Galactosylceramide sulfotransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:24240']},\n",
       "  'metabolites': {'pap_g': 1.0,\n",
       "   'digalside_hs_g': -1.0,\n",
       "   'paps_g': -1.0,\n",
       "   'digalsgalside_hs_g': 1.0},\n",
       "  'id': 'GAL3ST12',\n",
       "  'gene_reaction_rule': 'HGNC:24240'},\n",
       " {'subsystem': 'Keratan sulfate degradation',\n",
       "  'name': 'beta-galactosidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)']},\n",
       "  'metabolites': {'ksi_deg26_l': -1.0,\n",
       "   'ksi_deg27_l': 1.0,\n",
       "   'gal_l': 1.0,\n",
       "   'h2o_l': -1.0},\n",
       "  'id': 'GALASE10ly',\n",
       "  'gene_reaction_rule': '(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)'},\n",
       " {'subsystem': 'Keratan sulfate degradation',\n",
       "  'name': 'beta-galactosidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)']},\n",
       "  'metabolites': {'ksi_deg29_l': -1.0,\n",
       "   'gal_l': 1.0,\n",
       "   'ksi_deg30_l': 1.0,\n",
       "   'h2o_l': -1.0},\n",
       "  'id': 'GALASE11ly',\n",
       "  'gene_reaction_rule': '(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)'},\n",
       " {'subsystem': 'Keratan sulfate degradation',\n",
       "  'name': 'beta-galactosidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)']},\n",
       "  'metabolites': {'ksi_deg33_l': 1.0,\n",
       "   'ksi_deg32_l': -1.0,\n",
       "   'gal_l': 1.0,\n",
       "   'h2o_l': -1.0},\n",
       "  'id': 'GALASE12ly',\n",
       "  'gene_reaction_rule': '(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)'},\n",
       " {'subsystem': 'Keratan sulfate degradation',\n",
       "  'name': 'beta-galactosidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)']},\n",
       "  'metabolites': {'ksi_deg35_l': -1.0,\n",
       "   'gal_l': 1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   'ksi_deg36_l': 1.0},\n",
       "  'id': 'GALASE13ly',\n",
       "  'gene_reaction_rule': '(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)'},\n",
       " {'subsystem': 'Keratan sulfate degradation',\n",
       "  'name': 'beta-galactosidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)']},\n",
       "  'metabolites': {'gal_l': 1.0,\n",
       "   'ksi_deg39_l': 1.0,\n",
       "   'ksi_deg38_l': -1.0,\n",
       "   'h2o_l': -1.0},\n",
       "  'id': 'GALASE14ly',\n",
       "  'gene_reaction_rule': '(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)'},\n",
       " {'subsystem': 'Keratan sulfate degradation',\n",
       "  'name': 'beta-galactosidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)']},\n",
       "  'metabolites': {'gal_l': 1.0,\n",
       "   'ksi_deg40_l': -1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   'ksi_deg41_l': 1.0},\n",
       "  'id': 'GALASE15ly',\n",
       "  'gene_reaction_rule': '(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)'},\n",
       " {'subsystem': 'Keratan sulfate degradation',\n",
       "  'name': 'beta-galactosidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)']},\n",
       "  'metabolites': {'ksii_core2_deg3_l': 1.0,\n",
       "   'gal_l': 2.0,\n",
       "   'h2o_l': -2.0,\n",
       "   'ksii_core2_deg2_l': -1.0},\n",
       "  'id': 'GALASE16ly',\n",
       "  'gene_reaction_rule': '(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)'},\n",
       " {'subsystem': 'Keratan sulfate degradation',\n",
       "  'name': 'beta-galactosidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)']},\n",
       "  'metabolites': {'ksii_core2_deg6_l': 1.0,\n",
       "   'gal_l': 1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   'ksii_core2_deg5_l': -1.0},\n",
       "  'id': 'GALASE17ly',\n",
       "  'gene_reaction_rule': '(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)'},\n",
       " {'subsystem': 'Keratan sulfate degradation',\n",
       "  'name': 'beta-galactosidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)']},\n",
       "  'metabolites': {'ksii_core2_deg9_l': 1.0,\n",
       "   'ksii_core2_deg8_l': -1.0,\n",
       "   'gal_l': 1.0,\n",
       "   'h2o_l': -1.0},\n",
       "  'id': 'GALASE18ly',\n",
       "  'gene_reaction_rule': '(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)'},\n",
       " {'subsystem': 'Keratan sulfate degradation',\n",
       "  'name': 'beta-galactosidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)']},\n",
       "  'metabolites': {'gal_l': 1.0, 'f1a_l': -1.0, 'h2o_l': -1.0, 'core6_l': 1.0},\n",
       "  'id': 'GALASE19ly',\n",
       "  'gene_reaction_rule': '(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)'},\n",
       " {'subsystem': 'N-glycan degradation',\n",
       "  'name': 'beta-galactosidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)']},\n",
       "  'metabolites': {'l2n2m2mn_l': -1.0,\n",
       "   'gal_l': 2.0,\n",
       "   'h2o_l': -2.0,\n",
       "   'n2m2mn_l': 1.0},\n",
       "  'id': 'GALASE1ly',\n",
       "  'gene_reaction_rule': '(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)'},\n",
       " {'subsystem': 'Keratan sulfate degradation',\n",
       "  'name': 'beta-galactosidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)']},\n",
       "  'metabolites': {'gal_l': 2.0,\n",
       "   'h2o_l': -2.0,\n",
       "   'ksii_core4_deg3_l': 1.0,\n",
       "   'ksii_core4_deg2_l': -1.0},\n",
       "  'id': 'GALASE20ly',\n",
       "  'gene_reaction_rule': '(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)'},\n",
       " {'subsystem': 'Keratan sulfate degradation',\n",
       "  'name': 'beta-galactosidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)']},\n",
       "  'metabolites': {'ksi_deg5_l': -1.0,\n",
       "   'gal_l': 2.0,\n",
       "   'h2o_l': -2.0,\n",
       "   'ksi_deg6_l': 1.0},\n",
       "  'id': 'GALASE3ly',\n",
       "  'gene_reaction_rule': '(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)'},\n",
       " {'subsystem': 'Keratan sulfate degradation',\n",
       "  'name': 'beta-galactosidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)']},\n",
       "  'metabolites': {'gal_l': 1.0,\n",
       "   'ksi_deg9_l': 1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   'ksi_deg8_l': -1.0},\n",
       "  'id': 'GALASE4ly',\n",
       "  'gene_reaction_rule': '(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)'},\n",
       " {'subsystem': 'Keratan sulfate degradation',\n",
       "  'name': 'beta-galactosidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)']},\n",
       "  'metabolites': {'ksi_deg11_l': -1.0,\n",
       "   'gal_l': 1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   'ksi_deg12_l': 1.0},\n",
       "  'id': 'GALASE5ly',\n",
       "  'gene_reaction_rule': '(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)'},\n",
       " {'subsystem': 'Keratan sulfate degradation',\n",
       "  'name': 'beta-galactosidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)']},\n",
       "  'metabolites': {'gal_l': 1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   'ksi_deg15_l': 1.0,\n",
       "   'ksi_deg14_l': -1.0},\n",
       "  'id': 'GALASE6ly',\n",
       "  'gene_reaction_rule': '(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)'},\n",
       " {'subsystem': 'Keratan sulfate degradation',\n",
       "  'name': 'beta-galactosidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)']},\n",
       "  'metabolites': {'ksi_deg17_l': -1.0,\n",
       "   'ksi_deg18_l': 1.0,\n",
       "   'gal_l': 1.0,\n",
       "   'h2o_l': -1.0},\n",
       "  'id': 'GALASE7ly',\n",
       "  'gene_reaction_rule': '(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)'},\n",
       " {'subsystem': 'Keratan sulfate degradation',\n",
       "  'name': 'beta-galactosidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)']},\n",
       "  'metabolites': {'ksi_deg20_l': -1.0,\n",
       "   'ksi_deg21_l': 1.0,\n",
       "   'gal_l': 1.0,\n",
       "   'h2o_l': -1.0},\n",
       "  'id': 'GALASE8ly',\n",
       "  'gene_reaction_rule': '(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)'},\n",
       " {'subsystem': 'Keratan sulfate degradation',\n",
       "  'name': 'beta-galactosidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)']},\n",
       "  'metabolites': {'gal_l': 1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   'ksi_deg24_l': 1.0,\n",
       "   'ksi_deg23_l': -1.0},\n",
       "  'id': 'GALASE9ly',\n",
       "  'gene_reaction_rule': '(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'Galactocerebrosidase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4115']},\n",
       "  'metabolites': {'crm_hs_l': 1.0,\n",
       "   'gal_l': 1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   'galside_hs_l': -1.0},\n",
       "  'id': 'GALC',\n",
       "  'gene_reaction_rule': 'HGNC:4115'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'Beta-1,4 N-acetylgalactosaminyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4117']},\n",
       "  'metabolites': {'galgluside_hs_g': -1.0,\n",
       "   'h_g': 1.0,\n",
       "   'udpacgal_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'ga2_hs_g': 1.0},\n",
       "  'id': 'GALGT1',\n",
       "  'gene_reaction_rule': 'HGNC:4117'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'Beta-1,4 N-acetylgalactosaminyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4117']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'udpacgal_g': -1.0,\n",
       "   'gm2_hs_g': 1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'gm3_hs_g': -1.0},\n",
       "  'id': 'GALGT2',\n",
       "  'gene_reaction_rule': 'HGNC:4117'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'Beta-1,4 N-acetylgalactosaminyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4117']},\n",
       "  'metabolites': {'gd2_hs_g': 1.0,\n",
       "   'udpacgal_g': -1.0,\n",
       "   'gd3_hs_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'GALGT3',\n",
       "  'gene_reaction_rule': 'HGNC:4117'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'Beta-1,4 N-acetylgalactosaminyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4117']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'udpacgal_g': -1.0,\n",
       "   'gt3_hs_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'gt2_hs_g': 1.0},\n",
       "  'id': 'GALGT4',\n",
       "  'gene_reaction_rule': 'HGNC:4117'},\n",
       " {'subsystem': 'Galactose metabolism',\n",
       "  'name': 'galactokinase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['INVALID ECO TERM: IFH'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4118 or HGNC:4119']},\n",
       "  'metabolites': {'gal_c': -1.0,\n",
       "   'gal1p_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'adp_c': 1.0},\n",
       "  'id': 'GALK',\n",
       "  'gene_reaction_rule': 'HGNC:4118 or HGNC:4119'},\n",
       " {'subsystem': 'Chondroitin synthesis',\n",
       "  'name': 'UDPGal-chondroitin acetylgalactosaminyltransferase I, Golgi',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:24290 or HGNC:24292']},\n",
       "  'metabolites': {'cs_pre_g': 1.0,\n",
       "   'h_g': 1.0,\n",
       "   'udpacgal_g': -1.0,\n",
       "   'cs_hs_linkage_g': -1.0,\n",
       "   'udp_g': 1.0},\n",
       "  'id': 'GALNACT1g',\n",
       "  'gene_reaction_rule': 'HGNC:24290 or HGNC:24292'},\n",
       " {'subsystem': 'Chondroitin synthesis',\n",
       "  'name': 'uridine diphosphoacetylgalactosamine-chondroitin acetylgalactosaminyltransferase II, Golgi',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:17198 and HGNC:24291) or HGNC:24290 or HGNC:24292 or HGNC:24293']},\n",
       "  'metabolites': {'uacgam_g': -1.0,\n",
       "   'cs_a_b_pre2_g': -1.0,\n",
       "   'cs_a_b_pre3_g': 1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'GALNACT2g',\n",
       "  'gene_reaction_rule': '(HGNC:17198 and HGNC:24291) or HGNC:24290 or HGNC:24292 or HGNC:24293'},\n",
       " {'subsystem': 'Chondroitin synthesis',\n",
       "  'name': 'uridine diphosphoacetylgalactosamine-chondroitin acetylgalactosaminyltransferase II, Golgi',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:17198 and HGNC:24291) or HGNC:24290 or HGNC:24292 or HGNC:24293']},\n",
       "  'metabolites': {'cs_c_pre3_g': 1.0,\n",
       "   'h_g': 1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'cs_c_pre2_g': -1.0,\n",
       "   'udp_g': 1.0},\n",
       "  'id': 'GALNACT3g',\n",
       "  'gene_reaction_rule': '(HGNC:17198 and HGNC:24291) or HGNC:24290 or HGNC:24292 or HGNC:24293'},\n",
       " {'subsystem': 'Chondroitin synthesis',\n",
       "  'name': 'uridine diphosphoacetylgalactosamine-chondroitin acetylgalactosaminyltransferase II, Golgi',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:17198 and HGNC:24291) or HGNC:24290 or HGNC:24292 or HGNC:24293']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'cs_d_pre4_g': 1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'cs_d_pre3_g': -1.0},\n",
       "  'id': 'GALNACT4g',\n",
       "  'gene_reaction_rule': '(HGNC:17198 and HGNC:24291) or HGNC:24290 or HGNC:24292 or HGNC:24293'},\n",
       " {'subsystem': 'Chondroitin synthesis',\n",
       "  'name': 'uridine diphosphoacetylgalactosamine-chondroitin acetylgalactosaminyltransferase II, Golgi',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:17198 and HGNC:24291) or HGNC:24290 or HGNC:24292 or HGNC:24293']},\n",
       "  'metabolites': {'udp_g': 1.0,\n",
       "   'h_g': 1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'cs_e_pre4_g': 1.0,\n",
       "   'cs_e_pre3_g': -1.0},\n",
       "  'id': 'GALNACT5g',\n",
       "  'gene_reaction_rule': '(HGNC:17198 and HGNC:24291) or HGNC:24290 or HGNC:24292 or HGNC:24293'},\n",
       " {'subsystem': 'O-glycan synthesis',\n",
       "  'name': 'GalNAc transferase, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:19873 or HGNC:19875 or HGNC:19877 or HGNC:21531 or HGNC:22946 or HGNC:23242 or HGNC:4123 or HGNC:4124 or HGNC:4125 or HGNC:4126 or HGNC:4127 or HGNC:4128 or HGNC:4129 or HGNC:4130 or HGNC:4131']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'Ser_Thr_g': -1.0,\n",
       "   'udpacgal_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'Tn_antigen_g': 0.2},\n",
       "  'id': 'GALNTg',\n",
       "  'gene_reaction_rule': 'HGNC:19873 or HGNC:19875 or HGNC:19877 or HGNC:21531 or HGNC:22946 or HGNC:23242 or HGNC:4123 or HGNC:4124 or HGNC:4125 or HGNC:4126 or HGNC:4127 or HGNC:4128 or HGNC:4129 or HGNC:4130 or HGNC:4131'},\n",
       " {'subsystem': 'Galactose metabolism',\n",
       "  'name': 'D-Galactose:NADP+ 1-oxidoreductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'GENE ASSOCIATION': ['HGNC:380 or HGNC:381']},\n",
       "  'metabolites': {'gal_c': -1.0,\n",
       "   'nadph_c': -1.0,\n",
       "   'nadp_c': 1.0,\n",
       "   'galt_c': 1.0,\n",
       "   'h_c': -1.0},\n",
       "  'id': 'GALOR',\n",
       "  'gene_reaction_rule': 'HGNC:380 or HGNC:381'},\n",
       " {'subsystem': 'Galactose metabolism',\n",
       "  'name': 'galactose-1-phosphate uridylyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'], 'GENE ASSOCIATION': ['HGNC:4135']},\n",
       "  'metabolites': {'udpgal_c': 1.0,\n",
       "   'utp_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'h_c': -1.0,\n",
       "   'gal1p_c': -1.0},\n",
       "  'id': 'GALT',\n",
       "  'gene_reaction_rule': 'HGNC:4135'},\n",
       " {'subsystem': 'Chondroitin synthesis',\n",
       "  'name': 'UDP-D-galactose:galactosylxylose galactosyltransferase, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:17978']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'l2xser_g': 1.0,\n",
       "   'lxser_g': -1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0},\n",
       "  'id': 'GALT2g',\n",
       "  'gene_reaction_rule': 'HGNC:17978'},\n",
       " {'subsystem': 'Chondroitin synthesis',\n",
       "  'name': 'UDP-D-galactose:D-xylose galactosyltransferase, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:930']},\n",
       "  'metabolites': {'h2o_g': -1.0,\n",
       "   'lxser_g': 1.0,\n",
       "   'h_g': 1.0,\n",
       "   'udpgal_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'xser_g': -1.0},\n",
       "  'id': 'GALTg',\n",
       "  'gene_reaction_rule': 'HGNC:930'},\n",
       " {'subsystem': 'Galactose metabolism',\n",
       "  'name': 'UTP-glucose-1-phosphate uridylyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['Human Y/N: Y'],\n",
       "   'GENE ASSOCIATION': ['HGNC:12527 or HGNC:12527']},\n",
       "  'metabolites': {'utp_c': -1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'udpg_c': 1.0,\n",
       "   'h_c': -1.0,\n",
       "   'g1p_c': -1.0},\n",
       "  'id': 'GALU',\n",
       "  'gene_reaction_rule': 'HGNC:12527 or HGNC:12527'},\n",
       " {'subsystem': 'Starch and sucrose metabolism',\n",
       "  'name': 'glucoamylase, extracellular (glygn5 -> malt)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:7043']},\n",
       "  'metabolites': {'Tyr_ggn_e': 1.0,\n",
       "   'h2o_e': -1.0,\n",
       "   'malt_e': 1.0,\n",
       "   'glygn5_e': -1.0},\n",
       "  'id': 'GAMYe',\n",
       "  'gene_reaction_rule': 'HGNC:7043'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'Ganglioside O-acetylation',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['for humans - seq predicted SLC transporters have been found to have O-acetylation activity on gangliosides NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:95']},\n",
       "  'metabolites': {'coa_c': 1.0,\n",
       "   'gd3_hs_c': -1.0,\n",
       "   'oagd3_hs_c': 1.0,\n",
       "   'h_c': 2.0,\n",
       "   'accoa_c': -1.0},\n",
       "  'id': 'GAO1',\n",
       "  'gene_reaction_rule': 'HGNC:95'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'Ganglioside O-acetylation',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['for humans - seq predicted SLC transporters have been found to have O-acetylation activity on gangliosides NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:95']},\n",
       "  'metabolites': {'coa_g': 1.0,\n",
       "   'oagd3_hs_g': 1.0,\n",
       "   'accoa_g': -1.0,\n",
       "   'gd3_hs_g': -1.0,\n",
       "   'h_g': 2.0},\n",
       "  'id': 'GAO1g',\n",
       "  'gene_reaction_rule': 'HGNC:95'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'Ganglioside O-acetylation',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['for humans - seq predicted SLC transporters have been found to have O-acetylation activity on gangliosides NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:95']},\n",
       "  'metabolites': {'oagt3_hs_c': 1.0,\n",
       "   'gt3_hs_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'accoa_c': -1.0},\n",
       "  'id': 'GAO2',\n",
       "  'gene_reaction_rule': 'HGNC:95'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'Ganglioside O-acetylation',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['for humans - seq predicted SLC transporters have been found to have O-acetylation activity on gangliosides NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:95']},\n",
       "  'metabolites': {'coa_g': 1.0,\n",
       "   'oagt3_hs_g': 1.0,\n",
       "   'accoa_g': -1.0,\n",
       "   'gt3_hs_g': -1.0},\n",
       "  'id': 'GAO2g',\n",
       "  'gene_reaction_rule': 'HGNC:95'},\n",
       " {'subsystem': 'Glycolysis/gluconeogenesis',\n",
       "  'name': 'glyceraldehyde-3-phosphate dehydrogenase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['SMP. Same reaction as GAPDH, but localied to spermatids.  NOTE:  I think &quot;glyceroyl&quot; in this and many other reactions should be &quot;glycerol&quot;.  There are &quot;glyceroyl&quot; compounds, but I don&apos;t think that is what is intended here.'],\n",
       "   'GENE ASSOCIATION': ['HGNC:24864 or HGNC:4141']},\n",
       "  'metabolites': {'nadh_c': 1.0,\n",
       "   '13dpg_c': 1.0,\n",
       "   'g3p_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   'nad_c': -1.0,\n",
       "   'pi_c': -1.0},\n",
       "  'id': 'GAPD',\n",
       "  'gene_reaction_rule': 'HGNC:24864 or HGNC:4141'},\n",
       " {'subsystem': 'Purine synthesis',\n",
       "  'name': 'phosphoribosylglycinamide formyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:4163']},\n",
       "  'metabolites': {'gar_c': -1.0,\n",
       "   'fgam_c': 1.0,\n",
       "   'thf_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   '10fthf_c': -1.0},\n",
       "  'id': 'GARFT',\n",
       "  'gene_reaction_rule': 'HGNC:4163'},\n",
       " {'subsystem': 'N-glycan degradation',\n",
       "  'name': 'glycosylasparaginase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:318']},\n",
       "  'metabolites': {'n2m2nmasn_l': -1.0,\n",
       "   'n2m2nm_l': 1.0,\n",
       "   'Asn_X_Ser_Thr_l': 1.0,\n",
       "   'h2o_l': -1.0},\n",
       "  'id': 'GASNASE2ly',\n",
       "  'gene_reaction_rule': 'HGNC:318'},\n",
       " {'subsystem': 'Keratan sulfate degradation',\n",
       "  'name': 'glycosylasparaginase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:318']},\n",
       "  'metabolites': {'ksi_deg2_l': 1.0,\n",
       "   'ksi_deg1_l': -1.0,\n",
       "   'Asn_X_Ser_Thr_l': 1.0,\n",
       "   'h2o_l': -1.0},\n",
       "  'id': 'GASNASE3ly',\n",
       "  'gene_reaction_rule': 'HGNC:318'},\n",
       " {'subsystem': 'N-glycan degradation',\n",
       "  'name': 'glycosylasparaginase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:318']},\n",
       "  'metabolites': {'Asn_X_Ser_Thr_l': 1.0,\n",
       "   's2l2n2m2masn_l': -1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   's2l2n2m2m_l': 1.0},\n",
       "  'id': 'GASNASEly',\n",
       "  'gene_reaction_rule': 'HGNC:318'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'Glucosylceramidase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4177']},\n",
       "  'metabolites': {'glc_D_c': 1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'crm_hs_c': 1.0,\n",
       "   'gluside_hs_c': -1.0},\n",
       "  'id': 'GBA',\n",
       "  'gene_reaction_rule': 'HGNC:4177'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'Glucosylceramidase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4177']},\n",
       "  'metabolites': {'gluside_hs_l': -1.0,\n",
       "   'crm_hs_l': 1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   'glc_D_l': 1.0},\n",
       "  'id': 'GBAl',\n",
       "  'gene_reaction_rule': 'HGNC:4177'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'globoside alpha-1,3-N-acetylgalactosaminyltransferase 1',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:20460']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'udpacgal_g': -1.0,\n",
       "   'acgagbside_hs_g': 1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'gbside_hs_g': -1.0},\n",
       "  'id': 'GBGT1',\n",
       "  'gene_reaction_rule': 'HGNC:20460'},\n",
       " {'subsystem': 'Glyoxylate and dicarboxylate metabolism',\n",
       "  'name': 'Glycolaldehyde dehydrogenase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['JLR'],\n",
       "   'GENE ASSOCIATION': ['HGNC:15472 or HGNC:402 or HGNC:403 or HGNC:405 or HGNC:409 or HGNC:410 or HGNC:411 or HGNC:412 or HGNC:877']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   'glyclt_c': 1.0,\n",
       "   'h_c': 2.0,\n",
       "   'nad_c': -1.0,\n",
       "   'nadh_c': 1.0,\n",
       "   'gcald_c': -1.0},\n",
       "  'id': 'GCALDD',\n",
       "  'gene_reaction_rule': 'HGNC:15472 or HGNC:402 or HGNC:403 or HGNC:405 or HGNC:409 or HGNC:410 or HGNC:411 or HGNC:412 or HGNC:877'},\n",
       " {'subsystem': 'Glyoxylate and dicarboxylate metabolism',\n",
       "  'name': 'Glycolaldehyde dehydrogenase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['JLR'],\n",
       "   'GENE ASSOCIATION': ['HGNC:404 or HGNC:406 or HGNC:407 or HGNC:408 or HGNC:7179 or HGNC:9722']},\n",
       "  'metabolites': {'h2o_m': -1.0,\n",
       "   'glyclt_m': 1.0,\n",
       "   'gcald_m': -1.0,\n",
       "   'nadh_m': 1.0,\n",
       "   'nad_m': -1.0,\n",
       "   'h_m': 2.0},\n",
       "  'id': 'GCALDDm',\n",
       "  'gene_reaction_rule': 'HGNC:404 or HGNC:406 or HGNC:407 or HGNC:408 or HGNC:7179 or HGNC:9722'},\n",
       " {'subsystem': 'Glycine, serine, alanine and threonine metabolism',\n",
       "  'name': 'glycine-cleavage complex (lipoamide), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2898 and HGNC:4208 and HGNC:4313 and HGNC:473']},\n",
       "  'metabolites': {'lpam_m': -1.0,\n",
       "   'gly_m': -1.0,\n",
       "   'alpam_m': 1.0,\n",
       "   'co2_m': 1.0,\n",
       "   'h_m': -1.0},\n",
       "  'id': 'GCC2am',\n",
       "  'gene_reaction_rule': 'HGNC:2898 and HGNC:4208 and HGNC:4313 and HGNC:473'},\n",
       " {'subsystem': 'Glycine, serine, alanine and threonine metabolism',\n",
       "  'name': 'glycine-cleavage system (lipoamide) irreversible, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2898 and HGNC:4208 and HGNC:4313 and HGNC:473']},\n",
       "  'metabolites': {'thf_m': -1.0,\n",
       "   'mlthf_m': 1.0,\n",
       "   'nh4_m': 1.0,\n",
       "   'alpam_m': -1.0,\n",
       "   'dhlam_m': 1.0},\n",
       "  'id': 'GCC2bim',\n",
       "  'gene_reaction_rule': 'HGNC:2898 and HGNC:4208 and HGNC:4313 and HGNC:473'},\n",
       " {'subsystem': 'Glycine, serine, alanine and threonine metabolism',\n",
       "  'name': 'glycine-cleavage complex (lipoamide), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2898 and HGNC:4208 and HGNC:4313 and HGNC:473']},\n",
       "  'metabolites': {'lpam_m': 1.0,\n",
       "   'nad_m': -1.0,\n",
       "   'h_m': 1.0,\n",
       "   'dhlam_m': -1.0,\n",
       "   'nadh_m': 1.0},\n",
       "  'id': 'GCC2cm',\n",
       "  'gene_reaction_rule': 'HGNC:2898 and HGNC:4208 and HGNC:4313 and HGNC:473'},\n",
       " {'subsystem': 'Glycine, serine, alanine and threonine metabolism',\n",
       "  'name': 'glycine-cleavage complex (lipoylprotein), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2898 and HGNC:4208 and HGNC:4313 and HGNC:473']},\n",
       "  'metabolites': {'gly_m': -1.0, 'lpro_m': -1.0, 'co2_m': 1.0, 'alpro_m': 1.0},\n",
       "  'id': 'GCCam',\n",
       "  'gene_reaction_rule': 'HGNC:2898 and HGNC:4208 and HGNC:4313 and HGNC:473'},\n",
       " {'subsystem': 'Glycine, serine, alanine and threonine metabolism',\n",
       "  'name': 'glycine-cleavage complex (lipoylprotein) irreversible, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2898 and HGNC:4208 and HGNC:4313 and HGNC:473']},\n",
       "  'metabolites': {'h_m': -1.0,\n",
       "   'mlthf_m': 1.0,\n",
       "   'dhlpro_m': 1.0,\n",
       "   'nh4_m': 1.0,\n",
       "   'alpro_m': -1.0,\n",
       "   'thf_m': -1.0},\n",
       "  'id': 'GCCbim',\n",
       "  'gene_reaction_rule': 'HGNC:2898 and HGNC:4208 and HGNC:4313 and HGNC:473'},\n",
       " {'subsystem': 'Glycine, serine, alanine and threonine metabolism',\n",
       "  'name': 'glycine-cleavage complex (lipoylprotein), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:2898 and HGNC:4208 and HGNC:4313 and HGNC:473']},\n",
       "  'metabolites': {'nad_m': -1.0,\n",
       "   'dhlpro_m': -1.0,\n",
       "   'lpro_m': 1.0,\n",
       "   'h_m': 1.0,\n",
       "   'nadh_m': 1.0},\n",
       "  'id': 'GCCcm',\n",
       "  'gene_reaction_rule': 'HGNC:2898 and HGNC:4208 and HGNC:4313 and HGNC:473'},\n",
       " {'subsystem': 'Blood group synthesis',\n",
       "  'name': 'N-acetyllactosaminide beta-1,6-N-acetylglucosaminyl-transferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4204']},\n",
       "  'metabolites': {'galacglcgal14acglcgalgluside_hs_g': -1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'acglcgalacglcgal14acglcgalgluside_hs_g': 1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'GCNTg',\n",
       "  'gene_reaction_rule': 'HGNC:4204'},\n",
       " {'subsystem': 'Aminosugar metabolism',\n",
       "  'name': 'glutamine-fructose-6-phosphate transaminase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4241 or HGNC:4242']},\n",
       "  'metabolites': {'gam6p_c': 1.0,\n",
       "   'gln_L_c': -1.0,\n",
       "   'glu_L_c': 1.0,\n",
       "   'f6p_c': -1.0},\n",
       "  'id': 'GF6PTA',\n",
       "  'gene_reaction_rule': 'HGNC:4241 or HGNC:4242'},\n",
       " {'subsystem': 'Fructose and mannose metabolism',\n",
       "  'name': 'GDP-L-fucose synthase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['Both human and Escherichia coli enzymes can use NADH in place of  NADPH to a slight extent. IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:12390']},\n",
       "  'metabolites': {'gdpfuc_c': 1.0,\n",
       "   'nadp_c': 1.0,\n",
       "   'nadph_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'gdpddman_c': -1.0},\n",
       "  'id': 'GFUCS',\n",
       "  'gene_reaction_rule': 'HGNC:12390'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'Gamma-glutamyl hydrolase (10FTHF5GLU), extracellular',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['REMOVES EITHER ONE OR TWO GLUTAMYL RESIDUES WITH A REACTION (WILL ONLY REPRESENT 1) - last step from 5-glutamyl to monoglutamyl is in one step IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4248']},\n",
       "  'metabolites': {'glu_L_e': 4.0,\n",
       "   '10fthf_e': 1.0,\n",
       "   'h2o_e': -4.0,\n",
       "   '10fthf5glu_e': -1.0},\n",
       "  'id': 'GGH_10FTHF5GLUe',\n",
       "  'gene_reaction_rule': 'HGNC:4248'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'Gamma-glutamyl hydrolase (10FTHF5GLU), lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['REMOVES EITHER ONE OR TWO GLUTAMYL RESIDUES WITH A REACTION (WILL ONLY REPRESENT 1) - last step from 5-glutamyl to monoglutamyl is in one step IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4248']},\n",
       "  'metabolites': {'glu_L_l': 4.0,\n",
       "   '10fthf_l': 1.0,\n",
       "   'h2o_l': -4.0,\n",
       "   '10fthf5glu_l': -1.0},\n",
       "  'id': 'GGH_10FTHF5GLUl',\n",
       "  'gene_reaction_rule': 'HGNC:4248'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'Gamma-glutamyl hydrolase (10FTHF6GLU), extracellular',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['REMOVES EITHER ONE OR TWO GLUTAMYL RESIDUES WITH A REACTION (WILL ONLY REPRESENT 1) IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4248']},\n",
       "  'metabolites': {'glu_L_e': 1.0,\n",
       "   'h2o_e': -1.0,\n",
       "   '10fthf5glu_e': 1.0,\n",
       "   '10fthf6glu_e': -1.0},\n",
       "  'id': 'GGH_10FTHF6GLUe',\n",
       "  'gene_reaction_rule': 'HGNC:4248'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'Gamma-glutamyl hydrolase (10FTHF6GLU), lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['REMOVES EITHER ONE OR TWO GLUTAMYL RESIDUES WITH A REACTION (WILL ONLY REPRESENT 1) IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4248']},\n",
       "  'metabolites': {'glu_L_l': 1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   '10fthf6glu_l': -1.0,\n",
       "   '10fthf5glu_l': 1.0},\n",
       "  'id': 'GGH_10FTHF6GLUl',\n",
       "  'gene_reaction_rule': 'HGNC:4248'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'Gamma-glutamyl hydrolase (10FTHF7GLU), extracellular',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['REMOVES EITHER ONE OR TWO GLUTAMYL RESIDUES WITH A REACTION (WILL ONLY REPRESENT 1) IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4248']},\n",
       "  'metabolites': {'glu_L_e': 1.0,\n",
       "   'h2o_e': -1.0,\n",
       "   '10fthf7glu_e': -1.0,\n",
       "   '10fthf6glu_e': 1.0},\n",
       "  'id': 'GGH_10FTHF7GLUe',\n",
       "  'gene_reaction_rule': 'HGNC:4248'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'Gamma-glutamyl hydrolase (10FTHF7GLU), lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['REMOVES EITHER ONE OR TWO GLUTAMYL RESIDUES WITH A REACTION (WILL ONLY REPRESENT 1) IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4248']},\n",
       "  'metabolites': {'glu_L_l': 1.0,\n",
       "   '10fthf7glu_l': -1.0,\n",
       "   '10fthf6glu_l': 1.0,\n",
       "   'h2o_l': -1.0},\n",
       "  'id': 'GGH_10FTHF7GLUl',\n",
       "  'gene_reaction_rule': 'HGNC:4248'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'Gamma-glutamyl hydrolase (5DHF), extracellular',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['REMOVES EITHER ONE OR TWO GLUTAMYL RESIDUES WITH A REACTION (WILL ONLY REPRESENT 1) - last step from 5-glutamyl to monoglutamyl is in one step IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4248']},\n",
       "  'metabolites': {'glu_L_e': 4.0, '5dhf_e': -1.0, 'h2o_e': -4.0, 'dhf_e': 1.0},\n",
       "  'id': 'GGH_5DHFe',\n",
       "  'gene_reaction_rule': 'HGNC:4248'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'Gamma-glutamyl hydrolase (5DHF), lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['REMOVES EITHER ONE OR TWO GLUTAMYL RESIDUES WITH A REACTION (WILL ONLY REPRESENT 1) - last step from 5-glutamyl to monoglutamyl is in one step IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4248']},\n",
       "  'metabolites': {'glu_L_l': 4.0, 'dhf_l': 1.0, 'h2o_l': -4.0, '5dhf_l': -1.0},\n",
       "  'id': 'GGH_5DHFl',\n",
       "  'gene_reaction_rule': 'HGNC:4248'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'Gamma-glutamyl hydrolase (5THF), extracellular',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['REMOVES EITHER ONE OR TWO GLUTAMYL RESIDUES WITH A REACTION (WILL ONLY REPRESENT 1) - last step from 5-glutamyl to monoglutamyl is in one step IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4248']},\n",
       "  'metabolites': {'glu_L_e': 4.0, 'thf_e': 1.0, 'h2o_e': -4.0, '5thf_e': -1.0},\n",
       "  'id': 'GGH_5THFe',\n",
       "  'gene_reaction_rule': 'HGNC:4248'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'Gamma-glutamyl hydrolase (5THF), lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['REMOVES EITHER ONE OR TWO GLUTAMYL RESIDUES WITH A REACTION (WILL ONLY REPRESENT 1) - last step from 5-glutamyl to monoglutamyl is in one step IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4248']},\n",
       "  'metabolites': {'5thf_l': -1.0, 'glu_L_l': 4.0, 'thf_l': 1.0, 'h2o_l': -4.0},\n",
       "  'id': 'GGH_5THFl',\n",
       "  'gene_reaction_rule': 'HGNC:4248'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'Gamma-glutamyl hydrolase (6DHF), extracellular',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['REMOVES EITHER ONE OR TWO GLUTAMYL RESIDUES WITH A REACTION (WILL ONLY REPRESENT 1) IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4248']},\n",
       "  'metabolites': {'glu_L_e': 1.0,\n",
       "   '5dhf_e': 1.0,\n",
       "   'h2o_e': -1.0,\n",
       "   '6dhf_e': -1.0},\n",
       "  'id': 'GGH_6DHFe',\n",
       "  'gene_reaction_rule': 'HGNC:4248'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'Gamma-glutamyl hydrolase (6DHF), lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['REMOVES EITHER ONE OR TWO GLUTAMYL RESIDUES WITH A REACTION (WILL ONLY REPRESENT 1) IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4248']},\n",
       "  'metabolites': {'glu_L_l': 1.0,\n",
       "   '5dhf_l': 1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   '6dhf_l': -1.0},\n",
       "  'id': 'GGH_6DHFl',\n",
       "  'gene_reaction_rule': 'HGNC:4248'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'Gamma-glutamyl hydrolase (6THF), extracellular',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['REMOVES EITHER ONE OR TWO GLUTAMYL RESIDUES WITH A REACTION (WILL ONLY REPRESENT 1) IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4248']},\n",
       "  'metabolites': {'glu_L_e': 1.0,\n",
       "   '5thf_e': 1.0,\n",
       "   'h2o_e': -1.0,\n",
       "   '6thf_e': -1.0},\n",
       "  'id': 'GGH_6THFe',\n",
       "  'gene_reaction_rule': 'HGNC:4248'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'Gamma-glutamyl hydrolase (6THF), lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['REMOVES EITHER ONE OR TWO GLUTAMYL RESIDUES WITH A REACTION (WILL ONLY REPRESENT 1) IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4248']},\n",
       "  'metabolites': {'glu_L_l': 1.0,\n",
       "   '5thf_l': 1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   '6thf_l': -1.0},\n",
       "  'id': 'GGH_6THFl',\n",
       "  'gene_reaction_rule': 'HGNC:4248'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'Gamma-glutamyl hydrolase (7DHF), extracellular',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['REMOVES EITHER ONE OR TWO GLUTAMYL RESIDUES WITH A REACTION (WILL ONLY REPRESENT 1) IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4248']},\n",
       "  'metabolites': {'glu_L_e': 1.0,\n",
       "   'h2o_e': -1.0,\n",
       "   '7dhf_e': -1.0,\n",
       "   '6dhf_e': 1.0},\n",
       "  'id': 'GGH_7DHFe',\n",
       "  'gene_reaction_rule': 'HGNC:4248'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'Gamma-glutamyl hydrolase (7DHF), lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['REMOVES EITHER ONE OR TWO GLUTAMYL RESIDUES WITH A REACTION (WILL ONLY REPRESENT 1) IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4248']},\n",
       "  'metabolites': {'glu_L_l': 1.0,\n",
       "   '7dhf_l': -1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   '6dhf_l': 1.0},\n",
       "  'id': 'GGH_7DHFl',\n",
       "  'gene_reaction_rule': 'HGNC:4248'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'Gamma-glutamyl hydrolase (7THF), extracellular',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['REMOVES EITHER ONE OR TWO GLUTAMYL RESIDUES WITH A REACTION (WILL ONLY REPRESENT 1) IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4248']},\n",
       "  'metabolites': {'glu_L_e': 1.0,\n",
       "   'h2o_e': -1.0,\n",
       "   '7thf_e': -1.0,\n",
       "   '6thf_e': 1.0},\n",
       "  'id': 'GGH_7THFe',\n",
       "  'gene_reaction_rule': 'HGNC:4248'},\n",
       " {'subsystem': 'Folate metabolism',\n",
       "  'name': 'Gamma-glutamyl hydrolase (7THF), lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['REMOVES EITHER ONE OR TWO GLUTAMYL RESIDUES WITH A REACTION (WILL ONLY REPRESENT 1) IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4248']},\n",
       "  'metabolites': {'glu_L_l': 1.0,\n",
       "   '7thf_l': -1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   '6thf_l': 1.0},\n",
       "  'id': 'GGH_7THFl',\n",
       "  'gene_reaction_rule': 'HGNC:4248'},\n",
       " {'subsystem': 'Starch and sucrose metabolism',\n",
       "  'name': 'glycogenin self-glucosylation',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:4699 and HGNC:4706) or (HGNC:4699 and HGNC:4707) or (HGNC:4700 and HGNC:4706) or (HGNC:4700 and HGNC:4707)']},\n",
       "  'metabolites': {'udp_c': 8.0,\n",
       "   'ggn_c': 1.0,\n",
       "   'udpg_c': -8.0,\n",
       "   'h_c': 8.0,\n",
       "   'Tyr_ggn_c': -1.0},\n",
       "  'id': 'GGNG',\n",
       "  'gene_reaction_rule': '(HGNC:4699 and HGNC:4706) or (HGNC:4699 and HGNC:4707) or (HGNC:4700 and HGNC:4706) or (HGNC:4700 and HGNC:4707)'},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'geranylgeranyltransferase (uterus)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'], 'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'ttc_ggdp_c': -1.0,\n",
       "   'ppi_c': 3.0,\n",
       "   'dedoldp_U_c': 2.0,\n",
       "   'ipdp_c': -4.0,\n",
       "   'h_c': -6.0},\n",
       "  'id': 'GGT_U',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Eicosanoid metabolism',\n",
       "  'name': 'Gamma-glutamyltransferase 5',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Cleaves the gamma-glutamyl peptide bond of glutathione conjugates, but maybe not glutathione itself. Converts leukotriene C4 (LTC4) to leukotriene D4 (LTD4). NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4260']},\n",
       "  'metabolites': {'leuktrC4_r': -1.0,\n",
       "   'h2o_r': -1.0,\n",
       "   'leuktrD4_r': 1.0,\n",
       "   'glu_L_r': 1.0},\n",
       "  'id': 'GGT5r',\n",
       "  'gene_reaction_rule': 'HGNC:4260'},\n",
       " {'subsystem': 'Eicosanoid metabolism',\n",
       "  'name': 'Gamma-glutamyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['Converts leukotriene E4 (LTE4) to leukotriene D4 (LTE4). NJ'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'leuktrE4_c': -1.0,\n",
       "   'leuktrF4_c': 1.0,\n",
       "   'h2o_c': 1.0,\n",
       "   'glu_L_c': -1.0},\n",
       "  'id': 'GGT6',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Glycine, serine, alanine and threonine metabolism',\n",
       "  'name': 'glycine hydroxymethyltransferase, reversible',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:10850']},\n",
       "  'metabolites': {'ser_L_c': -1.0,\n",
       "   'h2o_c': 1.0,\n",
       "   'thf_c': -1.0,\n",
       "   'mlthf_c': 1.0,\n",
       "   'gly_c': 1.0},\n",
       "  'id': 'GHMT2r',\n",
       "  'gene_reaction_rule': 'HGNC:10850'},\n",
       " {'subsystem': 'Glycine, serine, alanine and threonine metabolism',\n",
       "  'name': 'glycine hydroxymethyltransferase, reversible, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:10850 or HGNC:10852']},\n",
       "  'metabolites': {'mlthf_m': 1.0,\n",
       "   'ser_L_m': -1.0,\n",
       "   'gly_m': 1.0,\n",
       "   'h2o_m': 1.0,\n",
       "   'thf_m': -1.0},\n",
       "  'id': 'GHMT2rm',\n",
       "  'gene_reaction_rule': 'HGNC:10850 or HGNC:10852'},\n",
       " {'subsystem': 'Lysine metabolism',\n",
       "  'name': 'glycine hydroxymethyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['MM-accoding to Hulse et. al (1978)'],\n",
       "   'GENE ASSOCIATION': ['HGNC:10850']},\n",
       "  'metabolites': {'4tmeabut_c': 1.0,\n",
       "   '3htmelys_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'gly_c': 1.0},\n",
       "  'id': 'GHMT3',\n",
       "  'gene_reaction_rule': 'HGNC:10850'},\n",
       " {'subsystem': 'Lysine metabolism',\n",
       "  'name': 'glycine hydroxymethyltransferase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['MM-according to Hulse et. al (1978)'],\n",
       "   'GENE ASSOCIATION': ['HGNC:10852']},\n",
       "  'metabolites': {'gly_m': 1.0,\n",
       "   'h_m': -1.0,\n",
       "   '3htmelys_m': -1.0,\n",
       "   '4tmeabut_m': 1.0},\n",
       "  'id': 'GHMT3m',\n",
       "  'gene_reaction_rule': 'HGNC:10852'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'guanylate kinase (GMP:ATP)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:4693']},\n",
       "  'metabolites': {'gmp_c': -1.0, 'gdp_c': 1.0, 'adp_c': 1.0, 'atp_c': -1.0},\n",
       "  'id': 'GK1',\n",
       "  'gene_reaction_rule': 'HGNC:4693'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'guanylate kinase (GMP:ATP), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['it. No evidence in the mitochondrion'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4693']},\n",
       "  'metabolites': {'adp_m': 1.0, 'atp_m': -1.0, 'gdp_m': 1.0, 'gmp_m': -1.0},\n",
       "  'id': 'GK1m',\n",
       "  'gene_reaction_rule': 'HGNC:4693'},\n",
       " {'subsystem': 'Vitamin C metabolism',\n",
       "  'name': 'D-Glucuronolactone:NAD+ oxidoreductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:15472 or HGNC:402 or HGNC:403 or HGNC:405 or HGNC:409 or HGNC:410 or HGNC:411 or HGNC:412']},\n",
       "  'metabolites': {'h2o_c': -2.0,\n",
       "   'glcr_c': 1.0,\n",
       "   'h_c': 3.0,\n",
       "   'nad_c': -1.0,\n",
       "   'nadh_c': 1.0,\n",
       "   'glac_c': -1.0},\n",
       "  'id': 'GLACO',\n",
       "  'gene_reaction_rule': 'HGNC:15472 or HGNC:402 or HGNC:403 or HGNC:405 or HGNC:409 or HGNC:410 or HGNC:411 or HGNC:412'},\n",
       " {'subsystem': 'Vitamin C metabolism',\n",
       "  'name': 'D-Glucuronolactone:NAD+ oxidoreductase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:404 or HGNC:406 or HGNC:407']},\n",
       "  'metabolites': {'glcr_m': 1.0,\n",
       "   'h2o_m': -2.0,\n",
       "   'glac_m': -1.0,\n",
       "   'nadh_m': 1.0,\n",
       "   'nad_m': -1.0,\n",
       "   'h_m': 3.0},\n",
       "  'id': 'GLACOm',\n",
       "  'gene_reaction_rule': 'HGNC:404 or HGNC:406 or HGNC:407'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'galactosidase, alpha',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4296']},\n",
       "  'metabolites': {'digalside_hs_l': -1.0,\n",
       "   'gal_l': 1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   'galside_hs_l': 1.0},\n",
       "  'id': 'GLAl',\n",
       "  'gene_reaction_rule': 'HGNC:4296'},\n",
       " {'subsystem': 'Sphingolipid metabolism',\n",
       "  'name': 'Beta-galactosidase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)']},\n",
       "  'metabolites': {'gluside_hs_l': 1.0,\n",
       "   'galgluside_hs_l': -1.0,\n",
       "   'gal_l': 1.0,\n",
       "   'h2o_l': -1.0},\n",
       "  'id': 'GLB1',\n",
       "  'gene_reaction_rule': '(HGNC:4122 and HGNC:4298 and HGNC:7758 and HGNC:9251) or (HGNC:4298 and HGNC:9251)'},\n",
       " {'subsystem': 'Starch and sucrose metabolism',\n",
       "  'name': '1,4-alpha-glucan branching enzyme (glygn1 -> glygn2)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4180']},\n",
       "  'metabolites': {'glygn2_c': 1.0, 'glygn1_c': -1.0},\n",
       "  'id': 'GLBRAN',\n",
       "  'gene_reaction_rule': 'HGNC:4180'},\n",
       " {'subsystem': 'Heparan sulfate degradation',\n",
       "  'name': 'beta-glucuronidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4696']},\n",
       "  'metabolites': {'hs_deg10_l': 1.0,\n",
       "   'hs_deg9_l': -1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   'glcur_l': 1.0},\n",
       "  'id': 'GLCAASE1ly',\n",
       "  'gene_reaction_rule': 'HGNC:4696'},\n",
       " {'subsystem': 'Chondroitin sulfate degradation',\n",
       "  'name': 'beta-glucuronidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4696']},\n",
       "  'metabolites': {'cs_a_deg3_l': 1.0,\n",
       "   'cs_a_deg2_l': -1.0,\n",
       "   'glcur_l': 1.0,\n",
       "   'h2o_l': -1.0},\n",
       "  'id': 'GLCAASE4ly',\n",
       "  'gene_reaction_rule': 'HGNC:4696'},\n",
       " {'subsystem': 'Chondroitin sulfate degradation',\n",
       "  'name': 'beta-glucuronidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4696']},\n",
       "  'metabolites': {'glcur_l': 1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   'cs_c_deg3_l': 1.0,\n",
       "   'cs_c_deg2_l': -1.0},\n",
       "  'id': 'GLCAASE5ly',\n",
       "  'gene_reaction_rule': 'HGNC:4696'},\n",
       " {'subsystem': 'Chondroitin sulfate degradation',\n",
       "  'name': 'beta-glucuronidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4696']},\n",
       "  'metabolites': {'cs_d_deg4_l': 1.0,\n",
       "   'cs_d_deg3_l': -1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   'glcur_l': 1.0},\n",
       "  'id': 'GLCAASE6ly',\n",
       "  'gene_reaction_rule': 'HGNC:4696'},\n",
       " {'subsystem': 'Chondroitin sulfate degradation',\n",
       "  'name': 'beta-glucuronidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4696']},\n",
       "  'metabolites': {'glcur_l': 1.0,\n",
       "   'cs_e_deg3_l': -1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   'cs_e_deg4_l': 1.0},\n",
       "  'id': 'GLCAASE7ly',\n",
       "  'gene_reaction_rule': 'HGNC:4696'},\n",
       " {'subsystem': 'Hyaluronan metabolism',\n",
       "  'name': 'beta-glucuronidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4696 or HGNC:5322']},\n",
       "  'metabolites': {'ha_deg1_l': 1.0,\n",
       "   'h_l': 2.0,\n",
       "   'glcur_l': 1.0,\n",
       "   'ha_l': -1.0,\n",
       "   'h2o_l': -1.0},\n",
       "  'id': 'GLCAASE8ly',\n",
       "  'gene_reaction_rule': 'HGNC:4696 or HGNC:5322'},\n",
       " {'subsystem': 'Hyaluronan metabolism',\n",
       "  'name': 'beta-glucuronidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4696']},\n",
       "  'metabolites': {'glcur_l': 1.0,\n",
       "   'ha_pre1_l': -1.0,\n",
       "   'h2o_l': -2.0,\n",
       "   'acgam_l': 1.0},\n",
       "  'id': 'GLCAASE9ly',\n",
       "  'gene_reaction_rule': 'HGNC:4696'},\n",
       " {'subsystem': 'Chondroitin synthesis',\n",
       "  'name': 'chondroitin-glucuronate 5-epimerase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'cs_b_pre4_g': 1.0, 'cs_a_b_pre3_g': -1.0},\n",
       "  'id': 'GLCAE1g',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Chondroitin synthesis',\n",
       "  'name': 'heparosan-N-sulfate-glucuronate 5-epimerase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:17855']},\n",
       "  'metabolites': {'hs_pre10_g': -1.0, 'hs_pre11_g': 1.0},\n",
       "  'id': 'GLCAE2g',\n",
       "  'gene_reaction_rule': 'HGNC:17855'},\n",
       " {'subsystem': 'Chondroitin synthesis',\n",
       "  'name': 'chondroitin glucuronyltransferase II, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['EC 2.4.1.226 NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:17198 and HGNC:24291) or HGNC:24293 or HGNC:29270']},\n",
       "  'metabolites': {'cs_a_b_e_pre1_g': -1.0,\n",
       "   'udpglcur_g': -1.0,\n",
       "   'cs_a_b_pre2_g': 1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'GLCAT2g',\n",
       "  'gene_reaction_rule': '(HGNC:17198 and HGNC:24291) or HGNC:24293 or HGNC:29270'},\n",
       " {'subsystem': 'Chondroitin synthesis',\n",
       "  'name': 'chondroitin glucuronyltransferase II, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['EC 2.4.1.226 NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:17198 and HGNC:24291) or HGNC:24293 or HGNC:29270']},\n",
       "  'metabolites': {'cs_c_d_e_pre1_g': -1.0,\n",
       "   'udpglcur_g': -1.0,\n",
       "   'cs_c_pre2_g': 1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'GLCAT3g',\n",
       "  'gene_reaction_rule': '(HGNC:17198 and HGNC:24291) or HGNC:24293 or HGNC:29270'},\n",
       " {'subsystem': 'Chondroitin synthesis',\n",
       "  'name': 'chondroitin glucuronyltransferase II, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['EC 2.4.1.226 NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:17198 and HGNC:24291) or HGNC:24293 or HGNC:29270']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'udpglcur_g': -1.0,\n",
       "   'cs_d_pre2_g': -1.0,\n",
       "   'cs_d_pre3_g': 1.0,\n",
       "   'udp_g': 1.0},\n",
       "  'id': 'GLCAT4g',\n",
       "  'gene_reaction_rule': '(HGNC:17198 and HGNC:24291) or HGNC:24293 or HGNC:29270'},\n",
       " {'subsystem': 'Chondroitin synthesis',\n",
       "  'name': 'chondroitin glucuronyltransferase II, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['EC 2.4.1.226 NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:17198 and HGNC:24291) or HGNC:24293 or HGNC:29270']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'udpglcur_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'cs_e_pre3_g': 1.0,\n",
       "   'cs_e_pre2_g': -1.0},\n",
       "  'id': 'GLCAT5g',\n",
       "  'gene_reaction_rule': '(HGNC:17198 and HGNC:24291) or HGNC:24293 or HGNC:29270'},\n",
       " {'subsystem': 'Chondroitin synthesis',\n",
       "  'name': 'heparan glucuronyltransferase II',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['EC 2.4.1.225 NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3512 and HGNC:3513']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'hs_pre2_g': 1.0,\n",
       "   'udpglcur_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'hs_pre1_g': -1.0},\n",
       "  'id': 'GLCAT6g',\n",
       "  'gene_reaction_rule': 'HGNC:3512 and HGNC:3513'},\n",
       " {'subsystem': 'Chondroitin synthesis',\n",
       "  'name': 'heparan glucuronyltransferase II',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['EC 2.4.1.225 NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3512 and HGNC:3513']},\n",
       "  'metabolites': {'hs_pre4_g': 1.0,\n",
       "   'h_g': 1.0,\n",
       "   'udpglcur_g': -1.0,\n",
       "   'hs_pre3_g': -1.0,\n",
       "   'udp_g': 1.0},\n",
       "  'id': 'GLCAT7g',\n",
       "  'gene_reaction_rule': 'HGNC:3512 and HGNC:3513'},\n",
       " {'subsystem': 'Chondroitin synthesis',\n",
       "  'name': 'heparan glucuronyltransferase II',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['EC 2.4.1.225 NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3512 and HGNC:3513']},\n",
       "  'metabolites': {'hs_pre5_g': -1.0,\n",
       "   'udpglcur_g': -1.0,\n",
       "   'hs_pre6_g': 1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'GLCAT8g',\n",
       "  'gene_reaction_rule': 'HGNC:3512 and HGNC:3513'},\n",
       " {'subsystem': 'Chondroitin synthesis',\n",
       "  'name': 'heparan glucuronyltransferase II',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['EC 2.4.1.225 NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3512 and HGNC:3513']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'udpglcur_g': -1.0,\n",
       "   'hs_pre8_g': 1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'hs_pre7_g': -1.0},\n",
       "  'id': 'GLCAT9g',\n",
       "  'gene_reaction_rule': 'HGNC:3512 and HGNC:3513'},\n",
       " {'subsystem': 'Chondroitin synthesis',\n",
       "  'name': 'UDP-D-glucuronate:galactosylgalactosylxylose glucuronosyltransferase, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:921 or HGNC:922 or HGNC:923']},\n",
       "  'metabolites': {'cs_hs_linkage_g': 1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'h2o_g': 1.0,\n",
       "   'udpglcur_g': -1.0,\n",
       "   'h_g': 1.0,\n",
       "   'l2xser_g': -1.0},\n",
       "  'id': 'GLCATg',\n",
       "  'gene_reaction_rule': 'HGNC:921 or HGNC:922 or HGNC:923'},\n",
       " {'subsystem': 'Heparan sulfate degradation',\n",
       "  'name': 'alpha-N-acetylglucosaminidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:7632']},\n",
       "  'metabolites': {'acgam_l': 1.0,\n",
       "   'hs_deg3_l': -1.0,\n",
       "   'hs_deg4_l': 1.0,\n",
       "   'h2o_l': -1.0},\n",
       "  'id': 'GLCNACASE1ly',\n",
       "  'gene_reaction_rule': 'HGNC:7632'},\n",
       " {'subsystem': 'Heparan sulfate degradation',\n",
       "  'name': 'alpha-N-acetylglucosaminidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:7632']},\n",
       "  'metabolites': {'hs_deg8_l': -1.0,\n",
       "   'hs_deg9_l': 1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   'acgam_l': 1.0},\n",
       "  'id': 'GLCNACASE2ly',\n",
       "  'gene_reaction_rule': 'HGNC:7632'},\n",
       " {'subsystem': 'Heparan sulfate degradation',\n",
       "  'name': 'alpha-N-acetylglucosaminidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:7632']},\n",
       "  'metabolites': {'hs_deg14_l': -1.0,\n",
       "   'hs_deg15_l': 1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   'acgam_l': 1.0},\n",
       "  'id': 'GLCNACASE3ly',\n",
       "  'gene_reaction_rule': 'HGNC:7632'},\n",
       " {'subsystem': 'Heparan sulfate degradation',\n",
       "  'name': 'alpha-N-acetylglucosaminidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:7632']},\n",
       "  'metabolites': {'hs_deg20_l': -1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   'hs_deg21_l': 1.0,\n",
       "   'acgam_l': 1.0},\n",
       "  'id': 'GLCNACASE4ly',\n",
       "  'gene_reaction_rule': 'HGNC:7632'},\n",
       " {'subsystem': 'Heparan sulfate degradation',\n",
       "  'name': 'alpha-N-acetylglucosaminidase, lysosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:7632']},\n",
       "  'metabolites': {'hs_deg25_l': 1.0,\n",
       "   'hs_deg24_l': -1.0,\n",
       "   'h2o_l': -1.0,\n",
       "   'acgam_l': 1.0},\n",
       "  'id': 'GLCNACASE5ly',\n",
       "  'gene_reaction_rule': 'HGNC:7632'},\n",
       " {'subsystem': 'Chondroitin synthesis',\n",
       "  'name': 'GlcNAc N-deacetylase / N-sulfotransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:20779 or HGNC:7680 or HGNC:7681 or HGNC:7682']},\n",
       "  'metabolites': {'pap_g': 4.0,\n",
       "   'h2o_g': -4.0,\n",
       "   'hs_pre10_g': 1.0,\n",
       "   'h_g': 4.0,\n",
       "   'paps_g': -4.0,\n",
       "   'hs_pre9_g': -1.0,\n",
       "   'ac_g': 4.0},\n",
       "  'id': 'GLCNACDASg',\n",
       "  'gene_reaction_rule': 'HGNC:20779 or HGNC:7680 or HGNC:7681 or HGNC:7682'},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'UDP-GlcNAc:dolichol-phosphate GlcNAc phosphotransferase (uterus)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'uacgam_c': -1.0,\n",
       "   'dolp_U_c': -1.0,\n",
       "   'h_c': -2.0,\n",
       "   'naglc2p_U_c': 1.0,\n",
       "   'ump_c': 1.0},\n",
       "  'id': 'GLCNACPT_U',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'N-glycan synthesis',\n",
       "  'name': 'UDP-GlcNAc:N-acetyl-D-glucosaminyl diphosphodolichol N-acetyl-D-glucosaminyltransferase (uterus)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'], 'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'uacgam_c': -1.0,\n",
       "   'udp_c': 1.0,\n",
       "   'chito2pdol_U_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'naglc2p_U_c': -1.0},\n",
       "  'id': 'GLCNACT_U',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Chondroitin synthesis',\n",
       "  'name': 'alpha-N-acetylglucosaminyltransferase I, Golgi apparatus',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['EC 2.4.1.223 NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3516 or HGNC:3518']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'cs_hs_linkage_g': -1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'hs_pre1_g': 1.0,\n",
       "   'udp_g': 1.0},\n",
       "  'id': 'GLCNACT1g',\n",
       "  'gene_reaction_rule': 'HGNC:3516 or HGNC:3518'},\n",
       " {'subsystem': 'Chondroitin synthesis',\n",
       "  'name': 'alpha-N-acetylglucosaminyltransferase II',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['EC 2.4.1.224 NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:3512 and HGNC:3513) or HGNC:3515 or HGNC:3518']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'hs_pre2_g': -1.0,\n",
       "   'hs_pre3_g': 1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'udp_g': 1.0},\n",
       "  'id': 'GLCNACT2g',\n",
       "  'gene_reaction_rule': '(HGNC:3512 and HGNC:3513) or HGNC:3515 or HGNC:3518'},\n",
       " {'subsystem': 'Chondroitin synthesis',\n",
       "  'name': 'alpha-N-acetylglucosaminyltransferase II',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['EC 2.4.1.224 NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:3512 and HGNC:3513) or HGNC:3515 or HGNC:3518']},\n",
       "  'metabolites': {'hs_pre4_g': -1.0,\n",
       "   'hs_pre5_g': 1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'GLCNACT3g',\n",
       "  'gene_reaction_rule': '(HGNC:3512 and HGNC:3513) or HGNC:3515 or HGNC:3518'},\n",
       " {'subsystem': 'Chondroitin synthesis',\n",
       "  'name': 'alpha-N-acetylglucosaminyltransferase II',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['EC 2.4.1.224 NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:3512 and HGNC:3513) or HGNC:3515 or HGNC:3518']},\n",
       "  'metabolites': {'h_g': 1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'hs_pre6_g': -1.0,\n",
       "   'hs_pre7_g': 1.0},\n",
       "  'id': 'GLCNACT4g',\n",
       "  'gene_reaction_rule': '(HGNC:3512 and HGNC:3513) or HGNC:3515 or HGNC:3518'},\n",
       " {'subsystem': 'Chondroitin synthesis',\n",
       "  'name': 'alpha-N-acetylglucosaminyltransferase II',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['EC 2.4.1.224 NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:3512 and HGNC:3513) or HGNC:3515 or HGNC:3518']},\n",
       "  'metabolites': {'hs_pre8_g': -1.0,\n",
       "   'uacgam_g': -1.0,\n",
       "   'hs_pre9_g': 1.0,\n",
       "   'udp_g': 1.0,\n",
       "   'h_g': 1.0},\n",
       "  'id': 'GLCNACT5g',\n",
       "  'gene_reaction_rule': '(HGNC:3512 and HGNC:3513) or HGNC:3515 or HGNC:3518'},\n",
       " {'subsystem': 'Starch and sucrose metabolism',\n",
       "  'name': 'glycogen debranching enzyme',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['lumped reaction, combination of 4-alpha-D-glucanotransferase (2.4.1.25) and amylo-alpha-1,6-glucosidase (3.2.1.33) NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:321']},\n",
       "  'metabolites': {'glc_D_c': 1.0,\n",
       "   'dxtrn_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'glygn3_c': 1.0},\n",
       "  'id': 'GLDBRAN',\n",
       "  'gene_reaction_rule': 'HGNC:321'},\n",
       " {'subsystem': 'Starch and sucrose metabolism',\n",
       "  'name': 'glycogen synthase (ggn -> glygn1)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:4699 and HGNC:4706) or (HGNC:4699 and HGNC:4707) or (HGNC:4700 and HGNC:4706) or (HGNC:4700 and HGNC:4707)']},\n",
       "  'metabolites': {'glygn1_c': 1.0,\n",
       "   'udp_c': 3.0,\n",
       "   'ggn_c': -1.0,\n",
       "   'udpg_c': -3.0,\n",
       "   'h_c': 3.0},\n",
       "  'id': 'GLGNS1',\n",
       "  'gene_reaction_rule': '(HGNC:4699 and HGNC:4706) or (HGNC:4699 and HGNC:4707) or (HGNC:4700 and HGNC:4706) or (HGNC:4700 and HGNC:4707)'},\n",
       " {'subsystem': 'Vitamin C metabolism',\n",
       "  'name': 'gluconolactonase, endoplasmic reticulum',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'gullac_r': 1.0, 'h2o_r': 1.0, 'h_r': -1.0, 'guln_r': -1.0},\n",
       "  'id': 'GLNLASEer',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Glutamate metabolism',\n",
       "  'name': 'glutamine synthetase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'GENE ASSOCIATION': ['HGNC:21016 or HGNC:4341']},\n",
       "  'metabolites': {'atp_c': -1.0,\n",
       "   'glu_L_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   'adp_c': 1.0,\n",
       "   'pi_c': 1.0,\n",
       "   'nh4_c': -1.0,\n",
       "   'gln_L_c': 1.0},\n",
       "  'id': 'GLNS',\n",
       "  'gene_reaction_rule': 'HGNC:21016 or HGNC:4341'},\n",
       " {'subsystem': 'Starch and sucrose metabolism',\n",
       "  'name': 'glycogen phosphorylase (glygn2 -> dxtrn)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:9723 or HGNC:9725 or HGNC:9726']},\n",
       "  'metabolites': {'dxtrn_c': 1.0,\n",
       "   'pi_c': -3.0,\n",
       "   'glygn2_c': -1.0,\n",
       "   'g1p_c': 3.0},\n",
       "  'id': 'GLPASE1',\n",
       "  'gene_reaction_rule': 'HGNC:9723 or HGNC:9725 or HGNC:9726'},\n",
       " {'subsystem': 'Starch and sucrose metabolism',\n",
       "  'name': 'glycogen phosphorylase (amyls -> glc-D)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:9723 or HGNC:9725 or HGNC:9726']},\n",
       "  'metabolites': {'glc_D_c': 7.0,\n",
       "   'h2o_c': -7.0,\n",
       "   'glygn3_c': -1.0,\n",
       "   'Tyr_ggn_c': 1.0},\n",
       "  'id': 'GLPASE2',\n",
       "  'gene_reaction_rule': 'HGNC:9723 or HGNC:9725 or HGNC:9726'},\n",
       " {'subsystem': 'Vitamin C metabolism',\n",
       "  'name': 'glucuronolactone reductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'nadp_c': -1.0,\n",
       "   'nadph_c': 1.0,\n",
       "   'glac_c': 1.0,\n",
       "   'gullac_c': -1.0,\n",
       "   'h_c': 1.0},\n",
       "  'id': 'GLRASE',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Urea cycle',\n",
       "  'name': 'glutamate 5-kinase (m)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['SAB m'],\n",
       "   'GENE ASSOCIATION': ['HGNC:9722']},\n",
       "  'metabolites': {'glu_L_m': -1.0,\n",
       "   'adp_m': 1.0,\n",
       "   'glu5p_m': 1.0,\n",
       "   'atp_m': -1.0},\n",
       "  'id': 'GLU5Km',\n",
       "  'gene_reaction_rule': 'HGNC:9722'},\n",
       " {'subsystem': 'Glutathione metabolism',\n",
       "  'name': 'Gamma-glutamylcysteine synthetase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4311 and HGNC:4312']},\n",
       "  'metabolites': {'cys_L_c': -1.0,\n",
       "   'glu_L_c': -1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'pi_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'adp_c': 1.0,\n",
       "   'glucys_c': 1.0},\n",
       "  'id': 'GLUCYS',\n",
       "  'gene_reaction_rule': 'HGNC:4311 and HGNC:4312'},\n",
       " {'subsystem': 'Glutamate metabolism',\n",
       "  'name': 'Glutamate Decarboxylase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['JLR- added h to balance'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4092 or HGNC:4093']},\n",
       "  'metabolites': {'co2_c': 1.0, 'h_c': -1.0, 'glu_L_c': -1.0, '4abut_c': 1.0},\n",
       "  'id': 'GLUDC',\n",
       "  'gene_reaction_rule': 'HGNC:4092 or HGNC:4093'},\n",
       " {'subsystem': 'Glutamate metabolism',\n",
       "  'name': 'glutamate dehydrogenase (NAD) (mitochondrial)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4335 or HGNC:4336']},\n",
       "  'metabolites': {'glu_L_m': -1.0,\n",
       "   'h2o_m': -1.0,\n",
       "   'akg_m': 1.0,\n",
       "   'nh4_m': 1.0,\n",
       "   'nadh_m': 1.0,\n",
       "   'nad_m': -1.0,\n",
       "   'h_m': 1.0},\n",
       "  'id': 'GLUDxm',\n",
       "  'gene_reaction_rule': 'HGNC:4335 or HGNC:4336'},\n",
       " {'subsystem': 'Glutamate metabolism',\n",
       "  'name': 'glutamate dehydrogenase (NADP), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4335 or HGNC:4336']},\n",
       "  'metabolites': {'glu_L_m': -1.0,\n",
       "   'nadph_m': 1.0,\n",
       "   'h2o_m': -1.0,\n",
       "   'nadp_m': -1.0,\n",
       "   'akg_m': 1.0,\n",
       "   'nh4_m': 1.0,\n",
       "   'h_m': 1.0},\n",
       "  'id': 'GLUDym',\n",
       "  'gene_reaction_rule': 'HGNC:4335 or HGNC:4336'},\n",
       " {'subsystem': 'Histidine metabolism',\n",
       "  'name': 'Glutamate formimidoyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3974']},\n",
       "  'metabolites': {'forglu_c': -1.0,\n",
       "   '5forthf_c': 1.0,\n",
       "   'h_c': -1.0,\n",
       "   'glu_L_c': 1.0,\n",
       "   'thf_c': -1.0},\n",
       "  'id': 'GluForTx',\n",
       "  'gene_reaction_rule': 'HGNC:3974'},\n",
       " {'subsystem': 'Glutamate metabolism',\n",
       "  'name': 'glutaminase (mitochondrial)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'GENE ASSOCIATION': ['HGNC:29570 or HGNC:4331']},\n",
       "  'metabolites': {'glu_L_m': 1.0,\n",
       "   'gln_L_m': -1.0,\n",
       "   'h2o_m': -1.0,\n",
       "   'nh4_m': 1.0},\n",
       "  'id': 'GLUNm',\n",
       "  'gene_reaction_rule': 'HGNC:29570 or HGNC:4331'},\n",
       " {'subsystem': 'Purine synthesis',\n",
       "  'name': 'glutamine phosphoribosyldiphosphate amidotransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:9238']},\n",
       "  'metabolites': {'pram_c': 1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'gln_L_c': -1.0,\n",
       "   'glu_L_c': 1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'prpp_c': -1.0},\n",
       "  'id': 'GLUPRT',\n",
       "  'gene_reaction_rule': 'HGNC:9238'},\n",
       " {'subsystem': 'Tryptophan metabolism',\n",
       "  'name': 'glutaryl-CoA dehydrogenase (mitochondria)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['SAB use of FAD is not certain'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4189']},\n",
       "  'metabolites': {'b2coa_m': 1.0,\n",
       "   'fadh2_m': 1.0,\n",
       "   'fad_m': -1.0,\n",
       "   'co2_m': 1.0,\n",
       "   'h_m': -1.0,\n",
       "   'glutcoa_m': -1.0},\n",
       "  'id': 'GLUTCOADHm',\n",
       "  'gene_reaction_rule': 'HGNC:4189'},\n",
       " {'subsystem': 'Glyoxylate and dicarboxylate metabolism',\n",
       "  'name': 'glyoxylate oxidase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:19708 or HGNC:21481 or HGNC:28335 or HGNC:30866 or (HGNC:6535 and HGNC:6541) or HGNC:6535 or HGNC:6541 or HGNC:6544']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   'oxa_c': 1.0,\n",
       "   'glx_c': -1.0,\n",
       "   'h_c': 2.0,\n",
       "   'nad_c': -1.0,\n",
       "   'nadh_c': 1.0},\n",
       "  'id': 'GLXO1',\n",
       "  'gene_reaction_rule': 'HGNC:19708 or HGNC:21481 or HGNC:28335 or HGNC:30866 or (HGNC:6535 and HGNC:6541) or HGNC:6535 or HGNC:6541 or HGNC:6544'},\n",
       " {'subsystem': 'Glyoxylate and dicarboxylate metabolism',\n",
       "  'name': 'glyoxylate oxidase, peroxisomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4809 or HGNC:4810']},\n",
       "  'metabolites': {'h2o2_x': 1.0,\n",
       "   'h_x': 1.0,\n",
       "   'glx_x': -1.0,\n",
       "   'oxa_x': 1.0,\n",
       "   'h2o_x': -1.0,\n",
       "   'o2_x': -1.0},\n",
       "  'id': 'GLXO2p',\n",
       "  'gene_reaction_rule': 'HGNC:4809 or HGNC:4810'},\n",
       " {'subsystem': 'Urea cycle',\n",
       "  'name': 'glycine amidinotransferase (c)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB c'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4175']},\n",
       "  'metabolites': {'arg_L_c': -1.0,\n",
       "   'orn_c': 1.0,\n",
       "   'gudac_c': 1.0,\n",
       "   'gly_c': -1.0},\n",
       "  'id': 'GLYAMDTRc',\n",
       "  'gene_reaction_rule': 'HGNC:4175'},\n",
       " {'subsystem': 'Glycine, serine, alanine and threonine metabolism',\n",
       "  'name': 'glycine C-acetyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'], 'GENE ASSOCIATION': ['HGNC:4188']},\n",
       "  'metabolites': {'accoa_m': -1.0,\n",
       "   '2aobut_m': 1.0,\n",
       "   'coa_m': 1.0,\n",
       "   'gly_m': -1.0},\n",
       "  'id': 'GLYATm',\n",
       "  'gene_reaction_rule': 'HGNC:4188'},\n",
       " {'subsystem': 'Glyoxylate and dicarboxylate metabolism',\n",
       "  'name': 'glycerate kinase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['JLR'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'2pg_c': 1.0,\n",
       "   'glyc_R_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'adp_c': 1.0},\n",
       "  'id': 'GLYCK2',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Glyoxylate and dicarboxylate metabolism',\n",
       "  'name': 'Glycolate dehydrogenase (NADP)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['JLR'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4570']},\n",
       "  'metabolites': {'glyclt_c': 1.0,\n",
       "   'nadp_c': 1.0,\n",
       "   'glx_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'nadph_c': -1.0},\n",
       "  'id': 'GLYCLTDy',\n",
       "  'gene_reaction_rule': 'HGNC:4570'},\n",
       " {'subsystem': 'Glyoxylate and dicarboxylate metabolism',\n",
       "  'name': 'Glycolate dehydrogenase (NADP), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'glx_m': -1.0,\n",
       "   'nadph_m': -1.0,\n",
       "   'h_m': -1.0,\n",
       "   'glyclt_m': 1.0,\n",
       "   'nadp_m': 1.0},\n",
       "  'id': 'GLYCLTDym',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Glyoxylate and dicarboxylate metabolism',\n",
       "  'name': 'Glycolate oxidase, peroxisome',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['tv NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4809 or HGNC:4810']},\n",
       "  'metabolites': {'glyclt_x': -1.0, 'h2o2_x': 1.0, 'o2_x': -1.0, 'glx_x': 1.0},\n",
       "  'id': 'GLYCTO1p',\n",
       "  'gene_reaction_rule': 'HGNC:4809 or HGNC:4810'},\n",
       " {'subsystem': 'Glycerophospholipid metabolism',\n",
       "  'name': 'glycerol kinase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4289 or HGNC:4291']},\n",
       "  'metabolites': {'adp_c': 1.0,\n",
       "   'glyc3p_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'glyc_c': -1.0},\n",
       "  'id': 'GLYK',\n",
       "  'gene_reaction_rule': 'HGNC:4289 or HGNC:4291'},\n",
       " {'subsystem': 'Glycerophospholipid metabolism',\n",
       "  'name': 'glycerol kinase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['TV'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4289 or HGNC:4291']},\n",
       "  'metabolites': {'adp_m': 1.0,\n",
       "   'atp_m': -1.0,\n",
       "   'glyc_m': -1.0,\n",
       "   'h_m': 1.0,\n",
       "   'glyc3p_m': 1.0},\n",
       "  'id': 'GLYKm',\n",
       "  'gene_reaction_rule': 'HGNC:4289 or HGNC:4291'},\n",
       " {'subsystem': 'Glycine, serine, alanine and threonine metabolism',\n",
       "  'name': 'glycine oxidase, perixosomal',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'gly_x': -1.0,\n",
       "   'nh4_x': 1.0,\n",
       "   'h2o2_x': 1.0,\n",
       "   'glx_x': 1.0,\n",
       "   'h2o_x': -1.0,\n",
       "   'o2_x': -1.0},\n",
       "  'id': 'GLYOp',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Pyruvate metabolism',\n",
       "  'name': 'hydroxyacylglutathione hydrolase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IF'],\n",
       "   'GENE ASSOCIATION': ['HGNC:14177 or HGNC:4805']},\n",
       "  'metabolites': {'gthrd_c': 1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'lac_D_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'lgt_S_c': -1.0},\n",
       "  'id': 'GLYOX',\n",
       "  'gene_reaction_rule': 'HGNC:14177 or HGNC:4805'},\n",
       " {'subsystem': 'Pyruvate metabolism',\n",
       "  'name': 'hydroxyacylglutathione hydrolase, mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IF'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4805']},\n",
       "  'metabolites': {'lgt_S_m': -1.0,\n",
       "   'gthrd_m': 1.0,\n",
       "   'h_m': 1.0,\n",
       "   'h2o_m': -1.0,\n",
       "   'lac_D_m': 1.0},\n",
       "  'id': 'GLYOXm',\n",
       "  'gene_reaction_rule': 'HGNC:4805'},\n",
       " {'subsystem': 'Fructose and mannose metabolism',\n",
       "  'name': 'GDP-D-mannose dehydratase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['JLR'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4369']},\n",
       "  'metabolites': {'h2o_c': 1.0, 'gdpddman_c': 1.0, 'gdpmann_c': -1.0},\n",
       "  'id': 'GMAND',\n",
       "  'gene_reaction_rule': 'HGNC:4369'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'GMP reductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4376 or HGNC:4377 or HGNC:4377 or HGNC:4377']},\n",
       "  'metabolites': {'nadp_c': 1.0,\n",
       "   'nadph_c': -1.0,\n",
       "   'imp_c': 1.0,\n",
       "   'gmp_c': -1.0,\n",
       "   'h_c': -2.0,\n",
       "   'nh4_c': 1.0},\n",
       "  'id': 'GMPR',\n",
       "  'gene_reaction_rule': 'HGNC:4376 or HGNC:4377 or HGNC:4377 or HGNC:4377'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'GMP synthase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:4378']},\n",
       "  'metabolites': {'xmp_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'gmp_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'glu_L_c': 1.0,\n",
       "   'ppi_c': 1.0,\n",
       "   'h_c': 2.0,\n",
       "   'amp_c': 1.0,\n",
       "   'gln_L_c': -1.0},\n",
       "  'id': 'GMPS2',\n",
       "  'gene_reaction_rule': 'HGNC:4378'},\n",
       " {'subsystem': 'Pentose phosphate pathway',\n",
       "  'name': 'phosphogluconate dehydrogenase, endoplasmic reticulum',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SMP Note no proton generation by the the reaction NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:8891']},\n",
       "  'metabolites': {'6pgc_r': -1.0,\n",
       "   'co2_r': 1.0,\n",
       "   'nadph_r': 1.0,\n",
       "   'nadp_r': -1.0,\n",
       "   'ru5p_D_r': 1.0},\n",
       "  'id': 'GNDer',\n",
       "  'gene_reaction_rule': 'HGNC:8891'},\n",
       " {'subsystem': 'Glycine, serine, alanine and threonine metabolism',\n",
       "  'name': 'glycine N-methyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:4415']},\n",
       "  'metabolites': {'amet_c': -1.0,\n",
       "   'ahcys_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'sarcs_c': 1.0,\n",
       "   'gly_c': -1.0},\n",
       "  'id': 'GNMT',\n",
       "  'gene_reaction_rule': 'HGNC:4415'},\n",
       " {'subsystem': 'Triacylglycerol synthesis',\n",
       "  'name': 'glycerol-3-phosphate acyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:24865 or HGNC:25193']},\n",
       "  'metabolites': {'h_c': 2.0,\n",
       "   'Rtotalcoa_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'glyc3p_c': -1.0,\n",
       "   'alpa_hs_c': 1.0},\n",
       "  'id': 'GPAM_hs',\n",
       "  'gene_reaction_rule': 'HGNC:24865 or HGNC:25193'},\n",
       " {'subsystem': 'Triacylglycerol synthesis',\n",
       "  'name': 'glycerol-3-phosphate acyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:20880 or HGNC:24865 or HGNC:27168 or HGNC:28157']},\n",
       "  'metabolites': {'glyc3p_m': -1.0,\n",
       "   'coa_m': 1.0,\n",
       "   'alpa_hs_m': 1.0,\n",
       "   'Rtotalcoa_m': -1.0,\n",
       "   'h_m': 2.0},\n",
       "  'id': 'GPAMm_hs',\n",
       "  'gene_reaction_rule': 'HGNC:20880 or HGNC:24865 or HGNC:27168 or HGNC:28157'},\n",
       " {'subsystem': 'Glycerophospholipid metabolism',\n",
       "  'name': 'Glycerophosphodiester phosphodiesterase (Glycerophosphocholine)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['JLR'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'chol_c': 1.0,\n",
       "   'glyc3p_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'g3pc_c': -1.0},\n",
       "  'id': 'GPDDA1',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Phosphatidylinositol phosphate metabolism',\n",
       "  'name': 'glucosaminylphosphatidyl inositol acetyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['EC 2.3.-.- NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:23213']},\n",
       "  'metabolites': {'pmtcoa_c': -1.0,\n",
       "   'gpail_hs_c': -1.0,\n",
       "   'coa_c': 1.0,\n",
       "   'gacpail_hs_c': 1.0},\n",
       "  'id': 'GPIAT',\n",
       "  'gene_reaction_rule': 'HGNC:23213'},\n",
       " {'subsystem': 'Phosphatidylinositol phosphate metabolism',\n",
       "  'name': 'glycophosphatidylinositol (GPI) deacylase, endoplasmic reticulum',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:25712']},\n",
       "  'metabolites': {'em2emgacpail_prot_hs_r': -1.0,\n",
       "   'h2o_r': -1.0,\n",
       "   'h_r': 1.0,\n",
       "   'hdca_r': 1.0,\n",
       "   'dem2emgacpail_prot_hs_r': 1.0},\n",
       "  'id': 'GPIDA2er',\n",
       "  'gene_reaction_rule': 'HGNC:25712'},\n",
       " {'subsystem': 'Phosphatidylinositol phosphate metabolism',\n",
       "  'name': 'glycophosphatidylinositol (GPI) deacylase, endoplasmic reticulum',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:25712']},\n",
       "  'metabolites': {'gpi_prot_hs_r': -1.0,\n",
       "   'dgpi_prot_hs_r': 1.0,\n",
       "   'h2o_r': -1.0,\n",
       "   'h_r': 1.0,\n",
       "   'hdca_r': 1.0},\n",
       "  'id': 'GPIDAer',\n",
       "  'gene_reaction_rule': 'HGNC:25712'},\n",
       " {'subsystem': 'Phosphatidylinositol phosphate metabolism',\n",
       "  'name': 'GlcN-acylPI mannosyltransferase, endoplasmic reticulum',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'dolp_U_r': 1.0,\n",
       "   'gacpail_hs_r': -1.0,\n",
       "   'mgacpail_hs_r': 1.0,\n",
       "   'h_r': 2.0,\n",
       "   'dolmanp_U_r': -1.0},\n",
       "  'id': 'GPIMTer_U',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Cholesterol metabolism',\n",
       "  'name': 'geranyltranstransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['Human Y/N: Y'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4249']},\n",
       "  'metabolites': {'ppi_c': 1.0, 'ipdp_c': -1.0, 'frdp_c': 1.0, 'grdp_c': -1.0},\n",
       "  'id': 'GRTT',\n",
       "  'gene_reaction_rule': 'HGNC:4249'},\n",
       " {'subsystem': 'Cholesterol metabolism',\n",
       "  'name': 'geranyltranstransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['peroxisomal version - NJ'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3631']},\n",
       "  'metabolites': {'ipdp_x': -1.0, 'ppi_x': 1.0, 'grdp_x': -1.0, 'frdp_x': 1.0},\n",
       "  'id': 'GRTTx',\n",
       "  'gene_reaction_rule': 'HGNC:3631'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'guanosine kinase (mitochondrial)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'adp_m': 1.0,\n",
       "   'gsn_m': -1.0,\n",
       "   'atp_m': -1.0,\n",
       "   'h_m': 1.0,\n",
       "   'gmp_m': 1.0},\n",
       "  'id': 'GSNKm',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Vitamin C metabolism',\n",
       "  'name': 'Glutathione dehydrogenase (dehydroascorbate reductase)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD. activity catalyzed by GSTO2-2'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4330 or HGNC:16065 or HGNC:16065']},\n",
       "  'metabolites': {'dhdascb_c': -1.0,\n",
       "   'ascb_L_c': 1.0,\n",
       "   'gthrd_c': -2.0,\n",
       "   'gthox_c': 1.0},\n",
       "  'id': 'GTHDH',\n",
       "  'gene_reaction_rule': 'HGNC:4330 or HGNC:16065 or HGNC:16065'},\n",
       " {'subsystem': 'Glutamate metabolism',\n",
       "  'name': 'glutathione oxidoreductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD, TV This EC (1.6.4.2) is obsolete.  The new EC is 1.8.1.7. http://www.genome.ad.jp/dbget-bin/www_bget?ec:1.6.4.2'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4623']},\n",
       "  'metabolites': {'nadph_c': -1.0,\n",
       "   'nadp_c': 1.0,\n",
       "   'gthrd_c': 2.0,\n",
       "   'h_c': -1.0,\n",
       "   'gthox_c': -1.0},\n",
       "  'id': 'GTHO',\n",
       "  'gene_reaction_rule': 'HGNC:4623'},\n",
       " {'subsystem': 'Glutamate metabolism',\n",
       "  'name': 'glutathione oxidoreductase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD, TV This EC (1.6.4.2) is obsolete.  The new EC is 1.8.1.7. http://www.genome.ad.jp/dbget-bin/www_bget?ec:1.6.4.2'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4623']},\n",
       "  'metabolites': {'gthox_m': -1.0,\n",
       "   'gthrd_m': 2.0,\n",
       "   'h_m': -1.0,\n",
       "   'nadp_m': 1.0,\n",
       "   'nadph_m': -1.0},\n",
       "  'id': 'GTHOm',\n",
       "  'gene_reaction_rule': 'HGNC:4623'},\n",
       " {'subsystem': 'Glutathione metabolism',\n",
       "  'name': 'glutathione peroxidase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4553 or HGNC:4554 or HGNC:4556 or HGNC:9352 or HGNC:9353']},\n",
       "  'metabolites': {'h2o2_c': -1.0,\n",
       "   'h2o_c': 2.0,\n",
       "   'gthrd_c': -2.0,\n",
       "   'gthox_c': 1.0},\n",
       "  'id': 'GTHP',\n",
       "  'gene_reaction_rule': 'HGNC:4553 or HGNC:4554 or HGNC:4556 or HGNC:9352 or HGNC:9353'},\n",
       " {'subsystem': 'Glutathione metabolism',\n",
       "  'name': 'glutathione peroxidase (e)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['SAB'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4553 or HGNC:4554 or HGNC:4555 or HGNC:4558 or HGNC:4559']},\n",
       "  'metabolites': {'gthox_e': 1.0,\n",
       "   'gthrd_e': -2.0,\n",
       "   'h2o_e': 2.0,\n",
       "   'h2o2_e': -1.0},\n",
       "  'id': 'GTHPe',\n",
       "  'gene_reaction_rule': 'HGNC:4553 or HGNC:4554 or HGNC:4555 or HGNC:4558 or HGNC:4559'},\n",
       " {'subsystem': 'Glutathione metabolism',\n",
       "  'name': 'glutathione peroxidase, mitochondria',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['TV (8-15-2003)'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4553 or HGNC:4556 or HGNC:9354']},\n",
       "  'metabolites': {'gthox_m': 1.0,\n",
       "   'gthrd_m': -2.0,\n",
       "   'h2o_m': 2.0,\n",
       "   'h2o2_m': -1.0},\n",
       "  'id': 'GTHPm',\n",
       "  'gene_reaction_rule': 'HGNC:4553 or HGNC:4556 or HGNC:9354'},\n",
       " {'subsystem': 'Glutathione metabolism',\n",
       "  'name': 'glutathione synthetase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:4624']},\n",
       "  'metabolites': {'glucys_c': -1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   'adp_c': 1.0,\n",
       "   'pi_c': 1.0,\n",
       "   'gthrd_c': 1.0,\n",
       "   'gly_c': -1.0},\n",
       "  'id': 'GTHS',\n",
       "  'gene_reaction_rule': 'HGNC:4624'},\n",
       " {'subsystem': 'Tetrahydrobiopterin metabolism',\n",
       "  'name': 'GTP cyclohydrolase I',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['The reaction involves hydrolysis of two C-N bonds and isomerization of the pentose unit; the recyclization may be non-enzymic'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4193']},\n",
       "  'metabolites': {'h2o_c': -1.0,\n",
       "   'gtp_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   'for_c': 1.0,\n",
       "   'ahdt_c': 1.0},\n",
       "  'id': 'GTPCI',\n",
       "  'gene_reaction_rule': 'HGNC:4193'},\n",
       " {'subsystem': 'Tetrahydrobiopterin metabolism',\n",
       "  'name': 'GTP cyclohydrolase I, nuclear',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['The reaction involves hydrolysis of two C-N bonds and isomerization of the pentose unit; the recyclization may be non-enzymic'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4193']},\n",
       "  'metabolites': {'h_n': 1.0,\n",
       "   'gtp_n': -1.0,\n",
       "   'ahdt_n': 1.0,\n",
       "   'for_n': 1.0,\n",
       "   'h2o_n': -1.0},\n",
       "  'id': 'GTPCIn',\n",
       "  'gene_reaction_rule': 'HGNC:4193'},\n",
       " {'subsystem': 'Nucleotide interconversion',\n",
       "  'name': 'guanylate cyclase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['IT'],\n",
       "   'GENE ASSOCIATION': ['(HGNC:4684 and HGNC:4687) or (HGNC:4685 and HGNC:4686) or (HGNC:4685 and HGNC:4687) or HGNC:4688 or HGNC:4689 or HGNC:4691 or HGNC:7943 or HGNC:7944']},\n",
       "  'metabolites': {'ppi_c': 1.0, 'gtp_c': -1.0, '35cgmp_c': 1.0},\n",
       "  'id': 'GUACYC',\n",
       "  'gene_reaction_rule': '(HGNC:4684 and HGNC:4687) or (HGNC:4685 and HGNC:4686) or (HGNC:4685 and HGNC:4687) or HGNC:4688 or HGNC:4689 or HGNC:4691 or HGNC:7943 or HGNC:7944'},\n",
       " {'subsystem': 'Purine catabolism',\n",
       "  'name': 'guanine deaminase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'], 'GENE ASSOCIATION': ['HGNC:4212']},\n",
       "  'metabolites': {'xan_c': 1.0,\n",
       "   'gua_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'nh4_c': 1.0},\n",
       "  'id': 'GUAD',\n",
       "  'gene_reaction_rule': 'HGNC:4212'},\n",
       " {'subsystem': 'Nucleotide salvage pathway',\n",
       "  'name': 'guanine phosphoribosyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Guanine and 6-mercaptopurine can replace hypoxanthine  EC added (2.4.2.7 makes the same reaction) IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:5157']},\n",
       "  'metabolites': {'gua_c': -1.0, 'gmp_c': 1.0, 'ppi_c': 1.0, 'prpp_c': -1.0},\n",
       "  'id': 'GUAPRT',\n",
       "  'gene_reaction_rule': 'HGNC:5157'},\n",
       " {'subsystem': 'Pentose phosphate pathway',\n",
       "  'name': 'L-gulonate 3-dehydrogenase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'nad_c': -1.0,\n",
       "   'nadh_c': 1.0,\n",
       "   'guln_c': -1.0,\n",
       "   'h_c': 1.0,\n",
       "   '3dhguln_c': 1.0},\n",
       "  'id': 'GULN3D',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Pentose phosphate pathway',\n",
       "  'name': 'gulonate dehydrogenase, endoplasmic reticulum',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'glcur_r': -1.0,\n",
       "   'nadp_r': 1.0,\n",
       "   'h_r': -1.0,\n",
       "   'nadph_r': -1.0,\n",
       "   'guln_r': 1.0},\n",
       "  'id': 'GULNDer',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Pentose phosphate pathway',\n",
       "  'name': 'Glucuronate 1-phosphate phosphatase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'glcur1p_c': -1.0,\n",
       "   'h2o_c': -1.0,\n",
       "   'pi_c': 1.0,\n",
       "   'glcur_c': 1.0},\n",
       "  'id': 'GUR1PP',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Miscellaneous',\n",
       "  'name': 'carboxylic acid dissociation',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['tv'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1368 or HGNC:1371 or HGNC:1372 or HGNC:1373 or HGNC:1374 or HGNC:1375 or HGNC:1380 or HGNC:1381 or HGNC:1383 or HGNC:14914']},\n",
       "  'metabolites': {'hco3_c': 1.0, 'h2o_c': -1.0, 'co2_c': -1.0, 'h_c': 1.0},\n",
       "  'id': 'H2CO3D',\n",
       "  'gene_reaction_rule': 'HGNC:1368 or HGNC:1371 or HGNC:1372 or HGNC:1373 or HGNC:1374 or HGNC:1375 or HGNC:1380 or HGNC:1381 or HGNC:1383 or HGNC:14914'},\n",
       " {'subsystem': 'Miscellaneous',\n",
       "  'name': 'carboxylic acid dissociation',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['tv'],\n",
       "   'GENE ASSOCIATION': ['HGNC:1377 or HGNC:1378']},\n",
       "  'metabolites': {'h_m': 1.0, 'h2o_m': -1.0, 'co2_m': -1.0, 'hco3_m': 1.0},\n",
       "  'id': 'H2CO3Dm',\n",
       "  'gene_reaction_rule': 'HGNC:1377 or HGNC:1378'},\n",
       " {'subsystem': 'Phosphatidylinositol phosphate metabolism',\n",
       "  'name': 'H2 phosphoethanolaminyl transferase, endoplasmic reticulum',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:8967']},\n",
       "  'metabolites': {'mgacpail_hs_r': -1.0,\n",
       "   'dag_hs_r': 1.0,\n",
       "   'emgacpail_hs_r': 1.0,\n",
       "   'pe_hs_r': -1.0},\n",
       "  'id': 'H2ETer',\n",
       "  'gene_reaction_rule': 'HGNC:8967'},\n",
       " {'subsystem': 'Phosphatidylinositol phosphate metabolism',\n",
       "  'name': 'H2 mannosyltransferase, endoplasmic reticulum',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'dolp_U_r': 1.0,\n",
       "   'mgacpail_hs_r': -1.0,\n",
       "   'h_r': 2.0,\n",
       "   'dolmanp_U_r': -1.0,\n",
       "   'm2gacpail_hs_r': 1.0},\n",
       "  'id': 'H2MTer_U',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Tyrosine metabolism',\n",
       "  'name': 'Hydrogen peroxide synthesis (NADPH dependent)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['SAB lumped reaction--first O2 goes to superoxide radical, then dismutes to H2O2'],\n",
       "   'GENE ASSOCIATION': ['HGNC:13273 or HGNC:3062']},\n",
       "  'metabolites': {'h2o2_c': 1.0,\n",
       "   'nadp_c': 1.0,\n",
       "   'nadph_c': -1.0,\n",
       "   'h_c': -1.0,\n",
       "   'o2_c': -1.0},\n",
       "  'id': 'H2O2syn',\n",
       "  'gene_reaction_rule': 'HGNC:13273 or HGNC:3062'},\n",
       " {'subsystem': 'Phosphatidylinositol phosphate metabolism',\n",
       "  'name': 'H3 phosphoethanolaminyl transferase, endoplasmic reticulum',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:8967']},\n",
       "  'metabolites': {'memgacpail_hs_r': 1.0,\n",
       "   'dag_hs_r': 1.0,\n",
       "   'pe_hs_r': -1.0,\n",
       "   'm2gacpail_hs_r': -1.0},\n",
       "  'id': 'H3ETer',\n",
       "  'gene_reaction_rule': 'HGNC:8967'},\n",
       " {'subsystem': 'Phosphatidylinositol phosphate metabolism',\n",
       "  'name': 'H3 mannosyltransferase, endoplasmic reticulum',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'dolp_U_r': 1.0,\n",
       "   'm3gacpail_hs_r': 1.0,\n",
       "   'h_r': 2.0,\n",
       "   'dolmanp_U_r': -1.0,\n",
       "   'm2gacpail_hs_r': -1.0},\n",
       "  'id': 'H3MTer_U',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Phosphatidylinositol phosphate metabolism',\n",
       "  'name': 'H4 phosphoethanolaminyl transferase, endoplasmic reticulum',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:23215 and HGNC:8962']},\n",
       "  'metabolites': {'m3gacpail_hs_r': -1.0,\n",
       "   'dag_hs_r': 1.0,\n",
       "   'em3gacpail_hs_r': 1.0,\n",
       "   'pe_hs_r': -1.0},\n",
       "  'id': 'H4ET3er',\n",
       "  'gene_reaction_rule': 'HGNC:23215 and HGNC:8962'},\n",
       " {'subsystem': 'Phosphatidylinositol phosphate metabolism',\n",
       "  'name': 'H4 phosphoethanolaminyl transferase, endoplasmic reticulum',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:8967']},\n",
       "  'metabolites': {'m3gacpail_hs_r': -1.0,\n",
       "   'dag_hs_r': 1.0,\n",
       "   'm2emgacpail_hs_r': 1.0,\n",
       "   'pe_hs_r': -1.0},\n",
       "  'id': 'H4ETer',\n",
       "  'gene_reaction_rule': 'HGNC:8967'},\n",
       " {'subsystem': 'Phosphatidylinositol phosphate metabolism',\n",
       "  'name': 'H5 mannosyltransferase, endoplasmic reticulum',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'memgacpail_hs_r': 1.0,\n",
       "   'dolp_U_r': 1.0,\n",
       "   'h_r': 2.0,\n",
       "   'emgacpail_hs_r': -1.0,\n",
       "   'dolmanp_U_r': -1.0},\n",
       "  'id': 'H5MTer_U',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Phosphatidylinositol phosphate metabolism',\n",
       "  'name': 'H6 phosphoethanolaminyl transferase, endoplasmic reticulum',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:25985 and HGNC:8962']},\n",
       "  'metabolites': {'dag_hs_r': 1.0,\n",
       "   'pe_hs_r': -1.0,\n",
       "   'em3gacpail_hs_r': -1.0,\n",
       "   'emem2gacpail_hs_r': 1.0},\n",
       "  'id': 'H6_ET2er',\n",
       "  'gene_reaction_rule': 'HGNC:25985 and HGNC:8962'},\n",
       " {'subsystem': 'Phosphatidylinositol phosphate metabolism',\n",
       "  'name': 'H6 phosphoethanolaminyl transferase, endoplasmic reticulum',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:23215 and HGNC:8962']},\n",
       "  'metabolites': {'m2emgacpail_hs_r': -1.0,\n",
       "   'pe_hs_r': -1.0,\n",
       "   'dag_hs_r': 1.0,\n",
       "   'em2emgacpail_hs_r': 1.0},\n",
       "  'id': 'H6ET3er',\n",
       "  'gene_reaction_rule': 'HGNC:23215 and HGNC:8962'},\n",
       " {'subsystem': 'Phosphatidylinositol phosphate metabolism',\n",
       "  'name': 'H6 phosphoethanolaminyl transferase, endoplasmic reticulum',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:8967']},\n",
       "  'metabolites': {'dag_hs_r': 1.0,\n",
       "   'pe_hs_r': -1.0,\n",
       "   'em3gacpail_hs_r': -1.0,\n",
       "   'em2emgacpail_hs_r': 1.0},\n",
       "  'id': 'H6_ETer',\n",
       "  'gene_reaction_rule': 'HGNC:8967'},\n",
       " {'subsystem': 'Phosphatidylinositol phosphate metabolism',\n",
       "  'name': 'H6 mannosyltransferase, endoplasmic reticulum',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'dolp_U_r': 1.0,\n",
       "   'h_r': 2.0,\n",
       "   'dolmanp_U_r': -1.0,\n",
       "   'm3emgacpail_hs_r': 1.0,\n",
       "   'm2emgacpail_hs_r': -1.0},\n",
       "  'id': 'H6MTer_U',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Phosphatidylinositol phosphate metabolism',\n",
       "  'name': 'H7 phosphoethanolaminyl transferase, endoplasmic reticulum',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:25985 and HGNC:8962']},\n",
       "  'metabolites': {'dag_hs_r': 1.0,\n",
       "   'pe_hs_r': -1.0,\n",
       "   'gpi_hs_r': 1.0,\n",
       "   'em2emgacpail_hs_r': -1.0},\n",
       "  'id': 'H7ET2er',\n",
       "  'gene_reaction_rule': 'HGNC:25985 and HGNC:8962'},\n",
       " {'subsystem': 'Phosphatidylinositol phosphate metabolism',\n",
       "  'name': 'H7 phosphoethanolaminyl transferase, endoplasmic reticulum',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:8967']},\n",
       "  'metabolites': {'dag_hs_r': 1.0,\n",
       "   'pe_hs_r': -1.0,\n",
       "   'gpi_hs_r': 1.0,\n",
       "   'emem2gacpail_hs_r': -1.0},\n",
       "  'id': 'H7_ETer',\n",
       "  'gene_reaction_rule': 'HGNC:8967'},\n",
       " {'subsystem': 'Phosphatidylinositol phosphate metabolism',\n",
       "  'name': 'H7 mannosyltransferase, endoplasmic reticulum',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['']},\n",
       "  'metabolites': {'dolp_U_r': 1.0,\n",
       "   'mem2emgacpail_hs_r': 1.0,\n",
       "   'h_r': 2.0,\n",
       "   'dolmanp_U_r': -1.0,\n",
       "   'em2emgacpail_hs_r': -1.0},\n",
       "  'id': 'H7MTer_U',\n",
       "  'gene_reaction_rule': ''},\n",
       " {'subsystem': 'Phosphatidylinositol phosphate metabolism',\n",
       "  'name': 'H7 transamidase, endoplasmic reticulum',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['EC 2.6.-.- NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:14937 and HGNC:14938 and HGNC:15791 and HGNC:4446 and HGNC:8965']},\n",
       "  'metabolites': {'pre_prot_r': -1.0,\n",
       "   'em2emgacpail_prot_hs_r': 1.0,\n",
       "   'gpi_sig_r': 1.0,\n",
       "   'em2emgacpail_hs_r': -1.0},\n",
       "  'id': 'H7_TAer',\n",
       "  'gene_reaction_rule': 'HGNC:14937 and HGNC:14938 and HGNC:15791 and HGNC:4446 and HGNC:8965'},\n",
       " {'subsystem': 'Phosphatidylinositol phosphate metabolism',\n",
       "  'name': 'H8 mannosyltransferase, endoplasmic reticulum',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:30596']},\n",
       "  'metabolites': {'m_em_3gacpail_r': 1.0,\n",
       "   'dolp_U_r': 1.0,\n",
       "   'h_r': 2.0,\n",
       "   'dolmanp_U_r': -1.0,\n",
       "   'gpi_hs_r': -1.0},\n",
       "  'id': 'H8MTer_L',\n",
       "  'gene_reaction_rule': 'HGNC:30596'},\n",
       " {'subsystem': 'Phosphatidylinositol phosphate metabolism',\n",
       "  'name': 'H8 mannosyltransferase, endoplasmic reticulum',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:30596']},\n",
       "  'metabolites': {'dolp_U_r': 1.0,\n",
       "   'm_em_3gacpail_hs_r': 1.0,\n",
       "   'h_r': 2.0,\n",
       "   'dolmanp_U_r': -1.0,\n",
       "   'gpi_hs_r': -1.0},\n",
       "  'id': 'H8MTer_U',\n",
       "  'gene_reaction_rule': 'HGNC:30596'},\n",
       " {'subsystem': 'Phosphatidylinositol phosphate metabolism',\n",
       "  'name': 'H8 transamidase, endoplasmic reticulum',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['EC 2.6.-.- NCD'],\n",
       "   'GENE ASSOCIATION': ['HGNC:14937 and HGNC:14938 and HGNC:15791 and HGNC:4446 and HGNC:8965']},\n",
       "  'metabolites': {'pre_prot_r': -1.0,\n",
       "   'gpi_prot_hs_r': 1.0,\n",
       "   'gpi_hs_r': -1.0,\n",
       "   'gpi_sig_r': 1.0},\n",
       "  'id': 'H8TAer',\n",
       "  'gene_reaction_rule': 'HGNC:14937 and HGNC:14938 and HGNC:15791 and HGNC:4446 and HGNC:8965'},\n",
       " {'subsystem': 'Tryptophan metabolism',\n",
       "  'name': '3-hydroxyacyl-CoA dehydrogenase (acetoacetyl-CoA) (mitochondria)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['Same as EC 1.1.1.157 SAB mito. some physiological'],\n",
       "   'GENE ASSOCIATION': ['HGNC:4799 or HGNC:4800 or (HGNC:4801 and HGNC:4803)']},\n",
       "  'metabolites': {'nad_m': 1.0,\n",
       "   'h_m': -1.0,\n",
       "   '3hbcoa_m': 1.0,\n",
       "   'nadh_m': -1.0,\n",
       "   'aacoa_m': -1.0},\n",
       "  'id': 'HACD1m',\n",
       "  'gene_reaction_rule': 'HGNC:4799 or HGNC:4800 or (HGNC:4801 and HGNC:4803)'},\n",
       " {'subsystem': 'Tryptophan metabolism',\n",
       "  'name': '3-hydroxyacyl-CoA dehydrogenase (acetoacetyl-CoA) (peroxisome)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['0'],\n",
       "   'NOTES': ['Same as EC 1.1.1.157 SAB x'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3247 or HGNC:5213']},\n",
       "  'metabolites': {'nadh_x': -1.0,\n",
       "   '3hbcoa_x': 1.0,\n",
       "   'aacoa_x': -1.0,\n",
       "   'nad_x': 1.0,\n",
       "   'h_x': -1.0},\n",
       "  'id': 'HACD1x',\n",
       "  'gene_reaction_rule': 'HGNC:3247 or HGNC:5213'},\n",
       " {'subsystem': 'Valine, leucine, and isoleucine metabolism',\n",
       "  'name': '3-hydroxyacyl-CoA dehydrogenase (2-Methylacetoacetyl-CoA), mitochondrial',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': -1000,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['MM. inborn deficiency enzyme causes accumulation of substrate 3-hydroxy-2-methylbutyric acid'],\n",
       "   'GENE ASSOCIATION': ['HGNC:3247 or HGNC:4799 or HGNC:4800']},\n",
       "  'metabolites': {'nad_m': -1.0,\n",
       "   'h_m': 1.0,\n",
       "   '2maacoa_m': 1.0,\n",
       "   '3hmbcoa_m': -1.0,\n",
       "   'nadh_m': 1.0},\n",
       "  'id': 'HACD9m',\n",
       "  'gene_reaction_rule': 'HGNC:3247 or HGNC:4799 or HGNC:4800'},\n",
       " {'subsystem': 'Ubiquinone synthesis',\n",
       "  'name': 'Hydroxybenzoate Decaprenyltransferase',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "   'NOTES': ['Used to make Ubiquinone-10 EC 2.5.1.- IT'],\n",
       "   'GENE ASSOCIATION': ['HGNC:25223']},\n",
       "  'metabolites': {'3dphb_m': 1.0,\n",
       "   'decdp_m': -1.0,\n",
       "   'ppi_m': 1.0,\n",
       "   '4hbz_m': -1.0},\n",
       "  'id': 'HBZOPT10m',\n",
       "  'gene_reaction_rule': 'HGNC:25223'},\n",
       " {'subsystem': 'Glycolysis/gluconeogenesis',\n",
       "  'name': 'hexokinase (D-glucose:ATP)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['2'],\n",
       "   'GENE ASSOCIATION': ['HGNC:23302 or HGNC:4195 or HGNC:4922 or HGNC:4923 or HGNC:4925']},\n",
       "  'metabolites': {'glc_D_c': -1.0,\n",
       "   'g6p_c': 1.0,\n",
       "   'h_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'adp_c': 1.0},\n",
       "  'id': 'HEX1',\n",
       "  'gene_reaction_rule': 'HGNC:23302 or HGNC:4195 or HGNC:4922 or HGNC:4923 or HGNC:4925'},\n",
       " {'subsystem': 'Aminosugar metabolism',\n",
       "  'name': 'hexokinase (D-glucosamine:ATP)',\n",
       "  'upper_bound': 1000,\n",
       "  'lower_bound': 0.0,\n",
       "  'notes': {'CONFIDENCE LEVEL': ['1'],\n",
       "   'NOTES': ['D-Glucose, D-mannose, D-fructose, sorbitol and D-glucosamine can act as acceptors; ITP and dATP can act as donors. The liver isoenzyme has sometimes been called glucokinase.'],\n",
       "   'GENE ASSOCIATION': ['HGNC:23302 or HGNC:4195 or HGNC:4922 or HGNC:4923 or HGNC:4925']},\n",
       "  'metabolites': {'gam6p_c': 1.0,\n",
       "   'gam_c': -1.0,\n",
       "   'adp_c': 1.0,\n",
       "   'atp_c': -1.0,\n",
       "   'h_c': 1.0},\n",
       "  'id': 'HEX10',\n",
       "  'gene_reaction_rule': 'HGNC:23302 or HGNC:4195 or HGNC:4922 or HGNC:4923 or HGNC:4925'},\n",
       " ...]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data[\"reactions\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'subsystem': 'Tyrosine metabolism',\n",
       " 'name': '3,4-Dihydroxyphenylethyleneglycol:NAD+ oxidoreductase',\n",
       " 'upper_bound': 1000,\n",
       " 'lower_bound': -1000,\n",
       " 'notes': {'CONFIDENCE LEVEL': ['4'],\n",
       "  'NOTES': ['SAB'],\n",
       "  'GENE ASSOCIATION': ['HGNC:16354 or (HGNC:249 and HGNC:250) or (HGNC:249 and HGNC:251) or (HGNC:250 and HGNC:251) or HGNC:252 or HGNC:253 or HGNC:255 or HGNC:256']},\n",
       " 'metabolites': {'34dhoxpeg_c': 1.0,\n",
       "  'nad_c': 1.0,\n",
       "  'nadh_c': -1.0,\n",
       "  'h_c': -1.0,\n",
       "  '34dhmald_c': -1.0},\n",
       " 'id': '34DHOXPEGOX',\n",
       " 'gene_reaction_rule': 'HGNC:16354 or (HGNC:249 and HGNC:250) or (HGNC:249 and HGNC:251) or (HGNC:250 and HGNC:251) or HGNC:252 or HGNC:253 or HGNC:255 or HGNC:256'}"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Accessing one reaction\n",
    "data[\"reactions\"][15]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### All  Recon reaction IDs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "recon_reaction_ids = [i['id'] for i in data['reactions']]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Cobra  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "#savin json file \n",
    "with open('Recon.json', 'w') as json_file:\n",
    "    json.dump(data, json_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "recon2 = cobra.io.load_json_model(join(\"Recon.json\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Exploring Cobra"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Reaction identifier</strong></td><td>34DHOXPEGOX</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>3,4-Dihydroxyphenylethyleneglycol:NAD+ oxidoreductase</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x019d9186b888</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Stoichiometry</strong></td>\n",
       "                <td>\n",
       "                    <p style='text-align:right'>34dhmald_c + h_c + nadh_c <=> 34dhoxpeg_c + nad_c</p>\n",
       "                    <p style='text-align:right'>3,4-dihydroxymandelaldehyde + proton + NADH(2-) <=> 3,4-dihydroxyphenylethyleneglycol + NAD(1-)</p>\n",
       "                </td>\n",
       "            </tr><tr>\n",
       "                <td><strong>GPR</strong></td><td>HGNC:16354 or (HGNC:249 and HGNC:250) or (HGNC:249 and HGNC:251) or (HGNC:250 and HGNC:251) or HG...</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Lower bound</strong></td><td>-1000</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Upper bound</strong></td><td>1000</td>\n",
       "            </tr>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<Reaction 34DHOXPEGOX at 0x19d9186b888>"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Find reaction by id\n",
    "recon2.reactions.get_by_id(\"34DHOXPEGOX\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Metabolite identifier</strong></td><td>34dhmald_c</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>3,4-dihydroxymandelaldehyde</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x019d916eb5c8</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Formula</strong></td><td>C8H8O4</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartment</strong></td><td>c</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>In 5 reaction(s)</strong></td><td>\n",
       "                    34DHXMANDACOX_NADP_, 34DHXMANDACOX, 34DHOXPEGOX, 41R2A1H12BOOX, 41R1H2MAE12BOOX</td>\n",
       "            </tr>\n",
       "        </table>"
      ],
      "text/plain": [
       "<Metabolite 34dhmald_c at 0x19d916eb5c8>"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Find metabolites by id\n",
    "recon2.metabolites.get_by_id(\"34dhmald_c\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Gene identifier</strong></td><td>HGNC:549</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td></td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x019d917e86c8</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Functional</strong></td><td>True</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>In 9 reaction(s)</strong></td><td>\n",
       "                    42A12BOOX, NMPTRCOX, PTRCOX1, HISTASE, 13DAMPPOX, TRYPTAOX, MHISOR, MAOX, PEAMNO</td>\n",
       "            </tr>\n",
       "        </table>"
      ],
      "text/plain": [
       "<Gene HGNC:549 at 0x19d917e86c8>"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Find gene by hgnc id\n",
    "recon2.genes.get_by_id(\"HGNC:549\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'HGNC:405 or HGNC:409 or HGNC:410 or HGNC:411'"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Get gene reaction rule of a reaction\n",
    "recon2.reactions.get_by_id('34DHPLACOX').gene_reaction_rule"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
